{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Finalize merged dataset w/o Sulfolobus\n",
    "* Incorporate phylogenetic tree into phyloseq object\n",
    "* Save master and split versions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: Rcpp\n",
      "Warning message:\n",
      "“package ‘Rcpp’ was built under R version 3.6.1”Registered S3 method overwritten by 'dplyr':\n",
      "  method               from  \n",
      "  as.data.frame.tbl_df tibble\n",
      "Warning message:\n",
      "“multiple methods tables found for ‘lengths’”Warning message:\n",
      "“multiple methods tables found for ‘rowSums’”Warning message:\n",
      "“multiple methods tables found for ‘colSums’”Warning message:\n",
      "“multiple methods tables found for ‘rowMeans’”Warning message:\n",
      "“multiple methods tables found for ‘colMeans’”\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "Warning message:\n",
      "“package ‘phyloseq’ was built under R version 3.6.1”Warning message:\n",
      "“package ‘ggplot2’ was built under R version 3.6.1”Warning message:\n",
      "“package ‘ape’ was built under R version 3.6.1”"
     ]
    }
   ],
   "source": [
    "library(dada2)\n",
    "library(tidyr)\n",
    "library(dplyr)\n",
    "library(phyloseq)\n",
    "library(ggplot2)\n",
    "library(ape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phyloseq-class experiment-level object\n",
       "otu_table()   OTU Table:         [ 15600 taxa and 434 samples ]\n",
       "sample_data() Sample Data:       [ 434 samples by 11 sample variables ]\n",
       "tax_table()   Taxonomy Table:    [ 15600 taxa by 6 taxonomic ranks ]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ps.thresh = readRDS('~/Hyphosphere/data/3Exp/phyloseq/3Exp_phyloseq_thresh.rds')\n",
    "ps.thresh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "Phylogenetic tree with 15430 tips and 15390 internal nodes.\n",
       "\n",
       "Tip labels:\n",
       "\tX90478, ASV5742, ASV6971, ASV2480, ASV15180, ASV9985, ...\n",
       "Node labels:\n",
       "\tRoot, 0.788, 0.546, 0.883.58, 0.952, 0.940, ...\n",
       "\n",
       "Rooted; includes branch lengths."
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "thresh.tree = read_tree(treefile = '~/Hyphosphere/data/3Exp/Fasttree/Master_wSulfo.tree')\n",
    "#is.rooted(thresh.tree)\n",
    "thresh.tree = root(thresh.tree, outgroup = 'X90478', r = TRUE)\n",
    "is.rooted(thresh.tree)\n",
    "thresh.tree\n",
    "\n",
    "# root.by.outgroup <- function(tree.unrooted) {\n",
    "#   longest.edge <- which.max(tree.unrooted$edge.length)\n",
    "#   long.nodes <- tree.unrooted$edge[longest.edge,]     #this should usually include one tip\n",
    "#   new.outgroup <- long.nodes[long.nodes < Ntip(tree.unrooted)]\n",
    "#   tree.rooted <- root(tree.unrooted, outgroup=new.outgroup, resolve.root=T)\n",
    "#   tree.rooted\n",
    "# }\n",
    "\n",
    "# require(ape)\n",
    "# thresh.tree = root.by.outgroup(thresh.tree)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'ASV28'</li>\n",
       "\t<li>'ASV31'</li>\n",
       "\t<li>'ASV82'</li>\n",
       "\t<li>'ASV90'</li>\n",
       "\t<li>'ASV128'</li>\n",
       "\t<li>'ASV130'</li>\n",
       "\t<li>'ASV140'</li>\n",
       "\t<li>'ASV149'</li>\n",
       "\t<li>'ASV152'</li>\n",
       "\t<li>'ASV186'</li>\n",
       "\t<li>'ASV201'</li>\n",
       "\t<li>'ASV331'</li>\n",
       "\t<li>'ASV353'</li>\n",
       "\t<li>'ASV391'</li>\n",
       "\t<li>'ASV432'</li>\n",
       "\t<li>'ASV547'</li>\n",
       "\t<li>'ASV594'</li>\n",
       "\t<li>'ASV650'</li>\n",
       "\t<li>'ASV728'</li>\n",
       "\t<li>'ASV763'</li>\n",
       "\t<li>'ASV922'</li>\n",
       "\t<li>'ASV937'</li>\n",
       "\t<li>'ASV995'</li>\n",
       "\t<li>'ASV1571'</li>\n",
       "\t<li>'ASV1732'</li>\n",
       "\t<li>'ASV1766'</li>\n",
       "\t<li>'ASV1777'</li>\n",
       "\t<li>'ASV1788'</li>\n",
       "\t<li>'ASV1841'</li>\n",
       "\t<li>'ASV1880'</li>\n",
       "\t<li>'ASV2159'</li>\n",
       "\t<li>'ASV2245'</li>\n",
       "\t<li>'ASV2277'</li>\n",
       "\t<li>'ASV2280'</li>\n",
       "\t<li>'ASV2631'</li>\n",
       "\t<li>'ASV2865'</li>\n",
       "\t<li>'ASV3045'</li>\n",
       "\t<li>'ASV3093'</li>\n",
       "\t<li>'ASV3275'</li>\n",
       "\t<li>'ASV3542'</li>\n",
       "\t<li>'ASV3574'</li>\n",
       "\t<li>'ASV3806'</li>\n",
       "\t<li>'ASV3933'</li>\n",
       "\t<li>'ASV4004'</li>\n",
       "\t<li>'ASV4213'</li>\n",
       "\t<li>'ASV4596'</li>\n",
       "\t<li>'ASV4612'</li>\n",
       "\t<li>'ASV4894'</li>\n",
       "\t<li>'ASV5214'</li>\n",
       "\t<li>'ASV5295'</li>\n",
       "\t<li>'ASV5384'</li>\n",
       "\t<li>'ASV5411'</li>\n",
       "\t<li>'ASV5507'</li>\n",
       "\t<li>'ASV5657'</li>\n",
       "\t<li>'ASV5728'</li>\n",
       "\t<li>'ASV5818'</li>\n",
       "\t<li>'ASV5833'</li>\n",
       "\t<li>'ASV5899'</li>\n",
       "\t<li>'ASV6118'</li>\n",
       "\t<li>'ASV6289'</li>\n",
       "\t<li>'ASV6295'</li>\n",
       "\t<li>'ASV6379'</li>\n",
       "\t<li>'ASV6385'</li>\n",
       "\t<li>'ASV6682'</li>\n",
       "\t<li>'ASV6702'</li>\n",
       "\t<li>'ASV7037'</li>\n",
       "\t<li>'ASV7510'</li>\n",
       "\t<li>'ASV7675'</li>\n",
       "\t<li>'ASV7777'</li>\n",
       "\t<li>'ASV7903'</li>\n",
       "\t<li>'ASV7968'</li>\n",
       "\t<li>'ASV8019'</li>\n",
       "\t<li>'ASV8024'</li>\n",
       "\t<li>'ASV8190'</li>\n",
       "\t<li>'ASV8213'</li>\n",
       "\t<li>'ASV8427'</li>\n",
       "\t<li>'ASV8548'</li>\n",
       "\t<li>'ASV8621'</li>\n",
       "\t<li>'ASV8702'</li>\n",
       "\t<li>'ASV8803'</li>\n",
       "\t<li>'ASV9071'</li>\n",
       "\t<li>'ASV9272'</li>\n",
       "\t<li>'ASV9294'</li>\n",
       "\t<li>'ASV9344'</li>\n",
       "\t<li>'ASV9473'</li>\n",
       "\t<li>'ASV9485'</li>\n",
       "\t<li>'ASV9574'</li>\n",
       "\t<li>'ASV9596'</li>\n",
       "\t<li>'ASV9635'</li>\n",
       "\t<li>'ASV9760'</li>\n",
       "\t<li>'ASV9850'</li>\n",
       "\t<li>'ASV10027'</li>\n",
       "\t<li>'ASV10200'</li>\n",
       "\t<li>'ASV10235'</li>\n",
       "\t<li>'ASV10508'</li>\n",
       "\t<li>'ASV10549'</li>\n",
       "\t<li>'ASV10555'</li>\n",
       "\t<li>'ASV10700'</li>\n",
       "\t<li>'ASV10831'</li>\n",
       "\t<li>'ASV10880'</li>\n",
       "\t<li>'ASV10963'</li>\n",
       "\t<li>'ASV11166'</li>\n",
       "\t<li>'ASV11385'</li>\n",
       "\t<li>'ASV11411'</li>\n",
       "\t<li>'ASV11468'</li>\n",
       "\t<li>'ASV11489'</li>\n",
       "\t<li>'ASV11520'</li>\n",
       "\t<li>'ASV11731'</li>\n",
       "\t<li>'ASV11735'</li>\n",
       "\t<li>'ASV12088'</li>\n",
       "\t<li>'ASV12128'</li>\n",
       "\t<li>'ASV12149'</li>\n",
       "\t<li>'ASV12265'</li>\n",
       "\t<li>'ASV12271'</li>\n",
       "\t<li>'ASV12304'</li>\n",
       "\t<li>'ASV12512'</li>\n",
       "\t<li>'ASV12586'</li>\n",
       "\t<li>'ASV12705'</li>\n",
       "\t<li>'ASV12811'</li>\n",
       "\t<li>'ASV12864'</li>\n",
       "\t<li>'ASV12912'</li>\n",
       "\t<li>'ASV12930'</li>\n",
       "\t<li>'ASV13282'</li>\n",
       "\t<li>'ASV13407'</li>\n",
       "\t<li>'ASV13408'</li>\n",
       "\t<li>'ASV13509'</li>\n",
       "\t<li>'ASV13562'</li>\n",
       "\t<li>'ASV13719'</li>\n",
       "\t<li>'ASV13774'</li>\n",
       "\t<li>'ASV13849'</li>\n",
       "\t<li>'ASV14065'</li>\n",
       "\t<li>'ASV14102'</li>\n",
       "\t<li>'ASV14169'</li>\n",
       "\t<li>'ASV14587'</li>\n",
       "\t<li>'ASV14609'</li>\n",
       "\t<li>'ASV14792'</li>\n",
       "\t<li>'ASV15054'</li>\n",
       "\t<li>'ASV15423'</li>\n",
       "\t<li>'ASV15499'</li>\n",
       "\t<li>'ASV15561'</li>\n",
       "\t<li>'ASV15798'</li>\n",
       "\t<li>'ASV16092'</li>\n",
       "\t<li>'ASV16488'</li>\n",
       "\t<li>'ASV16636'</li>\n",
       "\t<li>'ASV16641'</li>\n",
       "\t<li>'ASV16717'</li>\n",
       "\t<li>'ASV17165'</li>\n",
       "\t<li>'ASV17203'</li>\n",
       "\t<li>'ASV17265'</li>\n",
       "\t<li>'ASV17824'</li>\n",
       "\t<li>'ASV17953'</li>\n",
       "\t<li>'ASV17966'</li>\n",
       "\t<li>'ASV18062'</li>\n",
       "\t<li>'ASV18527'</li>\n",
       "\t<li>'ASV18598'</li>\n",
       "\t<li>'ASV19774'</li>\n",
       "\t<li>'ASV19904'</li>\n",
       "\t<li>'ASV20070'</li>\n",
       "\t<li>'ASV20075'</li>\n",
       "\t<li>'ASV20127'</li>\n",
       "\t<li>'ASV20134'</li>\n",
       "\t<li>'ASV20528'</li>\n",
       "\t<li>'ASV21179'</li>\n",
       "\t<li>'ASV21385'</li>\n",
       "\t<li>'ASV21513'</li>\n",
       "\t<li>'ASV21641'</li>\n",
       "\t<li>'ASV22556'</li>\n",
       "\t<li>'ASV22770'</li>\n",
       "\t<li>'ASV22801'</li>\n",
       "\t<li>'ASV23849'</li>\n",
       "\t<li>'ASV24855'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'ASV28'\n",
       "\\item 'ASV31'\n",
       "\\item 'ASV82'\n",
       "\\item 'ASV90'\n",
       "\\item 'ASV128'\n",
       "\\item 'ASV130'\n",
       "\\item 'ASV140'\n",
       "\\item 'ASV149'\n",
       "\\item 'ASV152'\n",
       "\\item 'ASV186'\n",
       "\\item 'ASV201'\n",
       "\\item 'ASV331'\n",
       "\\item 'ASV353'\n",
       "\\item 'ASV391'\n",
       "\\item 'ASV432'\n",
       "\\item 'ASV547'\n",
       "\\item 'ASV594'\n",
       "\\item 'ASV650'\n",
       "\\item 'ASV728'\n",
       "\\item 'ASV763'\n",
       "\\item 'ASV922'\n",
       "\\item 'ASV937'\n",
       "\\item 'ASV995'\n",
       "\\item 'ASV1571'\n",
       "\\item 'ASV1732'\n",
       "\\item 'ASV1766'\n",
       "\\item 'ASV1777'\n",
       "\\item 'ASV1788'\n",
       "\\item 'ASV1841'\n",
       "\\item 'ASV1880'\n",
       "\\item 'ASV2159'\n",
       "\\item 'ASV2245'\n",
       "\\item 'ASV2277'\n",
       "\\item 'ASV2280'\n",
       "\\item 'ASV2631'\n",
       "\\item 'ASV2865'\n",
       "\\item 'ASV3045'\n",
       "\\item 'ASV3093'\n",
       "\\item 'ASV3275'\n",
       "\\item 'ASV3542'\n",
       "\\item 'ASV3574'\n",
       "\\item 'ASV3806'\n",
       "\\item 'ASV3933'\n",
       "\\item 'ASV4004'\n",
       "\\item 'ASV4213'\n",
       "\\item 'ASV4596'\n",
       "\\item 'ASV4612'\n",
       "\\item 'ASV4894'\n",
       "\\item 'ASV5214'\n",
       "\\item 'ASV5295'\n",
       "\\item 'ASV5384'\n",
       "\\item 'ASV5411'\n",
       "\\item 'ASV5507'\n",
       "\\item 'ASV5657'\n",
       "\\item 'ASV5728'\n",
       "\\item 'ASV5818'\n",
       "\\item 'ASV5833'\n",
       "\\item 'ASV5899'\n",
       "\\item 'ASV6118'\n",
       "\\item 'ASV6289'\n",
       "\\item 'ASV6295'\n",
       "\\item 'ASV6379'\n",
       "\\item 'ASV6385'\n",
       "\\item 'ASV6682'\n",
       "\\item 'ASV6702'\n",
       "\\item 'ASV7037'\n",
       "\\item 'ASV7510'\n",
       "\\item 'ASV7675'\n",
       "\\item 'ASV7777'\n",
       "\\item 'ASV7903'\n",
       "\\item 'ASV7968'\n",
       "\\item 'ASV8019'\n",
       "\\item 'ASV8024'\n",
       "\\item 'ASV8190'\n",
       "\\item 'ASV8213'\n",
       "\\item 'ASV8427'\n",
       "\\item 'ASV8548'\n",
       "\\item 'ASV8621'\n",
       "\\item 'ASV8702'\n",
       "\\item 'ASV8803'\n",
       "\\item 'ASV9071'\n",
       "\\item 'ASV9272'\n",
       "\\item 'ASV9294'\n",
       "\\item 'ASV9344'\n",
       "\\item 'ASV9473'\n",
       "\\item 'ASV9485'\n",
       "\\item 'ASV9574'\n",
       "\\item 'ASV9596'\n",
       "\\item 'ASV9635'\n",
       "\\item 'ASV9760'\n",
       "\\item 'ASV9850'\n",
       "\\item 'ASV10027'\n",
       "\\item 'ASV10200'\n",
       "\\item 'ASV10235'\n",
       "\\item 'ASV10508'\n",
       "\\item 'ASV10549'\n",
       "\\item 'ASV10555'\n",
       "\\item 'ASV10700'\n",
       "\\item 'ASV10831'\n",
       "\\item 'ASV10880'\n",
       "\\item 'ASV10963'\n",
       "\\item 'ASV11166'\n",
       "\\item 'ASV11385'\n",
       "\\item 'ASV11411'\n",
       "\\item 'ASV11468'\n",
       "\\item 'ASV11489'\n",
       "\\item 'ASV11520'\n",
       "\\item 'ASV11731'\n",
       "\\item 'ASV11735'\n",
       "\\item 'ASV12088'\n",
       "\\item 'ASV12128'\n",
       "\\item 'ASV12149'\n",
       "\\item 'ASV12265'\n",
       "\\item 'ASV12271'\n",
       "\\item 'ASV12304'\n",
       "\\item 'ASV12512'\n",
       "\\item 'ASV12586'\n",
       "\\item 'ASV12705'\n",
       "\\item 'ASV12811'\n",
       "\\item 'ASV12864'\n",
       "\\item 'ASV12912'\n",
       "\\item 'ASV12930'\n",
       "\\item 'ASV13282'\n",
       "\\item 'ASV13407'\n",
       "\\item 'ASV13408'\n",
       "\\item 'ASV13509'\n",
       "\\item 'ASV13562'\n",
       "\\item 'ASV13719'\n",
       "\\item 'ASV13774'\n",
       "\\item 'ASV13849'\n",
       "\\item 'ASV14065'\n",
       "\\item 'ASV14102'\n",
       "\\item 'ASV14169'\n",
       "\\item 'ASV14587'\n",
       "\\item 'ASV14609'\n",
       "\\item 'ASV14792'\n",
       "\\item 'ASV15054'\n",
       "\\item 'ASV15423'\n",
       "\\item 'ASV15499'\n",
       "\\item 'ASV15561'\n",
       "\\item 'ASV15798'\n",
       "\\item 'ASV16092'\n",
       "\\item 'ASV16488'\n",
       "\\item 'ASV16636'\n",
       "\\item 'ASV16641'\n",
       "\\item 'ASV16717'\n",
       "\\item 'ASV17165'\n",
       "\\item 'ASV17203'\n",
       "\\item 'ASV17265'\n",
       "\\item 'ASV17824'\n",
       "\\item 'ASV17953'\n",
       "\\item 'ASV17966'\n",
       "\\item 'ASV18062'\n",
       "\\item 'ASV18527'\n",
       "\\item 'ASV18598'\n",
       "\\item 'ASV19774'\n",
       "\\item 'ASV19904'\n",
       "\\item 'ASV20070'\n",
       "\\item 'ASV20075'\n",
       "\\item 'ASV20127'\n",
       "\\item 'ASV20134'\n",
       "\\item 'ASV20528'\n",
       "\\item 'ASV21179'\n",
       "\\item 'ASV21385'\n",
       "\\item 'ASV21513'\n",
       "\\item 'ASV21641'\n",
       "\\item 'ASV22556'\n",
       "\\item 'ASV22770'\n",
       "\\item 'ASV22801'\n",
       "\\item 'ASV23849'\n",
       "\\item 'ASV24855'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'ASV28'\n",
       "2. 'ASV31'\n",
       "3. 'ASV82'\n",
       "4. 'ASV90'\n",
       "5. 'ASV128'\n",
       "6. 'ASV130'\n",
       "7. 'ASV140'\n",
       "8. 'ASV149'\n",
       "9. 'ASV152'\n",
       "10. 'ASV186'\n",
       "11. 'ASV201'\n",
       "12. 'ASV331'\n",
       "13. 'ASV353'\n",
       "14. 'ASV391'\n",
       "15. 'ASV432'\n",
       "16. 'ASV547'\n",
       "17. 'ASV594'\n",
       "18. 'ASV650'\n",
       "19. 'ASV728'\n",
       "20. 'ASV763'\n",
       "21. 'ASV922'\n",
       "22. 'ASV937'\n",
       "23. 'ASV995'\n",
       "24. 'ASV1571'\n",
       "25. 'ASV1732'\n",
       "26. 'ASV1766'\n",
       "27. 'ASV1777'\n",
       "28. 'ASV1788'\n",
       "29. 'ASV1841'\n",
       "30. 'ASV1880'\n",
       "31. 'ASV2159'\n",
       "32. 'ASV2245'\n",
       "33. 'ASV2277'\n",
       "34. 'ASV2280'\n",
       "35. 'ASV2631'\n",
       "36. 'ASV2865'\n",
       "37. 'ASV3045'\n",
       "38. 'ASV3093'\n",
       "39. 'ASV3275'\n",
       "40. 'ASV3542'\n",
       "41. 'ASV3574'\n",
       "42. 'ASV3806'\n",
       "43. 'ASV3933'\n",
       "44. 'ASV4004'\n",
       "45. 'ASV4213'\n",
       "46. 'ASV4596'\n",
       "47. 'ASV4612'\n",
       "48. 'ASV4894'\n",
       "49. 'ASV5214'\n",
       "50. 'ASV5295'\n",
       "51. 'ASV5384'\n",
       "52. 'ASV5411'\n",
       "53. 'ASV5507'\n",
       "54. 'ASV5657'\n",
       "55. 'ASV5728'\n",
       "56. 'ASV5818'\n",
       "57. 'ASV5833'\n",
       "58. 'ASV5899'\n",
       "59. 'ASV6118'\n",
       "60. 'ASV6289'\n",
       "61. 'ASV6295'\n",
       "62. 'ASV6379'\n",
       "63. 'ASV6385'\n",
       "64. 'ASV6682'\n",
       "65. 'ASV6702'\n",
       "66. 'ASV7037'\n",
       "67. 'ASV7510'\n",
       "68. 'ASV7675'\n",
       "69. 'ASV7777'\n",
       "70. 'ASV7903'\n",
       "71. 'ASV7968'\n",
       "72. 'ASV8019'\n",
       "73. 'ASV8024'\n",
       "74. 'ASV8190'\n",
       "75. 'ASV8213'\n",
       "76. 'ASV8427'\n",
       "77. 'ASV8548'\n",
       "78. 'ASV8621'\n",
       "79. 'ASV8702'\n",
       "80. 'ASV8803'\n",
       "81. 'ASV9071'\n",
       "82. 'ASV9272'\n",
       "83. 'ASV9294'\n",
       "84. 'ASV9344'\n",
       "85. 'ASV9473'\n",
       "86. 'ASV9485'\n",
       "87. 'ASV9574'\n",
       "88. 'ASV9596'\n",
       "89. 'ASV9635'\n",
       "90. 'ASV9760'\n",
       "91. 'ASV9850'\n",
       "92. 'ASV10027'\n",
       "93. 'ASV10200'\n",
       "94. 'ASV10235'\n",
       "95. 'ASV10508'\n",
       "96. 'ASV10549'\n",
       "97. 'ASV10555'\n",
       "98. 'ASV10700'\n",
       "99. 'ASV10831'\n",
       "100. 'ASV10880'\n",
       "101. 'ASV10963'\n",
       "102. 'ASV11166'\n",
       "103. 'ASV11385'\n",
       "104. 'ASV11411'\n",
       "105. 'ASV11468'\n",
       "106. 'ASV11489'\n",
       "107. 'ASV11520'\n",
       "108. 'ASV11731'\n",
       "109. 'ASV11735'\n",
       "110. 'ASV12088'\n",
       "111. 'ASV12128'\n",
       "112. 'ASV12149'\n",
       "113. 'ASV12265'\n",
       "114. 'ASV12271'\n",
       "115. 'ASV12304'\n",
       "116. 'ASV12512'\n",
       "117. 'ASV12586'\n",
       "118. 'ASV12705'\n",
       "119. 'ASV12811'\n",
       "120. 'ASV12864'\n",
       "121. 'ASV12912'\n",
       "122. 'ASV12930'\n",
       "123. 'ASV13282'\n",
       "124. 'ASV13407'\n",
       "125. 'ASV13408'\n",
       "126. 'ASV13509'\n",
       "127. 'ASV13562'\n",
       "128. 'ASV13719'\n",
       "129. 'ASV13774'\n",
       "130. 'ASV13849'\n",
       "131. 'ASV14065'\n",
       "132. 'ASV14102'\n",
       "133. 'ASV14169'\n",
       "134. 'ASV14587'\n",
       "135. 'ASV14609'\n",
       "136. 'ASV14792'\n",
       "137. 'ASV15054'\n",
       "138. 'ASV15423'\n",
       "139. 'ASV15499'\n",
       "140. 'ASV15561'\n",
       "141. 'ASV15798'\n",
       "142. 'ASV16092'\n",
       "143. 'ASV16488'\n",
       "144. 'ASV16636'\n",
       "145. 'ASV16641'\n",
       "146. 'ASV16717'\n",
       "147. 'ASV17165'\n",
       "148. 'ASV17203'\n",
       "149. 'ASV17265'\n",
       "150. 'ASV17824'\n",
       "151. 'ASV17953'\n",
       "152. 'ASV17966'\n",
       "153. 'ASV18062'\n",
       "154. 'ASV18527'\n",
       "155. 'ASV18598'\n",
       "156. 'ASV19774'\n",
       "157. 'ASV19904'\n",
       "158. 'ASV20070'\n",
       "159. 'ASV20075'\n",
       "160. 'ASV20127'\n",
       "161. 'ASV20134'\n",
       "162. 'ASV20528'\n",
       "163. 'ASV21179'\n",
       "164. 'ASV21385'\n",
       "165. 'ASV21513'\n",
       "166. 'ASV21641'\n",
       "167. 'ASV22556'\n",
       "168. 'ASV22770'\n",
       "169. 'ASV22801'\n",
       "170. 'ASV23849'\n",
       "171. 'ASV24855'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"ASV28\"    \"ASV31\"    \"ASV82\"    \"ASV90\"    \"ASV128\"   \"ASV130\"  \n",
       "  [7] \"ASV140\"   \"ASV149\"   \"ASV152\"   \"ASV186\"   \"ASV201\"   \"ASV331\"  \n",
       " [13] \"ASV353\"   \"ASV391\"   \"ASV432\"   \"ASV547\"   \"ASV594\"   \"ASV650\"  \n",
       " [19] \"ASV728\"   \"ASV763\"   \"ASV922\"   \"ASV937\"   \"ASV995\"   \"ASV1571\" \n",
       " [25] \"ASV1732\"  \"ASV1766\"  \"ASV1777\"  \"ASV1788\"  \"ASV1841\"  \"ASV1880\" \n",
       " [31] \"ASV2159\"  \"ASV2245\"  \"ASV2277\"  \"ASV2280\"  \"ASV2631\"  \"ASV2865\" \n",
       " [37] \"ASV3045\"  \"ASV3093\"  \"ASV3275\"  \"ASV3542\"  \"ASV3574\"  \"ASV3806\" \n",
       " [43] \"ASV3933\"  \"ASV4004\"  \"ASV4213\"  \"ASV4596\"  \"ASV4612\"  \"ASV4894\" \n",
       " [49] \"ASV5214\"  \"ASV5295\"  \"ASV5384\"  \"ASV5411\"  \"ASV5507\"  \"ASV5657\" \n",
       " [55] \"ASV5728\"  \"ASV5818\"  \"ASV5833\"  \"ASV5899\"  \"ASV6118\"  \"ASV6289\" \n",
       " [61] \"ASV6295\"  \"ASV6379\"  \"ASV6385\"  \"ASV6682\"  \"ASV6702\"  \"ASV7037\" \n",
       " [67] \"ASV7510\"  \"ASV7675\"  \"ASV7777\"  \"ASV7903\"  \"ASV7968\"  \"ASV8019\" \n",
       " [73] \"ASV8024\"  \"ASV8190\"  \"ASV8213\"  \"ASV8427\"  \"ASV8548\"  \"ASV8621\" \n",
       " [79] \"ASV8702\"  \"ASV8803\"  \"ASV9071\"  \"ASV9272\"  \"ASV9294\"  \"ASV9344\" \n",
       " [85] \"ASV9473\"  \"ASV9485\"  \"ASV9574\"  \"ASV9596\"  \"ASV9635\"  \"ASV9760\" \n",
       " [91] \"ASV9850\"  \"ASV10027\" \"ASV10200\" \"ASV10235\" \"ASV10508\" \"ASV10549\"\n",
       " [97] \"ASV10555\" \"ASV10700\" \"ASV10831\" \"ASV10880\" \"ASV10963\" \"ASV11166\"\n",
       "[103] \"ASV11385\" \"ASV11411\" \"ASV11468\" \"ASV11489\" \"ASV11520\" \"ASV11731\"\n",
       "[109] \"ASV11735\" \"ASV12088\" \"ASV12128\" \"ASV12149\" \"ASV12265\" \"ASV12271\"\n",
       "[115] \"ASV12304\" \"ASV12512\" \"ASV12586\" \"ASV12705\" \"ASV12811\" \"ASV12864\"\n",
       "[121] \"ASV12912\" \"ASV12930\" \"ASV13282\" \"ASV13407\" \"ASV13408\" \"ASV13509\"\n",
       "[127] \"ASV13562\" \"ASV13719\" \"ASV13774\" \"ASV13849\" \"ASV14065\" \"ASV14102\"\n",
       "[133] \"ASV14169\" \"ASV14587\" \"ASV14609\" \"ASV14792\" \"ASV15054\" \"ASV15423\"\n",
       "[139] \"ASV15499\" \"ASV15561\" \"ASV15798\" \"ASV16092\" \"ASV16488\" \"ASV16636\"\n",
       "[145] \"ASV16641\" \"ASV16717\" \"ASV17165\" \"ASV17203\" \"ASV17265\" \"ASV17824\"\n",
       "[151] \"ASV17953\" \"ASV17966\" \"ASV18062\" \"ASV18527\" \"ASV18598\" \"ASV19774\"\n",
       "[157] \"ASV19904\" \"ASV20070\" \"ASV20075\" \"ASV20127\" \"ASV20134\" \"ASV20528\"\n",
       "[163] \"ASV21179\" \"ASV21385\" \"ASV21513\" \"ASV21641\" \"ASV22556\" \"ASV22770\"\n",
       "[169] \"ASV22801\" \"ASV23849\" \"ASV24855\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# treenames = thresh.tree$tip.label\n",
    "# taxanames = taxa_names(ps.thresh)\n",
    "# toremove = setdiff(taxanames,treenames)\n",
    "# toremove"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phyloseq-class experiment-level object\n",
       "otu_table()   OTU Table:         [ 15429 taxa and 434 samples ]\n",
       "sample_data() Sample Data:       [ 434 samples by 11 sample variables ]\n",
       "tax_table()   Taxonomy Table:    [ 15429 taxa by 6 taxonomic ranks ]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# allTaxa = taxa_names(ps.thresh)\n",
    "# allTaxa <- allTaxa[!(allTaxa %in% toremove)]\n",
    "# ps.bac = prune_taxa(allTaxa, ps.thresh)\n",
    "# ps.bac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "FALSE"
      ],
      "text/latex": [
       "FALSE"
      ],
      "text/markdown": [
       "FALSE"
      ],
      "text/plain": [
       "[1] FALSE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# join bacterial tree with physeq object"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phyloseq-class experiment-level object\n",
       "otu_table()   OTU Table:         [ 15429 taxa and 434 samples ]\n",
       "sample_data() Sample Data:       [ 434 samples by 11 sample variables ]\n",
       "tax_table()   Taxonomy Table:    [ 15429 taxa by 6 taxonomic ranks ]\n",
       "phy_tree()    Phylogenetic Tree: [ 15429 tips and 15428 internal nodes ]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ps = merge_phyloseq(ps.thresh, thresh.tree)\n",
    "#force polytomies to dichotomies\n",
    "phy_tree(ps) = multi2di(phy_tree(ps), random = FALSE)\n",
    "ps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phyloseq-class experiment-level object\n",
       "otu_table()   OTU Table:         [ 15375 taxa and 434 samples ]\n",
       "sample_data() Sample Data:       [ 434 samples by 11 sample variables ]\n",
       "tax_table()   Taxonomy Table:    [ 15375 taxa by 6 taxonomic ranks ]\n",
       "phy_tree()    Phylogenetic Tree: [ 15375 tips and 15374 internal nodes ]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#remove mitochondria\n",
    "ps = subset_taxa(ps, Family != \"Mitochondria\" | is.na(Family))\n",
    "ps = subset_taxa(ps, Order != \"Chloroplast\" | is.na(Order))\n",
    "ps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(ps, '~/Hyphosphere/data/3Exp/phyloseq/3Exp_phyloseq_thresh_wtree.rds')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## save with contaminants removed see notebook 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "FC = c(\"ASV1\", \n",
    "\"ASV54\" ,\n",
    "\"ASV78\", \n",
    "\"ASV136\", \n",
    "\"ASV3\", \n",
    "\"ASV12\",\n",
    "\"ASV2516\",\n",
    "\"ASV681\", \n",
    "\"ASV447\",\n",
    "\"ASV49\", \n",
    "\"ASV12695\", \n",
    "\"ASV2395\", \n",
    "\"ASV284\", \n",
    "\"ASV1706\", \n",
    "\"ASV1900\", \n",
    "\"ASV26\", \n",
    "\"ASV14332\", \n",
    "\"ASV2962\",\n",
    "\"ASV9101\",\n",
    "\"ASV2180\", \n",
    "\"ASV8061\", \n",
    "\"ASV4437\", \n",
    "\"ASV6987\", \n",
    "\"ASV2714\",\n",
    "\"ASV2160\", \n",
    "\"ASV2117\", \n",
    "\"ASV2704\", \n",
    "\"ASV1002\", \n",
    "\"ASV4083\",\n",
    "\"ASV7871\", \n",
    "\"ASV1900\", \n",
    "\"ASV171\", \n",
    "\"ASV2516\", \n",
    "\"ASV8536\",\n",
    "\"ASV285\", \n",
    "\"ASV3225\", \n",
    "\"ASV5265\", \n",
    "\"ASV2962\", \n",
    "\"ASV14282\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phyloseq-class experiment-level object\n",
       "otu_table()   OTU Table:         [ 15339 taxa and 434 samples ]\n",
       "sample_data() Sample Data:       [ 434 samples by 11 sample variables ]\n",
       "tax_table()   Taxonomy Table:    [ 15339 taxa by 6 taxonomic ranks ]\n",
       "phy_tree()    Phylogenetic Tree: [ 15339 tips and 15338 internal nodes ]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "allTaxa = taxa_names(ps)\n",
    "allTaxa <- allTaxa[!(allTaxa %in% FC)]\n",
    "ps.sub = prune_taxa(allTaxa, ps)\n",
    "# new phyloseq object with just the taxa you kept.\n",
    "ps.sub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "saveRDS(ps.sub, '~/Hyphosphere/data/3Exp/phyloseq/3Exp_phyloseq_thresh_wtree_woContam.rds')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# now larger object without tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phyloseq-class experiment-level object\n",
       "otu_table()   OTU Table:         [ 15564 taxa and 434 samples ]\n",
       "sample_data() Sample Data:       [ 434 samples by 11 sample variables ]\n",
       "tax_table()   Taxonomy Table:    [ 15564 taxa by 6 taxonomic ranks ]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "allTaxa = taxa_names(ps.thresh)\n",
    "allTaxa <- allTaxa[!(allTaxa %in% FC)]\n",
    "ps.sub = prune_taxa(allTaxa, ps.thresh)\n",
    "# new phyloseq object with just the taxa you kept.\n",
    "ps.sub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phyloseq-class experiment-level object\n",
       "otu_table()   OTU Table:         [ 15495 taxa and 434 samples ]\n",
       "sample_data() Sample Data:       [ 434 samples by 11 sample variables ]\n",
       "tax_table()   Taxonomy Table:    [ 15495 taxa by 6 taxonomic ranks ]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#remove mitochondria\n",
    "ps.sub = subset_taxa(ps.sub, Family != \"Mitochondria\" | is.na(Family))\n",
    "ps.sub = subset_taxa(ps.sub, Order != \"Chloroplast\" | is.na(Order))\n",
    "ps.sub"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "saveRDS(ps.sub, '~/Hyphosphere/data/3Exp/phyloseq/3Exp_phyloseq_thresh_woTree_woContam.rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 3.6.0 (2019-04-26)\n",
       "Platform: x86_64-conda_cos6-linux-gnu (64-bit)\n",
       "Running under: Ubuntu 16.04.6 LTS\n",
       "\n",
       "Matrix products: default\n",
       "BLAS/LAPACK: /data/home/be68/anaconda3/envs/MyR/lib/R/lib/libRblas.so\n",
       "\n",
       "locale:\n",
       " [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              \n",
       " [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    \n",
       " [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   \n",
       " [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 \n",
       " [9] LC_ADDRESS=C               LC_TELEPHONE=C            \n",
       "[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       \n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       "[1] ape_5.3         ggplot2_3.2.1   phyloseq_1.28.0 dplyr_0.7.6    \n",
       "[5] tidyr_0.8.1     dada2_1.10.0    Rcpp_1.0.2     \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] Biobase_2.44.0              splines_3.6.0              \n",
       " [3] jsonlite_1.6                foreach_1.4.7              \n",
       " [5] RcppParallel_4.4.2          assertthat_0.2.1           \n",
       " [7] stats4_3.6.0                latticeExtra_0.6-28        \n",
       " [9] GenomeInfoDbData_1.2.1      Rsamtools_1.34.0           \n",
       "[11] pillar_1.4.2                lattice_0.20-38            \n",
       "[13] glue_1.3.1                  uuid_0.1-2                 \n",
       "[15] digest_0.6.20               GenomicRanges_1.34.0       \n",
       "[17] RColorBrewer_1.1-2          XVector_0.24.0             \n",
       "[19] colorspace_1.4-1            htmltools_0.3.6            \n",
       "[21] Matrix_1.2-17               plyr_1.8.4                 \n",
       "[23] pkgconfig_2.0.2             ShortRead_1.40.0           \n",
       "[25] zlibbioc_1.30.0             purrr_0.2.5                \n",
       "[27] scales_1.0.0                BiocParallel_1.16.6        \n",
       "[29] tibble_2.1.3                mgcv_1.8-28                \n",
       "[31] IRanges_2.18.2              withr_2.1.2                \n",
       "[33] SummarizedExperiment_1.12.0 repr_0.15.0                \n",
       "[35] BiocGenerics_0.30.0         lazyeval_0.2.2             \n",
       "[37] survival_2.44-1.1           magrittr_1.5               \n",
       "[39] crayon_1.3.4                evaluate_0.11              \n",
       "[41] nlme_3.1-141                MASS_7.3-51.4              \n",
       "[43] hwriter_1.3.2               vegan_2.5-5                \n",
       "[45] tools_3.6.0                 data.table_1.12.2          \n",
       "[47] matrixStats_0.54.0          stringr_1.4.0              \n",
       "[49] Rhdf5lib_1.6.0              S4Vectors_0.22.0           \n",
       "[51] munsell_0.5.0               cluster_2.1.0              \n",
       "[53] DelayedArray_0.8.0          bindrcpp_0.2.2             \n",
       "[55] Biostrings_2.52.0           ade4_1.7-13                \n",
       "[57] compiler_3.6.0              GenomeInfoDb_1.18.1        \n",
       "[59] rlang_0.4.0                 rhdf5_2.28.0               \n",
       "[61] grid_3.6.0                  RCurl_1.95-4.12            \n",
       "[63] pbdZMQ_0.3-3                iterators_1.0.12           \n",
       "[65] IRkernel_0.8.12             biomformat_1.12.0          \n",
       "[67] igraph_1.2.4.1              bitops_1.0-6               \n",
       "[69] base64enc_0.1-3             multtest_2.40.0            \n",
       "[71] gtable_0.3.0                codetools_0.2-16           \n",
       "[73] reshape2_1.4.3              R6_2.4.0                   \n",
       "[75] GenomicAlignments_1.18.1    bindr_0.1.1                \n",
       "[77] permute_0.9-5               stringi_1.4.3              \n",
       "[79] parallel_3.6.0              IRdisplay_0.5.0            \n",
       "[81] tidyselect_0.2.4           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
