{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Burkholderials OTUs and rdp type strain isolate tree\n",
    "* Align all Hyphal OTUs and Oxalobacteriaceae and Comamonadaceae type strains from RDP \n",
    "* Burkholderia denitrificans as an out group\n",
    "* !need to add working directory, currently doing work in the notebooks directory\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import sys, os, re, glob\n",
    "from Bio import SeqIO, Entrez\n",
    "from pandas import *\n",
    "import numpy as np\n",
    "#import entrez.direct\n",
    "from cogent.app.fasttree import build_tree_from_alignment\n",
    "from cogent import DNA, LoadSeqs\n",
    "\n",
    "Entrez.email = \"be68@cornell.edu\"\n",
    "nprocs = 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "FastaSource = '/home/be68/Hyphosphere/data/MiSeq/20180129/20180130.v2.otus.fa'\n",
    "OTUlist = '/home/be68/Hyphosphere/data/HyphalOTUs.txt'\n",
    "Group = 'o_Burkholderiales'\n",
    "\n",
    "RDPfile = '/home/be68/Hyphosphere/data/Burkholderiales/Burkholderiales_rdp_download_205seqs.fa'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OTU</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>Florence</th>\n",
       "      <th>N7</th>\n",
       "      <th>Pendelton</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Otu380</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp1</td>\n",
       "      <td>g_Gp1</td>\n",
       "      <td>uncl_g_Gp1</td>\n",
       "      <td>uncl_g_Gp1</td>\n",
       "      <td>uncl_g_Gp1</td>\n",
       "      <td>2.704108</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Otu2910</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp10</td>\n",
       "      <td>uncl_c_Acidobacteria_Gp10</td>\n",
       "      <td>uncl_c_Acidobacteria_Gp10</td>\n",
       "      <td>g_Gp10</td>\n",
       "      <td>uncl_g_Gp10</td>\n",
       "      <td>5.435838</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Otu754</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp10</td>\n",
       "      <td>uncl_c_Acidobacteria_Gp10</td>\n",
       "      <td>uncl_c_Acidobacteria_Gp10</td>\n",
       "      <td>g_Gp10</td>\n",
       "      <td>uncl_g_Gp10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.783544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Otu227</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp17</td>\n",
       "      <td>g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>4.332343</td>\n",
       "      <td>3.450006</td>\n",
       "      <td>3.156290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Otu411</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp17</td>\n",
       "      <td>g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.497636</td>\n",
       "      <td>3.737991</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       OTU           Phylum                 Class                      Order  \\\n",
       "1   Otu380  p_Acidobacteria   c_Acidobacteria_Gp1                      g_Gp1   \n",
       "2  Otu2910  p_Acidobacteria  c_Acidobacteria_Gp10  uncl_c_Acidobacteria_Gp10   \n",
       "3   Otu754  p_Acidobacteria  c_Acidobacteria_Gp10  uncl_c_Acidobacteria_Gp10   \n",
       "4   Otu227  p_Acidobacteria  c_Acidobacteria_Gp17                     g_Gp17   \n",
       "5   Otu411  p_Acidobacteria  c_Acidobacteria_Gp17                     g_Gp17   \n",
       "\n",
       "                      Family        Genus      Species  Florence        N7  \\\n",
       "1                 uncl_g_Gp1   uncl_g_Gp1   uncl_g_Gp1  2.704108       NaN   \n",
       "2  uncl_c_Acidobacteria_Gp10       g_Gp10  uncl_g_Gp10  5.435838       NaN   \n",
       "3  uncl_c_Acidobacteria_Gp10       g_Gp10  uncl_g_Gp10       NaN       NaN   \n",
       "4                uncl_g_Gp17  uncl_g_Gp17  uncl_g_Gp17  4.332343  3.450006   \n",
       "5                uncl_g_Gp17  uncl_g_Gp17  uncl_g_Gp17       NaN  3.497636   \n",
       "\n",
       "   Pendelton  \n",
       "1        NaN  \n",
       "2        NaN  \n",
       "3   3.783544  \n",
       "4   3.156290  \n",
       "5   3.737991  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pandas.read_csv(OTUlist, sep = \"\\t\")\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OTU</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>Florence</th>\n",
       "      <th>N7</th>\n",
       "      <th>Pendelton</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Otu380</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp1</td>\n",
       "      <td>g_Gp1</td>\n",
       "      <td>uncl_g_Gp1</td>\n",
       "      <td>uncl_g_Gp1</td>\n",
       "      <td>uncl_g_Gp1</td>\n",
       "      <td>2.704108</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Otu2910</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp10</td>\n",
       "      <td>uncl_c_Acidobacteria_Gp10</td>\n",
       "      <td>uncl_c_Acidobacteria_Gp10</td>\n",
       "      <td>g_Gp10</td>\n",
       "      <td>uncl_g_Gp10</td>\n",
       "      <td>5.435838</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Otu754</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp10</td>\n",
       "      <td>uncl_c_Acidobacteria_Gp10</td>\n",
       "      <td>uncl_c_Acidobacteria_Gp10</td>\n",
       "      <td>g_Gp10</td>\n",
       "      <td>uncl_g_Gp10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.783544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Otu227</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp17</td>\n",
       "      <td>g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>4.332343</td>\n",
       "      <td>3.450006</td>\n",
       "      <td>3.156290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Otu411</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp17</td>\n",
       "      <td>g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.497636</td>\n",
       "      <td>3.737991</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Otu2986</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp17</td>\n",
       "      <td>g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>uncl_g_Gp17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.105823</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Otu521</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp3</td>\n",
       "      <td>g_Gp3</td>\n",
       "      <td>uncl_g_Gp3</td>\n",
       "      <td>uncl_g_Gp3</td>\n",
       "      <td>uncl_g_Gp3</td>\n",
       "      <td>5.283346</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Otu225</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp6</td>\n",
       "      <td>g_Gp6</td>\n",
       "      <td>uncl_g_Gp6</td>\n",
       "      <td>uncl_g_Gp6</td>\n",
       "      <td>uncl_g_Gp6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.385309</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Otu5803</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp6</td>\n",
       "      <td>g_Gp6</td>\n",
       "      <td>uncl_g_Gp6</td>\n",
       "      <td>uncl_g_Gp6</td>\n",
       "      <td>uncl_g_Gp6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.786996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Otu153</td>\n",
       "      <td>p_Acidobacteria</td>\n",
       "      <td>c_Acidobacteria_Gp6</td>\n",
       "      <td>g_Gp6</td>\n",
       "      <td>uncl_g_Gp6</td>\n",
       "      <td>uncl_g_Gp6</td>\n",
       "      <td>uncl_g_Gp6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.510271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Otu608</td>\n",
       "      <td>p_Actinobacteria</td>\n",
       "      <td>c_Actinobacteria</td>\n",
       "      <td>o_Actinomycetales</td>\n",
       "      <td>f_Kineosporiaceae</td>\n",
       "      <td>uncl_f_Kineosporiaceae</td>\n",
       "      <td>uncl_f_Kineosporiaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.671189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Otu624</td>\n",
       "      <td>p_Actinobacteria</td>\n",
       "      <td>c_Actinobacteria</td>\n",
       "      <td>o_Actinomycetales</td>\n",
       "      <td>f_Micromonosporaceae</td>\n",
       "      <td>g_Dactylosporangium</td>\n",
       "      <td>s_Dactylosporangium_maewongense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.266740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Otu1048</td>\n",
       "      <td>p_Actinobacteria</td>\n",
       "      <td>c_Actinobacteria</td>\n",
       "      <td>o_Actinomycetales</td>\n",
       "      <td>f_Micromonosporaceae</td>\n",
       "      <td>g_Virgisporangium</td>\n",
       "      <td>s_Virgisporangium_aurantiacum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.468921</td>\n",
       "      <td>3.477135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Otu943</td>\n",
       "      <td>p_Actinobacteria</td>\n",
       "      <td>c_Actinobacteria</td>\n",
       "      <td>o_Actinomycetales</td>\n",
       "      <td>f_Micromonosporaceae</td>\n",
       "      <td>uncl_f_Micromonosporaceae</td>\n",
       "      <td>uncl_f_Micromonosporaceae</td>\n",
       "      <td>6.720825</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Otu69</td>\n",
       "      <td>p_Actinobacteria</td>\n",
       "      <td>c_Actinobacteria</td>\n",
       "      <td>o_Actinomycetales</td>\n",
       "      <td>uncl_o_Actinomycetales</td>\n",
       "      <td>uncl_o_Actinomycetales</td>\n",
       "      <td>uncl_o_Actinomycetales</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.188678</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Otu44</td>\n",
       "      <td>p_Bacteroidetes</td>\n",
       "      <td>c_Cytophagia</td>\n",
       "      <td>o_Cytophagales</td>\n",
       "      <td>f_Cytophagaceae</td>\n",
       "      <td>uncl_f_Cytophagaceae</td>\n",
       "      <td>uncl_f_Cytophagaceae</td>\n",
       "      <td>6.069690</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Otu252</td>\n",
       "      <td>p_Bacteroidetes</td>\n",
       "      <td>c_Cytophagia</td>\n",
       "      <td>o_Cytophagales</td>\n",
       "      <td>g_Chryseolinea</td>\n",
       "      <td>s_Chryseolinea_serpens</td>\n",
       "      <td>uncl_s_Chryseolinea_serpens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.463223</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Otu118</td>\n",
       "      <td>p_Bacteroidetes</td>\n",
       "      <td>c_Cytophagia</td>\n",
       "      <td>o_Cytophagales</td>\n",
       "      <td>g_Chryseolinea</td>\n",
       "      <td>s_Chryseolinea_serpens</td>\n",
       "      <td>uncl_s_Chryseolinea_serpens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.899348</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Otu90</td>\n",
       "      <td>p_Bacteroidetes</td>\n",
       "      <td>c_Cytophagia</td>\n",
       "      <td>o_Cytophagales</td>\n",
       "      <td>g_Ohtaekwangia</td>\n",
       "      <td>s_Ohtaekwangia_koreensis</td>\n",
       "      <td>uncl_s_Ohtaekwangia_koreensis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.690660</td>\n",
       "      <td>2.407355</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Otu188</td>\n",
       "      <td>p_Bacteroidetes</td>\n",
       "      <td>c_Cytophagia</td>\n",
       "      <td>o_Cytophagales</td>\n",
       "      <td>g_Ohtaekwangia</td>\n",
       "      <td>s_Ohtaekwangia_kribbensis</td>\n",
       "      <td>uncl_s_Ohtaekwangia_kribbensis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.069959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Otu70</td>\n",
       "      <td>p_Bacteroidetes</td>\n",
       "      <td>c_Cytophagia</td>\n",
       "      <td>o_Cytophagales</td>\n",
       "      <td>g_Ohtaekwangia</td>\n",
       "      <td>uncl_g_Ohtaekwangia</td>\n",
       "      <td>uncl_g_Ohtaekwangia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.262633</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Otu2966</td>\n",
       "      <td>p_Bacteroidetes</td>\n",
       "      <td>c_Cytophagia</td>\n",
       "      <td>o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>g_Chryseolinea</td>\n",
       "      <td>s_Chryseolinea_serpens</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.867672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Otu64</td>\n",
       "      <td>p_Bacteroidetes</td>\n",
       "      <td>c_Cytophagia</td>\n",
       "      <td>o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>g_Ohtaekwangia</td>\n",
       "      <td>s_Ohtaekwangia_koreensis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.716441</td>\n",
       "      <td>2.196016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Otu62</td>\n",
       "      <td>p_Bacteroidetes</td>\n",
       "      <td>c_Cytophagia</td>\n",
       "      <td>o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>g_Ohtaekwangia</td>\n",
       "      <td>s_Ohtaekwangia_koreensis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.476180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Otu38</td>\n",
       "      <td>p_Bacteroidetes</td>\n",
       "      <td>c_Cytophagia</td>\n",
       "      <td>o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>6.018164</td>\n",
       "      <td>7.128475</td>\n",
       "      <td>5.468153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Otu151</td>\n",
       "      <td>p_Bacteroidetes</td>\n",
       "      <td>c_Cytophagia</td>\n",
       "      <td>o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>5.130704</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Otu82</td>\n",
       "      <td>p_Bacteroidetes</td>\n",
       "      <td>c_Cytophagia</td>\n",
       "      <td>o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>4.581192</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Otu397</td>\n",
       "      <td>p_Bacteroidetes</td>\n",
       "      <td>c_Cytophagia</td>\n",
       "      <td>o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>uncl_o_Cytophagales</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.412532</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Otu330</td>\n",
       "      <td>p_Bacteroidetes</td>\n",
       "      <td>c_Sphingobacteriia</td>\n",
       "      <td>o_Sphingobacteriales</td>\n",
       "      <td>f_Chitinophagaceae</td>\n",
       "      <td>uncl_f_Chitinophagaceae</td>\n",
       "      <td>uncl_f_Chitinophagaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.272852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Otu2129</td>\n",
       "      <td>p_Chloroflexi</td>\n",
       "      <td>c_Chloroflexia</td>\n",
       "      <td>o_Chloroflexales</td>\n",
       "      <td>f_Oscillochloridaceae</td>\n",
       "      <td>g_Oscillochloris</td>\n",
       "      <td>uncl_g_Oscillochloris</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.314622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>Otu254</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>o_Enterobacteriales</td>\n",
       "      <td>f_Enterobacteriaceae</td>\n",
       "      <td>g_Escherichia/Shigella</td>\n",
       "      <td>s_Escherichia_fergusonii</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.159684</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>Otu7</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>o_Pseudomonadales</td>\n",
       "      <td>f_Pseudomonadaceae</td>\n",
       "      <td>g_Cellvibrio</td>\n",
       "      <td>s_Cellvibrio_ostraviensis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.010619</td>\n",
       "      <td>4.467566</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>Otu604</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>o_Pseudomonadales</td>\n",
       "      <td>f_Pseudomonadaceae</td>\n",
       "      <td>g_Permianibacter</td>\n",
       "      <td>s_Permianibacter_aggregans</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.402516</td>\n",
       "      <td>5.055835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>Otu80</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>o_Xanthomonadales</td>\n",
       "      <td>f_Sinobacteraceae</td>\n",
       "      <td>g_Povalibacter</td>\n",
       "      <td>s_Povalibacter_uvarum</td>\n",
       "      <td>2.879429</td>\n",
       "      <td>1.681169</td>\n",
       "      <td>2.925981</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>Otu2671</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>o_Xanthomonadales</td>\n",
       "      <td>f_Sinobacteraceae</td>\n",
       "      <td>g_Steroidobacter</td>\n",
       "      <td>uncl_g_Steroidobacter</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.639139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>Otu335</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>o_Xanthomonadales</td>\n",
       "      <td>f_Xanthomonadaceae</td>\n",
       "      <td>g_Arenimonas</td>\n",
       "      <td>s_Arenimonas_daechungensis</td>\n",
       "      <td>3.560817</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>Otu998</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>o_Xanthomonadales</td>\n",
       "      <td>f_Xanthomonadaceae</td>\n",
       "      <td>uncl_f_Xanthomonadaceae</td>\n",
       "      <td>uncl_f_Xanthomonadaceae</td>\n",
       "      <td>3.713112</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>Otu314</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>o_Xanthomonadales</td>\n",
       "      <td>f_Xanthomonadaceae</td>\n",
       "      <td>uncl_f_Xanthomonadaceae</td>\n",
       "      <td>uncl_f_Xanthomonadaceae</td>\n",
       "      <td>8.836342</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>Otu319</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>o_Xanthomonadales</td>\n",
       "      <td>f_Xanthomonadaceae</td>\n",
       "      <td>uncl_f_Xanthomonadaceae</td>\n",
       "      <td>uncl_f_Xanthomonadaceae</td>\n",
       "      <td>7.606011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>Otu196</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>o_Xanthomonadales</td>\n",
       "      <td>f_Xanthomonadaceae</td>\n",
       "      <td>uncl_f_Xanthomonadaceae</td>\n",
       "      <td>uncl_f_Xanthomonadaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.058131</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>Otu1009</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>o_Xanthomonadales</td>\n",
       "      <td>f_Xanthomonadaceae</td>\n",
       "      <td>uncl_f_Xanthomonadaceae</td>\n",
       "      <td>uncl_f_Xanthomonadaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.135916</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>Otu278</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>6.073368</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>Otu45</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>2.911140</td>\n",
       "      <td>3.794272</td>\n",
       "      <td>4.189808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>Otu84</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>7.757616</td>\n",
       "      <td>6.881337</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>Otu2974</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.843832</td>\n",
       "      <td>3.670155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>Otu677</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.186822</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>Otu373</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.090893</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>Otu560</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.939296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>Otu558</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>uncl_c_Gammaproteobacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.671159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>Otu24</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>8.862678</td>\n",
       "      <td>6.294408</td>\n",
       "      <td>6.224081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>Otu51</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>5.148824</td>\n",
       "      <td>8.815047</td>\n",
       "      <td>7.308821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>Otu396</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.758507</td>\n",
       "      <td>4.067799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>Otu519</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.914319</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>Otu527</td>\n",
       "      <td>p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>uncl_p_Proteobacteria</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.584186</td>\n",
       "      <td>6.520427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>Otu334</td>\n",
       "      <td>p_Spirochaetes</td>\n",
       "      <td>c_Spirochaetia</td>\n",
       "      <td>o_Spirochaetales</td>\n",
       "      <td>f_Leptospiraceae</td>\n",
       "      <td>g_Leptospira</td>\n",
       "      <td>uncl_g_Leptospira</td>\n",
       "      <td>6.769001</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>Otu515</td>\n",
       "      <td>p_Spirochaetes</td>\n",
       "      <td>c_Spirochaetia</td>\n",
       "      <td>o_Spirochaetales</td>\n",
       "      <td>f_Spirochaetaceae</td>\n",
       "      <td>g_Spirochaeta</td>\n",
       "      <td>s_Spirochaeta_aurantia_subsp._aurantia</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.933618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>Otu682</td>\n",
       "      <td>p_Verrucomicrobia</td>\n",
       "      <td>c_Spartobacteria</td>\n",
       "      <td>g_Spartobacteria_genera_incertae_sedis</td>\n",
       "      <td>uncl_g_Spartobacteria_genera_incertae_sedis</td>\n",
       "      <td>uncl_g_Spartobacteria_genera_incertae_sedis</td>\n",
       "      <td>uncl_g_Spartobacteria_genera_incertae_sedis</td>\n",
       "      <td>5.316732</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.023248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>Otu507</td>\n",
       "      <td>p_Verrucomicrobia</td>\n",
       "      <td>c_Spartobacteria</td>\n",
       "      <td>uncl_c_Spartobacteria</td>\n",
       "      <td>uncl_c_Spartobacteria</td>\n",
       "      <td>g_Spartobacteria_genera_incertae_sedis</td>\n",
       "      <td>uncl_g_Spartobacteria_genera_incertae_sedis</td>\n",
       "      <td>7.486376</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>Otu434</td>\n",
       "      <td>p_Verrucomicrobia</td>\n",
       "      <td>c_Subdivision3</td>\n",
       "      <td>g_Subdivision3_genera_incertae_sedis</td>\n",
       "      <td>uncl_g_Subdivision3_genera_incertae_sedis</td>\n",
       "      <td>uncl_g_Subdivision3_genera_incertae_sedis</td>\n",
       "      <td>uncl_g_Subdivision3_genera_incertae_sedis</td>\n",
       "      <td>2.324456</td>\n",
       "      <td>3.842743</td>\n",
       "      <td>2.911574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>Otu818</td>\n",
       "      <td>p_Verrucomicrobia</td>\n",
       "      <td>c_Subdivision3</td>\n",
       "      <td>g_Subdivision3_genera_incertae_sedis</td>\n",
       "      <td>uncl_g_Subdivision3_genera_incertae_sedis</td>\n",
       "      <td>uncl_g_Subdivision3_genera_incertae_sedis</td>\n",
       "      <td>uncl_g_Subdivision3_genera_incertae_sedis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.829366</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>126 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         OTU             Phylum                  Class  \\\n",
       "1     Otu380    p_Acidobacteria    c_Acidobacteria_Gp1   \n",
       "2    Otu2910    p_Acidobacteria   c_Acidobacteria_Gp10   \n",
       "3     Otu754    p_Acidobacteria   c_Acidobacteria_Gp10   \n",
       "4     Otu227    p_Acidobacteria   c_Acidobacteria_Gp17   \n",
       "5     Otu411    p_Acidobacteria   c_Acidobacteria_Gp17   \n",
       "6    Otu2986    p_Acidobacteria   c_Acidobacteria_Gp17   \n",
       "7     Otu521    p_Acidobacteria    c_Acidobacteria_Gp3   \n",
       "8     Otu225    p_Acidobacteria    c_Acidobacteria_Gp6   \n",
       "9    Otu5803    p_Acidobacteria    c_Acidobacteria_Gp6   \n",
       "10    Otu153    p_Acidobacteria    c_Acidobacteria_Gp6   \n",
       "11    Otu608   p_Actinobacteria       c_Actinobacteria   \n",
       "12    Otu624   p_Actinobacteria       c_Actinobacteria   \n",
       "13   Otu1048   p_Actinobacteria       c_Actinobacteria   \n",
       "14    Otu943   p_Actinobacteria       c_Actinobacteria   \n",
       "15     Otu69   p_Actinobacteria       c_Actinobacteria   \n",
       "16     Otu44    p_Bacteroidetes           c_Cytophagia   \n",
       "17    Otu252    p_Bacteroidetes           c_Cytophagia   \n",
       "18    Otu118    p_Bacteroidetes           c_Cytophagia   \n",
       "19     Otu90    p_Bacteroidetes           c_Cytophagia   \n",
       "20    Otu188    p_Bacteroidetes           c_Cytophagia   \n",
       "21     Otu70    p_Bacteroidetes           c_Cytophagia   \n",
       "22   Otu2966    p_Bacteroidetes           c_Cytophagia   \n",
       "23     Otu64    p_Bacteroidetes           c_Cytophagia   \n",
       "24     Otu62    p_Bacteroidetes           c_Cytophagia   \n",
       "25     Otu38    p_Bacteroidetes           c_Cytophagia   \n",
       "26    Otu151    p_Bacteroidetes           c_Cytophagia   \n",
       "27     Otu82    p_Bacteroidetes           c_Cytophagia   \n",
       "28    Otu397    p_Bacteroidetes           c_Cytophagia   \n",
       "29    Otu330    p_Bacteroidetes     c_Sphingobacteriia   \n",
       "30   Otu2129      p_Chloroflexi         c_Chloroflexia   \n",
       "..       ...                ...                    ...   \n",
       "97    Otu254   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "98      Otu7   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "99    Otu604   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "100    Otu80   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "101  Otu2671   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "102   Otu335   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "103   Otu998   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "104   Otu314   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "105   Otu319   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "106   Otu196   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "107  Otu1009   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "108   Otu278   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "109    Otu45   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "110    Otu84   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "111  Otu2974   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "112   Otu677   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "113   Otu373   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "114   Otu560   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "115   Otu558   p_Proteobacteria  c_Gammaproteobacteria   \n",
       "116    Otu24   p_Proteobacteria  uncl_p_Proteobacteria   \n",
       "117    Otu51   p_Proteobacteria  uncl_p_Proteobacteria   \n",
       "118   Otu396   p_Proteobacteria  uncl_p_Proteobacteria   \n",
       "119   Otu519   p_Proteobacteria  uncl_p_Proteobacteria   \n",
       "120   Otu527   p_Proteobacteria  uncl_p_Proteobacteria   \n",
       "121   Otu334     p_Spirochaetes         c_Spirochaetia   \n",
       "122   Otu515     p_Spirochaetes         c_Spirochaetia   \n",
       "123   Otu682  p_Verrucomicrobia       c_Spartobacteria   \n",
       "124   Otu507  p_Verrucomicrobia       c_Spartobacteria   \n",
       "125   Otu434  p_Verrucomicrobia         c_Subdivision3   \n",
       "126   Otu818  p_Verrucomicrobia         c_Subdivision3   \n",
       "\n",
       "                                      Order  \\\n",
       "1                                     g_Gp1   \n",
       "2                 uncl_c_Acidobacteria_Gp10   \n",
       "3                 uncl_c_Acidobacteria_Gp10   \n",
       "4                                    g_Gp17   \n",
       "5                                    g_Gp17   \n",
       "6                                    g_Gp17   \n",
       "7                                     g_Gp3   \n",
       "8                                     g_Gp6   \n",
       "9                                     g_Gp6   \n",
       "10                                    g_Gp6   \n",
       "11                        o_Actinomycetales   \n",
       "12                        o_Actinomycetales   \n",
       "13                        o_Actinomycetales   \n",
       "14                        o_Actinomycetales   \n",
       "15                        o_Actinomycetales   \n",
       "16                           o_Cytophagales   \n",
       "17                           o_Cytophagales   \n",
       "18                           o_Cytophagales   \n",
       "19                           o_Cytophagales   \n",
       "20                           o_Cytophagales   \n",
       "21                           o_Cytophagales   \n",
       "22                           o_Cytophagales   \n",
       "23                           o_Cytophagales   \n",
       "24                           o_Cytophagales   \n",
       "25                           o_Cytophagales   \n",
       "26                           o_Cytophagales   \n",
       "27                           o_Cytophagales   \n",
       "28                           o_Cytophagales   \n",
       "29                     o_Sphingobacteriales   \n",
       "30                         o_Chloroflexales   \n",
       "..                                      ...   \n",
       "97                      o_Enterobacteriales   \n",
       "98                        o_Pseudomonadales   \n",
       "99                        o_Pseudomonadales   \n",
       "100                       o_Xanthomonadales   \n",
       "101                       o_Xanthomonadales   \n",
       "102                       o_Xanthomonadales   \n",
       "103                       o_Xanthomonadales   \n",
       "104                       o_Xanthomonadales   \n",
       "105                       o_Xanthomonadales   \n",
       "106                       o_Xanthomonadales   \n",
       "107                       o_Xanthomonadales   \n",
       "108              uncl_c_Gammaproteobacteria   \n",
       "109              uncl_c_Gammaproteobacteria   \n",
       "110              uncl_c_Gammaproteobacteria   \n",
       "111              uncl_c_Gammaproteobacteria   \n",
       "112              uncl_c_Gammaproteobacteria   \n",
       "113              uncl_c_Gammaproteobacteria   \n",
       "114              uncl_c_Gammaproteobacteria   \n",
       "115              uncl_c_Gammaproteobacteria   \n",
       "116                   uncl_p_Proteobacteria   \n",
       "117                   uncl_p_Proteobacteria   \n",
       "118                   uncl_p_Proteobacteria   \n",
       "119                   uncl_p_Proteobacteria   \n",
       "120                   uncl_p_Proteobacteria   \n",
       "121                        o_Spirochaetales   \n",
       "122                        o_Spirochaetales   \n",
       "123  g_Spartobacteria_genera_incertae_sedis   \n",
       "124                   uncl_c_Spartobacteria   \n",
       "125    g_Subdivision3_genera_incertae_sedis   \n",
       "126    g_Subdivision3_genera_incertae_sedis   \n",
       "\n",
       "                                          Family  \\\n",
       "1                                     uncl_g_Gp1   \n",
       "2                      uncl_c_Acidobacteria_Gp10   \n",
       "3                      uncl_c_Acidobacteria_Gp10   \n",
       "4                                    uncl_g_Gp17   \n",
       "5                                    uncl_g_Gp17   \n",
       "6                                    uncl_g_Gp17   \n",
       "7                                     uncl_g_Gp3   \n",
       "8                                     uncl_g_Gp6   \n",
       "9                                     uncl_g_Gp6   \n",
       "10                                    uncl_g_Gp6   \n",
       "11                             f_Kineosporiaceae   \n",
       "12                          f_Micromonosporaceae   \n",
       "13                          f_Micromonosporaceae   \n",
       "14                          f_Micromonosporaceae   \n",
       "15                        uncl_o_Actinomycetales   \n",
       "16                               f_Cytophagaceae   \n",
       "17                                g_Chryseolinea   \n",
       "18                                g_Chryseolinea   \n",
       "19                                g_Ohtaekwangia   \n",
       "20                                g_Ohtaekwangia   \n",
       "21                                g_Ohtaekwangia   \n",
       "22                           uncl_o_Cytophagales   \n",
       "23                           uncl_o_Cytophagales   \n",
       "24                           uncl_o_Cytophagales   \n",
       "25                           uncl_o_Cytophagales   \n",
       "26                           uncl_o_Cytophagales   \n",
       "27                           uncl_o_Cytophagales   \n",
       "28                           uncl_o_Cytophagales   \n",
       "29                            f_Chitinophagaceae   \n",
       "30                         f_Oscillochloridaceae   \n",
       "..                                           ...   \n",
       "97                          f_Enterobacteriaceae   \n",
       "98                            f_Pseudomonadaceae   \n",
       "99                            f_Pseudomonadaceae   \n",
       "100                            f_Sinobacteraceae   \n",
       "101                            f_Sinobacteraceae   \n",
       "102                           f_Xanthomonadaceae   \n",
       "103                           f_Xanthomonadaceae   \n",
       "104                           f_Xanthomonadaceae   \n",
       "105                           f_Xanthomonadaceae   \n",
       "106                           f_Xanthomonadaceae   \n",
       "107                           f_Xanthomonadaceae   \n",
       "108                   uncl_c_Gammaproteobacteria   \n",
       "109                   uncl_c_Gammaproteobacteria   \n",
       "110                   uncl_c_Gammaproteobacteria   \n",
       "111                   uncl_c_Gammaproteobacteria   \n",
       "112                   uncl_c_Gammaproteobacteria   \n",
       "113                   uncl_c_Gammaproteobacteria   \n",
       "114                   uncl_c_Gammaproteobacteria   \n",
       "115                   uncl_c_Gammaproteobacteria   \n",
       "116                        uncl_p_Proteobacteria   \n",
       "117                        uncl_p_Proteobacteria   \n",
       "118                        uncl_p_Proteobacteria   \n",
       "119                        uncl_p_Proteobacteria   \n",
       "120                        uncl_p_Proteobacteria   \n",
       "121                             f_Leptospiraceae   \n",
       "122                            f_Spirochaetaceae   \n",
       "123  uncl_g_Spartobacteria_genera_incertae_sedis   \n",
       "124                        uncl_c_Spartobacteria   \n",
       "125    uncl_g_Subdivision3_genera_incertae_sedis   \n",
       "126    uncl_g_Subdivision3_genera_incertae_sedis   \n",
       "\n",
       "                                           Genus  \\\n",
       "1                                     uncl_g_Gp1   \n",
       "2                                         g_Gp10   \n",
       "3                                         g_Gp10   \n",
       "4                                    uncl_g_Gp17   \n",
       "5                                    uncl_g_Gp17   \n",
       "6                                    uncl_g_Gp17   \n",
       "7                                     uncl_g_Gp3   \n",
       "8                                     uncl_g_Gp6   \n",
       "9                                     uncl_g_Gp6   \n",
       "10                                    uncl_g_Gp6   \n",
       "11                        uncl_f_Kineosporiaceae   \n",
       "12                           g_Dactylosporangium   \n",
       "13                             g_Virgisporangium   \n",
       "14                     uncl_f_Micromonosporaceae   \n",
       "15                        uncl_o_Actinomycetales   \n",
       "16                          uncl_f_Cytophagaceae   \n",
       "17                        s_Chryseolinea_serpens   \n",
       "18                        s_Chryseolinea_serpens   \n",
       "19                      s_Ohtaekwangia_koreensis   \n",
       "20                     s_Ohtaekwangia_kribbensis   \n",
       "21                           uncl_g_Ohtaekwangia   \n",
       "22                                g_Chryseolinea   \n",
       "23                                g_Ohtaekwangia   \n",
       "24                                g_Ohtaekwangia   \n",
       "25                           uncl_o_Cytophagales   \n",
       "26                           uncl_o_Cytophagales   \n",
       "27                           uncl_o_Cytophagales   \n",
       "28                           uncl_o_Cytophagales   \n",
       "29                       uncl_f_Chitinophagaceae   \n",
       "30                              g_Oscillochloris   \n",
       "..                                           ...   \n",
       "97                        g_Escherichia/Shigella   \n",
       "98                                  g_Cellvibrio   \n",
       "99                              g_Permianibacter   \n",
       "100                               g_Povalibacter   \n",
       "101                             g_Steroidobacter   \n",
       "102                                 g_Arenimonas   \n",
       "103                      uncl_f_Xanthomonadaceae   \n",
       "104                      uncl_f_Xanthomonadaceae   \n",
       "105                      uncl_f_Xanthomonadaceae   \n",
       "106                      uncl_f_Xanthomonadaceae   \n",
       "107                      uncl_f_Xanthomonadaceae   \n",
       "108                   uncl_c_Gammaproteobacteria   \n",
       "109                   uncl_c_Gammaproteobacteria   \n",
       "110                   uncl_c_Gammaproteobacteria   \n",
       "111                   uncl_c_Gammaproteobacteria   \n",
       "112                   uncl_c_Gammaproteobacteria   \n",
       "113                   uncl_c_Gammaproteobacteria   \n",
       "114                   uncl_c_Gammaproteobacteria   \n",
       "115                   uncl_c_Gammaproteobacteria   \n",
       "116                        uncl_p_Proteobacteria   \n",
       "117                        uncl_p_Proteobacteria   \n",
       "118                        uncl_p_Proteobacteria   \n",
       "119                        uncl_p_Proteobacteria   \n",
       "120                        uncl_p_Proteobacteria   \n",
       "121                                 g_Leptospira   \n",
       "122                                g_Spirochaeta   \n",
       "123  uncl_g_Spartobacteria_genera_incertae_sedis   \n",
       "124       g_Spartobacteria_genera_incertae_sedis   \n",
       "125    uncl_g_Subdivision3_genera_incertae_sedis   \n",
       "126    uncl_g_Subdivision3_genera_incertae_sedis   \n",
       "\n",
       "                                         Species  Florence        N7  \\\n",
       "1                                     uncl_g_Gp1  2.704108       NaN   \n",
       "2                                    uncl_g_Gp10  5.435838       NaN   \n",
       "3                                    uncl_g_Gp10       NaN       NaN   \n",
       "4                                    uncl_g_Gp17  4.332343  3.450006   \n",
       "5                                    uncl_g_Gp17       NaN  3.497636   \n",
       "6                                    uncl_g_Gp17       NaN       NaN   \n",
       "7                                     uncl_g_Gp3  5.283346       NaN   \n",
       "8                                     uncl_g_Gp6       NaN  1.385309   \n",
       "9                                     uncl_g_Gp6       NaN       NaN   \n",
       "10                                    uncl_g_Gp6       NaN       NaN   \n",
       "11                        uncl_f_Kineosporiaceae       NaN       NaN   \n",
       "12               s_Dactylosporangium_maewongense       NaN       NaN   \n",
       "13                 s_Virgisporangium_aurantiacum       NaN  4.468921   \n",
       "14                     uncl_f_Micromonosporaceae  6.720825       NaN   \n",
       "15                        uncl_o_Actinomycetales       NaN  3.188678   \n",
       "16                          uncl_f_Cytophagaceae  6.069690       NaN   \n",
       "17                   uncl_s_Chryseolinea_serpens       NaN  5.463223   \n",
       "18                   uncl_s_Chryseolinea_serpens       NaN  1.899348   \n",
       "19                 uncl_s_Ohtaekwangia_koreensis       NaN  1.690660   \n",
       "20                uncl_s_Ohtaekwangia_kribbensis       NaN       NaN   \n",
       "21                           uncl_g_Ohtaekwangia       NaN  4.262633   \n",
       "22                        s_Chryseolinea_serpens       NaN       NaN   \n",
       "23                      s_Ohtaekwangia_koreensis       NaN  2.716441   \n",
       "24                      s_Ohtaekwangia_koreensis       NaN       NaN   \n",
       "25                           uncl_o_Cytophagales  6.018164  7.128475   \n",
       "26                           uncl_o_Cytophagales  5.130704       NaN   \n",
       "27                           uncl_o_Cytophagales  4.581192       NaN   \n",
       "28                           uncl_o_Cytophagales       NaN       NaN   \n",
       "29                       uncl_f_Chitinophagaceae       NaN       NaN   \n",
       "30                         uncl_g_Oscillochloris       NaN       NaN   \n",
       "..                                           ...       ...       ...   \n",
       "97                      s_Escherichia_fergusonii       NaN  8.159684   \n",
       "98                     s_Cellvibrio_ostraviensis       NaN  6.010619   \n",
       "99                    s_Permianibacter_aggregans       NaN  6.402516   \n",
       "100                        s_Povalibacter_uvarum  2.879429  1.681169   \n",
       "101                        uncl_g_Steroidobacter       NaN       NaN   \n",
       "102                   s_Arenimonas_daechungensis  3.560817       NaN   \n",
       "103                      uncl_f_Xanthomonadaceae  3.713112       NaN   \n",
       "104                      uncl_f_Xanthomonadaceae  8.836342       NaN   \n",
       "105                      uncl_f_Xanthomonadaceae  7.606011       NaN   \n",
       "106                      uncl_f_Xanthomonadaceae       NaN  2.058131   \n",
       "107                      uncl_f_Xanthomonadaceae       NaN       NaN   \n",
       "108                   uncl_c_Gammaproteobacteria  6.073368       NaN   \n",
       "109                   uncl_c_Gammaproteobacteria  2.911140  3.794272   \n",
       "110                   uncl_c_Gammaproteobacteria  7.757616  6.881337   \n",
       "111                   uncl_c_Gammaproteobacteria       NaN  3.843832   \n",
       "112                   uncl_c_Gammaproteobacteria       NaN  7.186822   \n",
       "113                   uncl_c_Gammaproteobacteria       NaN       NaN   \n",
       "114                   uncl_c_Gammaproteobacteria       NaN       NaN   \n",
       "115                   uncl_c_Gammaproteobacteria       NaN       NaN   \n",
       "116                        uncl_p_Proteobacteria  8.862678  6.294408   \n",
       "117                        uncl_p_Proteobacteria  5.148824  8.815047   \n",
       "118                        uncl_p_Proteobacteria       NaN  2.758507   \n",
       "119                        uncl_p_Proteobacteria       NaN  3.914319   \n",
       "120                        uncl_p_Proteobacteria       NaN  4.584186   \n",
       "121                            uncl_g_Leptospira  6.769001       NaN   \n",
       "122       s_Spirochaeta_aurantia_subsp._aurantia       NaN       NaN   \n",
       "123  uncl_g_Spartobacteria_genera_incertae_sedis  5.316732       NaN   \n",
       "124  uncl_g_Spartobacteria_genera_incertae_sedis  7.486376       NaN   \n",
       "125    uncl_g_Subdivision3_genera_incertae_sedis  2.324456  3.842743   \n",
       "126    uncl_g_Subdivision3_genera_incertae_sedis       NaN       NaN   \n",
       "\n",
       "     Pendelton  \n",
       "1          NaN  \n",
       "2          NaN  \n",
       "3     3.783544  \n",
       "4     3.156290  \n",
       "5     3.737991  \n",
       "6     3.105823  \n",
       "7          NaN  \n",
       "8          NaN  \n",
       "9     2.786996  \n",
       "10    3.510271  \n",
       "11    3.671189  \n",
       "12    2.266740  \n",
       "13    3.477135  \n",
       "14         NaN  \n",
       "15         NaN  \n",
       "16         NaN  \n",
       "17         NaN  \n",
       "18         NaN  \n",
       "19    2.407355  \n",
       "20    6.069959  \n",
       "21         NaN  \n",
       "22    2.867672  \n",
       "23    2.196016  \n",
       "24    4.476180  \n",
       "25    5.468153  \n",
       "26         NaN  \n",
       "27         NaN  \n",
       "28    4.412532  \n",
       "29    3.272852  \n",
       "30    6.314622  \n",
       "..         ...  \n",
       "97         NaN  \n",
       "98    4.467566  \n",
       "99    5.055835  \n",
       "100   2.925981  \n",
       "101   2.639139  \n",
       "102        NaN  \n",
       "103        NaN  \n",
       "104        NaN  \n",
       "105        NaN  \n",
       "106        NaN  \n",
       "107   4.135916  \n",
       "108        NaN  \n",
       "109   4.189808  \n",
       "110        NaN  \n",
       "111   3.670155  \n",
       "112        NaN  \n",
       "113   3.090893  \n",
       "114   2.939296  \n",
       "115   7.671159  \n",
       "116   6.224081  \n",
       "117   7.308821  \n",
       "118   4.067799  \n",
       "119        NaN  \n",
       "120   6.520427  \n",
       "121        NaN  \n",
       "122   4.933618  \n",
       "123   4.023248  \n",
       "124        NaN  \n",
       "125   2.911574  \n",
       "126   5.829366  \n",
       "\n",
       "[126 rows x 10 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Subset to Burkholderials"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(8, 10)\n",
      "        OTU            Phylum                 Class              Order  \\\n",
      "53   Otu142  p_Proteobacteria  c_Betaproteobacteria  o_Burkholderiales   \n",
      "54  Otu5063  p_Proteobacteria  c_Betaproteobacteria  o_Burkholderiales   \n",
      "55    Otu11  p_Proteobacteria  c_Betaproteobacteria  o_Burkholderiales   \n",
      "56    Otu27  p_Proteobacteria  c_Betaproteobacteria  o_Burkholderiales   \n",
      "57     Otu8  p_Proteobacteria  c_Betaproteobacteria  o_Burkholderiales   \n",
      "\n",
      "                    Family                   Genus  \\\n",
      "53        f_Comamonadaceae   uncl_f_Comamonadaceae   \n",
      "54        f_Comamonadaceae   uncl_f_Comamonadaceae   \n",
      "55      f_Oxalobacteraceae        g_Herbaspirillum   \n",
      "56  uncl_o_Burkholderiales  uncl_o_Burkholderiales   \n",
      "57  uncl_o_Burkholderiales  uncl_o_Burkholderiales   \n",
      "\n",
      "                              Species  Florence        N7  Pendelton  \n",
      "53              uncl_f_Comamonadaceae  2.858943       NaN   2.542253  \n",
      "54              uncl_f_Comamonadaceae       NaN       NaN   2.625196  \n",
      "55  s_Herbaspirillum_chlorophenolicum  3.873323       NaN        NaN  \n",
      "56             uncl_o_Burkholderiales  4.471465  2.155477   2.901648  \n",
      "57             uncl_o_Burkholderiales  1.997453  2.072334   2.510493  \n"
     ]
    }
   ],
   "source": [
    "is_Group =  df['Order']==Group\n",
    "Group_df = df[is_Group]\n",
    "print(Group_df.shape)\n",
    "(142, 6)\n",
    "print(Group_df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Otu142', 'Otu5063', 'Otu11', 'Otu27', 'Otu8', 'Otu87', 'Otu6713', 'Otu1027']"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "OTUoi = Group_df.OTU.tolist()\n",
    "OTUoi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "output = open(\"/home/be68/Hyphosphere/data/Burkholderiales/HyphalBurkholderiales.fa\", \"w\")\n",
    "\n",
    "for record in SeqIO.parse(open(FastaSource, \"rU\"), \"fasta\") :\n",
    "    if record.id in OTUoi:\n",
    "        output.write(\">\"+record.id+\"\\n\"+str(record.seq)+\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "OTUfasta = '/home/be68/Hyphosphere/data/Burkholderiales/HyphalBurkholderiales.fa'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Otu7\n",
      "TACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGTAGGTGGTTTGTTAAGCTAGCTGTGAAATCCCCGGGCTCAACCTGGGCACTGCAGTTAGAACTGGCAAGCTAGAGTAGGGTAGAGGGGTGTGGAATTCCAGGTGTAGCGGTGAAATGCGTAGATATCTGGAGGAACATCAGTGGCGAAGGCGACACCCTGGACTCATACTGACACTGAGGTGCGAAAGCGTGGGGAGCAAACAGG\n",
      ">Otu8\n",
      "TACGTAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGTGCGCAGGCGGTTATGCAAGACAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATTTGTGACTGCATGGCTAGAGTACGGCAGAGGGGGATGGAATTCCGCGTGTAGCAGTGAAATGCGTAGATATGCGGAGGAACACCGATGGCGAAGGCAATCCCCTGGGCCTGTACTGACGCTCATGCACGAAAGCGTGGGGAGCAAACAGG\n",
      ">Otu10\n",
      "TACGAAGGGTGCAAACGTTGCTCGGAATCATTGGGCGTAAAGCGCACGTAGGCGGCTTTGCAAGTCGGATGTGAAATCCCGTGGCTCAACCATGGACGTGCATTCGAAACTGCAAAGCTTGAGTATGGAAGAGGGTCGCGGAATTCCCGGTGTAGAGGTGAAATTCGTAGATATCGGGAGGAACACCAGTGGCGAAGGCGGCGACCTGGGCCAATACTGACGCTGAGGTGCGAAAGCGTGGGGAGCAAACAGG\n",
      ">Otu11\n",
      "TACGTAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGTGCGCAGGCGGTTGTGTAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAATTGCATTTGAGACTGCACGGCTAGAGTGTGTCAGAGGGGGGTAGAATTCCACGTGTAGCAGTGAAATGCGTAGATATGTGGAGGAATACCGATGGCGAAGGCAGCCCCCTGGGATAACACTGACGCTCATGCACGAAAGCGTGGGGAGCAAACAGG\n",
      ">Otu20\n",
      "TACGAAGGGGGCTAGCGTTGTTCGGAATTACTGGGCGTAAAGCGCACGTAGGCGGATATTTAAGTCAGGGGTGAAATCCCAGAGCTCAACTCTGGAACTGCCTTTGATACTGGGTATCTTGAGTATGGAAGAGGTAAGTGGAATTCCGAGTGTAGAGGTGAAATTCGTAGATATTCGGAGGAACACCAGTGGCGAAGGCGGCTTACTGGTCCATTACTGACGCTGAGGTGCGAAAGCGTGGGGAGCAAACAGG\n",
      ">Otu23\n",
      "TACGAAGGGGGCTAGCGTTGTTCGGATTTACTGGGCGTAAAGCGCACGTAGGCGGATCGTTAAGTCGGGGGTGAAATCCTGGAGCTCAACTCCAGAACTGCCTTCGATACTGGCGATCTTGAGTCCGGAAGAGGTGAGTGGAACTCCTAGTGTAGAGGTGGAATTCGTAGATATTAGGAAGAACACCAGTGGCGAAGGCGGCTCACTGGTCCGGTACTGACGCTGAGGTGCGAAAGCGTGGGGAGCAAACAGG\n",
      ">Otu24\n",
      "GACAGAGGGTGCAAACGTTGCTCGGAATCACTGGGCGTAAAGGGCGTGTAGGCGGGAAGGAAAGTCAGATGTGAAATCCCTGGGCTCAACCCAGGAACTGCATTTGAAACTCCCTTTCTTGAGTGTCGGAGAGGAAAGTGGAATTCCTGGTGTAGAGGTGAAATTCGTAGATATCAGGAGGAACACCGGTGGCGAAGGCGGCTTTCTGGACGACAACTGACGCTGAGACGCGAAAGCGTGGGGAGCAAACAGG\n",
      ">Otu26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "head: cannot open â€˜/home/be68/Hyphosphere/data/Burkholderiales/Burkholderiales_rdp_download_205seqs.faâ€™ for reading: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$OTUfasta\" \"$RDPfile\" \n",
    "#head $1 -n 15\n",
    "head $1 -n 15\n",
    "head $2 -n 8\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Using SSU-Align to seqs from hyphal samples and mask based on alignment posterior probabilities."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Align hyphal OTUs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# _ssu-prep :: prepare SSU rRNA sequences for parallel ssu-align jobs\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-prep -x -f -b 50 --dna --rfonly /home/be68/Hyphosphere/data/Burkholderiales/HyphalBurkholderiales.fa ssu_aln 1\n",
      "# date:    Mon Jun 25 14:55:31 2018\n",
      "#\n",
      "# Validating input sequence file ... done.\n",
      "#\n",
      "# Preparing 1 ssu-align jobs ...\n",
      "# Partitioning seqs with goal of equalizing total number of nucleotides per job ...\n",
      "#\n",
      "# output file name           description                                      \n",
      "# -------------------------  -------------------------------------------------\n",
      "  ssu_aln/HyphalBurkholderiales.fa.1  partition 1 FASTA sequence file (8 seqs; 2024 nt)\n",
      "  ssu_aln.ssu-align.sh       shell script that will execute 1 ssu-align jobs\n",
      "#\n",
      "################################################################################\n",
      "# To execute all 1 ssu-align jobs, run the shell script with the command:\n",
      "#\t\"ssu_aln.ssu-align.sh\"\n",
      "# (it is an executable file)\n",
      "################################################################################\n",
      "#\n",
      "# The final job will merge the output from all jobs once they are all finished.\n",
      "#\n",
      "# If one or more jobs fail: rerun the failed jobs, wait for them to finish,\n",
      "# and then perform manual merge from this directory by executing:\n",
      "#\tssu-merge ssu_aln\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln/ssu_aln.ssu-prep.log.\n",
      "# Output printed to the screen saved in:  ssu_aln/ssu_aln.ssu-prep.sum.\n",
      "#\n",
      "# CPU time:  00:00:00.29  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$OTUfasta\"\n",
    "ssu-prep -f -x -b 50 --rfonly --dna $1 ssu_aln 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Executing: ssu-align --merge 1 -b 50 --dna --rfonly ssu_aln/HyphalBurkholderiales.fa.1 ssu_aln/ssu_aln.1\n",
      "# _ssu-align :: align SSU rRNA sequences\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-align --merge 1 -b 50 --dna --rfonly ssu_aln/HyphalBurkholderiales.fa.1 ssu_aln/ssu_aln.1\n",
      "# date:    Mon Jun 25 14:55:51 2018\n",
      "#\n",
      "# Validating input sequence file ... done.\n",
      "#\n",
      "# Stage 1: Determining SSU start/end positions and best-matching models...\n",
      "#\n",
      "# output file name            description                                \n",
      "# --------------------------  -------------------------------------------\n",
      "  ssu_aln.1.tab               locations/scores of hits defined by HMM(s)\n",
      "  ssu_aln.1.bacteria.hitlist  list of sequences to align with bacteria CM\n",
      "  ssu_aln.1.bacteria.fa             8 sequences to align with bacteria CM\n",
      "#\n",
      "# Stage 2: Aligning each sequence to its best-matching model...\n",
      "#\n",
      "# output file name            description\n",
      "# --------------------------  ---------------------------------------\n",
      "  ssu_aln.1.bacteria.stk      bacteria alignment\n",
      "  ssu_aln.1.bacteria.cmalign  bacteria cmalign output\n",
      "  ssu_aln.1.bacteria.ifile    bacteria insert info\n",
      "  ssu_aln.1.scores            list of CM/HMM scores for each sequence\n",
      "#\n",
      "# Summary statistics:\n",
      "#\n",
      "# model or       number  fraction        average   average               \n",
      "# category      of seqs  of total         length  coverage    nucleotides\n",
      "# ------------  -------  --------  -------------  --------  -------------\n",
      "  *input*             8    1.0000         253.00    1.0000           2024\n",
      "#\n",
      "  archaea             0    0.0000              -         -              0\n",
      "  bacteria            8    1.0000         253.00    1.0000           2024\n",
      "  eukarya             0    0.0000              -         -              0\n",
      "#\n",
      "  *all-models*        8    1.0000         253.00    1.0000           2024\n",
      "  *no-models*         0    0.0000              -         -              0\n",
      "#\n",
      "# Speed statistics:\n",
      "#\n",
      "# stage      num seqs  seq/sec  seq/sec/model    nucleotides    nt/sec\n",
      "# ---------  --------  -------  -------------  -------------  --------\n",
      "  search            8    7.512          2.504           2024    1900.4\n",
      "  alignment         8    6.955          6.955           2024    1759.7\n",
      "#\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln.1.ssu-align.log.\n",
      "# Output printed to the screen saved in:  ssu_aln.1.ssu-align.sum.\n",
      "#\n",
      "# All output files created in directory ./ssu_aln/ssu_aln.1/\n",
      "#\n",
      "# CPU time (search):     00:00:01.07  (hh:mm:ss)\n",
      "# CPU time (alignment):  00:00:01.15  (hh:mm:ss)\n",
      "# CPU time (total):      00:00:02.43  (hh:mm:ss)\n",
      "#\n",
      "################################################################################\n",
      "#\n",
      "# Alignments successfully created.\n",
      "#\n",
      "# Checking/waiting for other jobs to finish before merging...\n",
      "#\n",
      "# Job 1 (dir: ssu_aln/ssu_aln.1) completed successfully! 0/1 still incomplete.\n",
      "#\n",
      "# All 1 jobs successfully completed.\n",
      "# About to merge their output:\n",
      "#\n",
      "# Executing:  ssu-merge ssu_aln\n",
      "#\n",
      "################################################################################\n",
      "#\n",
      "# _ssu-merge :: merge SSU rRNA alignments\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-merge ssu_aln\n",
      "# date:    Mon Jun 25 14:55:53 2018\n",
      "#\n",
      "# Merging files from 1 ssu-align runs...\n",
      "#\n",
      "#                                     # files     # seqs\n",
      "# merged file name          CM name    merged     merged\n",
      "# ------------------------  --------  -------  ---------\n",
      "  ssu_aln.tab               -               1          -\n",
      "  ssu_aln.scores            -               1          -\n",
      "  ssu_aln.ssu-align.sum     -               1          -\n",
      "  ssu_aln.ssu-align.log     -               1          -\n",
      "#\n",
      "  ssu_aln.bacteria.fa       bacteria        1          8\n",
      "  ssu_aln.bacteria.hitlist  bacteria        1          8\n",
      "  ssu_aln.bacteria.cmalign  bacteria        1          8\n",
      "  ssu_aln.bacteria.ifile    bacteria        1          8\n",
      "  ssu_aln.bacteria.stk      bacteria        1          8\n",
      "#\n",
      "#\n",
      "# Summary statistics:\n",
      "#\n",
      "# model or       number  fraction        average   average               \n",
      "# category      of seqs  of total         length  coverage    nucleotides\n",
      "# ------------  -------  --------  -------------  --------  -------------\n",
      "  *input*             8    1.0000         253.00    1.0000           2024\n",
      "#\n",
      "  archaea             0    0.0000              -         -              0\n",
      "  bacteria            8    1.0000         253.00    1.0000           2024\n",
      "  eukarya             0    0.0000              -         -              0\n",
      "#\n",
      "  *all-models*        8    1.0000         253.00    1.0000           2024\n",
      "  *no-models*         0    0.0000              -         -              0\n",
      "#\n",
      "# Speed statistics:\n",
      "#\n",
      "# stage      num seqs  seq/sec  seq/sec/model    nucleotides    nt/sec\n",
      "# ---------  --------  -------  -------------  -------------  --------\n",
      "  search            8    8.000          2.667           2024    2024.0\n",
      "  alignment         8    8.000          8.000           2024    2024.0\n",
      "#\n",
      "# Summed ssu-align CPU time (search):     00:00:00.00  (hh:mm:ss)\n",
      "# Summed ssu-align CPU time (alignment):  00:00:00.00  (hh:mm:ss)\n",
      "# Summed ssu-align CPU time (total):      00:00:00.00  (hh:mm:ss)\n",
      "#\n",
      "# Cleaning up by removing original files that were just merged ... done.\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln.ssu-merge.log.\n",
      "# Output printed to the screen saved in:  ssu_aln.ssu-merge.sum.\n",
      "#\n",
      "# All output files created in directory ./ssu_aln/\n",
      "#\n",
      "# CPU time:  00:00:00.09  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "./ssu_aln.ssu-align.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# _ssu-mask :: mask SSU rRNA alignments\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-mask --afa --dna ssu_aln\n",
      "# date:    Mon Jun 25 14:56:35 2018\n",
      "#\n",
      "# Masking alignments based on posterior probabilities...\n",
      "#\n",
      "#                                                     mask    \n",
      "#                                                 ------------\n",
      "# file name                  in/out  type  #cols  incl.  excl.\n",
      "# -------------------------  ------  ----  -----  -----  -----\n",
      "  ssu_aln.bacteria.stk        input   aln   1582      -      -\n",
      "  ssu_aln.bacteria.mask      output  mask   1582    253   1329\n",
      "  ssu_aln.bacteria.mask.pdf  output   pdf   1582    253   1329\n",
      "  ssu_aln.bacteria.mask.afa  output   aln    253      -      -\n",
      "#\n",
      "# All attempts to draw structure diagrams of masks were successful.\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln.ssu-mask.log.\n",
      "# Output printed to the screen saved in:  ssu_aln.ssu-mask.sum.\n",
      "#\n",
      "# All output files created in directory ./ssu_aln/\n",
      "#\n",
      "# CPU time:  00:00:01.17  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "!ssu-mask --dna --afa ssu_aln"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### align RDP sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# _ssu-prep :: prepare SSU rRNA sequences for parallel ssu-align jobs\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-prep -x -f -b 50 --dna --rfonly /home/be68/Hyphosphere/data/Burkholderiales/Burkholderiales_rdp_download_205seqs.fa ssr_aln 2\n",
      "# date:    Mon Jun 25 14:57:14 2018\n",
      "#\n",
      "# Validating input sequence file ... done.\n",
      "#\n",
      "# Preparing 2 ssu-align jobs ...\n",
      "# Partitioning seqs with goal of equalizing total number of nucleotides per job ...\n",
      "#\n",
      "# output file name                          description                                          \n",
      "# ----------------------------------------  -----------------------------------------------------\n",
      "  ssr_aln/Burkholderiales_rdp_download_205seqs.fa.1  partition 1 FASTA sequence file (103 seqs; 151488 nt)\n",
      "  ssr_aln/Burkholderiales_rdp_download_205seqs.fa.2  partition 2 FASTA sequence file (102 seqs; 148992 nt)\n",
      "  ssr_aln.ssu-align.sh                      shell script that will execute 2 ssu-align jobs\n",
      "#\n",
      "################################################################################\n",
      "# To execute all 2 ssu-align jobs, run the shell script with the command:\n",
      "#\t\"ssr_aln.ssu-align.sh\"\n",
      "# (it is an executable file)\n",
      "################################################################################\n",
      "#\n",
      "# The final job will merge the output from all jobs once they are all finished.\n",
      "#\n",
      "# If one or more jobs fail: rerun the failed jobs, wait for them to finish,\n",
      "# and then perform manual merge from this directory by executing:\n",
      "#\tssu-merge ssr_aln\n",
      "#\n",
      "# List of executed commands saved in:     ssr_aln/ssr_aln.ssu-prep.log.\n",
      "# Output printed to the screen saved in:  ssr_aln/ssr_aln.ssu-prep.sum.\n",
      "#\n",
      "# CPU time:  00:00:00.15  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$RDPfile\"\n",
    "ssu-prep -f -x -b 50 --rfonly --dna $1 ssr_aln 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Executing: ssu-align -b 50 --dna --rfonly ssr_aln/Burkholderiales_rdp_download_205seqs.fa.1 ssr_aln/ssr_aln.1 > /dev/null &\n",
      "# Executing: ssu-align --merge 2 -b 50 --dna --rfonly ssr_aln/Burkholderiales_rdp_download_205seqs.fa.2 ssr_aln/ssr_aln.2\n",
      "# _ssu-align :: align SSU rRNA sequences\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-align --merge 2 -b 50 --dna --rfonly ssr_aln/Burkholderiales_rdp_download_205seqs.fa.2 ssr_aln/ssr_aln.2\n",
      "# date:    Mon Jun 25 14:57:44 2018\n",
      "#\n",
      "# Validating input sequence file ... done.\n",
      "#\n",
      "# Stage 1: Determining SSU start/end positions and best-matching models...\n",
      "#\n",
      "# output file name            description                                \n",
      "# --------------------------  -------------------------------------------\n",
      "  ssr_aln.2.tab               locations/scores of hits defined by HMM(s)\n",
      "  ssr_aln.2.bacteria.hitlist  list of sequences to align with bacteria CM\n",
      "  ssr_aln.2.bacteria.fa           102 sequences to align with bacteria CM\n",
      "#\n",
      "# Stage 2: Aligning each sequence to its best-matching model...\n",
      "#\n",
      "# output file name            description\n",
      "# --------------------------  ---------------------------------------\n",
      "  ssr_aln.2.bacteria.stk      bacteria alignment\n",
      "  ssr_aln.2.bacteria.cmalign  bacteria cmalign output\n",
      "  ssr_aln.2.bacteria.ifile    bacteria insert info\n",
      "  ssr_aln.2.scores            list of CM/HMM scores for each sequence\n",
      "#\n",
      "# Summary statistics:\n",
      "#\n",
      "# model or       number  fraction        average   average               \n",
      "# category      of seqs  of total         length  coverage    nucleotides\n",
      "# ------------  -------  --------  -------------  --------  -------------\n",
      "  *input*           102    1.0000        1460.71    1.0000         148992\n",
      "#\n",
      "  archaea             0    0.0000              -         -              0\n",
      "  bacteria          102    1.0000        1457.37    0.9977         148652\n",
      "  eukarya             0    0.0000              -         -              0\n",
      "#\n",
      "  *all-models*      102    1.0000        1457.37    0.9977         148652\n",
      "  *no-models*         0    0.0000              -         -              0\n",
      "#\n",
      "# Speed statistics:\n",
      "#\n",
      "# stage      num seqs  seq/sec  seq/sec/model    nucleotides    nt/sec\n",
      "# ---------  --------  -------  -------------  -------------  --------\n",
      "  search          102    1.677          0.559         148992    2449.6\n",
      "  alignment       102    0.916          0.916         148652    1334.3\n",
      "#\n",
      "#\n",
      "# List of executed commands saved in:     ssr_aln.2.ssu-align.log.\n",
      "# Output printed to the screen saved in:  ssr_aln.2.ssu-align.sum.\n",
      "#\n",
      "# All output files created in directory ./ssr_aln/ssr_aln.2/\n",
      "#\n",
      "# CPU time (search):     00:01:00.82  (hh:mm:ss)\n",
      "# CPU time (alignment):  00:01:51.41  (hh:mm:ss)\n",
      "# CPU time (total):      00:02:52.42  (hh:mm:ss)\n",
      "#\n",
      "################################################################################\n",
      "#\n",
      "# Alignments successfully created.\n",
      "#\n",
      "# Checking/waiting for other jobs to finish before merging...\n",
      "#\n",
      "# Job 2 (dir: ssr_aln/ssr_aln.2) completed successfully! 1/2 still incomplete.\n",
      "# Job 1 (dir: ssr_aln/ssr_aln.1) completed successfully! 0/2 still incomplete.\n",
      "#\n",
      "# All 2 jobs successfully completed.\n",
      "# About to merge their output:\n",
      "#\n",
      "# Executing:  ssu-merge ssr_aln\n",
      "#\n",
      "################################################################################\n",
      "#\n",
      "# _ssu-merge :: merge SSU rRNA alignments\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-merge ssr_aln\n",
      "# date:    Mon Jun 25 15:00:40 2018\n",
      "#\n",
      "# Merging files from 2 ssu-align runs...\n",
      "#\n",
      "#                                     # files     # seqs\n",
      "# merged file name          CM name    merged     merged\n",
      "# ------------------------  --------  -------  ---------\n",
      "  ssr_aln.tab               -               2          -\n",
      "  ssr_aln.scores            -               2          -\n",
      "  ssr_aln.ssu-align.sum     -               2          -\n",
      "  ssr_aln.ssu-align.log     -               2          -\n",
      "#\n",
      "  ssr_aln.bacteria.fa       bacteria        2        205\n",
      "  ssr_aln.bacteria.hitlist  bacteria        2        205\n",
      "  ssr_aln.bacteria.cmalign  bacteria        2        205\n",
      "  ssr_aln.bacteria.ifile    bacteria        2        205\n",
      "  ssr_aln.bacteria.stk      bacteria        2        205\n",
      "#\n",
      "#\n",
      "# Summary statistics:\n",
      "#\n",
      "# model or       number  fraction        average   average               \n",
      "# category      of seqs  of total         length  coverage    nucleotides\n",
      "# ------------  -------  --------  -------------  --------  -------------\n",
      "  *input*           205    1.0000        1465.76    1.0000         300480\n",
      "#\n",
      "  archaea             0    0.0000              -         -              0\n",
      "  bacteria          205    1.0000        1463.18    0.9983         299951\n",
      "  eukarya             0    0.0000              -         -              0\n",
      "#\n",
      "  *all-models*      205    1.0000        1463.18    0.9983         299951\n",
      "  *no-models*         0    0.0000              -         -              0\n",
      "#\n",
      "# Speed statistics:\n",
      "#\n",
      "# stage      num seqs  seq/sec  seq/sec/model    nucleotides    nt/sec\n",
      "# ---------  --------  -------  -------------  -------------  --------\n",
      "  search          205  205.000         68.333         300480  300480.0\n",
      "  alignment       205  205.000        205.000         299951  299951.0\n",
      "#\n",
      "# Summed ssu-align CPU time (search):     00:00:00.00  (hh:mm:ss)\n",
      "# Summed ssu-align CPU time (alignment):  00:00:00.00  (hh:mm:ss)\n",
      "# Summed ssu-align CPU time (total):      00:00:00.00  (hh:mm:ss)\n",
      "#\n",
      "# Cleaning up by removing original files that were just merged ... done.\n",
      "#\n",
      "# List of executed commands saved in:     ssr_aln.ssu-merge.log.\n",
      "# Output printed to the screen saved in:  ssr_aln.ssu-merge.sum.\n",
      "#\n",
      "# All output files created in directory ./ssr_aln/\n",
      "#\n",
      "# CPU time:  00:00:00.09  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "./ssr_aln.ssu-align.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# _ssu-mask :: mask SSU rRNA alignments\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-mask -s ssu_aln/ssu_aln.bacteria.mask --afa --dna ssr_aln/\n",
      "# date:    Mon Jun 25 15:00:40 2018\n",
      "#\n",
      "# Masking alignments using pre-existing masks...\n",
      "#\n",
      "#                                                     mask    \n",
      "#                                                 ------------\n",
      "# file name                  in/out  type  #cols  incl.  excl.\n",
      "# -------------------------  ------  ----  -----  -----  -----\n",
      "  ssr_aln.bacteria.stk        input   aln   1582      -      -\n",
      "  ssu_aln.bacteria.mask       input  mask   1582    253   1329\n",
      "  ssr_aln.bacteria.mask.pdf  output   pdf   1582    253   1329\n",
      "  ssr_aln.bacteria.mask.afa  output   aln    253      -      -\n",
      "#\n",
      "# All attempts to draw structure diagrams of masks were successful.\n",
      "#\n",
      "# List of executed commands saved in:     ssr_aln.ssu-mask.log.\n",
      "# Output printed to the screen saved in:  ssr_aln.ssu-mask.sum.\n",
      "#\n",
      "# All output files created in directory ./ssr_aln/\n",
      "#\n",
      "# CPU time:  00:00:00.55  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "!ssu-mask -s ssu_aln/ssu_aln.bacteria.mask --dna --afa ssr_aln/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cat ssr_aln/ssr_aln.bacteria.mask.afa ssu_aln/ssu_aln.bacteria.mask.afa > aln_for_tree.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">S000017759 Acidovorax konjaci (T); DSM 7481; AJ420325\r\n",
      "TACGTAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGTGCGCAGGCGGTGATG\r\n",
      "TAAGACAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATTTGTGACTGCATCGCTG\r\n",
      "GAGTGCGGCAGAGGGGAGTGGAATTCCGCGTGTAGCACTGAAATGCGTANATATGCGGAA\r\n",
      "GAACACCGATGGCGAAGGCAATCCCCTGGGCCTGCACTGACGCTCATGCACGAAAGCGTG\r\n",
      "GGGAGCAAACAGG\r\n",
      ">S000020603 Acidovorax defluvii (T); BSB411; Y18616\r\n",
      "TACGTAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGTGCGCAGGCGGTTATA\r\n",
      "TAAGACAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATTTGTGACTGTATAGCTA\r\n",
      "GAGTACGGCAGAGGGGGATGGAATTCCGCGTGTAGCAGTGAAATGCGTAGATATGCGGAG\r\n",
      "GAACACCGATGGCGAAGGCAATCCCCTGGGCCTGTACTGACGCTCATGCACGAAAGCGTG\r\n",
      "GGGAGCAAACAGG\r\n",
      ">S000131286 Acidovorax anthurii (T); CFBP 3232; AJ007013\r\n",
      "TACGTAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGTGCGCAGGCGGTTATG\r\n",
      "TAAGACAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATTTGTGACTGCATAGCTA\r\n",
      "GAGTACGGTAGAGGGGGATGGAATTCCGCGTGTAGCAGTGAAATGCGTAGATATGCGGAG\r\n",
      "GAACACCGATGGCGAANGCAATCCCCTGGACCTGTACTGACGCTCATGCACGAAAGCGTG\r\n",
      "GGGAGCAAACAGG\r\n",
      ">S000133753 Acidovorax valerianellae (T); CFBP 4730; AJ431731\r\n",
      "TACGTAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGTGCGCAGGCGGTTATG\r\n"
     ]
    }
   ],
   "source": [
    "!head aln_for_tree.fasta -n 20"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Inferring and rooting the tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#aln = LoadSeqs(os.path.join(workDir, 'aln_for_tree.fasta'), moltype=DNA)\n",
    "aln = LoadSeqs('aln_for_tree.fasta', moltype=DNA)\n",
    "t_unroot = build_tree_from_alignment(aln, moltype=DNA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "t_unroot.writeToFile('MyxOTUs_wTypeStrains.tree')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
