{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Incorporate isolate sequences into master tree of DADA2 seqs\n",
    "* Align isolate seqs using SSU_align\n",
    "* Mask sequences based on same mask as used for sequences from Exp10 \n",
    "* Append sequences to create new fasta with both sets of sequences\n",
    "* Create master tree using FastTree\n",
    "* Use qiime environment and set kernel to Python 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "workDir = '/home/be68/Hyphosphere/Exp10/MasterTree/'\n",
    "maskDir = '/home/be68/Hyphosphere/Exp10/DADA2/'\n",
    "IsolateSeqDir = '/home/be68/Hyphosphere/Exp10/MasterTree/'\n",
    "IsolateSeqFile = 'IsolateSeqs.fasta'\n",
    "\n",
    "\n",
    "nprocs = 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import numpy as np\n",
    "#import entrez.direct\n",
    "from cogent.app.fasttree import build_tree_from_alignment\n",
    "from cogent import DNA, LoadSeqs\n",
    "from Bio import Entrez, SeqIO\n",
    "from pandas import *\n",
    "Entrez.email = \"be68@cornell.edu\"\n",
    "#from IPython.display import display, Image, SVG\n",
    "#from cogent3 import LoadSeqs, FastTree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "if not os.path.isdir(workDir):\n",
    "    os.mkdir(workDir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/home/be68/Hyphosphere/Exp10/MasterTree\n"
     ]
    }
   ],
   "source": [
    "%cd $workDir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ln: ‘/home/be68/Hyphosphere/Exp10/MasterTree/IsolateSeqs.fasta’ and ‘./IsolateSeqs.fasta’ are the same file\r\n"
     ]
    }
   ],
   "source": [
    "!cd $workDir; ln -f -s $IsolateSeqDir$IsolateSeqFile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">BE26\r",
      "\r\n",
      "AGCGGCGGACGGGTGAGTAACACGTGGGCAACCTGCCTGTAAGACTGGGATAACTTCGGGAAACCGAAGCTAATACCGGATAGGATCTTCTCCTTCATGGGAGATGATTGAAAGATGGTTTCGGCTATCACTTACAGATGGGCCCGCGGTGCATTAGCTAGTTGGTGAGGTAACGGCTCACCAAGGCAACGATGCATAGCCGACCTGAGAGGGTGATCGGCCACACTGGGACTGAGACACGGCCCAGACTCCTACGGGAGGCAGCAGTAGGGAATCTTCCGCAATGGACGAAAGTCTGACGGAGCAACGCCGCGTGAGTGATGAAGGCTTTCGGGTCGTAAAACTCTGTTGTTAGGGAAGAACAAGTACGAGAGTAACTGCTCGTACCTTGACGGTACCTAACCAGAAAGCCACGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGTGGCAAGCGTTATCCGGAATTATTGGGCGTAAAGCGCGCGCAGGCGGTTTCTTAAGTCTGATGTGAAAGCCCACGGCTCAACCGTGGAGGGTCATTGGAAACTGGGGAACTTGAGTGCAGAAGAGAAAAGCGGAATTCCACGTGTAGCGGTGAAATGCGTAGAGATGTGGAGGAACACCAGTGGCGAAGGCGGCTTTTTGGTCTGTAACTGACGCTGAGGCGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGAGTGCTAAGTGTTAGAGGGTTTCCGCCCTTTAGTGCTGCAGCTAACGCATTAAGCACTCCGCCTGGGGAGTACGGTCGCAAGACTGAAACTCAAAGGAATTGACGGGGGCCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGAAGCAACGCGAAGAACCTTACCAGGTCTTGACATCCTCTGACAACTCTAGAGATAGAGCGTTCCCCTTCGGGGGACAGAGTGACAGGTGGTGCATGGTTGTCGTCAGCTCGTGTCGTGAGATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTGATCTTAGTTGCCAGCATTTAGTTGGGCACTCTAAGGTGACTGCCGGTGACAAACCGGAGGAAGGTGGGGATGACGTCAAATCATCATGCCCCTTATGACCTGGGCTACACACGTGCTACAATGGATGGTACAAAGGGCTGCAAGACCGCGAGGTCAAGCCAATCCCATAAAACCATTCTCAGTTCGGATTGTAGGCTGCAACTCGCCTACATGAAGCTGGAATCGCTAGTAATCGCGGATCAGCATGCCGCGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCACGAGAGTTTG\r",
      "\r\n",
      ">BE27\r",
      "\r\n",
      "GACGTTAGCGGCGGACGGGTGAGTAACACGTGGGCAACCTGCCTGTAAGACTGGGATAACTTCGGGAAACCGAAGCTAATACCGGATAGGATCTTCTCCTTCATGGGAGATGATTGAAAGATGGTTTCGGCTATCACTTACAGATGGGCCCGCGGTGCATTAGCTAGTTGGTGAGGTAACGGCTCACCAAGGCAACGATGCATAGCCGACCTGAGAGGGTGATCGGCCACACTGGGACTGAGACACGGCCCAGACTCCTACGGGAGGCAGCAGTAGGGAATCTTCCGCAATGGACGAAAGTCTGACGGAGCAACGCCGCGTGAGTGATGAAGGCTTTCGGGTCGTAAAACTCTGTTGTTAGGGAAGAACAAGTACGAGAGTAACTGCTCGTACCTTGACGGTACCTAACCAGAAAGCCACGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGTGGCAAGCGTTATCCGGAATTATTGGGCGTAAAGCGCGCGCAGGCGGTTTCTTAAGTCTGATGTGAAAGCCCACGGCTCAACCGTGGAGGGTCATTGGAAACTGGGGAACTTGAGTGCAGAAGAGAAAAGCGGAATTCCACGTGTAGCGGTGAAATGCGTAGAGATGTGGAGGAACACCAGTGGCGAAGGCGGCTTTTTGGTCTGTAACTGACGCTGAGGCGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGAGTGCTAAGTGTTAGAGGGTTTCCGCCCTTTAGTGCTGCAGCTAACGCATTAAGCACTCCGCCTGGGGAGTACGGTCGCAAGACTGAAACTCAAAGGAATTGACGGGGGCCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGAAGCAACGCGAAGAACCTTACCAGGTCTTGACATCCTCTGACAACTCTAGAGATAGAGCGTTCCCCTTCGGGGGACAGAGTGACAGGTGGTGCATGGTTGTCGTCAGCTCGTGTCGTGAGATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTGATCTTAGTTGCCAGCATTTAGTTGGGCACTCTAAGGTGACTGCCGGTGACAAACCGGAGGAAGGTGGGGATGACGTCAAATCATCATGCCCCTTATGACCTGGGCTACACACGTGCTACAATGGATGGTACAAAGGGCTGCAAGACCGCGAGGTCAAGCCAATCCCATAAAACCATTCTCAGTTCGGATTGTAGGCTGCAACTCGCCTACATGAAGCTGGAATCGCTAGTAATCGCGGATCAGCATGCCGCGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCACGAGAGTTTG\r",
      "\r\n",
      ">BE28\r",
      "\r\n",
      "ATGACGTTAGCGGCGGACGGGTGAGTAACACGTGGGCAACCTGCCTGTAAGACTGGGATAACTTCGGGAAACCGAAGCTAATACCGGATAGGATCTTCTCCTTCATGGGAGATGATTGAAAGATGGTTTCGGCTATCACTTACAGATGGGCCCGCGGTGCATTAGCTAGTTGGTGAGGTAACGGCTCACCAAGGCAACGATGCATAGCCGACCTGAGAGGGTGATCGGCCACACTGGGACTGAGACACGGCCCAGACTCCTACGGGAGGCAGCAGTAGGGAATCTTCCGCAATGGACGAAAGTCTGACGGAGCAACGCCGCGTGAGTGATGAAGGCTTTCGGGTCGTAAAACTCTGTTGTTAGGGAAGAACAAGTACGAGAGTAACTGCTCGTACCTTGACGGTACCTAACCAGAAAGCCACGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGTGGCAAGCGTTATCCGGAATTATTGGGCGTAAAGCGCGCGCAGGCGGTTTCTTAAGTCTGATGTGAAAGCCCACGGCTCAACCGTGGAGGGTCATTGGAAACTGGGGAACTTGAGTGCAGAAGAGAAAAGCGGAATTCCACGTGTAGCGGTGAAATGCGTAGAGATGTGGAGGAACACCAGTGGCGAAGGCGGCTTTTTGGTCTGTAACTGACGCTGAGGCGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGATGAGTGCTAAGTGTTAGAGGGTTTCCGCCCTTTAGTGCTGCAGCTAACGCATTAAGCACTCCGCCTGGGGAGTACGGTCGCAAGACTGAAACTCAAAGGAATTGACGGGGGCCCGCACAAGCGGTGGAGCATGTGGTTTAATTCGAAGCAACGCGAAGAACCTTACCAGGTCTTGACATCCTCTGACAACTCTAGAGATAGAGCGTTCCCCTTCGGGGGACAGAGTGACAGGTGGTGCATGGTTGTCGTCAGCTCGTGTCGTGAGATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTGATCTTAGTTGCCAGCATTTAGTTGGGCACTCTAAGGTGACTGCCGGTGACAAACCGGAGGAAGGTGGGGATGACGTCAAATCATCATGCCCCTTATGACCTGGGCTACACACGTGCTACAATGGATGGTACAAAGGGCTGCAAGACCGCGAGGTCAAGCCAATCCCATAAAACCATTCTCAGTTCGGATTGTAGGCTGCAACTCGCCTACATGAAGCTGGAATCGCTAGTAATCGCGGATCAGCATGCCGCGGTGAATACGTTCCCGGGCCTTGTACACACCGCCCGTCACACCACGAGAGTTTG\r",
      "\r\n",
      ">BE29\r",
      "\r\n",
      "GTGGCGAACGGGTGAGTAATACATCGGAACGTGCCCAGTCGTGGGGGATAACTACTCGAAAGAGTAGCTAATACCGCATACGATCTGAGGATGAAAGCGGGGGACCTTCGGGCCTCGCGCGATTGGAGCGGCCGATGGCAGATTAGGTAGTTGGTGGGATAAAAGCTTACCAAGCCGACGATCTGTAGCTGGTCTGAGAGGACGACCAGCCACACTGGGACTGAGACACGGCCCAGACTCCTACGGGAGGCAGCAGTGGGGAATTTTGGACAATGGGCGAAAGCCTGATCCAGCAATGCCGCGTGCAGGATGAAGGCCTTCGGGTTGTAAACTGCTTTTGTACGGAACGAAAAAGCTCCTTCTAATACAGGGGGCCCATGACGGTACCGTAAGAATAAGCACCGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGGTGCGAGCGTTAATCGGAATTACTGGGCGTAAAGCGTGCGCAGGCGGTTATGTAAGACAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATTTGTGACTGCATGGCTAGAGTACGGTAGAGGGGGATGGAATTCCGCGTGTAGCAGTGAAATGCGTAGATATGCGGAGGAACACCGATGGCGAAGGCAATCCCCTGGACCTGTACTGACGCTCATGCACGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCCTAAACGATGTCAACTGGTTGTTGGGAATTAGTTTTCTCAGTAACGAAGCTAACGCGTGAAGTTGACCGCCTGGGGAGTACGGCCGCAAGGTTGAAACTCAAAGGAATTGACGGGGACCCGCACAAGCGGTGGATGATGTGGTTTAATTCGATGCAACGCGAAAAACCTTACCCACCTTTGACATGGCAGGAAGTTTCCAGAGATGGATTCGTGCTCGAAAGAGAACCTGCACACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTCGTGAGATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTGTCATTAGTTGCTACATTCAGTTGAGCACTCTAATGAGACTGCCGGTGACAAACCGGAGGAAGGTGGGGATGACGTCAAGTCCTCATGGCCCTTATAGGTGGGGCTACACACGTCATACAATGGCTGGTACAGAGGGTTGCCAACCCGCGAGGGGGAGCTAATCCCATAAAACCAGTCGTAGTCCGGATCGCAGTCTGCAACTCGACTGCGTGAAGTCGGAATCGCTAGTAATCGCGGATCAGCATGCCGCGGTGAATACGTTCCCGGGTCTTGTACACACCGCCCGTCACACCATGGGAGCGGGTCT\r",
      "\r\n",
      ">BE30\r",
      "\r\n",
      "GTGGCGAACGGGTGAGTAATACATCGGAACGTGCCCAGTCGTGGGGGATAACTACTCGAAAGAGTAGCTAATACCGCATACGATCTGAGGATGAAAGCGGGGGACCTTCGGGCCTCGCGCGATTGGAGCGGCCGATGGCAGATTAGGTAGTTGGTGGGATAAAAGCTTACCAAGCCGACGATCTGTAGCTGGTCTGAGAGGACGACCAGCCACACTGGGACTGAGACACGGCCCAGACTCCTACGGGAGGCAGCAGTGGGGAATTTTGGACAATGGGCGAAAGCCTGATCCAGCAATGCCGCGTGCAGGATGAAGGCCTTCGGGTTGTAAACTGCTTTTGTACGGAACGAAAAAGCTCCTTCTAATACAGGGGGCCCATGACGGTACCGTAAGAATAAGCACCGGCTAACTACGTGCCAGCAGCCGCGGTAATACGTAGGGTGCGAGCGTTAATCGGAATTACTGGGCGTAAAGCGTGCGCAGGCGGTTATGTAAGACAGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATTTGTGACTGCATGGCTAGAGTACGGTAGAGGGGGATGGAATTCCGCGTGTAGCAGTGAAATGCGTAGATATGCGGAGGAACACCGATGGCGAAGGCAATCCCCTGGACCTGTACTGACGCTCATGCACGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCCCTAAACGATGTCAACTGGTTGTTGGGAATTAGTTTTCTCAGTAACGAAGCTAACGCGTGAAGTTGACCGCCTGGGGAGTACGGCCGCAAGGTTGAAACTCAAAGGAATTGACGGGGACCCGCACAAGCGGTGGATGATGTGGTTTAATTCGATGCAACGCGAAAAACCTTACCCACCTTTGACATGGCAGGAAGTTTCCAGAGATGGATTCGTGCTCGAAAGAGAACCTGCACACAGGTGCTGCATGGCTGTCGTCAGCTCGTGTCGTGAGATGTTGGGTTAAGTCCCGCAACGAGCGCAACCCTTGTCATTAGTTGCTACATTCAGTTGAGCACTCTAATGAGACTGCCGGTGACAAACCGGAGGAAGGTGGGGATGACGTCAAGTCCTCATGGCCCTTATAGGTGGGGCTACACACGTCATACAATGGCTGGTACAGAGGGTTGCCAACCCGCGAGGGGGAGCTAATCCCATAAAACCAGTCGTAGTCCGGATCGCAGTCTGCAACTCGACTGCGTGAAGTCGGAATCGCTAGTAATCGCGGATCAGCATGCCGCGGTGAATACGTTCCCGGGTCTTGTACACACCGCCCGTCACACCATGGGAGCGGGTCT\r",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!cd $workDir; head $IsolateSeqFile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of OTUs in fasta: 170\n"
     ]
    }
   ],
   "source": [
    "!printf \"Number of OTUs in fasta: \"\n",
    "!cd $workDir; grep -c \">\" $IsolateSeqFile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Using SSU-Align to align seqs and masking based on alignment posterior probabilities of Exp10 - ASV set."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# _ssu-prep :: prepare SSU rRNA sequences for parallel ssu-align jobs\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-prep -x -f -b 50 --dna --rfonly IsolateSeqs.fasta ssu_aln 3\n",
      "# date:    Fri Apr 19 15:40:32 2019\n",
      "#\n",
      "# Validating input sequence file ... done.\n",
      "#\n",
      "# Preparing 3 ssu-align jobs ...\n",
      "# Partitioning seqs with goal of equalizing total number of nucleotides per job ...\n",
      "#\n",
      "# output file name      description                                        \n",
      "# --------------------  ---------------------------------------------------\n",
      "  ssu_aln/IsolateSeqs.fasta.1  partition 1 FASTA sequence file (55 seqs; 70726 nt)\n",
      "  ssu_aln/IsolateSeqs.fasta.2  partition 2 FASTA sequence file (56 seqs; 70104 nt)\n",
      "  ssu_aln/IsolateSeqs.fasta.3  partition 3 FASTA sequence file (54 seqs; 67718 nt)\n",
      "  ssu_aln.ssu-align.sh  shell script that will execute 3 ssu-align jobs\n",
      "#\n",
      "################################################################################\n",
      "# To execute all 3 ssu-align jobs, run the shell script with the command:\n",
      "#\t\"ssu_aln.ssu-align.sh\"\n",
      "# (it is an executable file)\n",
      "################################################################################\n",
      "#\n",
      "# The final job will merge the output from all jobs once they are all finished.\n",
      "#\n",
      "# If one or more jobs fail: rerun the failed jobs, wait for them to finish,\n",
      "# and then perform manual merge from this directory by executing:\n",
      "#\tssu-merge ssu_aln\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln/ssu_aln.ssu-prep.log.\n",
      "# Output printed to the screen saved in:  ssu_aln/ssu_aln.ssu-prep.sum.\n",
      "#\n",
      "# CPU time:  00:00:00.16  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "!cd $workDir; ssu-prep -f -x -b 50 --rfonly --dna $IsolateSeqFile ssu_aln 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Executing: ssu-align -b 50 --dna --rfonly ssu_aln/IsolateSeqs.fasta.1 ssu_aln/ssu_aln.1 > /dev/null &\n",
      "# Executing: ssu-align -b 50 --dna --rfonly ssu_aln/IsolateSeqs.fasta.2 ssu_aln/ssu_aln.2 > /dev/null &\n",
      "# Executing: ssu-align --merge 3 -b 50 --dna --rfonly ssu_aln/IsolateSeqs.fasta.3 ssu_aln/ssu_aln.3\n",
      "# _ssu-align :: align SSU rRNA sequences\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-align --merge 3 -b 50 --dna --rfonly ssu_aln/IsolateSeqs.fasta.3 ssu_aln/ssu_aln.3\n",
      "# date:    Fri Apr 19 15:40:58 2019\n",
      "#\n",
      "# Validating input sequence file ... done.\n",
      "#\n",
      "# Stage 1: Determining SSU start/end positions and best-matching models...\n",
      "#\n",
      "# output file name            description                                \n",
      "# --------------------------  -------------------------------------------\n",
      "  ssu_aln.3.tab               locations/scores of hits defined by HMM(s)\n",
      "  ssu_aln.3.bacteria.hitlist  list of sequences to align with bacteria CM\n",
      "  ssu_aln.3.bacteria.fa            54 sequences to align with bacteria CM\n",
      "#\n",
      "# Stage 2: Aligning each sequence to its best-matching model...\n",
      "#\n",
      "# output file name            description\n",
      "# --------------------------  ---------------------------------------\n",
      "  ssu_aln.3.bacteria.stk      bacteria alignment\n",
      "  ssu_aln.3.bacteria.cmalign  bacteria cmalign output\n",
      "  ssu_aln.3.bacteria.ifile    bacteria insert info\n",
      "  ssu_aln.3.scores            list of CM/HMM scores for each sequence\n",
      "#\n",
      "# Summary statistics:\n",
      "#\n",
      "# model or       number  fraction        average   average               \n",
      "# category      of seqs  of total         length  coverage    nucleotides\n",
      "# ------------  -------  --------  -------------  --------  -------------\n",
      "  *input*            54    1.0000        1254.04    1.0000          67718\n",
      "#\n",
      "  archaea             0    0.0000              -         -              0\n",
      "  bacteria           54    1.0000        1252.67    0.9989          67644\n",
      "  eukarya             0    0.0000              -         -              0\n",
      "#\n",
      "  *all-models*       54    1.0000        1252.67    0.9989          67644\n",
      "  *no-models*         0    0.0000              -         -              0\n",
      "#\n",
      "# Speed statistics:\n",
      "#\n",
      "# stage      num seqs  seq/sec  seq/sec/model    nucleotides    nt/sec\n",
      "# ---------  --------  -------  -------------  -------------  --------\n",
      "  search           54    1.963          0.654          67718    2462.2\n",
      "  alignment        54    1.086          1.086          67644    1360.9\n",
      "#\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln.3.ssu-align.log.\n",
      "# Output printed to the screen saved in:  ssu_aln.3.ssu-align.sum.\n",
      "#\n",
      "# All output files created in directory ./ssu_aln/ssu_aln.3/\n",
      "#\n",
      "# CPU time (search):     00:00:27.50  (hh:mm:ss)\n",
      "# CPU time (alignment):  00:00:49.71  (hh:mm:ss)\n",
      "# CPU time (total):      00:01:17.68  (hh:mm:ss)\n",
      "#\n",
      "################################################################################\n",
      "#\n",
      "# Alignments successfully created.\n",
      "#\n",
      "# Checking/waiting for other jobs to finish before merging...\n",
      "#\n",
      "# Job 3 (dir: ssu_aln/ssu_aln.3) completed successfully! 2/3 still incomplete.\n",
      "# Job 1 (dir: ssu_aln/ssu_aln.1) completed successfully! 1/3 still incomplete.\n",
      "# Job 2 (dir: ssu_aln/ssu_aln.2) completed successfully! 0/3 still incomplete.\n",
      "#\n",
      "# All 3 jobs successfully completed.\n",
      "# About to merge their output:\n",
      "#\n",
      "# Executing:  ssu-merge ssu_aln\n",
      "#\n",
      "################################################################################\n",
      "#\n",
      "# _ssu-merge :: merge SSU rRNA alignments\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-merge ssu_aln\n",
      "# date:    Fri Apr 19 15:42:22 2019\n",
      "#\n",
      "# Merging files from 3 ssu-align runs...\n",
      "#\n",
      "#                                     # files     # seqs\n",
      "# merged file name          CM name    merged     merged\n",
      "# ------------------------  --------  -------  ---------\n",
      "  ssu_aln.tab               -               3          -\n",
      "  ssu_aln.scores            -               3          -\n",
      "  ssu_aln.ssu-align.sum     -               3          -\n",
      "  ssu_aln.ssu-align.log     -               3          -\n",
      "#\n",
      "  ssu_aln.bacteria.fa       bacteria        3        165\n",
      "  ssu_aln.bacteria.hitlist  bacteria        3        165\n",
      "  ssu_aln.bacteria.cmalign  bacteria        3        165\n",
      "  ssu_aln.bacteria.ifile    bacteria        3        165\n",
      "  ssu_aln.bacteria.stk      bacteria        3        165\n",
      "#\n",
      "#\n",
      "# Summary statistics:\n",
      "#\n",
      "# model or       number  fraction        average   average               \n",
      "# category      of seqs  of total         length  coverage    nucleotides\n",
      "# ------------  -------  --------  -------------  --------  -------------\n",
      "  *input*           165    1.0000        1263.93    1.0000         208548\n",
      "#\n",
      "  archaea             0    0.0000              -         -              0\n",
      "  bacteria          165    1.0000        1262.46    0.9988         208306\n",
      "  eukarya             0    0.0000              -         -              0\n",
      "#\n",
      "  *all-models*      165    1.0000        1262.46    0.9988         208306\n",
      "  *no-models*         0    0.0000              -         -              0\n",
      "#\n",
      "# Speed statistics:\n",
      "#\n",
      "# stage      num seqs  seq/sec  seq/sec/model    nucleotides    nt/sec\n",
      "# ---------  --------  -------  -------------  -------------  --------\n",
      "  search          165  165.000         55.000         208548  208548.0\n",
      "  alignment       165  165.000        165.000         208306  208306.0\n",
      "#\n",
      "# Summed ssu-align CPU time (search):     00:00:00.00  (hh:mm:ss)\n",
      "# Summed ssu-align CPU time (alignment):  00:00:00.00  (hh:mm:ss)\n",
      "# Summed ssu-align CPU time (total):      00:00:00.00  (hh:mm:ss)\n",
      "#\n",
      "# Cleaning up by removing original files that were just merged ... done.\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln.ssu-merge.log.\n",
      "# Output printed to the screen saved in:  ssu_aln.ssu-merge.sum.\n",
      "#\n",
      "# All output files created in directory ./ssu_aln/\n",
      "#\n",
      "# CPU time:  00:00:00.11  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "!cd $workDir; ./ssu_aln.ssu-align.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# _ssu-mask :: mask SSU rRNA alignments\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-mask -s /home/be68/Hyphosphere/Exp10/DADA2//ssu_aln/ssu_aln.bacteria.mask --afa --dna ssu_aln/\n",
      "# date:    Fri Apr 19 15:44:03 2019\n",
      "#\n",
      "# Masking alignments using pre-existing masks...\n",
      "#\n",
      "#                                                     mask    \n",
      "#                                                 ------------\n",
      "# file name                  in/out  type  #cols  incl.  excl.\n",
      "# -------------------------  ------  ----  -----  -----  -----\n",
      "  ssu_aln.bacteria.stk        input   aln   1582      -      -\n",
      "  ssu_aln.bacteria.mask       input  mask   1582    248   1334\n",
      "  ssu_aln.bacteria.mask.pdf  output   pdf   1582    248   1334\n",
      "  ssu_aln.bacteria.mask.afa  output   aln    248      -      -\n",
      "#\n",
      "# All attempts to draw structure diagrams of masks were successful.\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln.ssu-mask.log.\n",
      "# Output printed to the screen saved in:  ssu_aln.ssu-mask.sum.\n",
      "#\n",
      "# All output files created in directory ./ssu_aln/\n",
      "#\n",
      "# CPU time:  00:00:01.39  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "!cd $workDir; ssu-mask -s $maskDir/ssu_aln/ssu_aln.bacteria.mask --dna --afa ssu_aln/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Append masked alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cd $workDir; cat $maskDir/ssu_aln/ssu_aln.bacteria.mask.afa ssu_aln/ssu_aln.bacteria.mask.afa > aln_for_tree.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAACATCTGATACTGGCAAGCTTGAG\r\n",
      "TCTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAA\r\n",
      "TACCGGTGGCGAAGGCGGCCCCCTGGACGAACTGACGCTCAGGTGCGAAAGCGTGGGGAG\r\n",
      "CAAACAGG\r\n",
      ">BE277\r\n",
      "TACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTTTGT\r\n",
      "TAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAACATCTGATACTGGCAAGCTTGAG\r\n",
      "TCTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAA\r\n",
      "TACCGGTGGCGAAGGCGGCCCCCTGGACGAACTGACGCTCAGGTGCGAAAGCGTGGGGAG\r\n",
      "CAAACAGG\r\n"
     ]
    }
   ],
   "source": [
    "!tail aln_for_tree.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Inferring and rooting the tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "aln = LoadSeqs(os.path.join(workDir, 'aln_for_tree.fasta'), moltype=DNA)\n",
    "t_unroot = build_tree_from_alignment(aln, moltype=DNA)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Using sulfalobus sso_acc = \"X90478\" as the outgroup leads to an error downstream when calculating unifrac distances, instead I checked the tree with sulfalobus as root and identified ASV1721 as an outgroup.  Rooted the tree to ASV1721 instead and received no errors during the analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "t_rooted = t_unroot.rootedWithTip('ASV1721')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "t_rooted.writeToFile(os.path.join(workDir, 'Master.tree'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## Subset tree with hyphal ASVs and top"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## get tip names of isolates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['BE26', 'BE27', 'BE28', 'BE29', 'BE30', 'BE31', 'BE32', 'BE33', 'BE34', 'BE35', 'BE36', 'BE38', 'BE39', 'BE40', 'BE41', 'BE42', 'BE43', 'BE44', 'BE45', 'BE46', 'BE47', 'BE48', 'BE49', 'BE50', 'BE51', 'BE52', 'BE53', 'BE54', 'BE55', 'BE56', 'BE57', 'BE59', 'BE60', 'BE61', 'BE62', 'BE63', 'BE64', 'BE65', 'BE65B', 'BE66', 'BE68', 'BE69', 'BE70', 'BE71', 'BE72', 'BE74', 'BE75', 'BE76', 'BE79', 'BE80', 'BE81', 'BE82', 'BE83', 'BE84', 'BE85', 'BE88', 'BE91', 'BE92', 'BE93', 'BE94', 'BE95', 'BE96', 'BE97', 'BE98', 'BE100', 'BE101', 'BE102', 'BE103', 'BE104', 'BE105', 'BE107', 'BE108', 'BE109', 'BE110', 'BE112', 'BE113', 'BE114', 'BE115', 'BE116', 'BE117', 'BE118', 'BE120', 'BE121', 'BE122', 'BE123', 'BE124', 'BE125', 'BE126', 'BE127', 'BE128', 'BE153', 'BE154', 'BE156', 'BE157', 'BE159', 'BE163', 'BE164', 'BE166', 'BE167', 'BE168', 'BE170', 'BE171', 'BE172', 'BE173', 'BE177', 'BE178', 'BE179', 'BE180', 'BE181', 'BE182', 'BE183', 'BE184', 'BE186', 'BE187', 'BE190', 'BE191', 'BE193', 'BE194', 'BE195', 'BE196', 'BE197', 'BE198', 'BE201', 'BE203', 'BE206', 'BE207', 'BE212', 'BE213', 'BE215', 'BE216', 'BE219', 'BE220', 'BE222', 'BE223', 'BE224', 'BE228', 'BE231', 'BE232', 'BE233', 'BE234', 'BE235', 'BE236', 'BE239', 'BE240', 'BE241', 'BE242', 'BE243', 'BE248', 'BE249', 'BE251', 'BE254', 'BE255', 'BE256', 'BE257', 'BE258', 'BE259', 'BE260', 'BE261', 'BE264', 'BE265', 'BE268', 'BE270', 'BE275', 'BE276', 'BE277']\n"
     ]
    }
   ],
   "source": [
    "file_to_open = os.path.join(IsolateSeqDir, IsolateSeqFile)\n",
    "IsolateList = []\n",
    "\n",
    "for record in SeqIO.parse(file_to_open, \"fasta\") :\n",
    "       IsolateList.append(record.id)#+\"\\n\"+str(record.seq)+\"\\n\")\n",
    "\n",
    "print(IsolateList)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import blast tablesb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BE26\tASV193\t100.00\t253\t0\t0\t444\t696\t1\t253\t3e-132\t  468\r\n",
      "BE27\tASV193\t100.00\t253\t0\t0\t450\t702\t1\t253\t3e-132\t  468\r\n",
      "BE28\tASV193\t100.00\t253\t0\t0\t452\t704\t1\t253\t3e-132\t  468\r\n",
      "BE29\tASV39\t100.00\t253\t0\t0\t433\t685\t1\t253\t3e-132\t  468\r\n",
      "BE30\tASV39\t100.00\t253\t0\t0\t433\t685\t1\t253\t3e-132\t  468\r\n",
      "BE31\tASV81\t100.00\t253\t0\t0\t417\t669\t1\t253\t3e-132\t  468\r\n",
      "BE32\tASV81\t100.00\t253\t0\t0\t417\t669\t1\t253\t3e-132\t  468\r\n",
      "BE33\tASV3899\t100.00\t253\t0\t0\t419\t671\t1\t253\t2e-132\t  468\r\n",
      "BE34\tASV81\t100.00\t253\t0\t0\t418\t670\t1\t253\t3e-132\t  468\r\n",
      "BE35\tASV4861\t98.81\t253\t3\t0\t417\t669\t1\t253\t2e-127\t  451\r\n"
     ]
    }
   ],
   "source": [
    "!head Isolate-ASV_blast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BE26</td>\n",
       "      <td>ASV193</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>444</td>\n",
       "      <td>696</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BE27</td>\n",
       "      <td>ASV193</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>450</td>\n",
       "      <td>702</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BE28</td>\n",
       "      <td>ASV193</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>452</td>\n",
       "      <td>704</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BE29</td>\n",
       "      <td>ASV39</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>433</td>\n",
       "      <td>685</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BE30</td>\n",
       "      <td>ASV39</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>433</td>\n",
       "      <td>685</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  qseqid  sseqid  pident  length  mismatch  gapopen  qstart  qend  sstart  \\\n",
       "0   BE26  ASV193   100.0     253         0        0     444   696       1   \n",
       "1   BE27  ASV193   100.0     253         0        0     450   702       1   \n",
       "2   BE28  ASV193   100.0     253         0        0     452   704       1   \n",
       "3   BE29   ASV39   100.0     253         0        0     433   685       1   \n",
       "4   BE30   ASV39   100.0     253         0        0     433   685       1   \n",
       "\n",
       "   send         evalue  bitscore  \n",
       "0   253  3.000000e-132       468  \n",
       "1   253  3.000000e-132       468  \n",
       "2   253  3.000000e-132       468  \n",
       "3   253  3.000000e-132       468  \n",
       "4   253  3.000000e-132       468  "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pandas.read_table('Isolate-ASV_blast', sep = \"\\t\", header=None)\n",
    "df.columns = ['qseqid','sseqid',\n",
    "'pident',\n",
    "'length',\n",
    "'mismatch',\n",
    "'gapopen',\n",
    "'qstart',\n",
    "'qend',\n",
    "'sstart',\n",
    "'send',\n",
    "'evalue',\n",
    "'bitscore']\n",
    "df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BE26</td>\n",
       "      <td>ASV193</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>444</td>\n",
       "      <td>696</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BE27</td>\n",
       "      <td>ASV193</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>450</td>\n",
       "      <td>702</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BE28</td>\n",
       "      <td>ASV193</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>452</td>\n",
       "      <td>704</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BE29</td>\n",
       "      <td>ASV39</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>433</td>\n",
       "      <td>685</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BE30</td>\n",
       "      <td>ASV39</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>433</td>\n",
       "      <td>685</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>BE31</td>\n",
       "      <td>ASV81</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>417</td>\n",
       "      <td>669</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>BE32</td>\n",
       "      <td>ASV81</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>417</td>\n",
       "      <td>669</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>BE33</td>\n",
       "      <td>ASV3899</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>419</td>\n",
       "      <td>671</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>2.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>BE34</td>\n",
       "      <td>ASV81</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>418</td>\n",
       "      <td>670</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>BE35</td>\n",
       "      <td>ASV4861</td>\n",
       "      <td>98.81</td>\n",
       "      <td>253</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>417</td>\n",
       "      <td>669</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>2.000000e-127</td>\n",
       "      <td>451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BE36</td>\n",
       "      <td>ASV5271</td>\n",
       "      <td>95.63</td>\n",
       "      <td>252</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>438</td>\n",
       "      <td>689</td>\n",
       "      <td>1</td>\n",
       "      <td>252</td>\n",
       "      <td>2.000000e-113</td>\n",
       "      <td>405</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>BE38</td>\n",
       "      <td>ASV91</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>436</td>\n",
       "      <td>688</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>BE39</td>\n",
       "      <td>ASV4861</td>\n",
       "      <td>98.81</td>\n",
       "      <td>253</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>419</td>\n",
       "      <td>671</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-127</td>\n",
       "      <td>451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>BE40</td>\n",
       "      <td>ASV1915</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>433</td>\n",
       "      <td>685</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>BE41</td>\n",
       "      <td>ASV188</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>411</td>\n",
       "      <td>663</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>BE42</td>\n",
       "      <td>ASV420</td>\n",
       "      <td>99.60</td>\n",
       "      <td>253</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>436</td>\n",
       "      <td>688</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-131</td>\n",
       "      <td>464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>BE43</td>\n",
       "      <td>ASV193</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>450</td>\n",
       "      <td>702</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>BE44</td>\n",
       "      <td>ASV5115</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>436</td>\n",
       "      <td>688</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>BE45</td>\n",
       "      <td>ASV193</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>448</td>\n",
       "      <td>700</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>BE46</td>\n",
       "      <td>ASV393</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>435</td>\n",
       "      <td>687</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>BE47</td>\n",
       "      <td>ASV1106</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>434</td>\n",
       "      <td>686</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>BE48</td>\n",
       "      <td>ASV452</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>442</td>\n",
       "      <td>694</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>BE49</td>\n",
       "      <td>ASV765</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>383</td>\n",
       "      <td>635</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>2.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>BE50</td>\n",
       "      <td>ASV4621</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>455</td>\n",
       "      <td>707</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>BE51</td>\n",
       "      <td>ASV430</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>416</td>\n",
       "      <td>668</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>2.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>BE52</td>\n",
       "      <td>ASV12</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>433</td>\n",
       "      <td>685</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>BE53</td>\n",
       "      <td>ASV1592</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>380</td>\n",
       "      <td>632</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>2.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>BE54</td>\n",
       "      <td>ASV188</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>409</td>\n",
       "      <td>661</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>BE55</td>\n",
       "      <td>ASV393</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>434</td>\n",
       "      <td>686</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>BE56</td>\n",
       "      <td>ASV1350</td>\n",
       "      <td>98.42</td>\n",
       "      <td>253</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>433</td>\n",
       "      <td>685</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>1.000000e-125</td>\n",
       "      <td>446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>134</th>\n",
       "      <td>BE228</td>\n",
       "      <td>ASV452</td>\n",
       "      <td>95.65</td>\n",
       "      <td>253</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>443</td>\n",
       "      <td>695</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>6.000000e-114</td>\n",
       "      <td>407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>BE231</td>\n",
       "      <td>ASV4308</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>448</td>\n",
       "      <td>700</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>BE232</td>\n",
       "      <td>ASV283</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>449</td>\n",
       "      <td>701</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>BE233</td>\n",
       "      <td>ASV39</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>432</td>\n",
       "      <td>684</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>BE234</td>\n",
       "      <td>ASV452</td>\n",
       "      <td>95.65</td>\n",
       "      <td>253</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>443</td>\n",
       "      <td>695</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>6.000000e-114</td>\n",
       "      <td>407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>BE235</td>\n",
       "      <td>ASV687</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>385</td>\n",
       "      <td>637</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>2.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>BE236</td>\n",
       "      <td>ASV4308</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>446</td>\n",
       "      <td>698</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>BE239</td>\n",
       "      <td>ASV90</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>383</td>\n",
       "      <td>635</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>2.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>BE240</td>\n",
       "      <td>ASV477</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>433</td>\n",
       "      <td>685</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>BE241</td>\n",
       "      <td>ASV171</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>437</td>\n",
       "      <td>689</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>BE242</td>\n",
       "      <td>ASV81</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>420</td>\n",
       "      <td>672</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>BE243</td>\n",
       "      <td>ASV1356</td>\n",
       "      <td>99.60</td>\n",
       "      <td>253</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>383</td>\n",
       "      <td>635</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>1.000000e-130</td>\n",
       "      <td>462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>BE248</td>\n",
       "      <td>ASV266</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>414</td>\n",
       "      <td>666</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>BE249</td>\n",
       "      <td>ASV687</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>383</td>\n",
       "      <td>635</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>2.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>148</th>\n",
       "      <td>BE251</td>\n",
       "      <td>ASV1356</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>380</td>\n",
       "      <td>632</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>2.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>BE254</td>\n",
       "      <td>ASV1030</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>381</td>\n",
       "      <td>633</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>2.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>BE255</td>\n",
       "      <td>ASV1030</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>384</td>\n",
       "      <td>636</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>2.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>BE256</td>\n",
       "      <td>ASV2669</td>\n",
       "      <td>99.60</td>\n",
       "      <td>253</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>381</td>\n",
       "      <td>633</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>1.000000e-130</td>\n",
       "      <td>462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>BE257</td>\n",
       "      <td>ASV90</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>380</td>\n",
       "      <td>632</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>1.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>BE258</td>\n",
       "      <td>ASV1356</td>\n",
       "      <td>99.60</td>\n",
       "      <td>253</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>379</td>\n",
       "      <td>631</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>1.000000e-130</td>\n",
       "      <td>462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154</th>\n",
       "      <td>BE259</td>\n",
       "      <td>ASV687</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>383</td>\n",
       "      <td>635</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>2.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>BE260</td>\n",
       "      <td>ASV1601</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>429</td>\n",
       "      <td>681</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>156</th>\n",
       "      <td>BE261</td>\n",
       "      <td>ASV19</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>412</td>\n",
       "      <td>664</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>BE264</td>\n",
       "      <td>ASV23</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>386</td>\n",
       "      <td>638</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>2.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>158</th>\n",
       "      <td>BE265</td>\n",
       "      <td>ASV360</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>433</td>\n",
       "      <td>685</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>BE268</td>\n",
       "      <td>ASV236</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>385</td>\n",
       "      <td>637</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>2.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>BE270</td>\n",
       "      <td>ASV150</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>383</td>\n",
       "      <td>635</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>1.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>BE275</td>\n",
       "      <td>ASV452</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>443</td>\n",
       "      <td>695</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>BE276</td>\n",
       "      <td>ASV452</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>441</td>\n",
       "      <td>693</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>BE277</td>\n",
       "      <td>ASV452</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>441</td>\n",
       "      <td>693</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>164 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    qseqid   sseqid  pident  length  mismatch  gapopen  qstart  qend  sstart  \\\n",
       "0     BE26   ASV193  100.00     253         0        0     444   696       1   \n",
       "1     BE27   ASV193  100.00     253         0        0     450   702       1   \n",
       "2     BE28   ASV193  100.00     253         0        0     452   704       1   \n",
       "3     BE29    ASV39  100.00     253         0        0     433   685       1   \n",
       "4     BE30    ASV39  100.00     253         0        0     433   685       1   \n",
       "5     BE31    ASV81  100.00     253         0        0     417   669       1   \n",
       "6     BE32    ASV81  100.00     253         0        0     417   669       1   \n",
       "7     BE33  ASV3899  100.00     253         0        0     419   671       1   \n",
       "8     BE34    ASV81  100.00     253         0        0     418   670       1   \n",
       "9     BE35  ASV4861   98.81     253         3        0     417   669       1   \n",
       "10    BE36  ASV5271   95.63     252        11        0     438   689       1   \n",
       "11    BE38    ASV91  100.00     253         0        0     436   688       1   \n",
       "12    BE39  ASV4861   98.81     253         3        0     419   671       1   \n",
       "13    BE40  ASV1915  100.00     253         0        0     433   685       1   \n",
       "14    BE41   ASV188  100.00     253         0        0     411   663       1   \n",
       "15    BE42   ASV420   99.60     253         1        0     436   688       1   \n",
       "16    BE43   ASV193  100.00     253         0        0     450   702       1   \n",
       "17    BE44  ASV5115  100.00     253         0        0     436   688       1   \n",
       "18    BE45   ASV193  100.00     253         0        0     448   700       1   \n",
       "19    BE46   ASV393  100.00     253         0        0     435   687       1   \n",
       "20    BE47  ASV1106  100.00     253         0        0     434   686       1   \n",
       "21    BE48   ASV452  100.00     253         0        0     442   694       1   \n",
       "22    BE49   ASV765  100.00     253         0        0     383   635       1   \n",
       "23    BE50  ASV4621  100.00     253         0        0     455   707       1   \n",
       "24    BE51   ASV430  100.00     253         0        0     416   668       1   \n",
       "25    BE52    ASV12  100.00     253         0        0     433   685       1   \n",
       "26    BE53  ASV1592  100.00     253         0        0     380   632       1   \n",
       "27    BE54   ASV188  100.00     253         0        0     409   661       1   \n",
       "28    BE55   ASV393  100.00     253         0        0     434   686       1   \n",
       "29    BE56  ASV1350   98.42     253         4        0     433   685       1   \n",
       "..     ...      ...     ...     ...       ...      ...     ...   ...     ...   \n",
       "134  BE228   ASV452   95.65     253        11        0     443   695       1   \n",
       "135  BE231  ASV4308  100.00     253         0        0     448   700       1   \n",
       "136  BE232   ASV283  100.00     253         0        0     449   701       1   \n",
       "137  BE233    ASV39  100.00     253         0        0     432   684       1   \n",
       "138  BE234   ASV452   95.65     253        11        0     443   695       1   \n",
       "139  BE235   ASV687  100.00     253         0        0     385   637       1   \n",
       "140  BE236  ASV4308  100.00     253         0        0     446   698       1   \n",
       "141  BE239    ASV90  100.00     253         0        0     383   635       1   \n",
       "142  BE240   ASV477  100.00     253         0        0     433   685       1   \n",
       "143  BE241   ASV171  100.00     253         0        0     437   689       1   \n",
       "144  BE242    ASV81  100.00     253         0        0     420   672       1   \n",
       "145  BE243  ASV1356   99.60     253         1        0     383   635       1   \n",
       "146  BE248   ASV266  100.00     253         0        0     414   666       1   \n",
       "147  BE249   ASV687  100.00     253         0        0     383   635       1   \n",
       "148  BE251  ASV1356  100.00     253         0        0     380   632       1   \n",
       "149  BE254  ASV1030  100.00     253         0        0     381   633       1   \n",
       "150  BE255  ASV1030  100.00     253         0        0     384   636       1   \n",
       "151  BE256  ASV2669   99.60     253         1        0     381   633       1   \n",
       "152  BE257    ASV90  100.00     253         0        0     380   632       1   \n",
       "153  BE258  ASV1356   99.60     253         1        0     379   631       1   \n",
       "154  BE259   ASV687  100.00     253         0        0     383   635       1   \n",
       "155  BE260  ASV1601  100.00     253         0        0     429   681       1   \n",
       "156  BE261    ASV19  100.00     253         0        0     412   664       1   \n",
       "157  BE264    ASV23  100.00     253         0        0     386   638       1   \n",
       "158  BE265   ASV360  100.00     253         0        0     433   685       1   \n",
       "159  BE268   ASV236  100.00     253         0        0     385   637       1   \n",
       "160  BE270   ASV150  100.00     253         0        0     383   635       1   \n",
       "161  BE275   ASV452  100.00     253         0        0     443   695       1   \n",
       "162  BE276   ASV452  100.00     253         0        0     441   693       1   \n",
       "163  BE277   ASV452  100.00     253         0        0     441   693       1   \n",
       "\n",
       "     send         evalue  bitscore  \n",
       "0     253  3.000000e-132       468  \n",
       "1     253  3.000000e-132       468  \n",
       "2     253  3.000000e-132       468  \n",
       "3     253  3.000000e-132       468  \n",
       "4     253  3.000000e-132       468  \n",
       "5     253  3.000000e-132       468  \n",
       "6     253  3.000000e-132       468  \n",
       "7     253  2.000000e-132       468  \n",
       "8     253  3.000000e-132       468  \n",
       "9     253  2.000000e-127       451  \n",
       "10    252  2.000000e-113       405  \n",
       "11    253  3.000000e-132       468  \n",
       "12    253  3.000000e-127       451  \n",
       "13    253  3.000000e-132       468  \n",
       "14    253  3.000000e-132       468  \n",
       "15    253  3.000000e-131       464  \n",
       "16    253  3.000000e-132       468  \n",
       "17    253  3.000000e-132       468  \n",
       "18    253  3.000000e-132       468  \n",
       "19    253  3.000000e-132       468  \n",
       "20    253  3.000000e-132       468  \n",
       "21    253  3.000000e-132       468  \n",
       "22    253  2.000000e-132       468  \n",
       "23    253  3.000000e-132       468  \n",
       "24    253  2.000000e-132       468  \n",
       "25    253  3.000000e-132       468  \n",
       "26    253  2.000000e-132       468  \n",
       "27    253  3.000000e-132       468  \n",
       "28    253  3.000000e-132       468  \n",
       "29    253  1.000000e-125       446  \n",
       "..    ...            ...       ...  \n",
       "134   253  6.000000e-114       407  \n",
       "135   253  3.000000e-132       468  \n",
       "136   253  3.000000e-132       468  \n",
       "137   253  3.000000e-132       468  \n",
       "138   253  6.000000e-114       407  \n",
       "139   253  2.000000e-132       468  \n",
       "140   253  3.000000e-132       468  \n",
       "141   253  2.000000e-132       468  \n",
       "142   253  3.000000e-132       468  \n",
       "143   253  3.000000e-132       468  \n",
       "144   253  3.000000e-132       468  \n",
       "145   253  1.000000e-130       462  \n",
       "146   253  3.000000e-132       468  \n",
       "147   253  2.000000e-132       468  \n",
       "148   253  2.000000e-132       468  \n",
       "149   253  2.000000e-132       468  \n",
       "150   253  2.000000e-132       468  \n",
       "151   253  1.000000e-130       462  \n",
       "152   253  1.000000e-132       468  \n",
       "153   253  1.000000e-130       462  \n",
       "154   253  2.000000e-132       468  \n",
       "155   253  3.000000e-132       468  \n",
       "156   253  3.000000e-132       468  \n",
       "157   253  2.000000e-132       468  \n",
       "158   253  3.000000e-132       468  \n",
       "159   253  2.000000e-132       468  \n",
       "160   253  1.000000e-132       468  \n",
       "161   253  3.000000e-132       468  \n",
       "162   253  3.000000e-132       468  \n",
       "163   253  3.000000e-132       468  \n",
       "\n",
       "[164 rows x 12 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# function to get unique values \n",
    "def unique(list1): \n",
    "    x = np.array(list1) \n",
    "    return(np.unique(x)) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ASV1030',\n",
       " 'ASV1106',\n",
       " 'ASV1114',\n",
       " 'ASV12',\n",
       " 'ASV13',\n",
       " 'ASV1350',\n",
       " 'ASV1356',\n",
       " 'ASV1379',\n",
       " 'ASV150',\n",
       " 'ASV151',\n",
       " 'ASV1592',\n",
       " 'ASV1601',\n",
       " 'ASV163',\n",
       " 'ASV171',\n",
       " 'ASV1811',\n",
       " 'ASV188',\n",
       " 'ASV19',\n",
       " 'ASV1915',\n",
       " 'ASV193',\n",
       " 'ASV2066',\n",
       " 'ASV2116',\n",
       " 'ASV2203',\n",
       " 'ASV2204',\n",
       " 'ASV23',\n",
       " 'ASV233',\n",
       " 'ASV236',\n",
       " 'ASV266',\n",
       " 'ASV2669',\n",
       " 'ASV270',\n",
       " 'ASV2719',\n",
       " 'ASV283',\n",
       " 'ASV2888',\n",
       " 'ASV296',\n",
       " 'ASV302',\n",
       " 'ASV321',\n",
       " 'ASV3270',\n",
       " 'ASV350',\n",
       " 'ASV3528',\n",
       " 'ASV3592',\n",
       " 'ASV360',\n",
       " 'ASV362',\n",
       " 'ASV3772',\n",
       " 'ASV3899',\n",
       " 'ASV39',\n",
       " 'ASV393',\n",
       " 'ASV420',\n",
       " 'ASV430',\n",
       " 'ASV4308',\n",
       " 'ASV452',\n",
       " 'ASV4604',\n",
       " 'ASV4621',\n",
       " 'ASV477',\n",
       " 'ASV481',\n",
       " 'ASV4861',\n",
       " 'ASV492',\n",
       " 'ASV5115',\n",
       " 'ASV522',\n",
       " 'ASV5271',\n",
       " 'ASV5417',\n",
       " 'ASV5572',\n",
       " 'ASV5578',\n",
       " 'ASV58',\n",
       " 'ASV600',\n",
       " 'ASV6114',\n",
       " 'ASV617',\n",
       " 'ASV665',\n",
       " 'ASV687',\n",
       " 'ASV7471',\n",
       " 'ASV765',\n",
       " 'ASV77',\n",
       " 'ASV792',\n",
       " 'ASV8',\n",
       " 'ASV81',\n",
       " 'ASV891',\n",
       " 'ASV90',\n",
       " 'ASV91']"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "IsolateMatches_list = df[\"sseqid\"].tolist()\n",
    "IsolateMatches_list = list(unique(IsolateMatches_list))\n",
    "IsolateMatches_list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Identify hyphal ASVs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Too many RH ASVs to include\n",
    "* Don't include for now, but use attributes to highlight any ASVs that make it into the tree otherwise"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>lfcSE</th>\n",
       "      <th>stat</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "      <th>ASV</th>\n",
       "      <th>Test</th>\n",
       "      <th>TimePoint</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASV11489</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.162168</td>\n",
       "      <td>1.455070</td>\n",
       "      <td>3.093253</td>\n",
       "      <td>0.470401</td>\n",
       "      <td>0.638068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASV11644</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.913138</td>\n",
       "      <td>-3.296519</td>\n",
       "      <td>3.034100</td>\n",
       "      <td>-1.086490</td>\n",
       "      <td>0.277262</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASV3656</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15.109037</td>\n",
       "      <td>-0.425264</td>\n",
       "      <td>2.894952</td>\n",
       "      <td>-0.146898</td>\n",
       "      <td>0.883212</td>\n",
       "      <td>0.972331</td>\n",
       "      <td>ASV562</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASV2718</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    baseMean  log2FoldChange     lfcSE      stat    pvalue      padj  \\\n",
       "1   0.000000             NaN       NaN       NaN       NaN       NaN   \n",
       "2   0.162168        1.455070  3.093253  0.470401  0.638068       NaN   \n",
       "3   0.913138       -3.296519  3.034100 -1.086490  0.277262       NaN   \n",
       "4  15.109037       -0.425264  2.894952 -0.146898  0.883212  0.972331   \n",
       "5   0.000000             NaN       NaN       NaN       NaN       NaN   \n",
       "\n",
       "        ASV   Test  TimePoint  \n",
       "1  ASV11489  CH-BS          1  \n",
       "2  ASV11644  CH-BS          1  \n",
       "3   ASV3656  CH-BS          1  \n",
       "4    ASV562  CH-BS          1  \n",
       "5   ASV2718  CH-BS          1  "
      ]
     },
     "execution_count": 206,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfl2fc_CHCS = pandas.read_table('/home/be68/Hyphosphere/Exp10/DESeq2/dfl2fc-CHCS.txt', sep = \"\\t\")\n",
    "dfl2fc_CHBS = pandas.read_table('/home/be68/Hyphosphere/Exp10/DESeq2/dfl2fc-CHBS.txt', sep = \"\\t\")\n",
    "dfl2fc_RHCS = pandas.read_table('/home/be68/Hyphosphere/Exp10/DESeq2/dfl2fc-RHCS.txt', sep = \"\\t\")\n",
    "dfl2fc = dfl2fc_CHBS.append(dfl2fc_CHCS)\n",
    "dfl2fc = dfl2fc.append(dfl2fc_RHCS)\n",
    "\n",
    "sigtab = dfl2fc_CHBS.append(dfl2fc_CHCS)\n",
    "sigtab.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "200"
      ]
     },
     "execution_count": 207,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "sigtab = sigtab.loc[(sigtab.padj < 0.05) & (sigtab.log2FoldChange > 0)]\n",
    "\n",
    "HA = sigtab.ASV.unique()\n",
    "HA = list(HA)\n",
    "#HA.sort()\n",
    "len(HA)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Identify top 100 ASVs for context"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OTU</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>BP</th>\n",
       "      <th>BS</th>\n",
       "      <th>CH</th>\n",
       "      <th>CS</th>\n",
       "      <th>RH</th>\n",
       "      <th>RT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ASV100</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Xanthobacteraceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003115</td>\n",
       "      <td>0.004161</td>\n",
       "      <td>0.002490</td>\n",
       "      <td>0.003742</td>\n",
       "      <td>0.000230</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASV1000</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>Verrucomicrobiae</td>\n",
       "      <td>Chthoniobacteraceae</td>\n",
       "      <td>Candidatus_Udaeobacter</td>\n",
       "      <td>0.000364</td>\n",
       "      <td>0.000745</td>\n",
       "      <td>0.000158</td>\n",
       "      <td>0.000268</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASV1002</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Microbacteriaceae</td>\n",
       "      <td>Agromyces</td>\n",
       "      <td>0.000508</td>\n",
       "      <td>0.000406</td>\n",
       "      <td>0.000361</td>\n",
       "      <td>0.000407</td>\n",
       "      <td>0.000142</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASV1003</td>\n",
       "      <td>Gemmatimonadetes</td>\n",
       "      <td>Gemmatimonadetes</td>\n",
       "      <td>Gemmatimonadaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000628</td>\n",
       "      <td>0.000274</td>\n",
       "      <td>0.000087</td>\n",
       "      <td>0.000534</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ASV10035</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Thermoleophilia</td>\n",
       "      <td>Solirubrobacteraceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>0.000013</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        OTU            Phylum                Class                Family  \\\n",
       "1    ASV100    Proteobacteria  Alphaproteobacteria     Xanthobacteraceae   \n",
       "2   ASV1000   Verrucomicrobia     Verrucomicrobiae   Chthoniobacteraceae   \n",
       "3   ASV1002    Actinobacteria       Actinobacteria     Microbacteriaceae   \n",
       "4   ASV1003  Gemmatimonadetes     Gemmatimonadetes     Gemmatimonadaceae   \n",
       "5  ASV10035    Actinobacteria      Thermoleophilia  Solirubrobacteraceae   \n",
       "\n",
       "                    Genus        BP        BS        CH        CS        RH  \\\n",
       "1                     NaN  0.003115  0.004161  0.002490  0.003742  0.000230   \n",
       "2  Candidatus_Udaeobacter  0.000364  0.000745  0.000158  0.000268  0.000031   \n",
       "3               Agromyces  0.000508  0.000406  0.000361  0.000407  0.000142   \n",
       "4                     NaN  0.000628  0.000274  0.000087  0.000534  0.000046   \n",
       "5                     NaN  0.000000  0.000000  0.000000  0.000000  0.000036   \n",
       "\n",
       "         RT  \n",
       "1  0.000000  \n",
       "2  0.000000  \n",
       "3  0.000000  \n",
       "4  0.000000  \n",
       "5  0.000013  "
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ASVabund = pandas.read_table('ASV_meanAbundances.txt', sep = \"\\t\")\n",
    "ASVabund.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OTU</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>BP</th>\n",
       "      <th>BS</th>\n",
       "      <th>CH</th>\n",
       "      <th>CS</th>\n",
       "      <th>RH</th>\n",
       "      <th>RT</th>\n",
       "      <th>meanAbund</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ASV100</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Xanthobacteraceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003115</td>\n",
       "      <td>0.004161</td>\n",
       "      <td>0.002490</td>\n",
       "      <td>0.003742</td>\n",
       "      <td>0.000230</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASV1000</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>Verrucomicrobiae</td>\n",
       "      <td>Chthoniobacteraceae</td>\n",
       "      <td>Candidatus_Udaeobacter</td>\n",
       "      <td>0.000364</td>\n",
       "      <td>0.000745</td>\n",
       "      <td>0.000158</td>\n",
       "      <td>0.000268</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASV1002</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Microbacteriaceae</td>\n",
       "      <td>Agromyces</td>\n",
       "      <td>0.000508</td>\n",
       "      <td>0.000406</td>\n",
       "      <td>0.000361</td>\n",
       "      <td>0.000407</td>\n",
       "      <td>0.000142</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASV1003</td>\n",
       "      <td>Gemmatimonadetes</td>\n",
       "      <td>Gemmatimonadetes</td>\n",
       "      <td>Gemmatimonadaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000628</td>\n",
       "      <td>0.000274</td>\n",
       "      <td>0.000087</td>\n",
       "      <td>0.000534</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ASV10035</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Thermoleophilia</td>\n",
       "      <td>Solirubrobacteraceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        OTU            Phylum                Class                Family  \\\n",
       "1    ASV100    Proteobacteria  Alphaproteobacteria     Xanthobacteraceae   \n",
       "2   ASV1000   Verrucomicrobia     Verrucomicrobiae   Chthoniobacteraceae   \n",
       "3   ASV1002    Actinobacteria       Actinobacteria     Microbacteriaceae   \n",
       "4   ASV1003  Gemmatimonadetes     Gemmatimonadetes     Gemmatimonadaceae   \n",
       "5  ASV10035    Actinobacteria      Thermoleophilia  Solirubrobacteraceae   \n",
       "\n",
       "                    Genus        BP        BS        CH        CS        RH  \\\n",
       "1                     NaN  0.003115  0.004161  0.002490  0.003742  0.000230   \n",
       "2  Candidatus_Udaeobacter  0.000364  0.000745  0.000158  0.000268  0.000031   \n",
       "3               Agromyces  0.000508  0.000406  0.000361  0.000407  0.000142   \n",
       "4                     NaN  0.000628  0.000274  0.000087  0.000534  0.000046   \n",
       "5                     NaN  0.000000  0.000000  0.000000  0.000000  0.000036   \n",
       "\n",
       "         RT  meanAbund  \n",
       "1  0.000000          0  \n",
       "2  0.000000          0  \n",
       "3  0.000000          0  \n",
       "4  0.000000          0  \n",
       "5  0.000013          0  "
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ASVabund['meanAbund'] = 0\n",
    "ASVabund.head()\n",
    "#np.mean(ASVabund[\"CS\"], ASVabund[\"CH\"]) #, ASVabund['RH'], ASVabund['CH'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ASV11',\n",
       " 'ASV17',\n",
       " 'ASV20',\n",
       " 'ASV21',\n",
       " 'ASV8',\n",
       " 'ASV19',\n",
       " 'ASV18',\n",
       " 'ASV6',\n",
       " 'ASV28',\n",
       " 'ASV22',\n",
       " 'ASV16',\n",
       " 'ASV13',\n",
       " 'ASV15',\n",
       " 'ASV12',\n",
       " 'ASV38',\n",
       " 'ASV41',\n",
       " 'ASV48',\n",
       " 'ASV42',\n",
       " 'ASV37',\n",
       " 'ASV45',\n",
       " 'ASV43',\n",
       " 'ASV99',\n",
       " 'ASV51',\n",
       " 'ASV56',\n",
       " 'ASV54',\n",
       " 'ASV26',\n",
       " 'ASV59',\n",
       " 'ASV60',\n",
       " 'ASV61',\n",
       " 'ASV68',\n",
       " 'ASV78',\n",
       " 'ASV35',\n",
       " 'ASV72',\n",
       " 'ASV80',\n",
       " 'ASV73',\n",
       " 'ASV79',\n",
       " 'ASV46',\n",
       " 'ASV89',\n",
       " 'ASV83',\n",
       " 'ASV57',\n",
       " 'ASV91',\n",
       " 'ASV23',\n",
       " 'ASV110',\n",
       " 'ASV100',\n",
       " 'ASV104',\n",
       " 'ASV113',\n",
       " 'ASV107',\n",
       " 'ASV161',\n",
       " 'ASV96',\n",
       " 'ASV74',\n",
       " 'ASV97',\n",
       " 'ASV114',\n",
       " 'ASV117',\n",
       " 'ASV71',\n",
       " 'ASV112',\n",
       " 'ASV39',\n",
       " 'ASV122',\n",
       " 'ASV123',\n",
       " 'ASV86',\n",
       " 'ASV101',\n",
       " 'ASV109',\n",
       " 'ASV140',\n",
       " 'ASV47',\n",
       " 'ASV82',\n",
       " 'ASV50',\n",
       " 'ASV111',\n",
       " 'ASV141',\n",
       " 'ASV227',\n",
       " 'ASV139',\n",
       " 'ASV169',\n",
       " 'ASV178',\n",
       " 'ASV76',\n",
       " 'ASV234',\n",
       " 'ASV129',\n",
       " 'ASV142',\n",
       " 'ASV125',\n",
       " 'ASV77',\n",
       " 'ASV115',\n",
       " 'ASV85',\n",
       " 'ASV134',\n",
       " 'ASV174',\n",
       " 'ASV189',\n",
       " 'ASV66',\n",
       " 'ASV144',\n",
       " 'ASV366',\n",
       " 'ASV152',\n",
       " 'ASV224',\n",
       " 'ASV167',\n",
       " 'ASV173',\n",
       " 'ASV162',\n",
       " 'ASV136',\n",
       " 'ASV180',\n",
       " 'ASV190',\n",
       " 'ASV44',\n",
       " 'ASV216',\n",
       " 'ASV219',\n",
       " 'ASV217',\n",
       " 'ASV143',\n",
       " 'ASV128',\n",
       " 'ASV263']"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfA = ASVabund\n",
    "dfA['sumAbund'] = dfA.BS + dfA.CS + dfA.CH + dfA.RH\n",
    "dfA = dfA.sort_values(['sumAbund'], ascending = False)[:100]\n",
    "TopASVs = list(dfA.OTU)\n",
    "TopASVs\n",
    "#df.sort(['A', 'B'], ascending=[1, 0])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Now join lists and prune tree\n",
    "* Top ASVs = TopASVs\n",
    "* Diff Abund asvs = HA\n",
    "* Isolates and top blast hits = IsolateList + IsolateMatches_list\n",
    "\n",
    "* master tree = t_rooted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['BE26',\n",
       " 'BE27',\n",
       " 'BE28',\n",
       " 'BE29',\n",
       " 'BE30',\n",
       " 'BE31',\n",
       " 'BE32',\n",
       " 'BE33',\n",
       " 'BE34',\n",
       " 'BE35',\n",
       " 'BE36',\n",
       " 'BE38',\n",
       " 'BE39',\n",
       " 'BE40',\n",
       " 'BE41',\n",
       " 'BE42',\n",
       " 'BE43',\n",
       " 'BE44',\n",
       " 'BE45',\n",
       " 'BE46',\n",
       " 'BE47',\n",
       " 'BE48',\n",
       " 'BE49',\n",
       " 'BE50',\n",
       " 'BE51',\n",
       " 'BE52',\n",
       " 'BE53',\n",
       " 'BE54',\n",
       " 'BE55',\n",
       " 'BE56',\n",
       " 'BE57',\n",
       " 'BE59',\n",
       " 'BE60',\n",
       " 'BE61',\n",
       " 'BE62',\n",
       " 'BE63',\n",
       " 'BE64',\n",
       " 'BE65',\n",
       " 'BE65B',\n",
       " 'BE66',\n",
       " 'BE68',\n",
       " 'BE69',\n",
       " 'BE70',\n",
       " 'BE71',\n",
       " 'BE72',\n",
       " 'BE74',\n",
       " 'BE75',\n",
       " 'BE76',\n",
       " 'BE79',\n",
       " 'BE80',\n",
       " 'BE81',\n",
       " 'BE82',\n",
       " 'BE83',\n",
       " 'BE84',\n",
       " 'BE85',\n",
       " 'BE88',\n",
       " 'BE91',\n",
       " 'BE92',\n",
       " 'BE93',\n",
       " 'BE94',\n",
       " 'BE95',\n",
       " 'BE96',\n",
       " 'BE97',\n",
       " 'BE98',\n",
       " 'BE100',\n",
       " 'BE101',\n",
       " 'BE102',\n",
       " 'BE103',\n",
       " 'BE104',\n",
       " 'BE105',\n",
       " 'BE107',\n",
       " 'BE108',\n",
       " 'BE109',\n",
       " 'BE110',\n",
       " 'BE112',\n",
       " 'BE113',\n",
       " 'BE114',\n",
       " 'BE115',\n",
       " 'BE116',\n",
       " 'BE117',\n",
       " 'BE118',\n",
       " 'BE120',\n",
       " 'BE121',\n",
       " 'BE122',\n",
       " 'BE123',\n",
       " 'BE124',\n",
       " 'BE125',\n",
       " 'BE126',\n",
       " 'BE127',\n",
       " 'BE128',\n",
       " 'BE153',\n",
       " 'BE154',\n",
       " 'BE156',\n",
       " 'BE157',\n",
       " 'BE159',\n",
       " 'BE163',\n",
       " 'BE164',\n",
       " 'BE166',\n",
       " 'BE167',\n",
       " 'BE168',\n",
       " 'BE170',\n",
       " 'BE171',\n",
       " 'BE172',\n",
       " 'BE173',\n",
       " 'BE177',\n",
       " 'BE178',\n",
       " 'BE179',\n",
       " 'BE180',\n",
       " 'BE181',\n",
       " 'BE182',\n",
       " 'BE183',\n",
       " 'BE184',\n",
       " 'BE186',\n",
       " 'BE187',\n",
       " 'BE190',\n",
       " 'BE191',\n",
       " 'BE193',\n",
       " 'BE194',\n",
       " 'BE195',\n",
       " 'BE196',\n",
       " 'BE197',\n",
       " 'BE198',\n",
       " 'BE201',\n",
       " 'BE203',\n",
       " 'BE206',\n",
       " 'BE207',\n",
       " 'BE212',\n",
       " 'BE213',\n",
       " 'BE215',\n",
       " 'BE216',\n",
       " 'BE219',\n",
       " 'BE220',\n",
       " 'BE222',\n",
       " 'BE223',\n",
       " 'BE224',\n",
       " 'BE228',\n",
       " 'BE231',\n",
       " 'BE232',\n",
       " 'BE233',\n",
       " 'BE234',\n",
       " 'BE235',\n",
       " 'BE236',\n",
       " 'BE239',\n",
       " 'BE240',\n",
       " 'BE241',\n",
       " 'BE242',\n",
       " 'BE243',\n",
       " 'BE248',\n",
       " 'BE249',\n",
       " 'BE251',\n",
       " 'BE254',\n",
       " 'BE255',\n",
       " 'BE256',\n",
       " 'BE257',\n",
       " 'BE258',\n",
       " 'BE259',\n",
       " 'BE260',\n",
       " 'BE261',\n",
       " 'BE264',\n",
       " 'BE265',\n",
       " 'BE268',\n",
       " 'BE270',\n",
       " 'BE275',\n",
       " 'BE276',\n",
       " 'BE277']"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "IsolateList"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "496"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MasterList = TopASVs + HA + IsolateList + IsolateMatches_list\n",
    "#len(MasterList)\n",
    "MasterList = list(set(MasterList))\n",
    "len(MasterList)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Prune tree with master list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "metadata": {},
   "outputs": [],
   "source": [
    "mt_rooted = t_rooted.getSubTree(MasterList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "metadata": {},
   "outputs": [],
   "source": [
    "mt_rooted.writeToFile(os.path.join(workDir, 'SeqofInt.tree'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Generate dataframes for iTOL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>lfcSE</th>\n",
       "      <th>stat</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "      <th>ASV</th>\n",
       "      <th>Test</th>\n",
       "      <th>TimePoint</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASV11489</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.162168</td>\n",
       "      <td>1.455070</td>\n",
       "      <td>3.093253</td>\n",
       "      <td>0.470401</td>\n",
       "      <td>0.638068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASV11644</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.913138</td>\n",
       "      <td>-3.296519</td>\n",
       "      <td>3.034100</td>\n",
       "      <td>-1.086490</td>\n",
       "      <td>0.277262</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASV3656</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15.109037</td>\n",
       "      <td>-0.425264</td>\n",
       "      <td>2.894952</td>\n",
       "      <td>-0.146898</td>\n",
       "      <td>0.883212</td>\n",
       "      <td>0.972331</td>\n",
       "      <td>ASV562</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASV2718</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    baseMean  log2FoldChange     lfcSE      stat    pvalue      padj  \\\n",
       "1   0.000000             NaN       NaN       NaN       NaN       NaN   \n",
       "2   0.162168        1.455070  3.093253  0.470401  0.638068       NaN   \n",
       "3   0.913138       -3.296519  3.034100 -1.086490  0.277262       NaN   \n",
       "4  15.109037       -0.425264  2.894952 -0.146898  0.883212  0.972331   \n",
       "5   0.000000             NaN       NaN       NaN       NaN       NaN   \n",
       "\n",
       "        ASV   Test  TimePoint  \n",
       "1  ASV11489  CH-BS          1  \n",
       "2  ASV11644  CH-BS          1  \n",
       "3   ASV3656  CH-BS          1  \n",
       "4    ASV562  CH-BS          1  \n",
       "5   ASV2718  CH-BS          1  "
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfl2fc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OTU</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>BP</th>\n",
       "      <th>BS</th>\n",
       "      <th>CH</th>\n",
       "      <th>CS</th>\n",
       "      <th>RH</th>\n",
       "      <th>RT</th>\n",
       "      <th>meanAbund</th>\n",
       "      <th>sumAbund</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ASV100</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Xanthobacteraceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003115</td>\n",
       "      <td>0.004161</td>\n",
       "      <td>0.002490</td>\n",
       "      <td>0.003742</td>\n",
       "      <td>0.000230</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007276</td>\n",
       "      <td>0.010624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASV1000</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>Verrucomicrobiae</td>\n",
       "      <td>Chthoniobacteraceae</td>\n",
       "      <td>Candidatus_Udaeobacter</td>\n",
       "      <td>0.000364</td>\n",
       "      <td>0.000745</td>\n",
       "      <td>0.000158</td>\n",
       "      <td>0.000268</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001109</td>\n",
       "      <td>0.001203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASV1002</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Microbacteriaceae</td>\n",
       "      <td>Agromyces</td>\n",
       "      <td>0.000508</td>\n",
       "      <td>0.000406</td>\n",
       "      <td>0.000361</td>\n",
       "      <td>0.000407</td>\n",
       "      <td>0.000142</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000914</td>\n",
       "      <td>0.001317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASV1003</td>\n",
       "      <td>Gemmatimonadetes</td>\n",
       "      <td>Gemmatimonadetes</td>\n",
       "      <td>Gemmatimonadaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000628</td>\n",
       "      <td>0.000274</td>\n",
       "      <td>0.000087</td>\n",
       "      <td>0.000534</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000902</td>\n",
       "      <td>0.000940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ASV10035</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Thermoleophilia</td>\n",
       "      <td>Solirubrobacteraceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000036</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        OTU            Phylum                Class                Family  \\\n",
       "1    ASV100    Proteobacteria  Alphaproteobacteria     Xanthobacteraceae   \n",
       "2   ASV1000   Verrucomicrobia     Verrucomicrobiae   Chthoniobacteraceae   \n",
       "3   ASV1002    Actinobacteria       Actinobacteria     Microbacteriaceae   \n",
       "4   ASV1003  Gemmatimonadetes     Gemmatimonadetes     Gemmatimonadaceae   \n",
       "5  ASV10035    Actinobacteria      Thermoleophilia  Solirubrobacteraceae   \n",
       "\n",
       "                    Genus        BP        BS        CH        CS        RH  \\\n",
       "1                     NaN  0.003115  0.004161  0.002490  0.003742  0.000230   \n",
       "2  Candidatus_Udaeobacter  0.000364  0.000745  0.000158  0.000268  0.000031   \n",
       "3               Agromyces  0.000508  0.000406  0.000361  0.000407  0.000142   \n",
       "4                     NaN  0.000628  0.000274  0.000087  0.000534  0.000046   \n",
       "5                     NaN  0.000000  0.000000  0.000000  0.000000  0.000036   \n",
       "\n",
       "         RT  meanAbund  sumAbund  \n",
       "1  0.000000   0.007276  0.010624  \n",
       "2  0.000000   0.001109  0.001203  \n",
       "3  0.000000   0.000914  0.001317  \n",
       "4  0.000000   0.000902  0.000940  \n",
       "5  0.000013   0.000000  0.000036  "
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ASVabund.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Process dataframes\n",
    "* Create a dataframe with names from MasterList\n",
    "* Left join abundance dataframe to sequences in Master List\n",
    "* generate dfl2fc indicators annd add attributes to dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ASV</th>\n",
       "      <th>OTU</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>BP</th>\n",
       "      <th>BS</th>\n",
       "      <th>CH</th>\n",
       "      <th>CS</th>\n",
       "      <th>RH</th>\n",
       "      <th>RT</th>\n",
       "      <th>meanAbund</th>\n",
       "      <th>sumAbund</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASV2719</td>\n",
       "      <td>ASV2719</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiaceae</td>\n",
       "      <td>Aminobacter</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.000155</td>\n",
       "      <td>0.000096</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.000193</td>\n",
       "      <td>0.000453</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BE42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASV2717</td>\n",
       "      <td>ASV2717</td>\n",
       "      <td>Armatimonadetes</td>\n",
       "      <td>Fimbriimonadia</td>\n",
       "      <td>Fimbriimonadaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000154</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000040</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000184</td>\n",
       "      <td>0.000286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BE41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BE47</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       ASV      OTU           Phylum                Class             Family  \\\n",
       "0  ASV2719  ASV2719   Proteobacteria  Alphaproteobacteria       Rhizobiaceae   \n",
       "1     BE42      NaN              NaN                  NaN                NaN   \n",
       "2  ASV2717  ASV2717  Armatimonadetes       Fimbriimonadia  Fimbriimonadaceae   \n",
       "3     BE41      NaN              NaN                  NaN                NaN   \n",
       "4     BE47      NaN              NaN                  NaN                NaN   \n",
       "\n",
       "         Genus        BP        BS        CH        CS        RH        RT  \\\n",
       "0  Aminobacter  0.000038  0.000155  0.000096  0.000025  0.000177  0.000013   \n",
       "1          NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "2          NaN  0.000154  0.000030  0.000187  0.000040  0.000029  0.000011   \n",
       "3          NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "4          NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "\n",
       "   meanAbund  sumAbund  \n",
       "0   0.000193  0.000453  \n",
       "1        NaN       NaN  \n",
       "2   0.000184  0.000286  \n",
       "3        NaN       NaN  \n",
       "4        NaN       NaN  "
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfM = pandas.DataFrame(MasterList, columns = [\"ASV\"])\n",
    "dfM = pandas.merge(dfM, ASVabund, left_on='ASV', right_on=\"OTU\", how='left')\n",
    "dfM.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Test</th>\n",
       "      <th>CH-BS</th>\n",
       "      <th>CH-CS</th>\n",
       "      <th>RH-CS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV101</th>\n",
       "      <td>2.995563e-02</td>\n",
       "      <td>0.034742</td>\n",
       "      <td>8.431885e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV1018</th>\n",
       "      <td>1.699569e-05</td>\n",
       "      <td>0.000863</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV1019</th>\n",
       "      <td>2.721585e-13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV102</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.708662e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV1022</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.102364e-03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Test            CH-BS     CH-CS         RH-CS\n",
       "ASV                                          \n",
       "ASV101   2.995563e-02  0.034742  8.431885e-11\n",
       "ASV1018  1.699569e-05  0.000863           NaN\n",
       "ASV1019  2.721585e-13       NaN           NaN\n",
       "ASV102            NaN       NaN  2.708662e-03\n",
       "ASV1022           NaN       NaN  1.102364e-03"
      ]
     },
     "execution_count": 229,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sigtab = dfl2fc.loc[(dfl2fc.padj < 0.05) & (dfl2fc.log2FoldChange > 0)]\n",
    "sigtab = sigtab.groupby(['ASV', 'Test'], as_index=False).agg({\"padj\": \"min\"})\n",
    "sigtab = sigtab.pivot(index = 'ASV', columns = \"Test\", values = \"padj\")\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "sigtab.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Test</th>\n",
       "      <th>CH-BS</th>\n",
       "      <th>CH-CS</th>\n",
       "      <th>RH-CS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV101</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV1018</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV1019</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV102</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV1022</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Test     CH-BS  CH-CS  RH-CS\n",
       "ASV                         \n",
       "ASV101     1.0    1.0    1.0\n",
       "ASV1018    1.0    1.0    NaN\n",
       "ASV1019    1.0    NaN    NaN\n",
       "ASV102     NaN    NaN    1.0\n",
       "ASV1022    NaN    NaN    1.0"
      ]
     },
     "execution_count": 232,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Note that this could be a good place to insert color codes into iTOL formated sheet\n",
    "mask = sigtab['CH-BS'] < 0.05\n",
    "column_name = 'CH-BS'\n",
    "sigtab.loc[mask, column_name] = 1\n",
    "\n",
    "mask = sigtab['CH-CS'] < 0.05\n",
    "column_name = 'CH-CS'\n",
    "sigtab.loc[mask, column_name] = 1\n",
    "\n",
    "mask = sigtab['RH-CS'] < 0.05\n",
    "column_name = 'RH-CS'\n",
    "sigtab.loc[mask, column_name] = 1\n",
    "\n",
    "sigtab.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ASV</th>\n",
       "      <th>OTU</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>BP</th>\n",
       "      <th>BS</th>\n",
       "      <th>CH</th>\n",
       "      <th>CS</th>\n",
       "      <th>RH</th>\n",
       "      <th>RT</th>\n",
       "      <th>meanAbund</th>\n",
       "      <th>sumAbund</th>\n",
       "      <th>CH-BS</th>\n",
       "      <th>CH-CS</th>\n",
       "      <th>RH-CS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASV2719</td>\n",
       "      <td>ASV2719</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiaceae</td>\n",
       "      <td>Aminobacter</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.000155</td>\n",
       "      <td>0.000096</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.000193</td>\n",
       "      <td>0.000453</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BE42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASV2717</td>\n",
       "      <td>ASV2717</td>\n",
       "      <td>Armatimonadetes</td>\n",
       "      <td>Fimbriimonadia</td>\n",
       "      <td>Fimbriimonadaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000154</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000040</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000184</td>\n",
       "      <td>0.000286</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BE41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BE47</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       ASV      OTU           Phylum                Class             Family  \\\n",
       "0  ASV2719  ASV2719   Proteobacteria  Alphaproteobacteria       Rhizobiaceae   \n",
       "1     BE42      NaN              NaN                  NaN                NaN   \n",
       "2  ASV2717  ASV2717  Armatimonadetes       Fimbriimonadia  Fimbriimonadaceae   \n",
       "3     BE41      NaN              NaN                  NaN                NaN   \n",
       "4     BE47      NaN              NaN                  NaN                NaN   \n",
       "\n",
       "         Genus        BP        BS        CH        CS        RH        RT  \\\n",
       "0  Aminobacter  0.000038  0.000155  0.000096  0.000025  0.000177  0.000013   \n",
       "1          NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "2          NaN  0.000154  0.000030  0.000187  0.000040  0.000029  0.000011   \n",
       "3          NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "4          NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "\n",
       "   meanAbund  sumAbund  CH-BS  CH-CS  RH-CS  \n",
       "0   0.000193  0.000453    NaN    NaN    NaN  \n",
       "1        NaN       NaN    NaN    NaN    NaN  \n",
       "2   0.000184  0.000286    1.0    NaN    NaN  \n",
       "3        NaN       NaN    NaN    NaN    NaN  \n",
       "4        NaN       NaN    NaN    NaN    NaN  "
      ]
     },
     "execution_count": 234,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfM = pandas.merge(dfM, sigtab, on = 'ASV', how='left')\n",
    "dfM.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### overall attribute table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "dfM.to_csv(os.path.join(workDir, 'SeqofInt_attributes.txt'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "### Individual tables for iTOL templates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfM['Taxonomy'] = dfM.Phylum + \" \" + dfM.Class + \" \" + dfM.Family"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ASV</th>\n",
       "      <th>Taxonomy</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASV2719</td>\n",
       "      <td>Proteobacteria Alphaproteobacteria Rhizobiaceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BE42</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASV2717</td>\n",
       "      <td>Armatimonadetes Fimbriimonadia Fimbriimonadaceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BE41</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BE47</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       ASV                                          Taxonomy\n",
       "0  ASV2719   Proteobacteria Alphaproteobacteria Rhizobiaceae\n",
       "1     BE42                                               NaN\n",
       "2  ASV2717  Armatimonadetes Fimbriimonadia Fimbriimonadaceae\n",
       "3     BE41                                               NaN\n",
       "4     BE47                                               NaN"
      ]
     },
     "execution_count": 241,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfT = dfM[['ASV', 'Taxonomy']]\n",
    "\n",
    "dfT.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfT.to_csv(os.path.join(workDir, 'SeqofInt_leaflabels.txt'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
