{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Grab sequences from OTUs of interest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import sys, os, re, glob\n",
    "from Bio import SeqIO, Entrez\n",
    "from pandas import *\n",
    "import numpy as np\n",
    "#import entrez.direct\n",
    "from cogent.app.fasttree import build_tree_from_alignment\n",
    "from cogent import DNA, LoadSeqs\n",
    "\n",
    "Entrez.email = \"be68@cornell.edu\"\n",
    "nprocs = 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "FastaSource = '/home/be68/SorghumAMF/otu.fasta'\n",
    "OTUlist = '/home/be68/SorghumAMF/Unknown_df.txt'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "OTUlist is a dataframe exported from R phyloseq object.  Here is was exported as a character vector so the column heading is only \"x\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Kingdom    object\n",
       "Phylum     object\n",
       "Class      object\n",
       "Order      object\n",
       "Family     object\n",
       "Genus      object\n",
       "OTU        object\n",
       "dtype: object"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pandas.read_csv(OTUlist, sep = \"\\t\")\n",
    "df.head()\n",
    "df.OTU = df.OTU.astype(object)\n",
    "df.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['225', '284', '328', '335']"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "OTUcheck = df.OTU.tolist()\n",
    "OTUcheck = [str(item) for item in OTUcheck]\n",
    "OTUcheck[0:4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "output = open(\"/home/be68/SorghumAMF/Unknown.fasta\", \"w\")\n",
    "\n",
    "\n",
    "for record in SeqIO.parse(open(FastaSource, \"rU\"), \"fasta\"):\n",
    "    if record.id in OTUcheck:\n",
    "        output.write(\">\"+record.id+\"\\n\"+str(record.seq)+\"\\n\")\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">225\n",
      "GACGGAGGGTGCAAACGTTGTTCGGAATCACTGGGCGTAAAGGGCGCGTAGGCGGACTTGTAAGTCGGGAGTGAAAACTCGGGGCTCAACCCCGAGCCTGCTTCCGAAACTGCGAGTCTTGAGGACCGGAGAGGAAAGTGGAATTCCTAGTGTAGCGGTGGAATGCGTAGATATTAGGAAGAACACCAGTGGCGAAGGCGACTTTCTGGACGGTTCCTGACGCTGAGGCGCGAAAGCTAGGGGAGCAAACAGG\n",
      ">284\n",
      "TACGGAGGGGGCGAGTGTTGTTCGGAATTATTGGGCGTAAAGGGCGCGCAGGCGGCTTGGCAAGTCGAGTGTGAAATCCCTTGGCTTAACTGAGGAATGGCGCTCGAAACTACCTCGCTAGAGGGCGGGAGAGGGAAGTGGAACTCTCGGTGTAGGGGTGAAATCTGTAGATATCGAGAGGAACACCGGTGGCGAAGGCGGCTTCCTGGCCCGCATCTGACGCTGAGGCGCGAAAGCGTGGGGAGCAAACAGG\n",
      ">328\n",
      "TACGAGGGGGGCAAATGTTGTTCGGAATCACTGGGCGTAAAGGGAGCGTAGGCGGACTTTTAAGTTAGGTGTTTAATCCCGGGGCCCAACCCCGGACAGGCACCTAAAACTGGAAGTCTTGAATAGTGGAGAGGACGGCGGAATTGGGGGTGTAGCGGTGGAATGCGTAGAGATCCCCAAGAACACCAGTTGCGAAGGCGGCCGTCTGGACACATATTGACGCTGAGGCTCGAAAGCATGGGGAGCAAACAGG\n",
      ">335\n",
      "AACGTAGGTGGCGAGCGTTATCCGGATTTACTGGGCGTAAAGGGTGTGCAGGCTGCCCCTCAAGTGGTGTATGAAAGGGCTCGGCTCAACCGGGCTAGGTTATGCCAGACTGAGGGGCTAGAGAGCGAGAGCGGGACGTGGAATTCCGGGTGTAGTGGTGAAATGCGTAGAGATCCGGAGGAACCCCAGAGGCGAAGGCGGCGTCCTGGCTCGCATCTGACGCTCAGACACGACAGCATGGGGAGCGAACGGG\n",
      ">362\n",
      "GACGTAGGATCCGAGCGTTGTCCGGAGTTACTGGGCGTAAAGCGCGCGCAGGCGGACACGCGCATCGGCGGTGAAAGCCCCCCGCTCAACGGGGGAGGGTCCGTCGAGATGGCGTGTCTGGAGGCAGGGAGAGGCCAGTGGAATTCCGGGTGTAGTGGTGAAATGCGTAGAGATCCGGAGGAACACCAGTGGCGAAGGCGGCTGGCTGGACCTGACCTGACGCTGAGGCGCGAAGGCGTGGGGAGCGAACGGG\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "head /home/be68/SorghumAMF/Unknown.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1541\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "grep -c \">\" /home/be68/SorghumAMF/Unknown.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Trim seqs to those that blast to MRE reference strains"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* re.fasta was prepared by downloading Moeniiplasma seqs from Desario paper\n",
    "* See Hyphosphere/MRE_Tree.ipnyb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "## Make blast database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "MREseqs = '/home/be68/Hyphosphere/data/MRE/ref.fasta'\n",
    "Unknowns = '/home/be68/SorghumAMF/Unknown.fasta'\n",
    "MREBlast = '/home/be68/SorghumAMF/Unknown-MRE_blast.txt'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">MG052969.1 Uncultured Mollicutes bacterium clone Msp-GBAus27b 16S ribosomal RNA gene, partial sequence\r\n",
      "AGAGTTTGATCCTGGCTCAGGATTAACGTTGGTAGTATGCCTAACACATGCAAGTCGAGCGGAAGAAAAT\r\n",
      "AGTAATATTTTTTTTAGCGGCGAACGGGTGAGTAATGCTTATCTAACTTGCCGATTAGTGGGGAATAACT\r\n",
      "ATGGGAAACCATTGCTAATACCGCATATTCGTTATTTAGTTAATGAAAGGAAATCGTAAGATTTTCGCTA\r\n",
      "ATCGATAGGGATAAGTCGTATTAGCTAGTTGGTGAGGTAATGGCTCACCAAGGCAATGATGCGTAGCCAG\r\n",
      "TCTGAGAGGATGAATGGTCACAAGGGAACTGAGACACGGTCCCTACTCCTACGGGAGGCAGCAGTGAGGA\r\n",
      "ATTTTCCACAATGGGGGAAACCCTGATGGAGCAATACTACGTGAAGGATGAAGGTCTAAGGATTGTAAAC\r\n",
      "TTCTGTTAGCGCGGAAGAAGGGTCGCAAGATCTTGACGGTACGCGCAGAGAAAGTCACGACTAACTACGT\r\n",
      "GCCAGCAGTCGCGGTAAGACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGAT\r\n",
      "GGCTTGGTAAGTTGGCGGTGGGAAATCGAGGCTCAACCTCGTGGAAGCCGGCAATACTACTAAGCTCGAG\r\n"
     ]
    }
   ],
   "source": [
    "!head /home/be68/Hyphosphere/data/MRE/ref.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "Building a new DB, current time: 11/06/2018 16:00:01\n",
      "New DB name:   /home/be68/Hyphosphere/data/MRE/ref.fasta\n",
      "New DB title:  /home/be68/Hyphosphere/data/MRE/ref.fasta\n",
      "Sequence type: Nucleotide\n",
      "Deleted existing BLAST database with identical name.\n",
      "Keep Linkouts: T\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 66 sequences in 0.0219531 seconds.\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$MREseqs\"\n",
    "makeblastdb -in $1 -parse_seqids -dbtype nucl\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%bash -s \"$MREseqs\" \"$Unknowns\" \"$MREBlast\"\n",
    "blastn -db $1 -query $2 -out $3 -max_target_seqs 1 -outfmt 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "225\tKC441811.1\t80.24\t248\t46\t3\t8\t253\t494\t740\t8e-49\t  183\n",
      "284\tKC441811.1\t78.12\t256\t50\t5\t1\t253\t488\t740\t5e-41\t  158\n",
      "363\tKP763351.1\t72.43\t243\t60\t5\t10\t248\t384\t623\t7e-15\t71.3\n",
      "403\tKC441811.1\t77.64\t246\t53\t2\t6\t250\t493\t737\t3e-38\t  148\n",
      "520\tKC441811.1\t78.78\t245\t52\t0\t6\t250\t493\t737\t3e-43\t  165\n",
      "683\tKC441811.1\t76.26\t257\t54\t7\t1\t254\t488\t740\t1e-32\t  130\n",
      "791\tKC441811.1\t79.52\t249\t47\t4\t8\t254\t494\t740\t5e-46\t  174\n",
      "887\tKC441811.1\t80.00\t250\t50\t0\t1\t250\t488\t737\t2e-49\t  185\n",
      "939\tKC441811.1\t77.73\t256\t51\t5\t1\t253\t488\t740\t2e-39\t  152\n",
      "991\tKC441811.1\t77.78\t243\t48\t6\t14\t253\t501\t740\t4e-37\t  145\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$MREseqs\" \"$Unknowns\" \"$MREBlast\"\n",
    "head $3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OTU</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>225</td>\n",
       "      <td>KC441811.1</td>\n",
       "      <td>80.24</td>\n",
       "      <td>248</td>\n",
       "      <td>46</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>253</td>\n",
       "      <td>494</td>\n",
       "      <td>740</td>\n",
       "      <td>8.000000e-49</td>\n",
       "      <td>183.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>284</td>\n",
       "      <td>KC441811.1</td>\n",
       "      <td>78.12</td>\n",
       "      <td>256</td>\n",
       "      <td>50</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>488</td>\n",
       "      <td>740</td>\n",
       "      <td>5.000000e-41</td>\n",
       "      <td>158.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>363</td>\n",
       "      <td>KP763351.1</td>\n",
       "      <td>72.43</td>\n",
       "      <td>243</td>\n",
       "      <td>60</td>\n",
       "      <td>5</td>\n",
       "      <td>10</td>\n",
       "      <td>248</td>\n",
       "      <td>384</td>\n",
       "      <td>623</td>\n",
       "      <td>7.000000e-15</td>\n",
       "      <td>71.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>403</td>\n",
       "      <td>KC441811.1</td>\n",
       "      <td>77.64</td>\n",
       "      <td>246</td>\n",
       "      <td>53</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>250</td>\n",
       "      <td>493</td>\n",
       "      <td>737</td>\n",
       "      <td>3.000000e-38</td>\n",
       "      <td>148.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>520</td>\n",
       "      <td>KC441811.1</td>\n",
       "      <td>78.78</td>\n",
       "      <td>245</td>\n",
       "      <td>52</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>250</td>\n",
       "      <td>493</td>\n",
       "      <td>737</td>\n",
       "      <td>3.000000e-43</td>\n",
       "      <td>165.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   OTU      sseqid  pident  length  mismatch  gapopen  qstart  qend  sstart  \\\n",
       "0  225  KC441811.1   80.24     248        46        3       8   253     494   \n",
       "1  284  KC441811.1   78.12     256        50        5       1   253     488   \n",
       "2  363  KP763351.1   72.43     243        60        5      10   248     384   \n",
       "3  403  KC441811.1   77.64     246        53        2       6   250     493   \n",
       "4  520  KC441811.1   78.78     245        52        0       6   250     493   \n",
       "\n",
       "   send        evalue  bitscore  \n",
       "0   740  8.000000e-49     183.0  \n",
       "1   740  5.000000e-41     158.0  \n",
       "2   623  7.000000e-15      71.3  \n",
       "3   737  3.000000e-38     148.0  \n",
       "4   737  3.000000e-43     165.0  "
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pandas.read_table(MREBlast, sep = \"\\t\", header=None, \n",
    "                       names = [\"OTU\", \"sseqid\", \"pident\", \"length\", \"mismatch\", \"gapopen\", \"qstart\",\n",
    "                                \"qend\", \"sstart\", \"send\", \"evalue\", \"bitscore\"])\n",
    "\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Filter to >85% match to MREs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OTU</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>2872</td>\n",
       "      <td>FJ984707.1</td>\n",
       "      <td>99.21</td>\n",
       "      <td>254</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>383</td>\n",
       "      <td>636</td>\n",
       "      <td>1.000000e-131</td>\n",
       "      <td>459.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>3001</td>\n",
       "      <td>KP763351.1</td>\n",
       "      <td>94.09</td>\n",
       "      <td>254</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>375</td>\n",
       "      <td>628</td>\n",
       "      <td>6.000000e-110</td>\n",
       "      <td>387.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>4190</td>\n",
       "      <td>FJ984646.1</td>\n",
       "      <td>98.82</td>\n",
       "      <td>254</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>475</td>\n",
       "      <td>727</td>\n",
       "      <td>2.000000e-129</td>\n",
       "      <td>451.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>4442</td>\n",
       "      <td>FJ984657.1</td>\n",
       "      <td>95.31</td>\n",
       "      <td>256</td>\n",
       "      <td>12</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>256</td>\n",
       "      <td>498</td>\n",
       "      <td>753</td>\n",
       "      <td>5.000000e-116</td>\n",
       "      <td>407.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>4562</td>\n",
       "      <td>KP763336.1</td>\n",
       "      <td>93.28</td>\n",
       "      <td>253</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>409</td>\n",
       "      <td>661</td>\n",
       "      <td>5.000000e-106</td>\n",
       "      <td>374.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>5222</td>\n",
       "      <td>FJ984653.1</td>\n",
       "      <td>91.34</td>\n",
       "      <td>254</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>475</td>\n",
       "      <td>728</td>\n",
       "      <td>3.000000e-98</td>\n",
       "      <td>348.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147</th>\n",
       "      <td>6878</td>\n",
       "      <td>FJ984654.1</td>\n",
       "      <td>98.82</td>\n",
       "      <td>255</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>255</td>\n",
       "      <td>480</td>\n",
       "      <td>734</td>\n",
       "      <td>2.000000e-130</td>\n",
       "      <td>455.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>7420</td>\n",
       "      <td>FJ984679.1</td>\n",
       "      <td>92.97</td>\n",
       "      <td>256</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>256</td>\n",
       "      <td>420</td>\n",
       "      <td>675</td>\n",
       "      <td>5.000000e-106</td>\n",
       "      <td>374.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>7660</td>\n",
       "      <td>FJ984707.1</td>\n",
       "      <td>95.67</td>\n",
       "      <td>254</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>383</td>\n",
       "      <td>636</td>\n",
       "      <td>1.000000e-116</td>\n",
       "      <td>409.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>175</th>\n",
       "      <td>7941</td>\n",
       "      <td>KP763351.1</td>\n",
       "      <td>94.88</td>\n",
       "      <td>254</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>375</td>\n",
       "      <td>628</td>\n",
       "      <td>3.000000e-113</td>\n",
       "      <td>398.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>186</th>\n",
       "      <td>8319</td>\n",
       "      <td>KP763286.1</td>\n",
       "      <td>97.64</td>\n",
       "      <td>254</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>387</td>\n",
       "      <td>640</td>\n",
       "      <td>6.000000e-125</td>\n",
       "      <td>436.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>190</th>\n",
       "      <td>8498</td>\n",
       "      <td>FJ984653.1</td>\n",
       "      <td>93.31</td>\n",
       "      <td>254</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>475</td>\n",
       "      <td>728</td>\n",
       "      <td>1.000000e-106</td>\n",
       "      <td>375.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>191</th>\n",
       "      <td>8522</td>\n",
       "      <td>KM594012.1</td>\n",
       "      <td>96.47</td>\n",
       "      <td>255</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>421</td>\n",
       "      <td>675</td>\n",
       "      <td>6.000000e-120</td>\n",
       "      <td>420.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>194</th>\n",
       "      <td>8660</td>\n",
       "      <td>FJ984635.1</td>\n",
       "      <td>97.66</td>\n",
       "      <td>256</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>256</td>\n",
       "      <td>421</td>\n",
       "      <td>676</td>\n",
       "      <td>4.000000e-126</td>\n",
       "      <td>440.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>8827</td>\n",
       "      <td>FJ984635.1</td>\n",
       "      <td>91.41</td>\n",
       "      <td>256</td>\n",
       "      <td>22</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>256</td>\n",
       "      <td>421</td>\n",
       "      <td>676</td>\n",
       "      <td>2.000000e-99</td>\n",
       "      <td>351.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>9232</td>\n",
       "      <td>FJ984658.1</td>\n",
       "      <td>95.67</td>\n",
       "      <td>254</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>476</td>\n",
       "      <td>729</td>\n",
       "      <td>1.000000e-116</td>\n",
       "      <td>409.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>9472</td>\n",
       "      <td>KP763291.1</td>\n",
       "      <td>98.44</td>\n",
       "      <td>256</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>256</td>\n",
       "      <td>407</td>\n",
       "      <td>662</td>\n",
       "      <td>2.000000e-129</td>\n",
       "      <td>451.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>9533</td>\n",
       "      <td>FJ984657.1</td>\n",
       "      <td>93.36</td>\n",
       "      <td>256</td>\n",
       "      <td>17</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>256</td>\n",
       "      <td>498</td>\n",
       "      <td>753</td>\n",
       "      <td>1.000000e-107</td>\n",
       "      <td>379.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>9550</td>\n",
       "      <td>FJ984657.1</td>\n",
       "      <td>92.58</td>\n",
       "      <td>256</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>256</td>\n",
       "      <td>498</td>\n",
       "      <td>753</td>\n",
       "      <td>2.000000e-104</td>\n",
       "      <td>368.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>10226</td>\n",
       "      <td>FJ984704.1</td>\n",
       "      <td>95.67</td>\n",
       "      <td>254</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>472</td>\n",
       "      <td>725</td>\n",
       "      <td>1.000000e-116</td>\n",
       "      <td>409.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>10910</td>\n",
       "      <td>KP763291.1</td>\n",
       "      <td>98.44</td>\n",
       "      <td>256</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>256</td>\n",
       "      <td>407</td>\n",
       "      <td>662</td>\n",
       "      <td>2.000000e-129</td>\n",
       "      <td>451.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>11169</td>\n",
       "      <td>KP763291.1</td>\n",
       "      <td>92.58</td>\n",
       "      <td>256</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>256</td>\n",
       "      <td>407</td>\n",
       "      <td>662</td>\n",
       "      <td>2.000000e-104</td>\n",
       "      <td>368.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>252</th>\n",
       "      <td>11600</td>\n",
       "      <td>KP763291.1</td>\n",
       "      <td>93.75</td>\n",
       "      <td>256</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>256</td>\n",
       "      <td>407</td>\n",
       "      <td>662</td>\n",
       "      <td>2.000000e-109</td>\n",
       "      <td>385.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>13308</td>\n",
       "      <td>FJ984654.1</td>\n",
       "      <td>93.73</td>\n",
       "      <td>255</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>255</td>\n",
       "      <td>480</td>\n",
       "      <td>734</td>\n",
       "      <td>8.000000e-109</td>\n",
       "      <td>383.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>13619</td>\n",
       "      <td>MG052968.1</td>\n",
       "      <td>91.83</td>\n",
       "      <td>257</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>257</td>\n",
       "      <td>415</td>\n",
       "      <td>670</td>\n",
       "      <td>5.000000e-101</td>\n",
       "      <td>357.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278</th>\n",
       "      <td>13848</td>\n",
       "      <td>FJ984657.1</td>\n",
       "      <td>91.73</td>\n",
       "      <td>254</td>\n",
       "      <td>19</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>498</td>\n",
       "      <td>750</td>\n",
       "      <td>2.000000e-99</td>\n",
       "      <td>351.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>279</th>\n",
       "      <td>13928</td>\n",
       "      <td>FJ984707.1</td>\n",
       "      <td>94.09</td>\n",
       "      <td>254</td>\n",
       "      <td>15</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>383</td>\n",
       "      <td>636</td>\n",
       "      <td>6.000000e-110</td>\n",
       "      <td>387.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280</th>\n",
       "      <td>14163</td>\n",
       "      <td>FJ984660.1</td>\n",
       "      <td>90.55</td>\n",
       "      <td>254</td>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>252</td>\n",
       "      <td>487</td>\n",
       "      <td>740</td>\n",
       "      <td>2.000000e-94</td>\n",
       "      <td>335.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>14224</td>\n",
       "      <td>FJ984709.1</td>\n",
       "      <td>91.73</td>\n",
       "      <td>254</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>407</td>\n",
       "      <td>660</td>\n",
       "      <td>6.000000e-100</td>\n",
       "      <td>353.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>14386</td>\n",
       "      <td>KP763336.1</td>\n",
       "      <td>91.02</td>\n",
       "      <td>256</td>\n",
       "      <td>21</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>409</td>\n",
       "      <td>664</td>\n",
       "      <td>4.000000e-97</td>\n",
       "      <td>344.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>14558</td>\n",
       "      <td>FJ984643.1</td>\n",
       "      <td>99.61</td>\n",
       "      <td>256</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>256</td>\n",
       "      <td>419</td>\n",
       "      <td>674</td>\n",
       "      <td>2.000000e-134</td>\n",
       "      <td>468.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>309</th>\n",
       "      <td>16770</td>\n",
       "      <td>FJ984660.1</td>\n",
       "      <td>96.06</td>\n",
       "      <td>254</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>254</td>\n",
       "      <td>487</td>\n",
       "      <td>740</td>\n",
       "      <td>3.000000e-118</td>\n",
       "      <td>414.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>21180</td>\n",
       "      <td>FJ984657.1</td>\n",
       "      <td>94.82</td>\n",
       "      <td>251</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>251</td>\n",
       "      <td>498</td>\n",
       "      <td>748</td>\n",
       "      <td>1.000000e-111</td>\n",
       "      <td>392.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333</th>\n",
       "      <td>21321</td>\n",
       "      <td>FJ984707.1</td>\n",
       "      <td>90.59</td>\n",
       "      <td>255</td>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>255</td>\n",
       "      <td>383</td>\n",
       "      <td>636</td>\n",
       "      <td>6.000000e-95</td>\n",
       "      <td>337.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       OTU      sseqid  pident  length  mismatch  gapopen  qstart  qend  \\\n",
       "41    2872  FJ984707.1   99.21     254         2        0       1   254   \n",
       "44    3001  KP763351.1   94.09     254        15        0       1   254   \n",
       "63    4190  FJ984646.1   98.82     254         2        1       1   254   \n",
       "66    4442  FJ984657.1   95.31     256        12        0       1   256   \n",
       "71    4562  KP763336.1   93.28     253        17        0       1   253   \n",
       "90    5222  FJ984653.1   91.34     254        22        0       1   254   \n",
       "147   6878  FJ984654.1   98.82     255         3        0       1   255   \n",
       "160   7420  FJ984679.1   92.97     256        18        0       1   256   \n",
       "171   7660  FJ984707.1   95.67     254        11        0       1   254   \n",
       "175   7941  KP763351.1   94.88     254        13        0       1   254   \n",
       "186   8319  KP763286.1   97.64     254         6        0       1   254   \n",
       "190   8498  FJ984653.1   93.31     254        17        0       1   254   \n",
       "191   8522  KM594012.1   96.47     255         8        1       1   254   \n",
       "194   8660  FJ984635.1   97.66     256         6        0       1   256   \n",
       "199   8827  FJ984635.1   91.41     256        22        0       1   256   \n",
       "205   9232  FJ984658.1   95.67     254        11        0       1   254   \n",
       "209   9472  KP763291.1   98.44     256         4        0       1   256   \n",
       "212   9533  FJ984657.1   93.36     256        17        0       1   256   \n",
       "213   9550  FJ984657.1   92.58     256        19        0       1   256   \n",
       "225  10226  FJ984704.1   95.67     254        11        0       1   254   \n",
       "237  10910  KP763291.1   98.44     256         4        0       1   256   \n",
       "246  11169  KP763291.1   92.58     256        19        0       1   256   \n",
       "252  11600  KP763291.1   93.75     256        16        0       1   256   \n",
       "270  13308  FJ984654.1   93.73     255        16        0       1   255   \n",
       "273  13619  MG052968.1   91.83     257        20        1       1   257   \n",
       "278  13848  FJ984657.1   91.73     254        19        2       1   253   \n",
       "279  13928  FJ984707.1   94.09     254        15        0       1   254   \n",
       "280  14163  FJ984660.1   90.55     254        22        1       1   252   \n",
       "281  14224  FJ984709.1   91.73     254        21        0       1   254   \n",
       "284  14386  KP763336.1   91.02     256        21        1       1   254   \n",
       "285  14558  FJ984643.1   99.61     256         1        0       1   256   \n",
       "309  16770  FJ984660.1   96.06     254        10        0       1   254   \n",
       "331  21180  FJ984657.1   94.82     251        13        0       1   251   \n",
       "333  21321  FJ984707.1   90.59     255        23        1       1   255   \n",
       "\n",
       "     sstart  send         evalue  bitscore  \n",
       "41      383   636  1.000000e-131     459.0  \n",
       "44      375   628  6.000000e-110     387.0  \n",
       "63      475   727  2.000000e-129     451.0  \n",
       "66      498   753  5.000000e-116     407.0  \n",
       "71      409   661  5.000000e-106     374.0  \n",
       "90      475   728   3.000000e-98     348.0  \n",
       "147     480   734  2.000000e-130     455.0  \n",
       "160     420   675  5.000000e-106     374.0  \n",
       "171     383   636  1.000000e-116     409.0  \n",
       "175     375   628  3.000000e-113     398.0  \n",
       "186     387   640  6.000000e-125     436.0  \n",
       "190     475   728  1.000000e-106     375.0  \n",
       "191     421   675  6.000000e-120     420.0  \n",
       "194     421   676  4.000000e-126     440.0  \n",
       "199     421   676   2.000000e-99     351.0  \n",
       "205     476   729  1.000000e-116     409.0  \n",
       "209     407   662  2.000000e-129     451.0  \n",
       "212     498   753  1.000000e-107     379.0  \n",
       "213     498   753  2.000000e-104     368.0  \n",
       "225     472   725  1.000000e-116     409.0  \n",
       "237     407   662  2.000000e-129     451.0  \n",
       "246     407   662  2.000000e-104     368.0  \n",
       "252     407   662  2.000000e-109     385.0  \n",
       "270     480   734  8.000000e-109     383.0  \n",
       "273     415   670  5.000000e-101     357.0  \n",
       "278     498   750   2.000000e-99     351.0  \n",
       "279     383   636  6.000000e-110     387.0  \n",
       "280     487   740   2.000000e-94     335.0  \n",
       "281     407   660  6.000000e-100     353.0  \n",
       "284     409   664   4.000000e-97     344.0  \n",
       "285     419   674  2.000000e-134     468.0  \n",
       "309     487   740  3.000000e-118     414.0  \n",
       "331     498   748  1.000000e-111     392.0  \n",
       "333     383   636   6.000000e-95     337.0  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mre = df.loc[(df['pident'] > 90) & (df['length'] > 240)]\n",
    "df_mre"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(34, 12)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_mre.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Grab seqs from fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "OTUcheck = df_mre.OTU.tolist()\n",
    "OTUcheck = [str(item) for item in OTUcheck]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "output = open(\"/home/be68/SorghumAMF/NebraskaMRE.fasta\", \"w\")\n",
    "\n",
    "#FastaSource same as defined above\n",
    "for record in SeqIO.parse(open(FastaSource, \"rU\"), \"fasta\"):\n",
    "    if record.id in OTUcheck:\n",
    "        output.write(\">\"+record.id+\"\\n\"+str(record.seq)+\"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">2872\r\n",
      "GACATAGGTGGCGAACGTTATCCGGAATTACTGGGCGTAAAGAGTGCGTAGGTGGTTCAATAAGTTACTGGTGGGAAATCGGGGCTCAACCCCGTGGAAGTCAGTAATACTATTAAACTAGAGGGTCGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTTAAATATATAGAGGAACACCAAACGGCGAAGGCAGTTAACTATTCGATTCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAGACGG\r\n",
      ">3001\r\n",
      "GACATAGGTGGCAAACGTTATCCGGAATTACTGGGCGTAAAGTGTGCGTAGGTGGTTGAATAAGTTACTGGTGGCAAATCGGGGCTCAACTTCGTGGAAGCCAGTAATACTATTCAACTCGAGGACCGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACATCAATCGGCGAAAGCAGTTAACTATTCGGTTCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAGACGG\r\n",
      ">4190\r\n",
      "GACATAGGTGGCGAACGTTATCCGGAATTACTGGGCGTAAAGAGTGCGTAGGTGGTTCAATAAGTTACTGGTGGCAAATCGGGGCCCAACCCCGTGGAAGTTGGTAATACTGTTGAACTAGAGGACCGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCAAACGGCGAAGGCAGTTAACTATTCGGTGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAGACGG\r\n",
      ">4442\r\n",
      "GACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCCTGACAAGTTAACAGTGGTAAATCGAGGCTCAACCTCGTGGAAACTGATAATACTGTCTGGCTTGAGTACAGAAGAGGTTAACGGAATTCCATGTGGAGCGGTAAAATGCACAGATATATGGAAGAACACCAATAACGGCGAAGGCAGTTAACTATTCTGTTACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">4562\r\n",
      "GACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATTGTATGGTAAGTTACTGGTGGCAAATCGAGGCTCAACCTCGTGGAAGTCAGTAATACTGCTATGCTAGAGTTCAGGAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACGCCAATAACGGCGAAGGCAGTTAACTATTCTGTTACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n"
     ]
    }
   ],
   "source": [
    "!head /home/be68/SorghumAMF/NebraskaMRE.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Build tree with Nebraska and Desario sequences"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Use SSU align to align both datasets and mask to smaller region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "NebraskaMREseqs = '/home/be68/SorghumAMF/NebraskaMRE.fasta'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "workDir = '/home/be68/SorghumAMF/NebraskaMREs'\n",
    "os.chdir(workDir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# _ssu-prep :: prepare SSU rRNA sequences for parallel ssu-align jobs\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-prep -x -f -b 50 --dna --rfonly /home/be68/SorghumAMF/NebraskaMRE.fasta ssu_aln 1\n",
      "# date:    Tue Nov  6 15:48:37 2018\n",
      "#\n",
      "# Validating input sequence file ... done.\n",
      "#\n",
      "# Preparing 1 ssu-align jobs ...\n",
      "# Partitioning seqs with goal of equalizing total number of nucleotides per job ...\n",
      "#\n",
      "# output file name      description                                       \n",
      "# --------------------  --------------------------------------------------\n",
      "  ssu_aln/NebraskaMRE.fasta.1  partition 1 FASTA sequence file (32 seqs; 8157 nt)\n",
      "  ssu_aln.ssu-align.sh  shell script that will execute 1 ssu-align jobs\n",
      "#\n",
      "################################################################################\n",
      "# To execute all 1 ssu-align jobs, run the shell script with the command:\n",
      "#\t\"ssu_aln.ssu-align.sh\"\n",
      "# (it is an executable file)\n",
      "################################################################################\n",
      "#\n",
      "# The final job will merge the output from all jobs once they are all finished.\n",
      "#\n",
      "# If one or more jobs fail: rerun the failed jobs, wait for them to finish,\n",
      "# and then perform manual merge from this directory by executing:\n",
      "#\tssu-merge ssu_aln\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln/ssu_aln.ssu-prep.log.\n",
      "# Output printed to the screen saved in:  ssu_aln/ssu_aln.ssu-prep.sum.\n",
      "#\n",
      "# CPU time:  00:00:00.36  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$NebraskaMREseqs\"\n",
    "ssu-prep -f -x -b 50 --rfonly --dna $1 ssu_aln 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Executing: ssu-align --merge 1 -b 50 --dna --rfonly ssu_aln/NebraskaMRE.fasta.1 ssu_aln/ssu_aln.1\n",
      "# _ssu-align :: align SSU rRNA sequences\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-align --merge 1 -b 50 --dna --rfonly ssu_aln/NebraskaMRE.fasta.1 ssu_aln/ssu_aln.1\n",
      "# date:    Tue Nov  6 15:49:01 2018\n",
      "#\n",
      "# Validating input sequence file ... done.\n",
      "#\n",
      "# Stage 1: Determining SSU start/end positions and best-matching models...\n",
      "#\n",
      "# output file name            description                                \n",
      "# --------------------------  -------------------------------------------\n",
      "  ssu_aln.1.tab               locations/scores of hits defined by HMM(s)\n",
      "  ssu_aln.1.bacteria.hitlist  list of sequences to align with bacteria CM\n",
      "  ssu_aln.1.bacteria.fa            32 sequences to align with bacteria CM\n",
      "#\n",
      "# Stage 2: Aligning each sequence to its best-matching model...\n",
      "#\n",
      "# output file name            description\n",
      "# --------------------------  ---------------------------------------\n",
      "  ssu_aln.1.bacteria.stk      bacteria alignment\n",
      "  ssu_aln.1.bacteria.cmalign  bacteria cmalign output\n",
      "  ssu_aln.1.bacteria.ifile    bacteria insert info\n",
      "  ssu_aln.1.scores            list of CM/HMM scores for each sequence\n",
      "#\n",
      "# Summary statistics:\n",
      "#\n",
      "# model or       number  fraction        average   average               \n",
      "# category      of seqs  of total         length  coverage    nucleotides\n",
      "# ------------  -------  --------  -------------  --------  -------------\n",
      "  *input*            32    1.0000         254.91    1.0000           8157\n",
      "#\n",
      "  archaea             0    0.0000              -         -              0\n",
      "  bacteria           32    1.0000         250.75    0.9837           8024\n",
      "  eukarya             0    0.0000              -         -              0\n",
      "#\n",
      "  *all-models*       32    1.0000         250.75    0.9837           8024\n",
      "  *no-models*         0    0.0000              -         -              0\n",
      "#\n",
      "# Speed statistics:\n",
      "#\n",
      "# stage      num seqs  seq/sec  seq/sec/model    nucleotides    nt/sec\n",
      "# ---------  --------  -------  -------------  -------------  --------\n",
      "  search           32    8.123          2.708           8157    2070.5\n",
      "  alignment        32    7.107          7.107           8024    1782.1\n",
      "#\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln.1.ssu-align.log.\n",
      "# Output printed to the screen saved in:  ssu_aln.1.ssu-align.sum.\n",
      "#\n",
      "# All output files created in directory ./ssu_aln/ssu_aln.1/\n",
      "#\n",
      "# CPU time (search):     00:00:03.94  (hh:mm:ss)\n",
      "# CPU time (alignment):  00:00:04.50  (hh:mm:ss)\n",
      "# CPU time (total):      00:00:08.66  (hh:mm:ss)\n",
      "#\n",
      "################################################################################\n",
      "#\n",
      "# Alignments successfully created.\n",
      "#\n",
      "# Checking/waiting for other jobs to finish before merging...\n",
      "#\n",
      "# Job 1 (dir: ssu_aln/ssu_aln.1) completed successfully! 0/1 still incomplete.\n",
      "#\n",
      "# All 1 jobs successfully completed.\n",
      "# About to merge their output:\n",
      "#\n",
      "# Executing:  ssu-merge ssu_aln\n",
      "#\n",
      "################################################################################\n",
      "#\n",
      "# _ssu-merge :: merge SSU rRNA alignments\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-merge ssu_aln\n",
      "# date:    Tue Nov  6 15:49:10 2018\n",
      "#\n",
      "# Merging files from 1 ssu-align runs...\n",
      "#\n",
      "#                                     # files     # seqs\n",
      "# merged file name          CM name    merged     merged\n",
      "# ------------------------  --------  -------  ---------\n",
      "  ssu_aln.tab               -               1          -\n",
      "  ssu_aln.scores            -               1          -\n",
      "  ssu_aln.ssu-align.sum     -               1          -\n",
      "  ssu_aln.ssu-align.log     -               1          -\n",
      "#\n",
      "  ssu_aln.bacteria.fa       bacteria        1         32\n",
      "  ssu_aln.bacteria.hitlist  bacteria        1         32\n",
      "  ssu_aln.bacteria.cmalign  bacteria        1         32\n",
      "  ssu_aln.bacteria.ifile    bacteria        1         32\n",
      "  ssu_aln.bacteria.stk      bacteria        1         32\n",
      "#\n",
      "#\n",
      "# Summary statistics:\n",
      "#\n",
      "# model or       number  fraction        average   average               \n",
      "# category      of seqs  of total         length  coverage    nucleotides\n",
      "# ------------  -------  --------  -------------  --------  -------------\n",
      "  *input*            32    1.0000         254.91    1.0000           8157\n",
      "#\n",
      "  archaea             0    0.0000              -         -              0\n",
      "  bacteria           32    1.0000         250.75    0.9837           8024\n",
      "  eukarya             0    0.0000              -         -              0\n",
      "#\n",
      "  *all-models*       32    1.0000         250.75    0.9837           8024\n",
      "  *no-models*         0    0.0000              -         -              0\n",
      "#\n",
      "# Speed statistics:\n",
      "#\n",
      "# stage      num seqs  seq/sec  seq/sec/model    nucleotides    nt/sec\n",
      "# ---------  --------  -------  -------------  -------------  --------\n",
      "  search           32   32.000         10.667           8157    8157.0\n",
      "  alignment        32   32.000         32.000           8024    8024.0\n",
      "#\n",
      "# Summed ssu-align CPU time (search):     00:00:00.00  (hh:mm:ss)\n",
      "# Summed ssu-align CPU time (alignment):  00:00:00.00  (hh:mm:ss)\n",
      "# Summed ssu-align CPU time (total):      00:00:00.00  (hh:mm:ss)\n",
      "#\n",
      "# Cleaning up by removing original files that were just merged ... done.\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln.ssu-merge.log.\n",
      "# Output printed to the screen saved in:  ssu_aln.ssu-merge.sum.\n",
      "#\n",
      "# All output files created in directory ./ssu_aln/\n",
      "#\n",
      "# CPU time:  00:00:00.11  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "./ssu_aln.ssu-align.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# _ssu-mask :: mask SSU rRNA alignments\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-mask --afa --dna ssu_aln\n",
      "# date:    Tue Nov  6 15:49:49 2018\n",
      "#\n",
      "# Masking alignments based on posterior probabilities...\n",
      "#\n",
      "#                                                     mask    \n",
      "#                                                 ------------\n",
      "# file name                  in/out  type  #cols  incl.  excl.\n",
      "# -------------------------  ------  ----  -----  -----  -----\n",
      "  ssu_aln.bacteria.stk        input   aln   1582      -      -\n",
      "  ssu_aln.bacteria.mask      output  mask   1582    225   1357\n",
      "  ssu_aln.bacteria.mask.pdf  output   pdf   1582    225   1357\n",
      "  ssu_aln.bacteria.mask.afa  output   aln    225      -      -\n",
      "#\n",
      "# All attempts to draw structure diagrams of masks were successful.\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln.ssu-mask.log.\n",
      "# Output printed to the screen saved in:  ssu_aln.ssu-mask.sum.\n",
      "#\n",
      "# All output files created in directory ./ssu_aln/\n",
      "#\n",
      "# CPU time:  00:00:01.58  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "!ssu-mask --dna --afa ssu_aln"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Align ref fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# _ssu-prep :: prepare SSU rRNA sequences for parallel ssu-align jobs\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-prep -x -f -b 50 --dna --rfonly /home/be68/Hyphosphere/data/MRE/ref.fasta ssr_aln 1\n",
      "# date:    Tue Nov  6 15:50:54 2018\n",
      "#\n",
      "# Validating input sequence file ... done.\n",
      "#\n",
      "# Preparing 1 ssu-align jobs ...\n",
      "# Partitioning seqs with goal of equalizing total number of nucleotides per job ...\n",
      "#\n",
      "# output file name      description                                        \n",
      "# --------------------  ---------------------------------------------------\n",
      "  ssr_aln/ref.fasta.1   partition 1 FASTA sequence file (66 seqs; 82218 nt)\n",
      "  ssr_aln.ssu-align.sh  shell script that will execute 1 ssu-align jobs\n",
      "#\n",
      "################################################################################\n",
      "# To execute all 1 ssu-align jobs, run the shell script with the command:\n",
      "#\t\"ssr_aln.ssu-align.sh\"\n",
      "# (it is an executable file)\n",
      "################################################################################\n",
      "#\n",
      "# The final job will merge the output from all jobs once they are all finished.\n",
      "#\n",
      "# If one or more jobs fail: rerun the failed jobs, wait for them to finish,\n",
      "# and then perform manual merge from this directory by executing:\n",
      "#\tssu-merge ssr_aln\n",
      "#\n",
      "# List of executed commands saved in:     ssr_aln/ssr_aln.ssu-prep.log.\n",
      "# Output printed to the screen saved in:  ssr_aln/ssr_aln.ssu-prep.sum.\n",
      "#\n",
      "# CPU time:  00:00:00.14  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$MREseqs\"\n",
    "ssu-prep -f -x -b 50 --rfonly --dna $1 ssr_aln 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Executing: ssu-align --merge 1 -b 50 --dna --rfonly ssr_aln/ref.fasta.1 ssr_aln/ssr_aln.1\n",
      "# _ssu-align :: align SSU rRNA sequences\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-align --merge 1 -b 50 --dna --rfonly ssr_aln/ref.fasta.1 ssr_aln/ssr_aln.1\n",
      "# date:    Tue Nov  6 15:51:09 2018\n",
      "#\n",
      "# Validating input sequence file ... done.\n",
      "#\n",
      "# Stage 1: Determining SSU start/end positions and best-matching models...\n",
      "#\n",
      "# output file name            description                                \n",
      "# --------------------------  -------------------------------------------\n",
      "  ssr_aln.1.tab               locations/scores of hits defined by HMM(s)\n",
      "  ssr_aln.1.bacteria.hitlist  list of sequences to align with bacteria CM\n",
      "  ssr_aln.1.bacteria.fa            66 sequences to align with bacteria CM\n",
      "#\n",
      "# Stage 2: Aligning each sequence to its best-matching model...\n",
      "#\n",
      "# output file name            description\n",
      "# --------------------------  ---------------------------------------\n",
      "  ssr_aln.1.bacteria.stk      bacteria alignment\n",
      "  ssr_aln.1.bacteria.cmalign  bacteria cmalign output\n",
      "  ssr_aln.1.bacteria.ifile    bacteria insert info\n",
      "  ssr_aln.1.scores            list of CM/HMM scores for each sequence\n",
      "#\n",
      "# Summary statistics:\n",
      "#\n",
      "# model or       number  fraction        average   average               \n",
      "# category      of seqs  of total         length  coverage    nucleotides\n",
      "# ------------  -------  --------  -------------  --------  -------------\n",
      "  *input*            66    1.0000        1245.73    1.0000          82218\n",
      "#\n",
      "  archaea             0    0.0000              -         -              0\n",
      "  bacteria           66    1.0000        1234.85    0.9913          81500\n",
      "  eukarya             0    0.0000              -         -              0\n",
      "#\n",
      "  *all-models*       66    1.0000        1234.85    0.9913          81500\n",
      "  *no-models*         0    0.0000              -         -              0\n",
      "#\n",
      "# Speed statistics:\n",
      "#\n",
      "# stage      num seqs  seq/sec  seq/sec/model    nucleotides    nt/sec\n",
      "# ---------  --------  -------  -------------  -------------  --------\n",
      "  search           66    1.556          0.519          82218    1938.2\n",
      "  alignment        66    1.046          1.046          81500    1291.4\n",
      "#\n",
      "#\n",
      "# List of executed commands saved in:     ssr_aln.1.ssu-align.log.\n",
      "# Output printed to the screen saved in:  ssr_aln.1.ssu-align.sum.\n",
      "#\n",
      "# All output files created in directory ./ssr_aln/ssr_aln.1/\n",
      "#\n",
      "# CPU time (search):     00:00:42.42  (hh:mm:ss)\n",
      "# CPU time (alignment):  00:01:03.11  (hh:mm:ss)\n",
      "# CPU time (total):      00:01:45.71  (hh:mm:ss)\n",
      "#\n",
      "################################################################################\n",
      "#\n",
      "# Alignments successfully created.\n",
      "#\n",
      "# Checking/waiting for other jobs to finish before merging...\n",
      "#\n",
      "# Job 1 (dir: ssr_aln/ssr_aln.1) completed successfully! 0/1 still incomplete.\n",
      "#\n",
      "# All 1 jobs successfully completed.\n",
      "# About to merge their output:\n",
      "#\n",
      "# Executing:  ssu-merge ssr_aln\n",
      "#\n",
      "################################################################################\n",
      "#\n",
      "# _ssu-merge :: merge SSU rRNA alignments\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-merge ssr_aln\n",
      "# date:    Tue Nov  6 15:52:55 2018\n",
      "#\n",
      "# Merging files from 1 ssu-align runs...\n",
      "#\n",
      "#                                     # files     # seqs\n",
      "# merged file name          CM name    merged     merged\n",
      "# ------------------------  --------  -------  ---------\n",
      "  ssr_aln.tab               -               1          -\n",
      "  ssr_aln.scores            -               1          -\n",
      "  ssr_aln.ssu-align.sum     -               1          -\n",
      "  ssr_aln.ssu-align.log     -               1          -\n",
      "#\n",
      "  ssr_aln.bacteria.fa       bacteria        1         66\n",
      "  ssr_aln.bacteria.hitlist  bacteria        1         66\n",
      "  ssr_aln.bacteria.cmalign  bacteria        1         66\n",
      "  ssr_aln.bacteria.ifile    bacteria        1         66\n",
      "  ssr_aln.bacteria.stk      bacteria        1         66\n",
      "#\n",
      "#\n",
      "# Summary statistics:\n",
      "#\n",
      "# model or       number  fraction        average   average               \n",
      "# category      of seqs  of total         length  coverage    nucleotides\n",
      "# ------------  -------  --------  -------------  --------  -------------\n",
      "  *input*            66    1.0000        1245.73    1.0000          82218\n",
      "#\n",
      "  archaea             0    0.0000              -         -              0\n",
      "  bacteria           66    1.0000        1234.85    0.9913          81500\n",
      "  eukarya             0    0.0000              -         -              0\n",
      "#\n",
      "  *all-models*       66    1.0000        1234.85    0.9913          81500\n",
      "  *no-models*         0    0.0000              -         -              0\n",
      "#\n",
      "# Speed statistics:\n",
      "#\n",
      "# stage      num seqs  seq/sec  seq/sec/model    nucleotides    nt/sec\n",
      "# ---------  --------  -------  -------------  -------------  --------\n",
      "  search           66   66.000         22.000          82218   82218.0\n",
      "  alignment        66   66.000         66.000          81500   81500.0\n",
      "#\n",
      "# Summed ssu-align CPU time (search):     00:00:00.00  (hh:mm:ss)\n",
      "# Summed ssu-align CPU time (alignment):  00:00:00.00  (hh:mm:ss)\n",
      "# Summed ssu-align CPU time (total):      00:00:00.00  (hh:mm:ss)\n",
      "#\n",
      "# Cleaning up by removing original files that were just merged ... done.\n",
      "#\n",
      "# List of executed commands saved in:     ssr_aln.ssu-merge.log.\n",
      "# Output printed to the screen saved in:  ssr_aln.ssu-merge.sum.\n",
      "#\n",
      "# All output files created in directory ./ssr_aln/\n",
      "#\n",
      "# CPU time:  00:00:00.07  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "./ssr_aln.ssu-align.sh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# _ssu-mask :: mask SSU rRNA alignments\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-mask -s ssu_aln/ssu_aln.bacteria.mask --afa --dna ssr_aln/\n",
      "# date:    Tue Nov  6 15:53:26 2018\n",
      "#\n",
      "# Masking alignments using pre-existing masks...\n",
      "#\n",
      "#                                                     mask    \n",
      "#                                                 ------------\n",
      "# file name                  in/out  type  #cols  incl.  excl.\n",
      "# -------------------------  ------  ----  -----  -----  -----\n",
      "  ssr_aln.bacteria.stk        input   aln   1582      -      -\n",
      "  ssu_aln.bacteria.mask       input  mask   1582    225   1357\n",
      "  ssr_aln.bacteria.mask.pdf  output   pdf   1582    225   1357\n",
      "  ssr_aln.bacteria.mask.afa  output   aln    225      -      -\n",
      "#\n",
      "# All attempts to draw structure diagrams of masks were successful.\n",
      "#\n",
      "# List of executed commands saved in:     ssr_aln.ssu-mask.log.\n",
      "# Output printed to the screen saved in:  ssr_aln.ssu-mask.sum.\n",
      "#\n",
      "# All output files created in directory ./ssr_aln/\n",
      "#\n",
      "# CPU time:  00:00:00.63  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "!ssu-mask -s ssu_aln/ssu_aln.bacteria.mask --dna --afa ssr_aln/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cat ssr_aln/ssr_aln.bacteria.mask.afa ssu_aln/ssu_aln.bacteria.mask.afa > NebraskMREaln_for_tree.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">14558\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGCGGCATGAT\r\n",
      "AAGTTATTGG-CTCAACCTAATAATACTGTCTTGCTTGAGTGCAGAAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGTGTAGATATATGGAAGAACATCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">16770\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGCAGAAT\r\n",
      "AAGTTACTGGTCTCAA-CTGGTAATACTGTTCTGCTAGAGGACAGGAGAGGTTGATGGAA\r\n",
      "TTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAATCAAC\r\n",
      "TATTCTGCCTGACATTGAGGCATGAAAGCGTGGGTAGCAAACCGG\r\n"
     ]
    }
   ],
   "source": [
    "!tail NebraskMREaln_for_tree.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#rename ref MREs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "NewName = {\"KP763319\":\"MRE from Claroideoglomus etunicatum\",\n",
    "           \"KP763328\":\"MRE from Claroideoglomus etunicatum\",\n",
    "          \"KP763360\":\"MRE from Claroideoglomus etunicatum\",\n",
    "           \"KP763286\":\"MRE from Claroideoglomus etunicatum\",\n",
    "           \"FJ984689\": \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"KP763351\" : \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"FJ984694\" : \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"KP763291\" : \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"KP763316\" : \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"KP763336\" : \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"FJ984660\" : \"MRE from Claroideoglomus claroideum\",\n",
    "           \"FJ984671\" : \"MRE from Claroideoglomus claroideum\",\n",
    "           \"FJ984658\" : \"MRE from Claroideoglomus claroideum\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'MRE from Claroideoglomus etunicatum'"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "NewName = {\"KP763319\" : \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"KP763328\" : \"MRE from Claroideoglomus etunicatum\",\n",
    "          \"KP763360\": \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"KP763286\" : \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"FJ984689\": \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"KP763351\" : \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"FJ984694\" : \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"KP763291\" : \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"KP763316\" : \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"KP763336\" : \"MRE from Claroideoglomus etunicatum\",\n",
    "           \"FJ984660\" : \"MRE from Claroideoglomus claroideum\",\n",
    "           \"FJ984671\" : \"MRE from Claroideoglomus claroideum\",\n",
    "           \"FJ984658\" : \"MRE from Claroideoglomus claroideum\",\n",
    "            \"KP763388\" : \"MRE from Gigaspora albida\", \n",
    "           \"KP763383\": \"MRE from Gigaspora albida\", \n",
    "             \"KF378701\":  \"MRE from Gigaspora margarita\",\n",
    "           \"KF378698\":  \"MRE from Gigaspora margarita\",\n",
    "           \"KF378694\" :  \"MRE from Gigaspora margarita\",\n",
    "           \"KP763409\" :  \"MRE from Gigaspora margarita\",\n",
    "           \"KF378704\":  \"MRE from Gigaspora margarita\",\n",
    "           \"FJ984709\": \"MRE from Funneliformis mosseae\",\n",
    "           \"FJ984707\": \"MRE from Funneliformis mosseae\",\n",
    "           \"FJ984704\" : \"MRE from Funneliformis mosseae\",\n",
    "           \"FJ984735\" : \"MRE from Scuttelospora gilmorei\",\n",
    "         \" FJ984646\": \"MRE from Funneliformis caledonius\",\n",
    "            \"FJ984653\":\"MRE from Funneliformis caledonius\",\n",
    "           \"FJ984698\":\"MRE from Funneliformis caledonius\",\n",
    "           \"FJ984654\":\"MRE from Funneliformis caledonius\",\n",
    "           \"FJ984652\":\"MRE from Funneliformis caledonius\",\n",
    "           \"FJ984657\": \"MRE from Funneliformis caledonius\",\n",
    "           \"KU992724\": \"MRE from Rhizophagus clarus\",\n",
    "           \"KU992754\":\"MRE from Rhizophagus clarus\",\n",
    "           \"KU992822\":\"MRE from Rhizophagus clarus\",\n",
    "           \"FJ984724\": \"MRE from Glomus versiforme\",\n",
    "           \"FJ984725\": \"MRE from Glomus versiforme\",\n",
    "            \"KP763367\": \"MRE from Dentiscutata\",\n",
    "\"KP763394\": \"MRE from Cetraspora pellucida\",  \n",
    "\"KM065895\": \"MRE from Racocetra verrucosa\",\n",
    " \"FJ984643\": \"MRE from Ambispora fennica\",\n",
    " \"FJ984635\": \"MRE from Ambispora appendicula\",\n",
    " \"FJ984679\": \"MRE from Geosiphon pyriformis\",\n",
    "           \"KM593997\":\"MRE from Jimgerdemannia lactiflua\",\n",
    "           \"KM594012\":\"MRE from Jimgerdemannia lactiflua\",\n",
    "           \"KM594000\":\"MRE from Jimgerdemannia lactiflua\",\n",
    "           \"KM594005\":\"MRE from Jimgerdemannia lactiflua\",\n",
    "           \"KM594004\":\"MRE from Jimgerdemannia lactiflua\",\n",
    "         \"KM594001\": \"MRE from Jimgerdemannia flammicorona\",\n",
    "           \"KM594003\":\"MRE from Jimgerdemannia flammicorona\",\n",
    "\"KM594011\":\"MRE from Endogone incrassata\",\n",
    "           \"KM594010\":\"MRE from Endogone incrassata\",\n",
    "         \"KM594002\":\"MRE from Endogone sp.\",\n",
    "         \"MG052969\": \"MRE from Mortierellomycotina sp.\",\n",
    "           \"MG052968\":  \"MRE from Mortierellomycotina sp.\",\n",
    " \"MG052970\": \"MRE from Mortierellomycotina rostafinskii\",\n",
    "  \"MG052966\": \"MRE from Mortierellomycotina selenospora\",    \n",
    "\"MG052967\":\"MRE from Lobosporangium transversale\" ,\n",
    "       \"MG052972\":\"MRE from Mortierella  gamsii\",     \n",
    "    \"MG052973\": \"MRE from Mortierella  capitata\" ,\n",
    " \"MG052974\": \"MRE from Mortierella  elongata\",\n",
    "           \"MG052971\": \"MRE from Mortierella  elongata\",\n",
    "\"MG052976\": \"MRE from Mortierella  minutissima\",\n",
    "\"MG052975\": \"MRE from Mortierella  humilis\", \n",
    " \"NR_044811\": \"Mycoplasma synoviae\",   \n",
    " \"KC441811\": \"Bacillus antracis\", \n",
    "\"NR_074620\": \"Mycoplasma mobile\"} \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "          "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "output = open(\"NebraskMREaln_for_tree_renamed.fasta\", \"w\")\n",
    "\n",
    "\n",
    "for record in SeqIO.parse(open('NebraskMREaln_for_tree.fasta', \"rU\"), \"fasta\"):\n",
    "    if record.id in NewName:\n",
    "        record.id = record.\n",
    "        \n",
    "        \n",
    "        out = open(os.path.join(workDir, 'X90478.fasta'), 'w')\n",
    "sso_fa_namestrip = sso_fa[1:]\n",
    "sso_fa_namestrip.insert(0,\">%s\\n\"%sso_acc)\n",
    "out.writelines(sso_fa_namestrip)\n",
    "out.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "output = open(\"NebraskMREaln_for_tree_renamed.fasta\", \"w\")\n",
    "\n",
    "for record in SeqIO.parse(open('NebraskMREaln_for_tree.fasta', \"rU\"), \"fasta\"):\n",
    "    acc = record.id.split(\".\")[0]\n",
    "    if acc in NewName:\n",
    "        record.id = NewName[acc] + \" \" + acc \n",
    "        output.write(\">\"+record.id+\"\\n\"+str(record.seq)+\"\\n\")\n",
    "    else:\n",
    "        output.write(\">\"+record.id+\"\\n\"+str(record.seq)+\"\\n\")\n",
    "   # print(acc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">MG052969.1 Uncultured Mollicutes bacterium clone Msp-GBAus27b 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCTTGGT\r\n",
      "AAGTTGGCGGTCTCAACCCGGCAATACTACTAAGCTCGAGCACAGGAGAGGTTAATGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCACAGATATATGGAAGAACACCGGCGAAGGCAATTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MG052970.1 Uncultured Mollicutes bacterium clone Mros-NRRL A-17819 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTTTAGT\r\n",
      "AAGTTGGCGGTCTCAACCCGACAATACTACTAAACTCGAGTACAGGAGAGGTTAATGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCACAGATATATGGAAGAACACCGGCGAAGGCAATTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MG052968.1 Uncultured Mollicutes bacterium clone Msp-AD185 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCTTGGT\r\n",
      "AAGTTGGCGGTCTCAACCCGGCAATACTACTAAGCTCGAGTACAGGAGAGGTTAATGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCATAGATATATGGAAGAACACCGGCGAAGGCAATTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MG052966.1 Uncultured Mollicutes bacterium clone Msel-CBS 811.68 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATCGTGCAAT\r\n",
      "AAGTTATCGGTCCCAACTCGGTAATACTGTTGCACTAGAGTTCAGGGGAGGTTGATGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAATGTCGGCGAAGGCAGTCAAC\r\n",
      "TACTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MG052967.1 Uncultured Mollicutes bacterium clone Ltra-NRRL 5525 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATTGTGYAAT\r\n",
      "AAGTTATCGGTCCCAACTCGGTAATACTGTTACACTACAGTTCAGGAGAGGTTGATGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAATGTCGGCGAAGGCAGTCAAC\r\n",
      "TATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MG052971.1 Uncultured Mollicutes bacterium clone Melo-NVP80 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAATGTTATCCGGAATTATTGGGCGTAAAGAATGCGTAGATTGTGTAAT\r\n",
      "AAGTCATTGGTCTCAACCTGGTGATACTGTTATGCTAGAGTGCAGGAGAGGTTGATGGAA\r\n",
      "TTCCATGTGGAGCGGTGAAATGCTTAGATATATGGAGGAACACCGGCGAAGGCAGTCAAC\r\n",
      "TATTCTGACTGACATTGAAGCATGAAAGCGTGGGGAGCAAATAGG\r\n",
      ">MG052972.1 Uncultured Mollicutes bacterium clone Mgam-AM1032 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAATGTTATCCGGAATTATTGGGCGTAAAGAATGCGTAGATTGTGTAAT\r\n",
      "AAGTCATTTGTCTCAACCTGGTGATACTGTTACGCTAGAGTGCAGGAGAGGTTGATGGAA\r\n",
      "TTCCATGTGGAGCGGTGAAATGCTTAGATATATGGAGGAACACCGGCGAAGGCAGTCAAC\r\n",
      "TATTCTGACTGACATTGAAGCATGAAAGCGTGGGGAGCAAATAGG\r\n",
      ">MG052973.1 Uncultured Mollicutes bacterium clone Mcap-AD192 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAATGTTATCCGGAATTATTGGGCGTAAAGAATGCGTAGATGGTGAAAT\r\n",
      "AAGTCATTGGTCTCAACCTGGTGATACTGTTTTTCTAGAGTGTAGGAGAGGTTGATGGAA\r\n",
      "TTCCATGTGGAGCGGTGAAATGCATAGATATATGGAGGAACACCGGCGAAGGCAGTCAAC\r\n",
      "TATTCTAACTGACATTGAAGCATGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MG052974.1 Uncultured Mollicutes bacterium clone Melo-AD073 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAATGTTATCCGGAATTATTGGGCGTAAAGAATGCGTAGATTGTTTAAT\r\n",
      "AAGTCATTGGTCTCAACCTGGTGATACTGTTAGACTAGAGTGCAGGAGAGGTTGATGGAA\r\n",
      "TTCCATGTGGAGCGGTGAAATGCTTAGATATATGGAGGAACACCGGCGAAGGCAGTCAAC\r\n",
      "TATTCTGACTGACATTGAAGCATGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MG052975.1 Uncultured Mollicutes bacterium clone Mhum-AD092 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAATGTTATCCGGAATTATTGGGCGTAAAGAATGCGTAGATTGTATAAT\r\n",
      "AAGTCATTGGTCTCAACCTGATGATACTGTTAAACTAGAGTGCAGGAGAGGTTGATGGAA\r\n",
      "TTCCATGTGGAGCGGTGAAATGCTTAGATATATGGAGGAACACCGGCGAAGGCAGTCAAC\r\n",
      "TATTCTGACTGACATTGAAGCATGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MG052976.1 Uncultured Mollicutes bacterium clone Mmin-KOD988 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAATGTTATCCGGAATTATTGGGCGTAAAGAATGCGTAGATTGTATAAT\r\n",
      "AAGTCATTGGTCTCAACCTGATGATACTGTTAAACTAGAGTGCAGGAGAGGTTGATGGAA\r\n",
      "TTCCATGTGGAGCGGTGAAATGCTTAGATATATGGAGGAACACCGGCGAAGGCAGTCAAC\r\n",
      "TATTCTGACTGACATTGAAGCATGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">KM593997.1 Uncultured bacterium clone 8 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTTGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTTTGAT\r\n",
      "AAGTTATCAGTCTCAACCTGGTAATACTGTCATACTCGAGTGCAGGAGAAGTTAACGGAA\r\n",
      "CTCTATATGGAGCGGTAAAATGTGACGATATATAGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KM594012.1 Uncultured bacterium clone 4 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTTGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTTTGAT\r\n",
      "AAGTTATCAGTCTCAACCTGGTAATACTGTCATACTCGAGTGCAGGAGAAGTTAACGGAA\r\n",
      "CTCTATATGGAGCGGTAAAATGTGACGATATATAGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KM594000.1 Uncultured bacterium clone 1 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTTGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTTTGAT\r\n",
      "AAGTTATCAGTCTCAACCTGGTAATACTGTCATACTCGAGTGCAGGAGAAGTTAACGGAA\r\n",
      "CTCTATATGGAGCGGTAAAATGTGACGATATATAGAGGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KM594001.1 Uncultured bacterium clone 13 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGACGGTGTGAT\r\n",
      "AAGTTATCGATCTTAACTTGGTAATACTGTCATACTTGAGGGTAGGAGAAGTTAACAGAA\r\n",
      "TTTCATATGGAGCGGTAAAATGTGACGATATATGAAGGAATATCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTAACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KM594003.1 Uncultured bacterium clone 5 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGACGGTGTGAT\r\n",
      "AAGTTATTGGTCTCAACCTGGTAATACTGTCATGCTCGAGTTCAGGAGGAGTTAGCGGAA\r\n",
      "TTCTATGTGGAATGGTAAAATATGCTAATATATAGAGGAACACCGGCGAAGGCAGCTAAC\r\n",
      "TATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KM594005.1 Uncultured bacterium clone 12 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTGTGAT\r\n",
      "AAGTTATCAGTCTCAACCTGGTAATACTGTTATACTTGAGTTCAGGAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KM594011.1 Uncultured bacterium clone 5 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTGTAAT\r\n",
      "AAGTTATCAGTCTCAACCTGGTAATACTGTTATACTTGAGTTCAGGAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KM594002.1 Uncultured bacterium clone 3 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTCGCAAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGCTGGAT\r\n",
      "AAGTTATCAGTCTCAACCTGGTAATACTGTTCGGCTTGAATACAGGAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACATCGGCGAAAGCAGTTAAC\r\n",
      "TATTCTGATTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KM594010.1 Uncultured bacterium clone 8 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTGCGAT\r\n",
      "AAGTTATCAGTCTCAACTTGGTAATACTGTCGTGCTTGAGTATAGGAGAGGTTAACGGAA\r\n",
      "CTTCATGTGGAGCGGTAAAATGTGTAGATATATGAAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTAACTGACATTGAGGCATGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">KM594004.1 Uncultured bacterium clone 15 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTATGAT\r\n",
      "AAGTTATCAGTCTCAACTTGGTAATACTGTCATACTAGAGTACAGAAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCGTAGATATATGGATGAACATCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGTCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">NR_044811.1 Mycoplasma synoviae strain WVU 1853 16S ribosomal RNA, partial sequence\r\n",
      "ACATAGGTTGCAAGCGTTATCCGGAATTATTGGGCGTAAAGCGTCTGTAGGTTGTTTGTT\r\n",
      "AAGTCTGGTGTCTCAACCATTGGATACTGGCAGACTAGAATTGTTTAGAGGTTAGCGGAA\r\n",
      "TTCCTTGTGAAGCGGTGGAATGCGTAGATATAAGGAAGAACACCGGCGAAGGCAGCTAAC\r\n",
      "TGGGAACATTGACACTGAGAGACGAAAGCGTGGGGAGCAAACAGG\r\n",
      ">KP763319.1 Candidatus Moeniiplasma glomeromycotorum isolate MRE_CE_CA301_1_16 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGAAT\r\n",
      "AAGTTATTGGTCTCAATCTGGTAATACTGTTCGGCTAGAGGGCAGGAGAAGTTGATGGAA\r\n",
      "TTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAGTCAAC\r\n",
      "TATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">KP763328.1 Candidatus Moeniiplasma glomeromycotorum isolate MRE_CE_CL372_1_5 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGAAT\r\n",
      "AAGTTATTGGTCTCAATCTGGTAATACTGTTCGGCTAGAGGGCAGGAGAAGTTGATGGAA\r\n",
      "TTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAGTCAAC\r\n",
      "TATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">KP763360.1 Candidatus Moeniiplasma glomeromycotorum isolate MRE_CE_MX916B_2_3 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGAAT\r\n",
      "AAGTTATTGGTCTCAATCTGGTAATACTGTTCGGCTAGAGGGCAGGAGAAGTTGATGGAA\r\n",
      "TTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAGTCAAC\r\n",
      "TATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">FJ984660.1 Uncultured bacterium clone MN_108_5 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGAAT\r\n",
      "AAGTTATTGGTCTCAATCTGGTAATACTGTTCTACTCGAGGGCAGGAGAGGTTGATGGAA\r\n",
      "TTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAGTCAAC\r\n",
      "TATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">KP763388.1 Candidatus Moeniiplasma glomeromycotorum isolate MRE_GA_BR601A_K1_12 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGGAT\r\n",
      "AAGTTACTGATCTCAATTCAGTAATACTGTTCTGCTAGAGGGCAGGAGAGGTTGATGGAA\r\n",
      "TTCTGTGTGGAGGGGTGAAATCCGTTGATCTGCAGAAGAACACCGGCGAAGGCACTCAAC\r\n",
      "TATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">KF378701.1 Uncultured bacterium clone 22 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGAAT\r\n",
      "AAGTTACTGATCTCAATTTGGTAATACTGTTCTGCTAGAGGGCAGAAGAGGTTGATGGAA\r\n",
      "TTCTGTGTGGAGCGGTGAAATGCGTTGATCTGCAGAGGAACACCGGCGAAGGCATTCAAC\r\n",
      "TATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">KF378698.1 Uncultured bacterium clone 91 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGAAT\r\n",
      "AAGTTACTGATCTCAATTTGGTAATACTGTTTTGCTAGAGGGCAGAAGAGGTTGATGGAA\r\n",
      "TTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCAGCGAAGGCATTCAAC\r\n",
      "TATTCTGCCTGACATTAAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">FJ984671.1 Uncultured bacterium clone MN_151_4 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGATGGTGGAAT\r\n",
      "AAGTTATTGGTCTCAACCTGATAATACTGTTCTACTTGAGGACAGGAGAGGTTGGCGGAA\r\n",
      "TTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAGCCAAC\r\n",
      "TATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">FJ984698.1 Uncultured bacterium clone MN_467_7 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGATGGTAGGAT\r\n",
      "AAGTTATTGGTCTCAACCTGATAATACTGTTCTGCTAGAAGAGAGGAGAGGTTAGTGGAA\r\n",
      "TTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAACTAAC\r\n",
      "TATTCTTCTTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">KP763286.1 Candidatus Moeniiplasma glomeromycotorum isolate MRE_CE_AZ201C_2_2 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGATGGTGGAAT\r\n",
      "AAGTTATTGGTCTCAACCTGGTAATACTGTTCTACTCGAGGACAGGAGAGGTTGGCGGAA\r\n",
      "TTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAGCCAAC\r\n",
      "TATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">FJ984689.1 Uncultured bacterium clone MN_74_2 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCAAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGATGGTGGAAC\r\n",
      "AAGTTATCGGTCTCAACCTGGTAATACTATTCTACTTGAGGACAGGAGAGGTTGGCGGAA\r\n",
      "TTCTGTGTGGAGCGGTGAAATGCGTTGATCTACANAGGAACACCGGCGAAGGCAGCCAAC\r\n",
      "TATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">FJ984658.1 Uncultured bacterium clone MN_108_3 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGGTGCGTAGATGGTAGAAT\r\n",
      "AAGTTATTAGCCTCAACTTAATAATACTGTTTTGCTAGAGAATGGAAGGAGTTAATGGAA\r\n",
      "TTCTATGTGTAGAGGTGAAATTCGTTGATCTATAGAGGACCACCGGCGAAGGCAATTAAC\r\n",
      "TATTCTATCTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">FJ984709.1 Uncultured bacterium clone MN_265_7 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCAAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGATGGCGGAAT\r\n",
      "AAGTTATCAGTCTCAACCTGATAATACTATTCTGCTAGAGGACAGGAGAGGTTGGCGGAA\r\n",
      "CTCTATGCGTAGGGGTGAAATCCATTGATCTATAGAGGACCACCGGCGAAGGCAGCCAAC\r\n",
      "TATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KF378694.1 Uncultured bacterium clone 70 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGAAT\r\n",
      "GAGTTACTGATCTCAACCTGATAATACTGTTCTGCTAGAGGAGAGGAGAGGTTAGTGGAA\r\n",
      "TTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAACTAAC\r\n",
      "TATTCTTCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">FJ984735.1 Uncultured bacterium clone MN_344_7 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGATGGTAGGAT\r\n",
      "AAGTTATTGGTCTCAACTTGATAATACTGTTCTGCTAGAGGAGAGGAGAGGTTAGTGGAA\r\n",
      "TTCTGTGTGTAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAACTAAC\r\n",
      "TATTCTTCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">FJ984646.1 Uncultured bacterium clone MN_107_3 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTACTGGGCGTAAAGAGTGCGTAGGTGGTTCAAT\r\n",
      "AAGTTACTGGTCCCAACTTGGTAATACTGATGAACTAGAGGACCGGAGAGGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCGGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAGACGG\r\n",
      ">FJ984653.1 Uncultured bacterium clone MN_105_08 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTACTGGGCGTAAAGAGTGCGTAGGTGGTTTAAT\r\n",
      "AAGTTACTGGTCCCAACTTGGTAATACTATTAGACTAGAGGACCGGAGAGGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCGGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAGACGG\r\n",
      ">KP763351.1 Candidatus Moeniiplasma glomeromycotorum isolate MRE_CE_KS887_4_2 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTACTGGGCGTAAAGGGTGCGTAGGTGGTTTAAT\r\n",
      "AAGTTACTGGTCCCAACTCAGTAATACTATTAAACTAGAGGGCCGGAGAGGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCGGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAGACGG\r\n",
      ">FJ984707.1 Uncultured bacterium clone MN_265_8 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGAGTGCGTAGGTGGTTTAAT\r\n",
      "AAGTTACTGGTCTCAACTCAGTAATACTATTAAACTAGAGGGTCGGAGAGGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGCTTAAATATATAGAGGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCGACCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAGACGG\r\n",
      ">FJ984704.1 Uncultured bacterium clone MN_269_6 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGAGTGCGTAGGTGGTTTAAT\r\n",
      "AAGTTACTGGTCTCAACTCAGTAATACTATTAAACTAGAGGGTCGGAGAAGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGCTTAAATATATAGAGGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCGACCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAGACGG\r\n",
      ">KU992724.1 Candidatus Glomeriplasma moeniorum clone MRE_RC_AU402B_1_1 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTACTGGGCGTAAAGGGTGCGTAGGTGGTTGAAT\r\n",
      "AAGTTACTGGTCTCAACTCAGTAATACTATCCAACTCGAGGGCTGGAGAGGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAAGCAGTTAAC\r\n",
      "TATTCAGCCTGACATTGAGGCACGAAAGCGTGGGGAGCAATACGG\r\n",
      ">KU992754.1 Candidatus Glomeriplasma moeniorum clone MRE_RC_CL156_2_2 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTACTGGACGTAAAGGGTGCGTAGGTGGTTGAAT\r\n",
      "AAGTTACTGGTCTCAACTCAGTAATACTATCCAACTCGAGGGCTGGAGAGGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAAGCAGTTAAC\r\n",
      "TATTCAGCCTGACATTGAGGCACGAAAGCGTGGGGAGCAATACGG\r\n",
      ">KU992822.1 Candidatus Glomeriplasma moeniorum clone MRE_RC_ND269B_1_1 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTACTGGGCGTAAAGGGTGCGTACGTGGTTGAAT\r\n",
      "AAGTTACTGGTCTCAACTCAGTAATACTATCCAACTCGAGGGCTGGAGAGGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAAGCAGTTAAC\r\n",
      "TATTCAGCCTGACATTGAGGCACGAAAGCGTGGGGAGCAATACGG\r\n",
      ">FJ984684.1 Uncultured bacterium clone MN_71_2 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTATGAT\r\n",
      "AAGTTAGCGGTCTCAACCTGGTAATACTATCGTGCTAGAGTGTAGGAGAGGTTAACGGAA\r\n",
      "CTCCCCGTGGAGCGGTGAAATGCATAGATATGGGGAAGAACACCGGCGAAGGCAGTCAAC\r\n",
      "TATTCTAACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">FJ984654.1 Uncultured bacterium clone MN_105_09 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTCGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTTCAGT\r\n",
      "AAGTTACTGGTCCTAGCTCAGTAATACTGTTGAACTTGAGTGCAGGAGAGGTTAACGGAA\r\n",
      "CTTCATGTGGAGCGGTAAAATGCGTAGATATATGAAAGAACATCAGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">FJ984694.1 Uncultured bacterium clone MN_377_2 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTATGAT\r\n",
      "AAGTTAGCGGTCTCAACCTGGTAATACTATCGTGCTAGAGTGTAGGAGAGGTTAACGGAA\r\n",
      "CTCCCCGTGGAGCGGTGAAATGCATAGATATGGGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTAACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">FJ984652.1 Uncultured bacterium clone MN_105_07 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCTTGGT\r\n",
      "AAGTTAGCAGTCTCAACCTGATAATACTGTCTGGCTAGAGTGCAGGAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGTGTAGATATATGGAAGAACACCGGCGAANGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCMAAACGG\r\n",
      ">FJ984657.1 Uncultured bacterium clone MN_105_02 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCTTGGT\r\n",
      "AAGTTAGCAGTCTCAACCTGATAATACTGTCTGGCTAGAGTGCAGGAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGTGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">FJ984724.1 Uncultured bacterium clone MN_101_2 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCAAACATTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGCGGCATGAT\r\n",
      "AAGTTGCTGGTCTCAACCTAGCAATACTGTCAAGCTAGAGGGCAGAAGAGGTTAACGGAA\r\n",
      "CTCTATGTGGAGCGGTAAAATGTGTAGATATATAGAAGAACATCGGCGAAGGCAGTTAAC\r\n",
      "TAGTCTGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">FJ984725.1 Uncultured bacterium clone MN_103_3 16S ribosomal RNA gene, partial sequence\r\n",
      "ACATAGGTGGCAAACATTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGCGGCATGAT\r\n",
      "AAGTTGCTGGTCTCAACCTAGCAATACTGTCAAGCTAGAGGGCAGAAGAGGTTAACGGAA\r\n",
      "CTCTATGTGGAGCGGTAAAATGTGTAGATATATAGAAGAACATCGGCGAAGGCAGTTAAC\r\n",
      "TAGTCTGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KP763367.1 Candidatus Moeniiplasma glomeromycotorum isolate MRE_DH_BR155A_A1_1 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGCGGCGGAAT\r\n",
      "AAGTTAGTGGTCTCAACTCATTAATACTGTTTTGCTAGAGGACAGGAGAGGTTAACGGAA\r\n",
      "TTCCCTGTGGAGCGGTAAAATGTGTAGATATAGGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAATAGG\r\n",
      ">KP763383.1 Candidatus Moeniiplasma glomeromycotorum isolate MRE_GA_BR601A_K1_4 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGACGGCAGAAT\r\n",
      "AAGTTAGTGGTCTCAACCCATTAATACTGTTTTGCTAGAGGGCAGAAGAGGTTAACGGAA\r\n",
      "CTCCCTGTGGAGCGGTAAAATGTGTAGATATAGGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAATAGG\r\n",
      ">KP763394.1 Candidatus Moeniiplasma glomeromycotorum isolate MRE_CP_CL750A_F2_2 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGTATCATTGGGCGTAAAGGGTGCGTAGATGGCCGAAT\r\n",
      "AAGTTATCGGTCTCAACCCGGTAATACTGTTTGGCTCGAGTGCGGGAGAGGTTAACGGAA\r\n",
      "CTCCTTGTGGAGCGGTAAAATGTGTAGATATAAGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCCGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAATAGG\r\n",
      ">KP763409.1 Candidatus Moeniiplasma glomeromycotorum isolate MRE_GM_JA201A_AP1_1 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTANATGGTCCGTT\r\n",
      "AAGTTAGCAGTCTCAACTTGCTAATACTGGCAGACTAGAGGACAGGAGAGGTTAACGGAA\r\n",
      "CTCCCGGTGGAGCGGTGAAATGCGTAGATATCGGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGCCTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KM065895.1 Candidatus Moeniiplasma glomeromycotorum isolate RV-B-1 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTCTGCT\r\n",
      "AAGTTAGCAGTCTCAACTTGCTAATACTAGCAGACTAGAGGACAGGAGAGGTTGACGGAA\r\n",
      "CTCCCGGTGGAGCGGTGAAATGCGTAGATATCGGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGCCTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KF378704.1 Uncultured bacterium clone 49 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCAGAAT\r\n",
      "AAGTTAGTGGTCTTAACCCATTAATACTGTTTTGCTAGAGGACGGGGGGGGTTAACGGAA\r\n",
      "CTCCCGGTGGAGCGGTGAAATGCGTAGATATCGGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGCCTGACATTGAGGCACGAAAGCGTGGGTAGCAAAGCGG\r\n",
      ">FJ984643.1 Uncultured bacterium clone MN_501_2 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGCGGCATGAT\r\n",
      "AAGTTATTGG-CTCAACCTGATAATACTGTCTTGCTTGAGTGCAGAAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGTGTAGATATATGGAAGAACATCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">KP763291.1 Candidatus Moeniiplasma glomeromycotorum isolate MRE_CE_AZ201C_2_7 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGACGGCATGAT\r\n",
      "AAGTTATTGG-CTCAACCTGATAATACTGTTTTGCTTGAGTACAGGAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGTGTAGATATATGGAAGAACATCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">FJ984635.1 Uncultured bacterium clone MN_443_23 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCCTGGT\r\n",
      "AAGTTAGCAGTCTCAACCTGGTAATACTGCTTGGCTTGAGTGCAGGAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">FJ984679.1 Uncultured bacterium clone MN_279_7 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTCTAAT\r\n",
      "AAGTTATCGGTCTCAACTCGGTAATACTGTTATACTCGAGTGCAGAAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACACCGGCGAAAGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KP763316.1 Candidatus Moeniiplasma glomeromycotorum isolate MRE_CE_CA301_1_12 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATCATTGGGCGTAAAGGGTGCGTAGATGGCATGAT\r\n",
      "AAGTTATTAGTCTCAACCTAGTAATACTGTTTTGCTTGAGTACAGAAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KP763336.1 Candidatus Moeniiplasma glomeromycotorum isolate MRE_CE_IA205_1_2 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATCATTGGGCGTAAAGGGTGCGTAGATGGCATGAT\r\n",
      "AAGTTATTAGTCTCAACCTAGTAATACTGTTTTGCTTGAGTACAGAAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">KC441811.1 Bacillus anthracis strain N34 16S ribosomal RNA gene, partial sequence\r\n",
      "ACGTAGGTGGCAAGCGTTATCCGGAATTATTGGGCGTAAAGCGCGCGCAGGTGGTTTCTT\r\n",
      "AAGTCTGATGTCTCAACCATTGGAAACTGGGAGACTTGAGTGCAGAAGAGGAAAGTGGAA\r\n",
      "TTCCATGTGTAGCGGTGAAATGCGTAGAGATATGGAGGAACACCGGCGAAGGCGACTTTC\r\n",
      "TGGTCTGACTGACACTGAGGCGCGAAAGCGTGGGGAGCAAACAGG\r\n",
      ">NR_074620.2 Mycoplasma mobile strain 163K 16S ribosomal RNA, complete sequence\r\n",
      "ACATAGGTCGCAAGCGTTATCCGGAATTATTGGGCGTAAAGCGTTCGTAGGCGGTTTATT\r\n",
      "AAGTCTGGAGTCTCAACCTTTGGATACTGGTAAACTAGAGTTATGTAGAGGTAAGCGGAA\r\n",
      "TTCCATGTGAAGTGGTGGAATACGTAGATATATGGAAGAACACCGGCGAAGGCAGCTTAC\r\n",
      "TGGACATACTGACGCTCAGGAACGAAAGCGTGGGGAGCAAACAGG\r\n",
      ">8319\r\n",
      "ACATAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGATGGTGGAAT\r\n",
      "AAGTTATTGGTCTCAACCTGGTAATACTGTTCTACTCGAGGACAGGAGAGGTTGGCGGAA\r\n",
      "TTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAGCCAAC\r\n",
      "TATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">8660\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCCTGAC\r\n",
      "AAGTTAGCAGTCTCAACCTGGTAATACTGTCTGGCTCGAGTGCAGGAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">14224\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGGTGGTAGAAT\r\n",
      "AAGTTATCAGTCTCAACCTGGTAAAACTATTCTGCTAGAGGAAAGGAGAGGTTAGCGGAA\r\n",
      "CTCTATGAGTAAGGGTGAAATCTATTGATCCATAGAGGACCACCGGCGAAGGCAGCTAAC\r\n",
      "TATTCTACCTGACACTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">2872\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTACTGGGCGTAAAGAGTGCGTAGGTGGTTCAAT\r\n",
      "AAGTTACTGGTCTCAACTCAGTAATACTATTAAACTAGAGGGTCGGAGAGGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGCTTAAATATATAGAGGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCGACCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">3001\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTACTGGGCGTAAAGTGTGCGTAGGTGGTTGAAT\r\n",
      "AAGTTACTGGTCTCAACCCAGTAATACTATTCAACTCGAGGACCGGAGAGGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACATCGGCGAAAGCAGTTAAC\r\n",
      "TATTCGGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">4190\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTACTGGGCGTAAAGAGTGCGTAGGTGGTTCAAT\r\n",
      "AAGTTACTGGTCCCAACTTGGTAATACTGTTGAACTAGAGGACCGGAGAGGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCGGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">4442\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCCTGAC\r\n",
      "AAGTTAACAGTCTCAACCTGATAATACTGTCTGGCTTGAGTACAGAAGAGGTTAACGGAA\r\n",
      "TTCCATGTGGAGCGGTAAAATGCACAGATATATGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">4562\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATTGTATGGT\r\n",
      "AAGTTACTGGTCTCAACTCAGTAATACTGCTATGCTAGAGTTCAGGAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACGCCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">5222\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTACTGGGCGTAAAGGGTGCGTAGGTGGTTTAAT\r\n",
      "AAGTTATCGGTCCCAACTCGGTAATACTATTAGACTAGAGGGCTGGAGAGGTTAACGGAA\r\n",
      "TGGTTCATGTAGGGGTAAAATCCGTAAATATGAATTGGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCAGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">6878\r\n",
      "ACATAGGTCGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTTCAGT\r\n",
      "AAGTTACTGGTCCTAACTCAGTAATACTGTTGAACTTGAGTGCAGGAGAGGTTAACGGAA\r\n",
      "CTTCATGTGGAGCGGTAAAATGCGTAGATATATGAAAGAACATCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTAAGGCATGAAAGCGTGGGGAGCAAA----\r\n",
      ">7420\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTTTAAT\r\n",
      "AAGTTAGCGGTCTCAACTCGCTAATACTATCAAACTGGAGTGCAGGAGAGGTTAACGGAA\r\n",
      "CTCCATGTGTAGCGGTAAAATGCGACGATATATGGATGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">7660\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGAGTGCGTAGGTGGTTTAAT\r\n",
      "AAGTTACTGGTCTCAACTCAGTAATACTGTTAGACTAGAGGATCGGAGGGGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGCTCAGATACATAGAGGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCGACCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">7941\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTACTGGGCGTAAAGAGTGCGTAGGTGGTTTAAT\r\n",
      "AAGTTATTGGTCTCAACCCAGTAATACTATTGAACTAGAGGGCTGGAGAGGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCAGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">8498\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGTGTGCGTAGGTGGTTTAAT\r\n",
      "AAGTTACTGGTCTCAACTCGGTAATACTATTAGACTAGAGGATCGGAGAGGTTAACGGAA\r\n",
      "TGGTTCGTGTAGGGGTAAAATCCTCAAATATGAATTGGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCGGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">8522\r\n",
      "ACATAGGTTGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGCTTGAT\r\n",
      "AAGTTATCAGTCTTAACCTGGTAATACTGTCATGCTTGAGTGCAGGAGAGGTTAACGGAA\r\n",
      "CTCTATGTGGAGCGGTAAAATGTGACGATATATAGAAGAACACCGGCGAAAGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAA----\r\n",
      ">8827\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTCGAAT\r\n",
      "AAGTTAGCAGTCCTAACCTGGTAATACTGTTCGACTGGAGGGCAGGAGAGGTTAATGGAA\r\n",
      "CTCTATGTGGAACGGTAAAATGTGACGATATATAGATGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGCCTGACATTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">9232\r\n",
      "ACATAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGGTGCGTAGATGGTAGAAC\r\n",
      "AAGTTA----TCTCAACTTAATAATACTGTTTTGCTAGAGAATGGAAGAGGTTAATGGAA\r\n",
      "TTCTATGTGTAGAGGTGAAATTCGTTGATCCATAGAGGACCACCGGCGAAGGCAATTAAC\r\n",
      "TATTCCATCTGACATTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">9472\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGCGGCATGAT\r\n",
      "AAGTTATTGG-CTCAACCTGATAATACTGTTTTGCTTGAGTACAGGAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGTGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACGCTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">9533\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTTTGAT\r\n",
      "AAGTTAGCGGTCTCAACTCGCTAATACTATCAAACTAGAGGGCAGGAGAGGTTAACGGAA\r\n",
      "CTCTATGTGGAGCGGTAAAATGTATAGATATATAGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">9550\r\n",
      "ACATAGGTCGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTGTGAT\r\n",
      "AAGTTATTGGTCTCAACCTGATAATACTGTCACACTAGAGTGTAGGAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACATCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTAACTGACATTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">10226\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGAGTGCGTAGGTGGTTTAAT\r\n",
      "AAGTTATCGGTCCCAACTCGATAATACTATTAAACTAGAGGGCTGGAGAAGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGTGTAAATATATAGAGGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCGGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">10910\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGACGGCAGGAT\r\n",
      "AAGTTATTGGTCTCAACCTGATAATACTGTTTTGCTTGAGTACAGGAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACATCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">11169\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGTGGCCTGAC\r\n",
      "AAGTTATCAGTCTCAACCTGGTAATACTGTCCGGCTTGAGTACAGGAGAGGTTAACGGAA\r\n",
      "CTCTATGTGGAGCGGTAAAATGCACAGATATATAGAAGAACATCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACGCTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">11600\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATCATTGGGCGTAAAGGGTGCGTAGGCGGCAGGAT\r\n",
      "AAGTTATTGG-CTCAACCTGATAATACTGTCTTGCTTGAGTGCAGGAGAGGTTAACGGAA\r\n",
      "TTCTTTGTGGAGGGGTAAAATCCAAAGATATAAAGGAGAACATCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">13308\r\n",
      "ACATAGGTCGCAAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTTCAGT\r\n",
      "AAGTTACTGGTCTCAACTCAGTAATACTGCTGAACTTGAGTGCAGGAGAGGTTAACGGAA\r\n",
      "CTCTATGTGTAGGGGTAAAATCCATAGATATATAGATGAACATCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAA----\r\n",
      ">13619\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCTTGAC\r\n",
      "AAGTTAGTGGTCTCAACTCACTAATACTATCAAACTAGAGTGCAGGAGAGGTTAATGGAA\r\n",
      "CTCCATGTGTAGCGGTAAAATGCGACGATATATGGATGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">13848\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTTGGAA\r\n",
      "AAGTTAGTGGTCCCAACCTATTAATACTATCTGACTTGAGAACAGGAGAGGTTAACGGAA\r\n",
      "CTCTATGTGGAGCGGTAAAATGCGACGATATATAGAAGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGTCTGACATTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">13928\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGAGTGCGTAGGTGGTTTAAT\r\n",
      "AAGTTAGTAGTCTCAACTTACTAATACTATTAAACTAGAGGATCGGAGAGGTTAACGGAA\r\n",
      "TTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCGACCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">14163\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAAAAT\r\n",
      "AAGTTACCGGTCTCAATCTAGTAATACTATTTTGCTAGAGGATAGGAGAGGTTGATGGAA\r\n",
      "CTCTATGCGTAGGGGTGAAATCCGTTGATCTATAGAGGAACACCGGCGAAGGCAATCAAC\r\n",
      "TATTCTACCTGACATTGAGGCATGAAAGCGTGGGTAGCAAACCGG\r\n",
      ">14386\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTTTGAT\r\n",
      "AAGTTATTAGTCTCAACTTAGTAATACTATTAGACTAGAGTACAGGAGAGGTTAATGGAA\r\n",
      "CTCTATGTTTAGGGGTAAAATCCGCAGATATA--GATGAACACCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">14558\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGCGGCATGAT\r\n",
      "AAGTTATTGG-CTCAACCTAATAATACTGTCTTGCTTGAGTGCAGAAGAGGTTAACGGAA\r\n",
      "CTCCATGTGGAGCGGTAAAATGTGTAGATATATGGAAGAACATCGGCGAAGGCAGTTAAC\r\n",
      "TATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">16770\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGCAGAAT\r\n",
      "AAGTTACTGGTCTCAA-CTGGTAATACTGTTCTGCTAGAGGACAGGAGAGGTTGATGGAA\r\n",
      "TTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAATCAAC\r\n",
      "TATTCTGCCTGACATTGAGGCATGAAAGCGTGGGTAGCAAACCGG\r\n"
     ]
    }
   ],
   "source": [
    "!cat NebraskMREaln_for_tree.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">MRE from Mortierellomycotina sp. MG052969\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCTTGGTAAGTTGGCGGTCTCAACCCGGCAATACTACTAAGCTCGAGCACAGGAGAGGTTAATGGAACTCCATGTGGAGCGGTAAAATGCACAGATATATGGAAGAACACCGGCGAAGGCAATTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Mortierellomycotina rostafinskii MG052970\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTTTAGTAAGTTGGCGGTCTCAACCCGACAATACTACTAAACTCGAGTACAGGAGAGGTTAATGGAACTCCATGTGGAGCGGTAAAATGCACAGATATATGGAAGAACACCGGCGAAGGCAATTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Mortierellomycotina sp. MG052968\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCTTGGTAAGTTGGCGGTCTCAACCCGGCAATACTACTAAGCTCGAGTACAGGAGAGGTTAATGGAACTCCATGTGGAGCGGTAAAATGCATAGATATATGGAAGAACACCGGCGAAGGCAATTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Mortierellomycotina selenospora MG052966\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATCGTGCAATAAGTTATCGGTCCCAACTCGGTAATACTGTTGCACTAGAGTTCAGGGGAGGTTGATGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAATGTCGGCGAAGGCAGTCAACTACTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MRE from Lobosporangium transversale MG052967\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATTGTGYAATAAGTTATCGGTCCCAACTCGGTAATACTGTTACACTACAGTTCAGGAGAGGTTGATGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAATGTCGGCGAAGGCAGTCAACTATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MRE from Mortierella  elongata MG052971\r\n",
      "ACGTAGGTGGCGAATGTTATCCGGAATTATTGGGCGTAAAGAATGCGTAGATTGTGTAATAAGTCATTGGTCTCAACCTGGTGATACTGTTATGCTAGAGTGCAGGAGAGGTTGATGGAATTCCATGTGGAGCGGTGAAATGCTTAGATATATGGAGGAACACCGGCGAAGGCAGTCAACTATTCTGACTGACATTGAAGCATGAAAGCGTGGGGAGCAAATAGG\r\n",
      ">MRE from Mortierella  gamsii MG052972\r\n",
      "ACGTAGGTGGCGAATGTTATCCGGAATTATTGGGCGTAAAGAATGCGTAGATTGTGTAATAAGTCATTTGTCTCAACCTGGTGATACTGTTACGCTAGAGTGCAGGAGAGGTTGATGGAATTCCATGTGGAGCGGTGAAATGCTTAGATATATGGAGGAACACCGGCGAAGGCAGTCAACTATTCTGACTGACATTGAAGCATGAAAGCGTGGGGAGCAAATAGG\r\n",
      ">MRE from Mortierella  capitata MG052973\r\n",
      "ACGTAGGTGGCGAATGTTATCCGGAATTATTGGGCGTAAAGAATGCGTAGATGGTGAAATAAGTCATTGGTCTCAACCTGGTGATACTGTTTTTCTAGAGTGTAGGAGAGGTTGATGGAATTCCATGTGGAGCGGTGAAATGCATAGATATATGGAGGAACACCGGCGAAGGCAGTCAACTATTCTAACTGACATTGAAGCATGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MRE from Mortierella  elongata MG052974\r\n",
      "ACGTAGGTGGCGAATGTTATCCGGAATTATTGGGCGTAAAGAATGCGTAGATTGTTTAATAAGTCATTGGTCTCAACCTGGTGATACTGTTAGACTAGAGTGCAGGAGAGGTTGATGGAATTCCATGTGGAGCGGTGAAATGCTTAGATATATGGAGGAACACCGGCGAAGGCAGTCAACTATTCTGACTGACATTGAAGCATGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MRE from Mortierella  humilis MG052975\r\n",
      "ACGTAGGTGGCGAATGTTATCCGGAATTATTGGGCGTAAAGAATGCGTAGATTGTATAATAAGTCATTGGTCTCAACCTGATGATACTGTTAAACTAGAGTGCAGGAGAGGTTGATGGAATTCCATGTGGAGCGGTGAAATGCTTAGATATATGGAGGAACACCGGCGAAGGCAGTCAACTATTCTGACTGACATTGAAGCATGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MRE from Mortierella  minutissima MG052976\r\n",
      "ACGTAGGTGGCGAATGTTATCCGGAATTATTGGGCGTAAAGAATGCGTAGATTGTATAATAAGTCATTGGTCTCAACCTGATGATACTGTTAAACTAGAGTGCAGGAGAGGTTGATGGAATTCCATGTGGAGCGGTGAAATGCTTAGATATATGGAGGAACACCGGCGAAGGCAGTCAACTATTCTGACTGACATTGAAGCATGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MRE from Jimgerdemannia lactiflua KM593997\r\n",
      "ACATAGGTTGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTTTGATAAGTTATCAGTCTCAACCTGGTAATACTGTCATACTCGAGTGCAGGAGAAGTTAACGGAACTCTATATGGAGCGGTAAAATGTGACGATATATAGAAGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Jimgerdemannia lactiflua KM594012\r\n",
      "ACATAGGTTGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTTTGATAAGTTATCAGTCTCAACCTGGTAATACTGTCATACTCGAGTGCAGGAGAAGTTAACGGAACTCTATATGGAGCGGTAAAATGTGACGATATATAGAAGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Jimgerdemannia lactiflua KM594000\r\n",
      "ACATAGGTTGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTTTGATAAGTTATCAGTCTCAACCTGGTAATACTGTCATACTCGAGTGCAGGAGAAGTTAACGGAACTCTATATGGAGCGGTAAAATGTGACGATATATAGAGGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Jimgerdemannia flammicorona KM594001\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGACGGTGTGATAAGTTATCGATCTTAACTTGGTAATACTGTCATACTTGAGGGTAGGAGAAGTTAACAGAATTTCATATGGAGCGGTAAAATGTGACGATATATGAAGGAATATCGGCGAAGGCAGTTAACTATTCTAACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Jimgerdemannia flammicorona KM594003\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGACGGTGTGATAAGTTATTGGTCTCAACCTGGTAATACTGTCATGCTCGAGTTCAGGAGGAGTTAGCGGAATTCTATGTGGAATGGTAAAATATGCTAATATATAGAGGAACACCGGCGAAGGCAGCTAACTATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Jimgerdemannia lactiflua KM594005\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTGTGATAAGTTATCAGTCTCAACCTGGTAATACTGTTATACTTGAGTTCAGGAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Endogone incrassata KM594011\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTGTAATAAGTTATCAGTCTCAACCTGGTAATACTGTTATACTTGAGTTCAGGAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Endogone sp. KM594002\r\n",
      "ACATAGGTCGCAAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGCTGGATAAGTTATCAGTCTCAACCTGGTAATACTGTTCGGCTTGAATACAGGAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACATCGGCGAAAGCAGTTAACTATTCTGATTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Endogone incrassata KM594010\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTGCGATAAGTTATCAGTCTCAACTTGGTAATACTGTCGTGCTTGAGTATAGGAGAGGTTAACGGAACTTCATGTGGAGCGGTAAAATGTGTAGATATATGAAAGAACACCGGCGAAGGCAGTTAACTATTCTAACTGACATTGAGGCATGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MRE from Jimgerdemannia lactiflua KM594004\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTATGATAAGTTATCAGTCTCAACTTGGTAATACTGTCATACTAGAGTACAGAAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGATGAACATCGGCGAAGGCAGTTAACTATTCTGTCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">Mycoplasma synoviae NR_044811\r\n",
      "ACATAGGTTGCAAGCGTTATCCGGAATTATTGGGCGTAAAGCGTCTGTAGGTTGTTTGTTAAGTCTGGTGTCTCAACCATTGGATACTGGCAGACTAGAATTGTTTAGAGGTTAGCGGAATTCCTTGTGAAGCGGTGGAATGCGTAGATATAAGGAAGAACACCGGCGAAGGCAGCTAACTGGGAACATTGACACTGAGAGACGAAAGCGTGGGGAGCAAACAGG\r\n",
      ">MRE from Claroideoglomus etunicatum KP763319\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGAATAAGTTATTGGTCTCAATCTGGTAATACTGTTCGGCTAGAGGGCAGGAGAAGTTGATGGAATTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAGTCAACTATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">MRE from Claroideoglomus etunicatum KP763328\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGAATAAGTTATTGGTCTCAATCTGGTAATACTGTTCGGCTAGAGGGCAGGAGAAGTTGATGGAATTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAGTCAACTATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">MRE from Claroideoglomus etunicatum KP763360\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGAATAAGTTATTGGTCTCAATCTGGTAATACTGTTCGGCTAGAGGGCAGGAGAAGTTGATGGAATTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAGTCAACTATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">MRE from Claroideoglomus claroideum FJ984660\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGAATAAGTTATTGGTCTCAATCTGGTAATACTGTTCTACTCGAGGGCAGGAGAGGTTGATGGAATTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAGTCAACTATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">MRE from Gigaspora albida KP763388\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGGATAAGTTACTGATCTCAATTCAGTAATACTGTTCTGCTAGAGGGCAGGAGAGGTTGATGGAATTCTGTGTGGAGGGGTGAAATCCGTTGATCTGCAGAAGAACACCGGCGAAGGCACTCAACTATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">MRE from Gigaspora margarita KF378701\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGAATAAGTTACTGATCTCAATTTGGTAATACTGTTCTGCTAGAGGGCAGAAGAGGTTGATGGAATTCTGTGTGGAGCGGTGAAATGCGTTGATCTGCAGAGGAACACCGGCGAAGGCATTCAACTATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">MRE from Gigaspora margarita KF378698\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGAATAAGTTACTGATCTCAATTTGGTAATACTGTTTTGCTAGAGGGCAGAAGAGGTTGATGGAATTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCAGCGAAGGCATTCAACTATTCTGCCTGACATTAAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">MRE from Claroideoglomus claroideum FJ984671\r\n",
      "ACATAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGATGGTGGAATAAGTTATTGGTCTCAACCTGATAATACTGTTCTACTTGAGGACAGGAGAGGTTGGCGGAATTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAGCCAACTATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">MRE from Funneliformis caledonius FJ984698\r\n",
      "ACATAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGATGGTAGGATAAGTTATTGGTCTCAACCTGATAATACTGTTCTGCTAGAAGAGAGGAGAGGTTAGTGGAATTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAACTAACTATTCTTCTTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">MRE from Claroideoglomus etunicatum KP763286\r\n",
      "ACATAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGATGGTGGAATAAGTTATTGGTCTCAACCTGGTAATACTGTTCTACTCGAGGACAGGAGAGGTTGGCGGAATTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAGCCAACTATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">MRE from Claroideoglomus etunicatum FJ984689\r\n",
      "ACATAGGTGGCAAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGATGGTGGAACAAGTTATCGGTCTCAACCTGGTAATACTATTCTACTTGAGGACAGGAGAGGTTGGCGGAATTCTGTGTGGAGCGGTGAAATGCGTTGATCTACANAGGAACACCGGCGAAGGCAGCCAACTATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">MRE from Claroideoglomus claroideum FJ984658\r\n",
      "ACATAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGGTGCGTAGATGGTAGAATAAGTTATTAGCCTCAACTTAATAATACTGTTTTGCTAGAGAATGGAAGGAGTTAATGGAATTCTATGTGTAGAGGTGAAATTCGTTGATCTATAGAGGACCACCGGCGAAGGCAATTAACTATTCTATCTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Funneliformis mosseae FJ984709\r\n",
      "ACATAGGTGGCAAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGATGGCGGAATAAGTTATCAGTCTCAACCTGATAATACTATTCTGCTAGAGGACAGGAGAGGTTGGCGGAACTCTATGCGTAGGGGTGAAATCCATTGATCTATAGAGGACCACCGGCGAAGGCAGCCAACTATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Gigaspora margarita KF378694\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAGAATGAGTTACTGATCTCAACCTGATAATACTGTTCTGCTAGAGGAGAGGAGAGGTTAGTGGAATTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAACTAACTATTCTTCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">MRE from Scuttelospora gilmorei FJ984735\r\n",
      "ACATAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGATGGTAGGATAAGTTATTGGTCTCAACTTGATAATACTGTTCTGCTAGAGGAGAGGAGAGGTTAGTGGAATTCTGTGTGTAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAACTAACTATTCTTCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">FJ984646.1\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTACTGGGCGTAAAGAGTGCGTAGGTGGTTCAATAAGTTACTGGTCCCAACTTGGTAATACTGATGAACTAGAGGACCGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAGGCAGTTAACTATTCGGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAGACGG\r\n",
      ">MRE from Funneliformis caledonius FJ984653\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTACTGGGCGTAAAGAGTGCGTAGGTGGTTTAATAAGTTACTGGTCCCAACTTGGTAATACTATTAGACTAGAGGACCGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAGGCAGTTAACTATTCGGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAGACGG\r\n",
      ">MRE from Claroideoglomus etunicatum KP763351\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTACTGGGCGTAAAGGGTGCGTAGGTGGTTTAATAAGTTACTGGTCCCAACTCAGTAATACTATTAAACTAGAGGGCCGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAGGCAGTTAACTATTCGGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAGACGG\r\n",
      ">MRE from Funneliformis mosseae FJ984707\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGAGTGCGTAGGTGGTTTAATAAGTTACTGGTCTCAACTCAGTAATACTATTAAACTAGAGGGTCGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTTAAATATATAGAGGAACACCGGCGAAGGCAGTTAACTATTCGACCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAGACGG\r\n",
      ">MRE from Funneliformis mosseae FJ984704\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGAGTGCGTAGGTGGTTTAATAAGTTACTGGTCTCAACTCAGTAATACTATTAAACTAGAGGGTCGGAGAAGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTTAAATATATAGAGGAACACCGGCGAAGGCAGTTAACTATTCGACCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAGACGG\r\n",
      ">MRE from Rhizophagus clarus KU992724\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTACTGGGCGTAAAGGGTGCGTAGGTGGTTGAATAAGTTACTGGTCTCAACTCAGTAATACTATCCAACTCGAGGGCTGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAAGCAGTTAACTATTCAGCCTGACATTGAGGCACGAAAGCGTGGGGAGCAATACGG\r\n",
      ">MRE from Rhizophagus clarus KU992754\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTACTGGACGTAAAGGGTGCGTAGGTGGTTGAATAAGTTACTGGTCTCAACTCAGTAATACTATCCAACTCGAGGGCTGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAAGCAGTTAACTATTCAGCCTGACATTGAGGCACGAAAGCGTGGGGAGCAATACGG\r\n",
      ">MRE from Rhizophagus clarus KU992822\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTACTGGGCGTAAAGGGTGCGTACGTGGTTGAATAAGTTACTGGTCTCAACTCAGTAATACTATCCAACTCGAGGGCTGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAAGCAGTTAACTATTCAGCCTGACATTGAGGCACGAAAGCGTGGGGAGCAATACGG\r\n",
      ">FJ984684.1\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTATGATAAGTTAGCGGTCTCAACCTGGTAATACTATCGTGCTAGAGTGTAGGAGAGGTTAACGGAACTCCCCGTGGAGCGGTGAAATGCATAGATATGGGGAAGAACACCGGCGAAGGCAGTCAACTATTCTAACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MRE from Funneliformis caledonius FJ984654\r\n",
      "ACATAGGTCGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTTCAGTAAGTTACTGGTCCTAGCTCAGTAATACTGTTGAACTTGAGTGCAGGAGAGGTTAACGGAACTTCATGTGGAGCGGTAAAATGCGTAGATATATGAAAGAACATCAGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Claroideoglomus etunicatum FJ984694\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTATGATAAGTTAGCGGTCTCAACCTGGTAATACTATCGTGCTAGAGTGTAGGAGAGGTTAACGGAACTCCCCGTGGAGCGGTGAAATGCATAGATATGGGGAAGAACACCGGCGAAGGCAGTTAACTATTCTAACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MRE from Funneliformis caledonius FJ984652\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCTTGGTAAGTTAGCAGTCTCAACCTGATAATACTGTCTGGCTAGAGTGCAGGAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGTGTAGATATATGGAAGAACACCGGCGAANGCAGTTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCMAAACGG\r\n",
      ">MRE from Funneliformis caledonius FJ984657\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCTTGGTAAGTTAGCAGTCTCAACCTGATAATACTGTCTGGCTAGAGTGCAGGAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGTGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Glomus versiforme FJ984724\r\n",
      "ACATAGGTGGCAAACATTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGCGGCATGATAAGTTGCTGGTCTCAACCTAGCAATACTGTCAAGCTAGAGGGCAGAAGAGGTTAACGGAACTCTATGTGGAGCGGTAAAATGTGTAGATATATAGAAGAACATCGGCGAAGGCAGTTAACTAGTCTGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Glomus versiforme FJ984725\r\n",
      "ACATAGGTGGCAAACATTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGCGGCATGATAAGTTGCTGGTCTCAACCTAGCAATACTGTCAAGCTAGAGGGCAGAAGAGGTTAACGGAACTCTATGTGGAGCGGTAAAATGTGTAGATATATAGAAGAACATCGGCGAAGGCAGTTAACTAGTCTGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Dentiscutata KP763367\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGCGGCGGAATAAGTTAGTGGTCTCAACTCATTAATACTGTTTTGCTAGAGGACAGGAGAGGTTAACGGAATTCCCTGTGGAGCGGTAAAATGTGTAGATATAGGGAAGAACACCGGCGAAGGCAGTTAACTATTCTGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAATAGG\r\n",
      ">MRE from Gigaspora albida KP763383\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGACGGCAGAATAAGTTAGTGGTCTCAACCCATTAATACTGTTTTGCTAGAGGGCAGAAGAGGTTAACGGAACTCCCTGTGGAGCGGTAAAATGTGTAGATATAGGGAAGAACACCGGCGAAGGCAGTTAACTATTCTGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAAATAGG\r\n",
      ">MRE from Cetraspora pellucida KP763394\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGTATCATTGGGCGTAAAGGGTGCGTAGATGGCCGAATAAGTTATCGGTCTCAACCCGGTAATACTGTTTGGCTCGAGTGCGGGAGAGGTTAACGGAACTCCTTGTGGAGCGGTAAAATGTGTAGATATAAGGAAGAACACCGGCGAAGGCAGTTAACTATTCCGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAATAGG\r\n",
      ">MRE from Gigaspora margarita KP763409\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTANATGGTCCGTTAAGTTAGCAGTCTCAACTTGCTAATACTGGCAGACTAGAGGACAGGAGAGGTTAACGGAACTCCCGGTGGAGCGGTGAAATGCGTAGATATCGGGAAGAACACCGGCGAAGGCAGTTAACTATTCTGCCTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Racocetra verrucosa KM065895\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTCTGCTAAGTTAGCAGTCTCAACTTGCTAATACTAGCAGACTAGAGGACAGGAGAGGTTGACGGAACTCCCGGTGGAGCGGTGAAATGCGTAGATATCGGGAAGAACACCGGCGAAGGCAGTTAACTATTCTGCCTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Gigaspora margarita KF378704\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCAGAATAAGTTAGTGGTCTTAACCCATTAATACTGTTTTGCTAGAGGACGGGGGGGGTTAACGGAACTCCCGGTGGAGCGGTGAAATGCGTAGATATCGGGAAGAACACCGGCGAAGGCAGTTAACTATTCTGCCTGACATTGAGGCACGAAAGCGTGGGTAGCAAAGCGG\r\n",
      ">MRE from Ambispora fennica FJ984643\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGCGGCATGATAAGTTATTGG-CTCAACCTGATAATACTGTCTTGCTTGAGTGCAGAAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGTGTAGATATATGGAAGAACATCGGCGAAGGCAGTTAACTATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MRE from Claroideoglomus etunicatum KP763291\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGACGGCATGATAAGTTATTGG-CTCAACCTGATAATACTGTTTTGCTTGAGTACAGGAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGTGTAGATATATGGAAGAACATCGGCGAAGGCAGTTAACTATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Ambispora appendicula FJ984635\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCCTGGTAAGTTAGCAGTCTCAACCTGGTAATACTGCTTGGCTTGAGTGCAGGAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">MRE from Geosiphon pyriformis FJ984679\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTCTAATAAGTTATCGGTCTCAACTCGGTAATACTGTTATACTCGAGTGCAGAAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACACCGGCGAAAGCAGTTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Claroideoglomus etunicatum KP763316\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATCATTGGGCGTAAAGGGTGCGTAGATGGCATGATAAGTTATTAGTCTCAACCTAGTAATACTGTTTTGCTTGAGTACAGAAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">MRE from Claroideoglomus etunicatum KP763336\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATCATTGGGCGTAAAGGGTGCGTAGATGGCATGATAAGTTATTAGTCTCAACCTAGTAATACTGTTTTGCTTGAGTACAGAAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAACGG\r\n",
      ">Bacillus antracis KC441811\r\n",
      "ACGTAGGTGGCAAGCGTTATCCGGAATTATTGGGCGTAAAGCGCGCGCAGGTGGTTTCTTAAGTCTGATGTCTCAACCATTGGAAACTGGGAGACTTGAGTGCAGAAGAGGAAAGTGGAATTCCATGTGTAGCGGTGAAATGCGTAGAGATATGGAGGAACACCGGCGAAGGCGACTTTCTGGTCTGACTGACACTGAGGCGCGAAAGCGTGGGGAGCAAACAGG\r\n",
      ">Mycoplasma mobile NR_074620\r\n",
      "ACATAGGTCGCAAGCGTTATCCGGAATTATTGGGCGTAAAGCGTTCGTAGGCGGTTTATTAAGTCTGGAGTCTCAACCTTTGGATACTGGTAAACTAGAGTTATGTAGAGGTAAGCGGAATTCCATGTGAAGTGGTGGAATACGTAGATATATGGAAGAACACCGGCGAAGGCAGCTTACTGGACATACTGACGCTCAGGAACGAAAGCGTGGGGAGCAAACAGG\r\n",
      ">8319\r\n",
      "ACATAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGATGGTGGAATAAGTTATTGGTCTCAACCTGGTAATACTGTTCTACTCGAGGACAGGAGAGGTTGGCGGAATTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAGCCAACTATTCTGCCTGACATTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">8660\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCCTGACAAGTTAGCAGTCTCAACCTGGTAATACTGTCTGGCTCGAGTGCAGGAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">14224\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGATGCGTAGGTGGTAGAATAAGTTATCAGTCTCAACCTGGTAAAACTATTCTGCTAGAGGAAAGGAGAGGTTAGCGGAACTCTATGAGTAAGGGTGAAATCTATTGATCCATAGAGGACCACCGGCGAAGGCAGCTAACTATTCTACCTGACACTGAGGCATGAAAGCGTGGGGAGCAAACCGG\r\n",
      ">2872\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTACTGGGCGTAAAGAGTGCGTAGGTGGTTCAATAAGTTACTGGTCTCAACTCAGTAATACTATTAAACTAGAGGGTCGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTTAAATATATAGAGGAACACCGGCGAAGGCAGTTAACTATTCGACCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">3001\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTACTGGGCGTAAAGTGTGCGTAGGTGGTTGAATAAGTTACTGGTCTCAACCCAGTAATACTATTCAACTCGAGGACCGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACATCGGCGAAAGCAGTTAACTATTCGGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">4190\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTACTGGGCGTAAAGAGTGCGTAGGTGGTTCAATAAGTTACTGGTCCCAACTTGGTAATACTGTTGAACTAGAGGACCGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAGGCAGTTAACTATTCGGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">4442\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCCTGACAAGTTAACAGTCTCAACCTGATAATACTGTCTGGCTTGAGTACAGAAGAGGTTAACGGAATTCCATGTGGAGCGGTAAAATGCACAGATATATGGAAGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">4562\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATTGTATGGTAAGTTACTGGTCTCAACTCAGTAATACTGCTATGCTAGAGTTCAGGAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACGCCGGCGAAGGCAGTTAACTATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">5222\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTACTGGGCGTAAAGGGTGCGTAGGTGGTTTAATAAGTTATCGGTCCCAACTCGGTAATACTATTAGACTAGAGGGCTGGAGAGGTTAACGGAATGGTTCATGTAGGGGTAAAATCCGTAAATATGAATTGGAACACCGGCGAAGGCAGTTAACTATTCAGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">6878\r\n",
      "ACATAGGTCGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTTCAGTAAGTTACTGGTCCTAACTCAGTAATACTGTTGAACTTGAGTGCAGGAGAGGTTAACGGAACTTCATGTGGAGCGGTAAAATGCGTAGATATATGAAAGAACATCGGCGAAGGCAGTTAACTATTCTGACTGACATTAAGGCATGAAAGCGTGGGGAGCAAA----\r\n",
      ">7420\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTTTAATAAGTTAGCGGTCTCAACTCGCTAATACTATCAAACTGGAGTGCAGGAGAGGTTAACGGAACTCCATGTGTAGCGGTAAAATGCGACGATATATGGATGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">7660\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGAGTGCGTAGGTGGTTTAATAAGTTACTGGTCTCAACTCAGTAATACTGTTAGACTAGAGGATCGGAGGGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTCAGATACATAGAGGAACACCGGCGAAGGCAGTTAACTATTCGACCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">7941\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTACTGGGCGTAAAGAGTGCGTAGGTGGTTTAATAAGTTATTGGTCTCAACCCAGTAATACTATTGAACTAGAGGGCTGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAGGCAGTTAACTATTCAGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">8498\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGTGTGCGTAGGTGGTTTAATAAGTTACTGGTCTCAACTCGGTAATACTATTAGACTAGAGGATCGGAGAGGTTAACGGAATGGTTCGTGTAGGGGTAAAATCCTCAAATATGAATTGGAACACCGGCGAAGGCAGTTAACTATTCGGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">8522\r\n",
      "ACATAGGTTGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGCTTGATAAGTTATCAGTCTTAACCTGGTAATACTGTCATGCTTGAGTGCAGGAGAGGTTAACGGAACTCTATGTGGAGCGGTAAAATGTGACGATATATAGAAGAACACCGGCGAAAGCAGTTAACTATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAA----\r\n",
      ">8827\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTCGAATAAGTTAGCAGTCCTAACCTGGTAATACTGTTCGACTGGAGGGCAGGAGAGGTTAATGGAACTCTATGTGGAACGGTAAAATGTGACGATATATAGATGAACACCGGCGAAGGCAGTTAACTATTCTGCCTGACATTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">9232\r\n",
      "ACATAGGTGGCGAACGTTATCCGGATTTATTGGGCGTAAAGGGTGCGTAGATGGTAGAACAAGTTA----TCTCAACTTAATAATACTGTTTTGCTAGAGAATGGAAGAGGTTAATGGAATTCTATGTGTAGAGGTGAAATTCGTTGATCCATAGAGGACCACCGGCGAAGGCAATTAACTATTCCATCTGACATTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">9472\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGCGGCATGATAAGTTATTGG-CTCAACCTGATAATACTGTTTTGCTTGAGTACAGGAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGTGTAGATATATGGAAGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACGCTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">9533\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTTTGATAAGTTAGCGGTCTCAACTCGCTAATACTATCAAACTAGAGGGCAGGAGAGGTTAACGGAACTCTATGTGGAGCGGTAAAATGTATAGATATATAGAAGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">9550\r\n",
      "ACATAGGTCGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTGTGATAAGTTATTGGTCTCAACCTGATAATACTGTCACACTAGAGTGTAGGAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACATCGGCGAAGGCAGTTAACTATTCTAACTGACATTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">10226\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGAGTGCGTAGGTGGTTTAATAAGTTATCGGTCCCAACTCGATAATACTATTAAACTAGAGGGCTGGAGAAGTTAACGGAATTCTATGTGTAGCGGTAAAATGTGTAAATATATAGAGGAACACCGGCGAAGGCAGTTAACTATTCGGCCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">10910\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGACGGCAGGATAAGTTATTGGTCTCAACCTGATAATACTGTTTTGCTTGAGTACAGGAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGCGTAGATATATGGAAGAACATCGGCGAAGGCAGTTAACTATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">11169\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGTGGCCTGACAAGTTATCAGTCTCAACCTGGTAATACTGTCCGGCTTGAGTACAGGAGAGGTTAACGGAACTCTATGTGGAGCGGTAAAATGCACAGATATATAGAAGAACATCGGCGAAGGCAGTTAACTATTCTGACTGACGCTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">11600\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATCATTGGGCGTAAAGGGTGCGTAGGCGGCAGGATAAGTTATTGG-CTCAACCTGATAATACTGTCTTGCTTGAGTGCAGGAGAGGTTAACGGAATTCTTTGTGGAGGGGTAAAATCCAAAGATATAAAGGAGAACATCGGCGAAGGCAGTTAACTATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">13308\r\n",
      "ACATAGGTCGCAAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTTCAGTAAGTTACTGGTCTCAACTCAGTAATACTGCTGAACTTGAGTGCAGGAGAGGTTAACGGAACTCTATGTGTAGGGGTAAAATCCATAGATATATAGATGAACATCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCATGAAAGCGTGGGGAGCAAA----\r\n",
      ">13619\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGCTTGACAAGTTAGTGGTCTCAACTCACTAATACTATCAAACTAGAGTGCAGGAGAGGTTAATGGAACTCCATGTGTAGCGGTAAAATGCGACGATATATGGATGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">13848\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTTGGAAAAGTTAGTGGTCCCAACCTATTAATACTATCTGACTTGAGAACAGGAGAGGTTAACGGAACTCTATGTGGAGCGGTAAAATGCGACGATATATAGAAGAACACCGGCGAAGGCAGTTAACTATTCTGTCTGACATTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">13928\r\n",
      "ACATAGGTGGCAAACGTTATCCGGAATTATTGGGCGTAAAGAGTGCGTAGGTGGTTTAATAAGTTAGTAGTCTCAACTTACTAATACTATTAAACTAGAGGATCGGAGAGGTTAACGGAATTCTATGTGTAGCGGTAAAATGCTCAAATATATAGAGGAACACCGGCGAAGGCAGTTAACTATTCGACCTGACGTTGAGGCACGAAAGCGTGGGGAGCAA-----\r\n",
      ">14163\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGTAAAATAAGTTACCGGTCTCAATCTAGTAATACTATTTTGCTAGAGGATAGGAGAGGTTGATGGAACTCTATGCGTAGGGGTGAAATCCGTTGATCTATAGAGGAACACCGGCGAAGGCAATCAACTATTCTACCTGACATTGAGGCATGAAAGCGTGGGTAGCAAACCGG\r\n",
      ">14386\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGATGGTTTGATAAGTTATTAGTCTCAACTTAGTAATACTATTAGACTAGAGTACAGGAGAGGTTAATGGAACTCTATGTTTAGGGGTAAAATCCGCAGATATA--GATGAACACCGGCGAAGGCAGTTAACTATTCTGACTGACATTGAGGCACGAAAGCGTGGGGAGCAAA----\r\n",
      ">14558\r\n",
      "ACGTAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGCGGCATGATAAGTTATTGG-CTCAACCTAATAATACTGTCTTGCTTGAGTGCAGAAGAGGTTAACGGAACTCCATGTGGAGCGGTAAAATGTGTAGATATATGGAAGAACATCGGCGAAGGCAGTTAACTATTCTGACTGACGTTGAGGCACGAAAGCGTGGGGAGCAAAAAGG\r\n",
      ">16770\r\n",
      "ACATAGGTGGCGAACGTTATCCGGAATTATTGGGCGTAAAGGATGCGTAGATGGCAGAATAAGTTACTGGTCTCAA-CTGGTAATACTGTTCTGCTAGAGGACAGGAGAGGTTGATGGAATTCTGTGTGGAGCGGTGAAATGCGTTGATCTACAGAGGAACACCGGCGAAGGCAATCAACTATTCTGCCTGACATTGAGGCATGAAAGCGTGGGTAGCAAACCGG\r\n"
     ]
    }
   ],
   "source": [
    "!cat NebraskMREaln_for_tree_renamed.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Inferring and rooting the tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "aln = LoadSeqs('NebraskMREaln_for_tree_renamed.fasta', moltype=DNA)\n",
    "t_unroot = build_tree_from_alignment(aln, moltype=DNA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Bacillus antracis KC441811\r\n"
     ]
    }
   ],
   "source": [
    "!grep \"KC441811\" NebraskMREaln_for_tree_renamed.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "t_rooted = t_unroot.rootedWithTip(\"Bacillus antracis KC441811\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "t_rooted.writeToFile('NebraskaMRE.tree')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
