{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Incorporate isolate sequences into master tree of DADA2 seqs\n",
    "* Align isolate seqs using SSU_align\n",
    "* Mask sequences based on same mask as used for sequences from Exp10 \n",
    "* Append sequences to create new fasta with both sets of sequences\n",
    "* Create master tree using FastTree\n",
    "* Use qiime environment and set kernel to Python 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "workDir = '/home/be68/Hyphosphere/data/3Exp/MasterTree/'\n",
    "maskDir = '/home/be68/Hyphosphere/data/3Exp/Fasttree'\n",
    "IsolateSeqDir = '/home/be68/Hyphosphere/data/3Exp/'\n",
    "IsolateSeqFile = 'IsolateSeqs.fasta'\n",
    "\n",
    "\n",
    "nprocs = 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import numpy as np\n",
    "#import entrez.direct\n",
    "from cogent.app.fasttree import build_tree_from_alignment\n",
    "from cogent import DNA, LoadSeqs\n",
    "from Bio import Entrez, SeqIO\n",
    "from pandas import *\n",
    "Entrez.email = \"be68@cornell.edu\"\n",
    "#from IPython.display import display, Image, SVG\n",
    "#from cogent3 import LoadSeqs, FastTree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.isdir(workDir):\n",
    "    os.mkdir(workDir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/data/home/be68/Hyphosphere/data/3Exp/MasterTree\n"
     ]
    }
   ],
   "source": [
    "%cd $workDir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cd $workDir; ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cd $workDir; ln -f -s $IsolateSeqDir$IsolateSeqFile"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of OTUs in fasta: 165\n"
     ]
    }
   ],
   "source": [
    "!printf \"Number of OTUs in fasta: \"\n",
    "!cd $workDir; grep -c \">\" $IsolateSeqFile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Using SSU-Align to align seqs and masking based on alignment posterior probabilities of 3Exp - ASV set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# _ssu-prep :: prepare SSU rRNA sequences for parallel ssu-align jobs\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-prep -x -f -b 50 --dna --rfonly IsolateSeqs.fasta ssu_aln 3\n",
      "# date:    Sat Oct 26 11:56:03 2019\n",
      "#\n",
      "# Validating input sequence file ... done.\n",
      "#\n",
      "# Preparing 3 ssu-align jobs ...\n",
      "# Partitioning seqs with goal of equalizing total number of nucleotides per job ...\n",
      "#\n",
      "# output file name      description                                        \n",
      "# --------------------  ---------------------------------------------------\n",
      "  ssu_aln/IsolateSeqs.fasta.1  partition 1 FASTA sequence file (55 seqs; 70726 nt)\n",
      "  ssu_aln/IsolateSeqs.fasta.2  partition 2 FASTA sequence file (56 seqs; 70104 nt)\n",
      "  ssu_aln/IsolateSeqs.fasta.3  partition 3 FASTA sequence file (54 seqs; 67718 nt)\n",
      "  ssu_aln.ssu-align.sh  shell script that will execute 3 ssu-align jobs\n",
      "#\n",
      "################################################################################\n",
      "# To execute all 3 ssu-align jobs, run the shell script with the command:\n",
      "#\t\"ssu_aln.ssu-align.sh\"\n",
      "# (it is an executable file)\n",
      "################################################################################\n",
      "#\n",
      "# The final job will merge the output from all jobs once they are all finished.\n",
      "#\n",
      "# If one or more jobs fail: rerun the failed jobs, wait for them to finish,\n",
      "# and then perform manual merge from this directory by executing:\n",
      "#\tssu-merge ssu_aln\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln/ssu_aln.ssu-prep.log.\n",
      "# Output printed to the screen saved in:  ssu_aln/ssu_aln.ssu-prep.sum.\n",
      "#\n",
      "# CPU time:  00:00:00.29  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "!cd $workDir; ssu-prep -f -x -b 50 --rfonly --dna $IsolateSeqFile ssu_aln 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Executing: ssu-align -b 50 --dna --rfonly ssu_aln/IsolateSeqs.fasta.1 ssu_aln/ssu_aln.1 > /dev/null &\n",
      "# Executing: ssu-align -b 50 --dna --rfonly ssu_aln/IsolateSeqs.fasta.2 ssu_aln/ssu_aln.2 > /dev/null &\n",
      "# Executing: ssu-align --merge 3 -b 50 --dna --rfonly ssu_aln/IsolateSeqs.fasta.3 ssu_aln/ssu_aln.3\n",
      "# _ssu-align :: align SSU rRNA sequences\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-align --merge 3 -b 50 --dna --rfonly ssu_aln/IsolateSeqs.fasta.3 ssu_aln/ssu_aln.3\n",
      "# date:    Sat Oct 26 11:56:11 2019\n",
      "#\n",
      "# Validating input sequence file ... done.\n",
      "#\n",
      "# Stage 1: Determining SSU start/end positions and best-matching models...\n",
      "#\n",
      "# output file name            description                                \n",
      "# --------------------------  -------------------------------------------\n",
      "  ssu_aln.3.tab               locations/scores of hits defined by HMM(s)\n",
      "  ssu_aln.3.bacteria.hitlist  list of sequences to align with bacteria CM\n",
      "  ssu_aln.3.bacteria.fa            54 sequences to align with bacteria CM\n",
      "#\n",
      "# Stage 2: Aligning each sequence to its best-matching model...\n",
      "#\n",
      "# output file name            description\n",
      "# --------------------------  ---------------------------------------\n",
      "  ssu_aln.3.bacteria.stk      bacteria alignment\n",
      "  ssu_aln.3.bacteria.cmalign  bacteria cmalign output\n",
      "  ssu_aln.3.bacteria.ifile    bacteria insert info\n",
      "  ssu_aln.3.scores            list of CM/HMM scores for each sequence\n",
      "#\n",
      "# Summary statistics:\n",
      "#\n",
      "# model or       number  fraction        average   average               \n",
      "# category      of seqs  of total         length  coverage    nucleotides\n",
      "# ------------  -------  --------  -------------  --------  -------------\n",
      "  *input*            54    1.0000        1254.04    1.0000          67718\n",
      "#\n",
      "  archaea             0    0.0000              -         -              0\n",
      "  bacteria           54    1.0000        1252.67    0.9989          67644\n",
      "  eukarya             0    0.0000              -         -              0\n",
      "#\n",
      "  *all-models*       54    1.0000        1252.67    0.9989          67644\n",
      "  *no-models*         0    0.0000              -         -              0\n",
      "#\n",
      "# Speed statistics:\n",
      "#\n",
      "# stage      num seqs  seq/sec  seq/sec/model    nucleotides    nt/sec\n",
      "# ---------  --------  -------  -------------  -------------  --------\n",
      "  search           54    1.959          0.653          67718    2457.1\n",
      "  alignment        54    1.111          1.111          67644    1391.4\n",
      "#\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln.3.ssu-align.log.\n",
      "# Output printed to the screen saved in:  ssu_aln.3.ssu-align.sum.\n",
      "#\n",
      "# All output files created in directory ./ssu_aln/ssu_aln.3/\n",
      "#\n",
      "# CPU time (search):     00:00:27.56  (hh:mm:ss)\n",
      "# CPU time (alignment):  00:00:48.62  (hh:mm:ss)\n",
      "# CPU time (total):      00:01:16.40  (hh:mm:ss)\n",
      "#\n",
      "################################################################################\n",
      "#\n",
      "# Alignments successfully created.\n",
      "#\n",
      "# Checking/waiting for other jobs to finish before merging...\n",
      "#\n",
      "# Job 3 (dir: ssu_aln/ssu_aln.3) completed successfully! 2/3 still incomplete.\n",
      "# Job 1 (dir: ssu_aln/ssu_aln.1) completed successfully! 1/3 still incomplete.\n",
      "# Job 2 (dir: ssu_aln/ssu_aln.2) completed successfully! 0/3 still incomplete.\n",
      "#\n",
      "# All 3 jobs successfully completed.\n",
      "# About to merge their output:\n",
      "#\n",
      "# Executing:  ssu-merge ssu_aln\n",
      "#\n",
      "################################################################################\n",
      "#\n",
      "# _ssu-merge :: merge SSU rRNA alignments\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-merge ssu_aln\n",
      "# date:    Sat Oct 26 11:57:33 2019\n",
      "#\n",
      "# Merging files from 3 ssu-align runs...\n",
      "#\n",
      "#                                     # files     # seqs\n",
      "# merged file name          CM name    merged     merged\n",
      "# ------------------------  --------  -------  ---------\n",
      "  ssu_aln.tab               -               3          -\n",
      "  ssu_aln.scores            -               3          -\n",
      "  ssu_aln.ssu-align.sum     -               3          -\n",
      "  ssu_aln.ssu-align.log     -               3          -\n",
      "#\n",
      "  ssu_aln.bacteria.fa       bacteria        3        165\n",
      "  ssu_aln.bacteria.hitlist  bacteria        3        165\n",
      "  ssu_aln.bacteria.cmalign  bacteria        3        165\n",
      "  ssu_aln.bacteria.ifile    bacteria        3        165\n",
      "  ssu_aln.bacteria.stk      bacteria        3        165\n",
      "#\n",
      "#\n",
      "# Summary statistics:\n",
      "#\n",
      "# model or       number  fraction        average   average               \n",
      "# category      of seqs  of total         length  coverage    nucleotides\n",
      "# ------------  -------  --------  -------------  --------  -------------\n",
      "  *input*           165    1.0000        1263.93    1.0000         208548\n",
      "#\n",
      "  archaea             0    0.0000              -         -              0\n",
      "  bacteria          165    1.0000        1262.46    0.9988         208306\n",
      "  eukarya             0    0.0000              -         -              0\n",
      "#\n",
      "  *all-models*      165    1.0000        1262.46    0.9988         208306\n",
      "  *no-models*         0    0.0000              -         -              0\n",
      "#\n",
      "# Speed statistics:\n",
      "#\n",
      "# stage      num seqs  seq/sec  seq/sec/model    nucleotides    nt/sec\n",
      "# ---------  --------  -------  -------------  -------------  --------\n",
      "  search          165  165.000         55.000         208548  208548.0\n",
      "  alignment       165  165.000        165.000         208306  208306.0\n",
      "#\n",
      "# Summed ssu-align CPU time (search):     00:00:00.00  (hh:mm:ss)\n",
      "# Summed ssu-align CPU time (alignment):  00:00:00.00  (hh:mm:ss)\n",
      "# Summed ssu-align CPU time (total):      00:00:00.00  (hh:mm:ss)\n",
      "#\n",
      "# Cleaning up by removing original files that were just merged ... done.\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln.ssu-merge.log.\n",
      "# Output printed to the screen saved in:  ssu_aln.ssu-merge.sum.\n",
      "#\n",
      "# All output files created in directory ./ssu_aln/\n",
      "#\n",
      "# CPU time:  00:00:00.10  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "!cd $workDir; ./ssu_aln.ssu-align.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## mask alignments based on 3Exp ASVs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# _ssu-mask :: mask SSU rRNA alignments\n",
      "# SSU-ALIGN 0.1.1 (Feb 2016)\n",
      "# Copyright (C) 2016 Howard Hughes Medical Institute\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# command: _ssu-mask -s /home/be68/Hyphosphere/data/3Exp/Fasttree//ssu_aln/ssu_aln.bacteria.mask --afa --dna ssu_aln/\n",
      "# date:    Sat Oct 26 12:07:01 2019\n",
      "#\n",
      "# Masking alignments using pre-existing masks...\n",
      "#\n",
      "#                                                     mask    \n",
      "#                                                 ------------\n",
      "# file name                  in/out  type  #cols  incl.  excl.\n",
      "# -------------------------  ------  ----  -----  -----  -----\n",
      "  ssu_aln.bacteria.stk        input   aln   1582      -      -\n",
      "  ssu_aln.bacteria.mask       input  mask   1582    247   1335\n",
      "  ssu_aln.bacteria.mask.pdf  output   pdf   1582    247   1335\n",
      "  ssu_aln.bacteria.mask.afa  output   aln    247      -      -\n",
      "#\n",
      "# All attempts to draw structure diagrams of masks were successful.\n",
      "#\n",
      "# List of executed commands saved in:     ssu_aln.ssu-mask.log.\n",
      "# Output printed to the screen saved in:  ssu_aln.ssu-mask.sum.\n",
      "#\n",
      "# All output files created in directory ./ssu_aln/\n",
      "#\n",
      "# CPU time:  00:00:00.55  (hh:mm:ss)\n",
      "# \n"
     ]
    }
   ],
   "source": [
    "!cd $workDir; ssu-mask -s $maskDir/ssu_aln/ssu_aln.bacteria.mask --dna --afa ssu_aln/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Append masked alignments and include outgroup from notebook #4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "!cd $workDir; cat $maskDir/ssu_aln/ssu_aln.bacteria.mask.afa $maskDir/sso_aln/sso_aln.bacteria.mask.afa ssu_aln/ssu_aln.bacteria.mask.afa > aln_for_tree.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">ASV1\n",
      "TACGAAGGGGGCTAGCGTTGTTCGGATTTACTGGGCGTAAAGCGCACGTAGGCGGACTAT\n",
      "TAAGTCAGGGGTGAAATCCCGGGGCTCAACCCCGGAACTTTGATACTGGTAGTCTTGAGT\n",
      "TCGAGAGAGGTGAGTGGAATTCCGAGTGTAGAGGTGAAATTCGTAGATATTCGGAGGAAC\n",
      "ACCAGTGGCGAAGGCGGCTCACTGGCTCGACTGACGCTGAGGTGCGAAAGCGTGGGGAGC\n",
      "AAACAGG\n",
      ">ASV2\n",
      "-ACATAGGTGGCAAACATTATCCGGAATTATTGGGCGTAAAGGGTGCGTAGGCGGCATGA\n",
      "TAAGTTGCTGGTGGGAAATCAAGGCTCAACCTTGTGGTAGCAATACTGTCAAGCTAGAGG\n",
      "GCAGAAGAGGTTAACGGAACTCTATGTGGAGCGGTAAAATGTGTAGATATATAGAAGAAC\n",
      "TAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAAATCTGATACTGGCAAGCTTGAGT\n",
      "CTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAAT\n",
      "ACCGGTGGCGAAGGCGGCCCCCTGGACGAACTGACGCTCAGGTGCGAAAGCGTGGGGAGC\n",
      "AAACAGG\n",
      ">BE277\n",
      "TACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTTTGT\n",
      "TAAGTCAGATGTGAAATCCCCGGGCTCAACCTGGGAAATCTGATACTGGCAAGCTTGAGT\n",
      "CTCGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGCGTAGAGATCTGGAGGAAT\n",
      "ACCGGTGGCGAAGGCGGCCCCCTGGACGAACTGACGCTCAGGTGCGAAAGCGTGGGGAGC\n",
      "AAACAGG\n"
     ]
    }
   ],
   "source": [
    "!head aln_for_tree.fasta\n",
    "!tail aln_for_tree.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Inferring and rooting the tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "aln = LoadSeqs(os.path.join(workDir, 'aln_for_tree.fasta'), moltype=DNA)\n",
    "t_unroot = build_tree_from_alignment(aln, moltype=DNA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "t_rooted = t_unroot.rootedWithTip('X90478')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "t_rooted.writeToFile(os.path.join(workDir, 'Master.tree'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## Subset tree with hyphal ASVs and top"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "## get tip names of isolates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['BE26', 'BE27', 'BE28', 'BE29', 'BE30', 'BE31', 'BE32', 'BE33', 'BE34', 'BE35', 'BE36', 'BE38', 'BE39', 'BE40', 'BE41', 'BE42', 'BE43', 'BE44', 'BE45', 'BE46', 'BE47', 'BE48', 'BE49', 'BE50', 'BE51', 'BE52', 'BE53', 'BE54', 'BE55', 'BE56', 'BE57', 'BE59', 'BE60', 'BE61', 'BE62', 'BE63', 'BE64', 'BE65', 'BE65B', 'BE66', 'BE68', 'BE69', 'BE70', 'BE71', 'BE72', 'BE74', 'BE75', 'BE76', 'BE79', 'BE80', 'BE81', 'BE82', 'BE83', 'BE84', 'BE85', 'BE88', 'BE91', 'BE92', 'BE93', 'BE94', 'BE95', 'BE96', 'BE97', 'BE98', 'BE100', 'BE101', 'BE102', 'BE103', 'BE104', 'BE105', 'BE107', 'BE108', 'BE109', 'BE110', 'BE112', 'BE113', 'BE114', 'BE115', 'BE116', 'BE117', 'BE118', 'BE120', 'BE121', 'BE122', 'BE123', 'BE124', 'BE125', 'BE126', 'BE127', 'BE128', 'BE153', 'BE154', 'BE156', 'BE157', 'BE159', 'BE163', 'BE164', 'BE166', 'BE167', 'BE168', 'BE170', 'BE171', 'BE172', 'BE173', 'BE177', 'BE178', 'BE179', 'BE180', 'BE181', 'BE182', 'BE183', 'BE184', 'BE186', 'BE187', 'BE190', 'BE191', 'BE193', 'BE194', 'BE195', 'BE196', 'BE197', 'BE198', 'BE201', 'BE203', 'BE206', 'BE207', 'BE212', 'BE213', 'BE215', 'BE216', 'BE219', 'BE220', 'BE222', 'BE223', 'BE224', 'BE228', 'BE231', 'BE232', 'BE233', 'BE234', 'BE235', 'BE236', 'BE239', 'BE240', 'BE241', 'BE242', 'BE243', 'BE248', 'BE249', 'BE251', 'BE254', 'BE255', 'BE256', 'BE257', 'BE258', 'BE259', 'BE260', 'BE261', 'BE264', 'BE265', 'BE268', 'BE270', 'BE275', 'BE276', 'BE277']\n"
     ]
    }
   ],
   "source": [
    "file_to_open = os.path.join(IsolateSeqDir, IsolateSeqFile)\n",
    "IsolateList = []\n",
    "\n",
    "for record in SeqIO.parse(file_to_open, \"fasta\") :\n",
    "       IsolateList.append(record.id)#+\"\\n\"+str(record.seq)+\"\\n\")\n",
    "\n",
    "print(IsolateList)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import blast tables\n",
    "* blast database created from seqs_thresh.fasta with the command \"makeblastdb -in seqs_thresh.fasta -parse_seqids -dbtype nucl\"\n",
    "* blast table created with: 'blastn -db phyloseq/seqs_thresh.fasta -query IsolateSeqs.fasta -out Isolate_ASV_blast.out -outfmt 6 -max_target_seqs 1'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BE26\tASV97\t100.00\t253\t0\t0\t444\t696\t1\t253\t5e-132\t  468\r\n",
      "BE27\tASV97\t100.00\t253\t0\t0\t450\t702\t1\t253\t5e-132\t  468\r\n",
      "BE28\tASV97\t100.00\t253\t0\t0\t452\t704\t1\t253\t5e-132\t  468\r\n",
      "BE29\tASV34\t100.00\t253\t0\t0\t433\t685\t1\t253\t5e-132\t  468\r\n",
      "BE30\tASV34\t100.00\t253\t0\t0\t433\t685\t1\t253\t5e-132\t  468\r\n",
      "BE31\tASV120\t100.00\t253\t0\t0\t417\t669\t1\t253\t5e-132\t  468\r\n",
      "BE32\tASV120\t100.00\t253\t0\t0\t417\t669\t1\t253\t5e-132\t  468\r\n",
      "BE33\tASV4392\t100.00\t253\t0\t0\t419\t671\t1\t253\t4e-132\t  468\r\n",
      "BE34\tASV120\t100.00\t253\t0\t0\t418\t670\t1\t253\t5e-132\t  468\r\n",
      "BE35\tASV636\t100.00\t253\t0\t0\t417\t669\t1\t253\t4e-132\t  468\r\n"
     ]
    }
   ],
   "source": [
    "!head ../Isolate_ASV_blast.out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BE26</td>\n",
       "      <td>ASV97</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>444</td>\n",
       "      <td>696</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>5.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BE27</td>\n",
       "      <td>ASV97</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>450</td>\n",
       "      <td>702</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>5.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BE28</td>\n",
       "      <td>ASV97</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>452</td>\n",
       "      <td>704</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>5.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BE29</td>\n",
       "      <td>ASV34</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>433</td>\n",
       "      <td>685</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>5.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BE30</td>\n",
       "      <td>ASV34</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>433</td>\n",
       "      <td>685</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>5.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  qseqid sseqid  pident  length  mismatch  gapopen  qstart  qend  sstart  \\\n",
       "0   BE26  ASV97   100.0     253         0        0     444   696       1   \n",
       "1   BE27  ASV97   100.0     253         0        0     450   702       1   \n",
       "2   BE28  ASV97   100.0     253         0        0     452   704       1   \n",
       "3   BE29  ASV34   100.0     253         0        0     433   685       1   \n",
       "4   BE30  ASV34   100.0     253         0        0     433   685       1   \n",
       "\n",
       "   send         evalue  bitscore  \n",
       "0   253  5.000000e-132       468  \n",
       "1   253  5.000000e-132       468  \n",
       "2   253  5.000000e-132       468  \n",
       "3   253  5.000000e-132       468  \n",
       "4   253  5.000000e-132       468  "
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pandas.read_table('/home/be68/Hyphosphere/data/3Exp/Isolate_ASV_blast.out', sep = \"\\t\", header=None)\n",
    "df.columns = ['qseqid','sseqid',\n",
    "'pident',\n",
    "'length',\n",
    "'mismatch',\n",
    "'gapopen',\n",
    "'qstart',\n",
    "'qend',\n",
    "'sstart',\n",
    "'send',\n",
    "'evalue',\n",
    "'bitscore']\n",
    "df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>BE268</td>\n",
       "      <td>ASV298</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>385</td>\n",
       "      <td>637</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>5.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>BE270</td>\n",
       "      <td>ASV173</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>383</td>\n",
       "      <td>635</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>BE275</td>\n",
       "      <td>ASV681</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>443</td>\n",
       "      <td>695</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>5.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>BE276</td>\n",
       "      <td>ASV681</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>441</td>\n",
       "      <td>693</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>5.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>BE277</td>\n",
       "      <td>ASV681</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>441</td>\n",
       "      <td>693</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>5.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    qseqid  sseqid  pident  length  mismatch  gapopen  qstart  qend  sstart  \\\n",
       "159  BE268  ASV298   100.0     253         0        0     385   637       1   \n",
       "160  BE270  ASV173   100.0     253         0        0     383   635       1   \n",
       "161  BE275  ASV681   100.0     253         0        0     443   695       1   \n",
       "162  BE276  ASV681   100.0     253         0        0     441   693       1   \n",
       "163  BE277  ASV681   100.0     253         0        0     441   693       1   \n",
       "\n",
       "     send         evalue  bitscore  \n",
       "159   253  5.000000e-132       468  \n",
       "160   253  3.000000e-132       468  \n",
       "161   253  5.000000e-132       468  \n",
       "162   253  5.000000e-132       468  \n",
       "163   253  5.000000e-132       468  "
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "# function to get unique values \n",
    "def unique(list1): \n",
    "    x = np.array(list1) \n",
    "    return(np.unique(x)) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ASV1012',\n",
       " 'ASV103',\n",
       " 'ASV105',\n",
       " 'ASV120',\n",
       " 'ASV12433',\n",
       " 'ASV1296',\n",
       " 'ASV131',\n",
       " 'ASV1492',\n",
       " 'ASV15',\n",
       " 'ASV1578',\n",
       " 'ASV158',\n",
       " 'ASV159',\n",
       " 'ASV1622',\n",
       " 'ASV17',\n",
       " 'ASV173',\n",
       " 'ASV183',\n",
       " 'ASV1967',\n",
       " 'ASV206',\n",
       " 'ASV2102',\n",
       " 'ASV2103',\n",
       " 'ASV2123',\n",
       " 'ASV2155',\n",
       " 'ASV2227',\n",
       " 'ASV2250',\n",
       " 'ASV2258',\n",
       " 'ASV231',\n",
       " 'ASV242',\n",
       " 'ASV2449',\n",
       " 'ASV248',\n",
       " 'ASV275',\n",
       " 'ASV2857',\n",
       " 'ASV2948',\n",
       " 'ASV296',\n",
       " 'ASV298',\n",
       " 'ASV3069',\n",
       " 'ASV3108',\n",
       " 'ASV320',\n",
       " 'ASV3225',\n",
       " 'ASV34',\n",
       " 'ASV3446',\n",
       " 'ASV3647',\n",
       " 'ASV371',\n",
       " 'ASV395',\n",
       " 'ASV404',\n",
       " 'ASV407',\n",
       " 'ASV43',\n",
       " 'ASV4392',\n",
       " 'ASV44',\n",
       " 'ASV443',\n",
       " 'ASV447',\n",
       " 'ASV490',\n",
       " 'ASV4965',\n",
       " 'ASV5',\n",
       " 'ASV514',\n",
       " 'ASV523',\n",
       " 'ASV627',\n",
       " 'ASV636',\n",
       " 'ASV64',\n",
       " 'ASV664',\n",
       " 'ASV673',\n",
       " 'ASV681',\n",
       " 'ASV6820',\n",
       " 'ASV70',\n",
       " 'ASV7019',\n",
       " 'ASV744',\n",
       " 'ASV773',\n",
       " 'ASV8',\n",
       " 'ASV8301',\n",
       " 'ASV85',\n",
       " 'ASV864',\n",
       " 'ASV9039',\n",
       " 'ASV9045',\n",
       " 'ASV9153',\n",
       " 'ASV93',\n",
       " 'ASV953',\n",
       " 'ASV97',\n",
       " 'ASV9759',\n",
       " 'ASV9844',\n",
       " 'ASV99']"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "IsolateMatches_list = df[\"sseqid\"].tolist()\n",
    "IsolateMatches_list = list(unique(IsolateMatches_list))\n",
    "IsolateMatches_list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Identify hyphal ASVs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* Stick with CH_CS and BH_BS for now"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/be68/anaconda3/envs/ipykernel_py2/lib/python2.7/site-packages/pandas/core/computation/expressions.py:180: UserWarning: evaluating in Python space because the '*' operator is not supported by numexpr for the bool dtype, use '&' instead\n",
      "  .format(op=op_str, alt_op=unsupported[op_str]))\n"
     ]
    }
   ],
   "source": [
    "dfl2fc_1 = pandas.read_table('/home/be68/Hyphosphere/data/3Exp/DESeq/dfl2fc-Exp1.txt', sep = \"\\t\")\n",
    "sigtab1 = dfl2fc_1.loc[(dfl2fc_1.padj < 0.05) & (dfl2fc_1.log2FoldChange > 0) & (dfl2fc_1.Contrast == 'CH_CS') & (dfl2fc_1.Kingdom == \"Bacteria\")]\n",
    "del dfl2fc_1 \n",
    "\n",
    "dfl2fc_2 = pandas.read_table('/home/be68/Hyphosphere/data/3Exp/DESeq/dfl2fc-Exp2.txt', sep = \"\\t\")\n",
    "sigtab2 = dfl2fc_2.loc[(dfl2fc_2.padj < 0.05) & (dfl2fc_2.log2FoldChange > 0) & (dfl2fc_2.Contrast == 'CH_CS') * (dfl2fc_2.Kingdom == \"Bacteria\")]\n",
    "del dfl2fc_2\n",
    "\n",
    "dfl2fc_3 = pandas.read_table('/home/be68/Hyphosphere/data/3Exp/DESeq/dfl2fc-Exp3.txt', sep = \"\\t\")\n",
    "sigtab3 = dfl2fc_3.loc[(dfl2fc_3.padj < 0.05) & (dfl2fc_3.log2FoldChange > 0) & (dfl2fc_3.Contrast == 'CH_CS') & (dfl2fc_3.Kingdom == \"Bacteria\")]\n",
    "del dfl2fc_3\n",
    "\n",
    "# dfl2fc = dfl2fc.append(dfl2fc_CHCS)\n",
    "# dfl2fc = dfl2fc.append(dfl2fc_RHCS)\n",
    "\n",
    "# sigtab = dfl2fc_CHBS.append(dfl2fc_CHCS)\n",
    "# sigtab.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "536"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "sigtab = sigtab1.append(sigtab2) \n",
    "sigtab = sigtab.append(sigtab3)\n",
    "\n",
    "HA = sigtab.OTU.unique()\n",
    "HA = list(HA)\n",
    "#HA.sort()\n",
    "len(HA)\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Identify top 200 ASVs for context"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfl2fc_1 = pandas.read_table('/home/be68/Hyphosphere/data/3Exp/DESeq/dfl2fc-Exp1.txt', sep = \"\\t\")\n",
    "Abund1 = dfl2fc_1.loc[(dfl2fc_1.Contrast == 'CH_CS') & (dfl2fc_1.Kingdom == \"Bacteria\")]\n",
    "del dfl2fc_1 \n",
    "\n",
    "dfl2fc_2 = pandas.read_table('/home/be68/Hyphosphere/data/3Exp/DESeq/dfl2fc-Exp2.txt', sep = \"\\t\")\n",
    "Abund2 = dfl2fc_2.loc[(dfl2fc_2.Contrast == 'CH_CS') & (dfl2fc_2.Kingdom == \"Bacteria\")]\n",
    "del dfl2fc_2\n",
    "\n",
    "dfl2fc_3 = pandas.read_table('/home/be68/Hyphosphere/data/3Exp/DESeq/dfl2fc-Exp3.txt', sep = \"\\t\")\n",
    "Abund3 = dfl2fc_3.loc[(dfl2fc_3.Contrast == 'CH_CS') & (dfl2fc_3.Kingdom == \"Bacteria\")]\n",
    "del dfl2fc_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "Abund = Abund1.append(Abund2)\n",
    "Abund = Abund2.append(Abund3)\n",
    "\n",
    "Abund = Abund.groupby('OTU').agg({'baseMean': 'mean'})\n",
    "Abund = Abund.sort_values(by='baseMean', ascending=False)[0:199]\n",
    "TopASVs = list(Abund.index.values)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Now join lists and prune tree\n",
    "* Top ASVs = TopASVs\n",
    "* Diff Abund asvs = HA\n",
    "* Isolates and top blast hits = IsolateList + IsolateMatches_list\n",
    "\n",
    "* master tree = t_rooted"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['BE26',\n",
       " 'BE27',\n",
       " 'BE28',\n",
       " 'BE29',\n",
       " 'BE30',\n",
       " 'BE31',\n",
       " 'BE32',\n",
       " 'BE33',\n",
       " 'BE34',\n",
       " 'BE35',\n",
       " 'BE36',\n",
       " 'BE38',\n",
       " 'BE39',\n",
       " 'BE40',\n",
       " 'BE41',\n",
       " 'BE42',\n",
       " 'BE43',\n",
       " 'BE44',\n",
       " 'BE45',\n",
       " 'BE46',\n",
       " 'BE47',\n",
       " 'BE48',\n",
       " 'BE49',\n",
       " 'BE50',\n",
       " 'BE51',\n",
       " 'BE52',\n",
       " 'BE53',\n",
       " 'BE54',\n",
       " 'BE55',\n",
       " 'BE56',\n",
       " 'BE57',\n",
       " 'BE59',\n",
       " 'BE60',\n",
       " 'BE61',\n",
       " 'BE62',\n",
       " 'BE63',\n",
       " 'BE64',\n",
       " 'BE65',\n",
       " 'BE65B',\n",
       " 'BE66',\n",
       " 'BE68',\n",
       " 'BE69',\n",
       " 'BE70',\n",
       " 'BE71',\n",
       " 'BE72',\n",
       " 'BE74',\n",
       " 'BE75',\n",
       " 'BE76',\n",
       " 'BE79',\n",
       " 'BE80',\n",
       " 'BE81',\n",
       " 'BE82',\n",
       " 'BE83',\n",
       " 'BE84',\n",
       " 'BE85',\n",
       " 'BE88',\n",
       " 'BE91',\n",
       " 'BE92',\n",
       " 'BE93',\n",
       " 'BE94',\n",
       " 'BE95',\n",
       " 'BE96',\n",
       " 'BE97',\n",
       " 'BE98',\n",
       " 'BE100',\n",
       " 'BE101',\n",
       " 'BE102',\n",
       " 'BE103',\n",
       " 'BE104',\n",
       " 'BE105',\n",
       " 'BE107',\n",
       " 'BE108',\n",
       " 'BE109',\n",
       " 'BE110',\n",
       " 'BE112',\n",
       " 'BE113',\n",
       " 'BE114',\n",
       " 'BE115',\n",
       " 'BE116',\n",
       " 'BE117',\n",
       " 'BE118',\n",
       " 'BE120',\n",
       " 'BE121',\n",
       " 'BE122',\n",
       " 'BE123',\n",
       " 'BE124',\n",
       " 'BE125',\n",
       " 'BE126',\n",
       " 'BE127',\n",
       " 'BE128',\n",
       " 'BE153',\n",
       " 'BE154',\n",
       " 'BE156',\n",
       " 'BE157',\n",
       " 'BE159',\n",
       " 'BE163',\n",
       " 'BE164',\n",
       " 'BE166',\n",
       " 'BE167',\n",
       " 'BE168',\n",
       " 'BE170',\n",
       " 'BE171',\n",
       " 'BE172',\n",
       " 'BE173',\n",
       " 'BE177',\n",
       " 'BE178',\n",
       " 'BE179',\n",
       " 'BE180',\n",
       " 'BE181',\n",
       " 'BE182',\n",
       " 'BE183',\n",
       " 'BE184',\n",
       " 'BE186',\n",
       " 'BE187',\n",
       " 'BE190',\n",
       " 'BE191',\n",
       " 'BE193',\n",
       " 'BE194',\n",
       " 'BE195',\n",
       " 'BE196',\n",
       " 'BE197',\n",
       " 'BE198',\n",
       " 'BE201',\n",
       " 'BE203',\n",
       " 'BE206',\n",
       " 'BE207',\n",
       " 'BE212',\n",
       " 'BE213',\n",
       " 'BE215',\n",
       " 'BE216',\n",
       " 'BE219',\n",
       " 'BE220',\n",
       " 'BE222',\n",
       " 'BE223',\n",
       " 'BE224',\n",
       " 'BE228',\n",
       " 'BE231',\n",
       " 'BE232',\n",
       " 'BE233',\n",
       " 'BE234',\n",
       " 'BE235',\n",
       " 'BE236',\n",
       " 'BE239',\n",
       " 'BE240',\n",
       " 'BE241',\n",
       " 'BE242',\n",
       " 'BE243',\n",
       " 'BE248',\n",
       " 'BE249',\n",
       " 'BE251',\n",
       " 'BE254',\n",
       " 'BE255',\n",
       " 'BE256',\n",
       " 'BE257',\n",
       " 'BE258',\n",
       " 'BE259',\n",
       " 'BE260',\n",
       " 'BE261',\n",
       " 'BE264',\n",
       " 'BE265',\n",
       " 'BE268',\n",
       " 'BE270',\n",
       " 'BE275',\n",
       " 'BE276',\n",
       " 'BE277']"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "IsolateList"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "866"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MasterList = TopASVs + HA + IsolateList + IsolateMatches_list\n",
    "#len(MasterList)\n",
    "MasterList = list(set(MasterList))\n",
    "len(MasterList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "mt_rooted = t_rooted.getSubTree(MasterList)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "mt_rooted.writeToFile(os.path.join(workDir, 'SeqofInt.tree'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Generate dataframes for iTOL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>baseMean</th>\n",
       "      <th>log2FoldChange</th>\n",
       "      <th>lfcSE</th>\n",
       "      <th>stat</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>padj</th>\n",
       "      <th>ASV</th>\n",
       "      <th>Test</th>\n",
       "      <th>TimePoint</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASV11489</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.162168</td>\n",
       "      <td>1.455070</td>\n",
       "      <td>3.093253</td>\n",
       "      <td>0.470401</td>\n",
       "      <td>0.638068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASV11644</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.913138</td>\n",
       "      <td>-3.296519</td>\n",
       "      <td>3.034100</td>\n",
       "      <td>-1.086490</td>\n",
       "      <td>0.277262</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASV3656</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15.109037</td>\n",
       "      <td>-0.425264</td>\n",
       "      <td>2.894952</td>\n",
       "      <td>-0.146898</td>\n",
       "      <td>0.883212</td>\n",
       "      <td>0.972331</td>\n",
       "      <td>ASV562</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASV2718</td>\n",
       "      <td>CH-BS</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    baseMean  log2FoldChange     lfcSE      stat    pvalue      padj  \\\n",
       "1   0.000000             NaN       NaN       NaN       NaN       NaN   \n",
       "2   0.162168        1.455070  3.093253  0.470401  0.638068       NaN   \n",
       "3   0.913138       -3.296519  3.034100 -1.086490  0.277262       NaN   \n",
       "4  15.109037       -0.425264  2.894952 -0.146898  0.883212  0.972331   \n",
       "5   0.000000             NaN       NaN       NaN       NaN       NaN   \n",
       "\n",
       "        ASV   Test  TimePoint  \n",
       "1  ASV11489  CH-BS          1  \n",
       "2  ASV11644  CH-BS          1  \n",
       "3   ASV3656  CH-BS          1  \n",
       "4    ASV562  CH-BS          1  \n",
       "5   ASV2718  CH-BS          1  "
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfl2fc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OTU</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>BP</th>\n",
       "      <th>BS</th>\n",
       "      <th>CH</th>\n",
       "      <th>CS</th>\n",
       "      <th>RH</th>\n",
       "      <th>RT</th>\n",
       "      <th>meanAbund</th>\n",
       "      <th>sumAbund</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ASV100</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Xanthobacteraceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003115</td>\n",
       "      <td>0.004161</td>\n",
       "      <td>0.002490</td>\n",
       "      <td>0.003742</td>\n",
       "      <td>0.000230</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007276</td>\n",
       "      <td>0.010624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASV1000</td>\n",
       "      <td>Verrucomicrobia</td>\n",
       "      <td>Verrucomicrobiae</td>\n",
       "      <td>Chthoniobacteraceae</td>\n",
       "      <td>Candidatus_Udaeobacter</td>\n",
       "      <td>0.000364</td>\n",
       "      <td>0.000745</td>\n",
       "      <td>0.000158</td>\n",
       "      <td>0.000268</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001109</td>\n",
       "      <td>0.001203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASV1002</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Microbacteriaceae</td>\n",
       "      <td>Agromyces</td>\n",
       "      <td>0.000508</td>\n",
       "      <td>0.000406</td>\n",
       "      <td>0.000361</td>\n",
       "      <td>0.000407</td>\n",
       "      <td>0.000142</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000914</td>\n",
       "      <td>0.001317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASV1003</td>\n",
       "      <td>Gemmatimonadetes</td>\n",
       "      <td>Gemmatimonadetes</td>\n",
       "      <td>Gemmatimonadaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000628</td>\n",
       "      <td>0.000274</td>\n",
       "      <td>0.000087</td>\n",
       "      <td>0.000534</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000902</td>\n",
       "      <td>0.000940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ASV10035</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Thermoleophilia</td>\n",
       "      <td>Solirubrobacteraceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000036</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        OTU            Phylum                Class                Family  \\\n",
       "1    ASV100    Proteobacteria  Alphaproteobacteria     Xanthobacteraceae   \n",
       "2   ASV1000   Verrucomicrobia     Verrucomicrobiae   Chthoniobacteraceae   \n",
       "3   ASV1002    Actinobacteria       Actinobacteria     Microbacteriaceae   \n",
       "4   ASV1003  Gemmatimonadetes     Gemmatimonadetes     Gemmatimonadaceae   \n",
       "5  ASV10035    Actinobacteria      Thermoleophilia  Solirubrobacteraceae   \n",
       "\n",
       "                    Genus        BP        BS        CH        CS        RH  \\\n",
       "1                     NaN  0.003115  0.004161  0.002490  0.003742  0.000230   \n",
       "2  Candidatus_Udaeobacter  0.000364  0.000745  0.000158  0.000268  0.000031   \n",
       "3               Agromyces  0.000508  0.000406  0.000361  0.000407  0.000142   \n",
       "4                     NaN  0.000628  0.000274  0.000087  0.000534  0.000046   \n",
       "5                     NaN  0.000000  0.000000  0.000000  0.000000  0.000036   \n",
       "\n",
       "         RT  meanAbund  sumAbund  \n",
       "1  0.000000   0.007276  0.010624  \n",
       "2  0.000000   0.001109  0.001203  \n",
       "3  0.000000   0.000914  0.001317  \n",
       "4  0.000000   0.000902  0.000940  \n",
       "5  0.000013   0.000000  0.000036  "
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ASVabund.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Process dataframes\n",
    "* Create a dataframe with names from MasterList\n",
    "* Left join abundance dataframe to sequences in Master List\n",
    "* generate dfl2fc indicators annd add attributes to dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ASV</th>\n",
       "      <th>OTU</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>BP</th>\n",
       "      <th>BS</th>\n",
       "      <th>CH</th>\n",
       "      <th>CS</th>\n",
       "      <th>RH</th>\n",
       "      <th>RT</th>\n",
       "      <th>meanAbund</th>\n",
       "      <th>sumAbund</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASV2719</td>\n",
       "      <td>ASV2719</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiaceae</td>\n",
       "      <td>Aminobacter</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.000155</td>\n",
       "      <td>0.000096</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.000193</td>\n",
       "      <td>0.000453</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BE42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASV2717</td>\n",
       "      <td>ASV2717</td>\n",
       "      <td>Armatimonadetes</td>\n",
       "      <td>Fimbriimonadia</td>\n",
       "      <td>Fimbriimonadaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000154</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000040</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000184</td>\n",
       "      <td>0.000286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BE41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASV2481</td>\n",
       "      <td>ASV2481</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Haliangiaceae</td>\n",
       "      <td>Haliangium</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000325</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>0.000339</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       ASV      OTU           Phylum                Class             Family  \\\n",
       "0  ASV2719  ASV2719   Proteobacteria  Alphaproteobacteria       Rhizobiaceae   \n",
       "1     BE42      NaN              NaN                  NaN                NaN   \n",
       "2  ASV2717  ASV2717  Armatimonadetes       Fimbriimonadia  Fimbriimonadaceae   \n",
       "3     BE41      NaN              NaN                  NaN                NaN   \n",
       "4  ASV2481  ASV2481   Proteobacteria  Deltaproteobacteria      Haliangiaceae   \n",
       "\n",
       "         Genus        BP        BS        CH        CS        RH        RT  \\\n",
       "0  Aminobacter  0.000038  0.000155  0.000096  0.000025  0.000177  0.000013   \n",
       "1          NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "2          NaN  0.000154  0.000030  0.000187  0.000040  0.000029  0.000011   \n",
       "3          NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "4   Haliangium  0.000176  0.000000  0.000325  0.000014  0.000000  0.000000   \n",
       "\n",
       "   meanAbund  sumAbund  \n",
       "0   0.000193  0.000453  \n",
       "1        NaN       NaN  \n",
       "2   0.000184  0.000286  \n",
       "3        NaN       NaN  \n",
       "4   0.000176  0.000339  "
      ]
     },
     "execution_count": 293,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfM = pandas.DataFrame(MasterList, columns = [\"ASV\"])\n",
    "dfM = pandas.merge(dfM, ASVabund, left_on='ASV', right_on=\"OTU\", how='left')\n",
    "dfM.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Test</th>\n",
       "      <th>CH-BS</th>\n",
       "      <th>CH-CS</th>\n",
       "      <th>RH-CS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV101</th>\n",
       "      <td>2.995563e-02</td>\n",
       "      <td>0.034742</td>\n",
       "      <td>8.431885e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV1018</th>\n",
       "      <td>1.699569e-05</td>\n",
       "      <td>0.000863</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV1019</th>\n",
       "      <td>2.721585e-13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV102</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.708662e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV1022</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.102364e-03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Test            CH-BS     CH-CS         RH-CS\n",
       "ASV                                          \n",
       "ASV101   2.995563e-02  0.034742  8.431885e-11\n",
       "ASV1018  1.699569e-05  0.000863           NaN\n",
       "ASV1019  2.721585e-13       NaN           NaN\n",
       "ASV102            NaN       NaN  2.708662e-03\n",
       "ASV1022           NaN       NaN  1.102364e-03"
      ]
     },
     "execution_count": 294,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sigtab = dfl2fc.loc[(dfl2fc.padj < 0.05) & (dfl2fc.log2FoldChange > 0)]\n",
    "sigtab = sigtab.groupby(['ASV', 'Test'], as_index=False).agg({\"padj\": \"min\"})\n",
    "sigtab = sigtab.pivot(index = 'ASV', columns = \"Test\", values = \"padj\")\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "sigtab.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Test</th>\n",
       "      <th>CH-BS</th>\n",
       "      <th>CH-CS</th>\n",
       "      <th>RH-CS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ASV101</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV1018</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV1019</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV102</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASV1022</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Test     CH-BS  CH-CS  RH-CS\n",
       "ASV                         \n",
       "ASV101     1.0    1.0    1.0\n",
       "ASV1018    1.0    1.0    NaN\n",
       "ASV1019    1.0    NaN    NaN\n",
       "ASV102     NaN    NaN    1.0\n",
       "ASV1022    NaN    NaN    1.0"
      ]
     },
     "execution_count": 295,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Note that this could be a good place to insert color codes into iTOL formated sheet\n",
    "mask = sigtab['CH-BS'] < 0.05\n",
    "column_name = 'CH-BS'\n",
    "sigtab.loc[mask, column_name] = 1\n",
    "\n",
    "mask = sigtab['CH-CS'] < 0.05\n",
    "column_name = 'CH-CS'\n",
    "sigtab.loc[mask, column_name] = 1\n",
    "\n",
    "mask = sigtab['RH-CS'] < 0.05\n",
    "column_name = 'RH-CS'\n",
    "sigtab.loc[mask, column_name] = 1\n",
    "\n",
    "sigtab.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ASV</th>\n",
       "      <th>OTU</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>BP</th>\n",
       "      <th>BS</th>\n",
       "      <th>CH</th>\n",
       "      <th>CS</th>\n",
       "      <th>RH</th>\n",
       "      <th>RT</th>\n",
       "      <th>meanAbund</th>\n",
       "      <th>sumAbund</th>\n",
       "      <th>CH-BS</th>\n",
       "      <th>CH-CS</th>\n",
       "      <th>RH-CS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASV2719</td>\n",
       "      <td>ASV2719</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiaceae</td>\n",
       "      <td>Aminobacter</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.000155</td>\n",
       "      <td>0.000096</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.000193</td>\n",
       "      <td>0.000453</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BE42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASV2717</td>\n",
       "      <td>ASV2717</td>\n",
       "      <td>Armatimonadetes</td>\n",
       "      <td>Fimbriimonadia</td>\n",
       "      <td>Fimbriimonadaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000154</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000040</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000184</td>\n",
       "      <td>0.000286</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BE41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASV2481</td>\n",
       "      <td>ASV2481</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Haliangiaceae</td>\n",
       "      <td>Haliangium</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000325</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>0.000339</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       ASV      OTU           Phylum                Class             Family  \\\n",
       "0  ASV2719  ASV2719   Proteobacteria  Alphaproteobacteria       Rhizobiaceae   \n",
       "1     BE42      NaN              NaN                  NaN                NaN   \n",
       "2  ASV2717  ASV2717  Armatimonadetes       Fimbriimonadia  Fimbriimonadaceae   \n",
       "3     BE41      NaN              NaN                  NaN                NaN   \n",
       "4  ASV2481  ASV2481   Proteobacteria  Deltaproteobacteria      Haliangiaceae   \n",
       "\n",
       "         Genus        BP        BS        CH        CS        RH        RT  \\\n",
       "0  Aminobacter  0.000038  0.000155  0.000096  0.000025  0.000177  0.000013   \n",
       "1          NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "2          NaN  0.000154  0.000030  0.000187  0.000040  0.000029  0.000011   \n",
       "3          NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "4   Haliangium  0.000176  0.000000  0.000325  0.000014  0.000000  0.000000   \n",
       "\n",
       "   meanAbund  sumAbund  CH-BS  CH-CS  RH-CS  \n",
       "0   0.000193  0.000453    NaN    NaN    NaN  \n",
       "1        NaN       NaN    NaN    NaN    NaN  \n",
       "2   0.000184  0.000286    1.0    NaN    NaN  \n",
       "3        NaN       NaN    NaN    NaN    NaN  \n",
       "4   0.000176  0.000339    NaN    1.0    NaN  "
      ]
     },
     "execution_count": 296,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfM = pandas.merge(dfM, sigtab, on = 'ASV', how='left')\n",
    "dfM.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### overall attribute table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "dfM.to_csv(os.path.join(workDir, 'SeqofInt_attributes.txt'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "### Individual tables for iTOL templates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfM['Taxonomy'] = dfM.Phylum + \" \" + dfM.Class + \" \" + dfM.Family"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ASV</th>\n",
       "      <th>Taxonomy</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASV2719</td>\n",
       "      <td>Proteobacteria Alphaproteobacteria Rhizobiaceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BE42</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASV2717</td>\n",
       "      <td>Armatimonadetes Fimbriimonadia Fimbriimonadaceae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BE41</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BE47</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       ASV                                          Taxonomy\n",
       "0  ASV2719   Proteobacteria Alphaproteobacteria Rhizobiaceae\n",
       "1     BE42                                               NaN\n",
       "2  ASV2717  Armatimonadetes Fimbriimonadia Fimbriimonadaceae\n",
       "3     BE41                                               NaN\n",
       "4     BE47                                               NaN"
      ]
     },
     "execution_count": 241,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfT = dfM[['ASV', 'Taxonomy']]\n",
    "\n",
    "dfT.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {},
   "outputs": [],
   "source": [
    "dfT.to_csv(os.path.join(workDir, 'SeqofInt_leaflabels.txt'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Responder color strips"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 324,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "TreeDir = 'iTOL'\n",
    "if not os.path.isdir('TreeDir'):\n",
    "    os.mkdir(TreeDir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* CHCS "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 325,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/home/be68/anaconda3/envs/ipykernel_py2/lib/python2.7/site-packages/ipykernel_launcher.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  if __name__ == '__main__':\n"
     ]
    }
   ],
   "source": [
    "#Replace responder value '1.0' with colorstrip RGB\n",
    "dfR = dfM[['ASV', 'CH-CS']]\n",
    "mask = dfR['CH-CS'] == 1.0\n",
    "column_name = 'CH-CS'\n",
    "dfR.loc[mask, column_name] = '#1b9e77'\n",
    "dfR.head()\n",
    "\n",
    "#Replace NaN values with white RGB\n",
    "dfR['CH-CS'] = dfR['CH-CS'].fillna('#ffffff')\n",
    "\n",
    "#dfR = dfR[['ASV', 'CH-CS']]\n",
    "dfR.head()\n",
    "\n",
    "\n",
    "dfR.to_csv(os.path.join(TreeDir, 'CHCS_colorstrip.txt'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* CHBS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 326,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/home/be68/anaconda3/envs/ipykernel_py2/lib/python2.7/site-packages/ipykernel_launcher.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  if __name__ == '__main__':\n"
     ]
    }
   ],
   "source": [
    "#Replace responder value '1.0' with colorstrip RGB\n",
    "dfR = dfM[['ASV', 'CH-BS']]\n",
    "mask = dfR['CH-BS'] == 1.0\n",
    "column_name = 'CH-BS'\n",
    "dfR.loc[mask, column_name] = '#7570b3'\n",
    "dfR.head()\n",
    "\n",
    "#Replace NaN values with white RGB\n",
    "dfR['CH-BS'] = dfR['CH-BS'].fillna('#ffffff')\n",
    "\n",
    "#dfR = dfR[['ASV', 'CH-CS']]\n",
    "dfR.head()\n",
    "\n",
    "\n",
    "dfR.to_csv(os.path.join(TreeDir, 'CHBS_colorstrip.txt'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 327,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/data/home/be68/anaconda3/envs/ipykernel_py2/lib/python2.7/site-packages/ipykernel_launcher.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  if __name__ == '__main__':\n"
     ]
    }
   ],
   "source": [
    "#Replace responder value '1.0' with colorstrip RGB\n",
    "dfR = dfM[['ASV', 'RH-CS']]\n",
    "mask = dfR['RH-CS'] == 1.0\n",
    "column_name = 'RH-CS'\n",
    "dfR.loc[mask, column_name] = '#d95f02'\n",
    "dfR.head()\n",
    "\n",
    "#Replace NaN values with white RGB\n",
    "dfR['RH-CS'] = dfR['RH-CS'].fillna('#ffffff')\n",
    "\n",
    "#dfR = dfR[['ASV', 'CH-CS']]\n",
    "#dfR.head()\n",
    "\n",
    "\n",
    "dfR.to_csv(os.path.join(TreeDir, 'RHCS_colorstrip.txt'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Identify isolates closely related to responders\n",
    "* Perhaps one color for CHBS or CHCS and another color for isolates only close to RHCS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 386,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASV1</td>\n",
       "      <td>BE256</td>\n",
       "      <td>95.29</td>\n",
       "      <td>255</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>381</td>\n",
       "      <td>633</td>\n",
       "      <td>5.000000e-114</td>\n",
       "      <td>401.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ASV3</td>\n",
       "      <td>BE215</td>\n",
       "      <td>99.21</td>\n",
       "      <td>253</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>384</td>\n",
       "      <td>636</td>\n",
       "      <td>1.000000e-130</td>\n",
       "      <td>457.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASV6</td>\n",
       "      <td>BE96</td>\n",
       "      <td>90.91</td>\n",
       "      <td>253</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>434</td>\n",
       "      <td>686</td>\n",
       "      <td>1.000000e-95</td>\n",
       "      <td>340.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ASV8</td>\n",
       "      <td>BE183</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>382</td>\n",
       "      <td>634</td>\n",
       "      <td>5.000000e-134</td>\n",
       "      <td>468.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASV11</td>\n",
       "      <td>BE193</td>\n",
       "      <td>97.63</td>\n",
       "      <td>253</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>429</td>\n",
       "      <td>681</td>\n",
       "      <td>5.000000e-124</td>\n",
       "      <td>435.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  qseqid sseqid  pident  length  mismatch  gapopen  qstart  qend  sstart  \\\n",
       "0   ASV1  BE256   95.29     255         8        4       1   253     381   \n",
       "1   ASV3  BE215   99.21     253         2        0       1   253     384   \n",
       "2   ASV6   BE96   90.91     253        23        0       1   253     434   \n",
       "3   ASV8  BE183  100.00     253         0        0       1   253     382   \n",
       "4  ASV11  BE193   97.63     253         6        0       1   253     429   \n",
       "\n",
       "   send         evalue  bitscore  \n",
       "0   633  5.000000e-114     401.0  \n",
       "1   636  1.000000e-130     457.0  \n",
       "2   686   1.000000e-95     340.0  \n",
       "3   634  5.000000e-134     468.0  \n",
       "4   681  5.000000e-124     435.0  "
      ]
     },
     "execution_count": 386,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pandas.read_table('ASV-Isolate_blast', sep = \"\\t\", header=None)\n",
    "df.columns = ['qseqid','sseqid',\n",
    "'pident',\n",
    "'length',\n",
    "'mismatch',\n",
    "'gapopen',\n",
    "'qstart',\n",
    "'qend',\n",
    "'sstart',\n",
    "'send',\n",
    "'evalue',\n",
    "'bitscore']\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 401,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>ASV261</td>\n",
       "      <td>BE127</td>\n",
       "      <td>98.81</td>\n",
       "      <td>253</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>435</td>\n",
       "      <td>687</td>\n",
       "      <td>5.000000e-129</td>\n",
       "      <td>451.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>ASV263</td>\n",
       "      <td>BE127</td>\n",
       "      <td>98.02</td>\n",
       "      <td>253</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>435</td>\n",
       "      <td>687</td>\n",
       "      <td>1.000000e-125</td>\n",
       "      <td>440.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>302</th>\n",
       "      <td>ASV361</td>\n",
       "      <td>BE127</td>\n",
       "      <td>98.02</td>\n",
       "      <td>253</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>435</td>\n",
       "      <td>687</td>\n",
       "      <td>1.000000e-125</td>\n",
       "      <td>440.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>477</th>\n",
       "      <td>ASV582</td>\n",
       "      <td>BE127</td>\n",
       "      <td>99.60</td>\n",
       "      <td>253</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>435</td>\n",
       "      <td>687</td>\n",
       "      <td>2.000000e-132</td>\n",
       "      <td>462.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>854</th>\n",
       "      <td>ASV1069</td>\n",
       "      <td>BE127</td>\n",
       "      <td>98.81</td>\n",
       "      <td>253</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>435</td>\n",
       "      <td>687</td>\n",
       "      <td>5.000000e-129</td>\n",
       "      <td>451.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1043</th>\n",
       "      <td>ASV1320</td>\n",
       "      <td>BE127</td>\n",
       "      <td>98.42</td>\n",
       "      <td>253</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>435</td>\n",
       "      <td>687</td>\n",
       "      <td>2.000000e-127</td>\n",
       "      <td>446.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1397</th>\n",
       "      <td>ASV1796</td>\n",
       "      <td>BE127</td>\n",
       "      <td>97.63</td>\n",
       "      <td>253</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>435</td>\n",
       "      <td>687</td>\n",
       "      <td>5.000000e-124</td>\n",
       "      <td>435.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1638</th>\n",
       "      <td>ASV2116</td>\n",
       "      <td>BE127</td>\n",
       "      <td>100.00</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>435</td>\n",
       "      <td>687</td>\n",
       "      <td>5.000000e-134</td>\n",
       "      <td>468.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2227</th>\n",
       "      <td>ASV2910</td>\n",
       "      <td>BE127</td>\n",
       "      <td>97.63</td>\n",
       "      <td>253</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>435</td>\n",
       "      <td>687</td>\n",
       "      <td>5.000000e-124</td>\n",
       "      <td>435.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2253</th>\n",
       "      <td>ASV2943</td>\n",
       "      <td>BE127</td>\n",
       "      <td>98.42</td>\n",
       "      <td>253</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>435</td>\n",
       "      <td>687</td>\n",
       "      <td>2.000000e-127</td>\n",
       "      <td>446.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3125</th>\n",
       "      <td>ASV4193</td>\n",
       "      <td>BE127</td>\n",
       "      <td>97.23</td>\n",
       "      <td>253</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>435</td>\n",
       "      <td>687</td>\n",
       "      <td>2.000000e-122</td>\n",
       "      <td>429.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3636</th>\n",
       "      <td>ASV4992</td>\n",
       "      <td>BE127</td>\n",
       "      <td>97.63</td>\n",
       "      <td>253</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>435</td>\n",
       "      <td>687</td>\n",
       "      <td>5.000000e-124</td>\n",
       "      <td>435.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       qseqid sseqid  pident  length  mismatch  gapopen  qstart  qend  sstart  \\\n",
       "220    ASV261  BE127   98.81     253         3        0       1   253     435   \n",
       "222    ASV263  BE127   98.02     253         5        0       1   253     435   \n",
       "302    ASV361  BE127   98.02     253         5        0       1   253     435   \n",
       "477    ASV582  BE127   99.60     253         1        0       1   253     435   \n",
       "854   ASV1069  BE127   98.81     253         3        0       1   253     435   \n",
       "1043  ASV1320  BE127   98.42     253         4        0       1   253     435   \n",
       "1397  ASV1796  BE127   97.63     253         6        0       1   253     435   \n",
       "1638  ASV2116  BE127  100.00     253         0        0       1   253     435   \n",
       "2227  ASV2910  BE127   97.63     253         6        0       1   253     435   \n",
       "2253  ASV2943  BE127   98.42     253         4        0       1   253     435   \n",
       "3125  ASV4193  BE127   97.23     253         7        0       1   253     435   \n",
       "3636  ASV4992  BE127   97.63     253         6        0       1   253     435   \n",
       "\n",
       "      send         evalue  bitscore  \n",
       "220    687  5.000000e-129     451.0  \n",
       "222    687  1.000000e-125     440.0  \n",
       "302    687  1.000000e-125     440.0  \n",
       "477    687  2.000000e-132     462.0  \n",
       "854    687  5.000000e-129     451.0  \n",
       "1043   687  2.000000e-127     446.0  \n",
       "1397   687  5.000000e-124     435.0  \n",
       "1638   687  5.000000e-134     468.0  \n",
       "2227   687  5.000000e-124     435.0  \n",
       "2253   687  2.000000e-127     446.0  \n",
       "3125   687  2.000000e-122     429.0  \n",
       "3636   687  5.000000e-124     435.0  "
      ]
     },
     "execution_count": 401,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pandas.read_table('ASV-Isolate_blast', sep = \"\\t\", header=None)\n",
    "df.columns = ['qseqid','sseqid',\n",
    "'pident',\n",
    "'length',\n",
    "'mismatch',\n",
    "'gapopen',\n",
    "'qstart',\n",
    "'qend',\n",
    "'sstart',\n",
    "'send',\n",
    "'evalue',\n",
    "'bitscore']\n",
    "df = df.loc[(df.sseqid == 'BE127')& (df.pident > 97)]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 418,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ASV</th>\n",
       "      <th>OTU</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>BP</th>\n",
       "      <th>BS</th>\n",
       "      <th>CH</th>\n",
       "      <th>CS</th>\n",
       "      <th>RH</th>\n",
       "      <th>RT</th>\n",
       "      <th>meanAbund</th>\n",
       "      <th>sumAbund</th>\n",
       "      <th>CH-BS</th>\n",
       "      <th>CH-CS</th>\n",
       "      <th>RH-CS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASV2481</td>\n",
       "      <td>ASV2481</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Haliangiaceae</td>\n",
       "      <td>Haliangium</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000325</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>0.000339</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ASV1404</td>\n",
       "      <td>ASV1404</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Dongiaceae</td>\n",
       "      <td>Dongia</td>\n",
       "      <td>0.000412</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.000301</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>0.000078</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000449</td>\n",
       "      <td>0.000467</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ASV231</td>\n",
       "      <td>ASV231</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Haliangiaceae</td>\n",
       "      <td>Haliangium</td>\n",
       "      <td>0.000073</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0.005937</td>\n",
       "      <td>0.000103</td>\n",
       "      <td>0.000301</td>\n",
       "      <td>0.000095</td>\n",
       "      <td>0.000081</td>\n",
       "      <td>0.006348</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ASV480</td>\n",
       "      <td>ASV480</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>Leptothrix</td>\n",
       "      <td>0.000544</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.003459</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.000149</td>\n",
       "      <td>0.000061</td>\n",
       "      <td>0.000581</td>\n",
       "      <td>0.003657</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ASV897</td>\n",
       "      <td>ASV897</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001839</td>\n",
       "      <td>0.000145</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001985</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ASV2621</td>\n",
       "      <td>ASV2621</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Haliangiaceae</td>\n",
       "      <td>Haliangium</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000638</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000638</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ASV4705</td>\n",
       "      <td>ASV4705</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Haliangiaceae</td>\n",
       "      <td>Haliangium</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000290</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000290</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>ASV153</td>\n",
       "      <td>ASV153</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Unknown_Family</td>\n",
       "      <td>Acidibacter</td>\n",
       "      <td>0.004421</td>\n",
       "      <td>0.000623</td>\n",
       "      <td>0.002689</td>\n",
       "      <td>0.000266</td>\n",
       "      <td>0.000401</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005045</td>\n",
       "      <td>0.003980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>ASV150</td>\n",
       "      <td>ASV150</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Sphingomonadaceae</td>\n",
       "      <td>Sphingomonas</td>\n",
       "      <td>0.000089</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000445</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.000542</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000089</td>\n",
       "      <td>0.000996</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>ASV1975</td>\n",
       "      <td>ASV1975</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Microscillaceae</td>\n",
       "      <td>Ohtaekwangia</td>\n",
       "      <td>0.000079</td>\n",
       "      <td>0.000045</td>\n",
       "      <td>0.000625</td>\n",
       "      <td>0.000039</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000125</td>\n",
       "      <td>0.000747</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>ASV2028</td>\n",
       "      <td>ASV2028</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Xanthomonadaceae</td>\n",
       "      <td>Arenimonas</td>\n",
       "      <td>0.000242</td>\n",
       "      <td>0.000090</td>\n",
       "      <td>0.000290</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000333</td>\n",
       "      <td>0.000380</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>ASV147</td>\n",
       "      <td>ASV147</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>Ideonella</td>\n",
       "      <td>0.000186</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.004734</td>\n",
       "      <td>0.000445</td>\n",
       "      <td>0.001105</td>\n",
       "      <td>0.003055</td>\n",
       "      <td>0.000231</td>\n",
       "      <td>0.006330</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>ASV1904</td>\n",
       "      <td>ASV1904</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Microscillaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000074</td>\n",
       "      <td>0.000294</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>0.000143</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>0.000074</td>\n",
       "      <td>0.000551</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>ASV473</td>\n",
       "      <td>ASV473</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Cytophagaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001652</td>\n",
       "      <td>0.000026</td>\n",
       "      <td>0.002235</td>\n",
       "      <td>0.000144</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.003913</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>ASV977</td>\n",
       "      <td>ASV977</td>\n",
       "      <td>Chloroflexi</td>\n",
       "      <td>Chloroflexia</td>\n",
       "      <td>Herpetosiphonaceae</td>\n",
       "      <td>Herpetosiphon</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001564</td>\n",
       "      <td>0.000188</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001752</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>ASV276</td>\n",
       "      <td>ASV276</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Microscillaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002595</td>\n",
       "      <td>0.000632</td>\n",
       "      <td>0.001314</td>\n",
       "      <td>0.000365</td>\n",
       "      <td>0.000178</td>\n",
       "      <td>0.000236</td>\n",
       "      <td>0.003228</td>\n",
       "      <td>0.002490</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>ASV3134</td>\n",
       "      <td>ASV3134</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000663</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000663</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>ASV1357</td>\n",
       "      <td>ASV1357</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Steroidobacteraceae</td>\n",
       "      <td>Steroidobacter</td>\n",
       "      <td>0.000267</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.000634</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0.000048</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>0.000273</td>\n",
       "      <td>0.000713</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>ASV663</td>\n",
       "      <td>ASV663</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Micromonosporaceae</td>\n",
       "      <td>Plantactinospora</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.002858</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.002907</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>ASV1018</td>\n",
       "      <td>ASV1018</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Sandaracinaceae</td>\n",
       "      <td>Sandaracinus</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000618</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000747</td>\n",
       "      <td>0.000114</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.001366</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>ASV1681</td>\n",
       "      <td>ASV1681</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Rhodanobacteraceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>0.000656</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000221</td>\n",
       "      <td>0.000720</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>ASV1122</td>\n",
       "      <td>ASV1122</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Intrasporangiaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000443</td>\n",
       "      <td>0.000077</td>\n",
       "      <td>0.000566</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.000092</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000520</td>\n",
       "      <td>0.000774</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>ASV1125</td>\n",
       "      <td>ASV1125</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>BIrii41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000970</td>\n",
       "      <td>0.000075</td>\n",
       "      <td>0.000373</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001045</td>\n",
       "      <td>0.000515</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>ASV2108</td>\n",
       "      <td>ASV2108</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Chitinophagaceae</td>\n",
       "      <td>Terrimonas</td>\n",
       "      <td>0.000267</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000267</td>\n",
       "      <td>0.000238</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>ASV2109</td>\n",
       "      <td>ASV2109</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Nitrosomonadaceae</td>\n",
       "      <td>Ellin6067</td>\n",
       "      <td>0.000128</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>0.000709</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000154</td>\n",
       "      <td>0.000736</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>ASV2101</td>\n",
       "      <td>ASV2101</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000854</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000870</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>ASV1547</td>\n",
       "      <td>ASV1547</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Haliangiaceae</td>\n",
       "      <td>Haliangium</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000891</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000122</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001012</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>ASV101</td>\n",
       "      <td>ASV101</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>0319-6G20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000356</td>\n",
       "      <td>0.000050</td>\n",
       "      <td>0.008504</td>\n",
       "      <td>0.002939</td>\n",
       "      <td>0.000019</td>\n",
       "      <td>0.008910</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>ASV46</td>\n",
       "      <td>ASV46</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Unknown_Family</td>\n",
       "      <td>Acidibacter</td>\n",
       "      <td>0.009776</td>\n",
       "      <td>0.002728</td>\n",
       "      <td>0.006229</td>\n",
       "      <td>0.002639</td>\n",
       "      <td>0.000806</td>\n",
       "      <td>0.000026</td>\n",
       "      <td>0.012503</td>\n",
       "      <td>0.012403</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>ASV43</td>\n",
       "      <td>ASV43</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiaceae</td>\n",
       "      <td>Allorhizobium-Neorhizobium-Pararhizobium-Rhizo...</td>\n",
       "      <td>0.000276</td>\n",
       "      <td>0.000418</td>\n",
       "      <td>0.001289</td>\n",
       "      <td>0.000256</td>\n",
       "      <td>0.015225</td>\n",
       "      <td>0.004310</td>\n",
       "      <td>0.000694</td>\n",
       "      <td>0.017188</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>360</th>\n",
       "      <td>ASV1414</td>\n",
       "      <td>ASV1414</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Flavobacteriaceae</td>\n",
       "      <td>Flavobacterium</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000590</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000590</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361</th>\n",
       "      <td>ASV222</td>\n",
       "      <td>ASV222</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Sandaracinaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001571</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003563</td>\n",
       "      <td>0.000381</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>0.005133</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>362</th>\n",
       "      <td>ASV224</td>\n",
       "      <td>ASV224</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Nitrosomonadaceae</td>\n",
       "      <td>Ellin6067</td>\n",
       "      <td>0.000093</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007127</td>\n",
       "      <td>0.000280</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000093</td>\n",
       "      <td>0.007407</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>363</th>\n",
       "      <td>ASV3140</td>\n",
       "      <td>ASV3140</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Acidiferrobacteraceae</td>\n",
       "      <td>Sulfurifustis</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000665</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000665</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>364</th>\n",
       "      <td>ASV228</td>\n",
       "      <td>ASV228</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Haliangiaceae</td>\n",
       "      <td>Haliangium</td>\n",
       "      <td>0.000048</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>0.003646</td>\n",
       "      <td>0.000113</td>\n",
       "      <td>0.002181</td>\n",
       "      <td>0.001061</td>\n",
       "      <td>0.000066</td>\n",
       "      <td>0.005959</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365</th>\n",
       "      <td>ASV1384</td>\n",
       "      <td>ASV1384</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>Noviherbaspirillum</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001063</td>\n",
       "      <td>0.000208</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001271</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368</th>\n",
       "      <td>ASV737</td>\n",
       "      <td>ASV737</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Rhodanobacteraceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000118</td>\n",
       "      <td>0.000064</td>\n",
       "      <td>0.002014</td>\n",
       "      <td>0.000107</td>\n",
       "      <td>0.000040</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000181</td>\n",
       "      <td>0.002224</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>370</th>\n",
       "      <td>ASV888</td>\n",
       "      <td>ASV888</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Cytophagaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001714</td>\n",
       "      <td>0.000053</td>\n",
       "      <td>0.000609</td>\n",
       "      <td>0.000075</td>\n",
       "      <td>0.000004</td>\n",
       "      <td>0.002375</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379</th>\n",
       "      <td>ASV2644</td>\n",
       "      <td>ASV2644</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Acidiferrobacteraceae</td>\n",
       "      <td>Sulfurifustis</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000550</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000550</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>ASV644</td>\n",
       "      <td>ASV644</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004489</td>\n",
       "      <td>0.000085</td>\n",
       "      <td>0.000366</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.004941</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>405</th>\n",
       "      <td>ASV547</td>\n",
       "      <td>ASV547</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Chitinophagaceae</td>\n",
       "      <td>Niastella</td>\n",
       "      <td>0.000640</td>\n",
       "      <td>0.000306</td>\n",
       "      <td>0.000851</td>\n",
       "      <td>0.000299</td>\n",
       "      <td>0.000099</td>\n",
       "      <td>0.000297</td>\n",
       "      <td>0.000946</td>\n",
       "      <td>0.001554</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406</th>\n",
       "      <td>ASV541</td>\n",
       "      <td>ASV541</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Haliangiaceae</td>\n",
       "      <td>Haliangium</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002865</td>\n",
       "      <td>0.000044</td>\n",
       "      <td>0.000181</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000049</td>\n",
       "      <td>0.003091</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>ASV4447</td>\n",
       "      <td>ASV4447</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Haliangiaceae</td>\n",
       "      <td>Haliangium</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000373</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.000384</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>411</th>\n",
       "      <td>ASV99</td>\n",
       "      <td>ASV99</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>Rhizobacter</td>\n",
       "      <td>0.000665</td>\n",
       "      <td>0.000751</td>\n",
       "      <td>0.014062</td>\n",
       "      <td>0.001257</td>\n",
       "      <td>0.000572</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001416</td>\n",
       "      <td>0.016643</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>422</th>\n",
       "      <td>ASV8</td>\n",
       "      <td>ASV8</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Caulobacteraceae</td>\n",
       "      <td>Asticcacaulis</td>\n",
       "      <td>0.000339</td>\n",
       "      <td>0.000023</td>\n",
       "      <td>0.001922</td>\n",
       "      <td>0.000378</td>\n",
       "      <td>0.039888</td>\n",
       "      <td>0.074301</td>\n",
       "      <td>0.000362</td>\n",
       "      <td>0.042211</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>ASV263</td>\n",
       "      <td>ASV263</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000267</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.004147</td>\n",
       "      <td>0.000415</td>\n",
       "      <td>0.001757</td>\n",
       "      <td>0.000120</td>\n",
       "      <td>0.000453</td>\n",
       "      <td>0.006506</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>444</th>\n",
       "      <td>ASV1443</td>\n",
       "      <td>ASV1443</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Actinobacteria</td>\n",
       "      <td>Streptomycetaceae</td>\n",
       "      <td>Streptomyces</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000734</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000048</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000782</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>445</th>\n",
       "      <td>ASV1113</td>\n",
       "      <td>ASV1113</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Sandaracinaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000444</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000515</td>\n",
       "      <td>0.000258</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000959</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>446</th>\n",
       "      <td>ASV602</td>\n",
       "      <td>ASV602</td>\n",
       "      <td>Spirochaetes</td>\n",
       "      <td>Spirochaetia</td>\n",
       "      <td>Spirochaetaceae</td>\n",
       "      <td>Salinispira</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.003294</td>\n",
       "      <td>0.000140</td>\n",
       "      <td>0.000393</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>0.003840</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>450</th>\n",
       "      <td>ASV161</td>\n",
       "      <td>ASV161</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006203</td>\n",
       "      <td>0.000252</td>\n",
       "      <td>0.003774</td>\n",
       "      <td>0.000048</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.010230</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>453</th>\n",
       "      <td>ASV2111</td>\n",
       "      <td>ASV2111</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Microscillaceae</td>\n",
       "      <td>Chryseolinea</td>\n",
       "      <td>0.000329</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>0.000170</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000355</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>457</th>\n",
       "      <td>ASV1231</td>\n",
       "      <td>ASV1231</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Haliangiaceae</td>\n",
       "      <td>Haliangium</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001224</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>0.001224</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>463</th>\n",
       "      <td>ASV1652</td>\n",
       "      <td>ASV1652</td>\n",
       "      <td>Acidobacteria</td>\n",
       "      <td>Blastocatellia_(Subgroup_4)</td>\n",
       "      <td>Pyrinomonadaceae</td>\n",
       "      <td>RB41</td>\n",
       "      <td>0.000341</td>\n",
       "      <td>0.000113</td>\n",
       "      <td>0.000232</td>\n",
       "      <td>0.000202</td>\n",
       "      <td>0.000057</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000454</td>\n",
       "      <td>0.000604</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>470</th>\n",
       "      <td>ASV1931</td>\n",
       "      <td>ASV1931</td>\n",
       "      <td>Bacteroidetes</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Chitinophagaceae</td>\n",
       "      <td>Flavihumibacter</td>\n",
       "      <td>0.000274</td>\n",
       "      <td>0.000034</td>\n",
       "      <td>0.000393</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000308</td>\n",
       "      <td>0.000470</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>471</th>\n",
       "      <td>ASV693</td>\n",
       "      <td>ASV693</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Haliangiaceae</td>\n",
       "      <td>Haliangium</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.000094</td>\n",
       "      <td>0.003530</td>\n",
       "      <td>0.000168</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000116</td>\n",
       "      <td>0.003827</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476</th>\n",
       "      <td>ASV597</td>\n",
       "      <td>ASV597</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>Rhodoferax</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002571</td>\n",
       "      <td>0.000075</td>\n",
       "      <td>0.000612</td>\n",
       "      <td>0.000032</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.003258</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>485</th>\n",
       "      <td>ASV341</td>\n",
       "      <td>ASV341</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Haliangiaceae</td>\n",
       "      <td>Haliangium</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>0.004539</td>\n",
       "      <td>0.000125</td>\n",
       "      <td>0.000256</td>\n",
       "      <td>0.000048</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>0.004953</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>488</th>\n",
       "      <td>ASV3875</td>\n",
       "      <td>ASV3875</td>\n",
       "      <td>Gemmatimonadetes</td>\n",
       "      <td>Gemmatimonadetes</td>\n",
       "      <td>Gemmatimonadaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>0.000095</td>\n",
       "      <td>0.000220</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000110</td>\n",
       "      <td>0.000315</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>489</th>\n",
       "      <td>ASV184</td>\n",
       "      <td>ASV184</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Unknown_Family</td>\n",
       "      <td>Acidibacter</td>\n",
       "      <td>0.004056</td>\n",
       "      <td>0.000142</td>\n",
       "      <td>0.002350</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>0.000381</td>\n",
       "      <td>0.000059</td>\n",
       "      <td>0.004198</td>\n",
       "      <td>0.002939</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>490</th>\n",
       "      <td>ASV187</td>\n",
       "      <td>ASV187</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Burkholderiaceae</td>\n",
       "      <td>Rhizobacter</td>\n",
       "      <td>0.000448</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001353</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000948</td>\n",
       "      <td>0.005231</td>\n",
       "      <td>0.000448</td>\n",
       "      <td>0.002301</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>122 rows  17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ASV      OTU            Phylum                        Class  \\\n",
       "4    ASV2481  ASV2481    Proteobacteria          Deltaproteobacteria   \n",
       "7    ASV1404  ASV1404    Proteobacteria          Alphaproteobacteria   \n",
       "11    ASV231   ASV231    Proteobacteria          Deltaproteobacteria   \n",
       "13    ASV480   ASV480    Proteobacteria          Gammaproteobacteria   \n",
       "17    ASV897   ASV897    Proteobacteria          Gammaproteobacteria   \n",
       "18   ASV2621  ASV2621    Proteobacteria          Deltaproteobacteria   \n",
       "28   ASV4705  ASV4705    Proteobacteria          Deltaproteobacteria   \n",
       "32    ASV153   ASV153    Proteobacteria          Gammaproteobacteria   \n",
       "33    ASV150   ASV150    Proteobacteria          Alphaproteobacteria   \n",
       "37   ASV1975  ASV1975     Bacteroidetes                  Bacteroidia   \n",
       "47   ASV2028  ASV2028    Proteobacteria          Gammaproteobacteria   \n",
       "48    ASV147   ASV147    Proteobacteria          Gammaproteobacteria   \n",
       "63   ASV1904  ASV1904     Bacteroidetes                  Bacteroidia   \n",
       "65    ASV473   ASV473     Bacteroidetes                  Bacteroidia   \n",
       "72    ASV977   ASV977       Chloroflexi                 Chloroflexia   \n",
       "78    ASV276   ASV276     Bacteroidetes                  Bacteroidia   \n",
       "79   ASV3134  ASV3134    Proteobacteria          Gammaproteobacteria   \n",
       "81   ASV1357  ASV1357    Proteobacteria          Gammaproteobacteria   \n",
       "83    ASV663   ASV663    Actinobacteria               Actinobacteria   \n",
       "87   ASV1018  ASV1018    Proteobacteria          Deltaproteobacteria   \n",
       "89   ASV1681  ASV1681    Proteobacteria          Gammaproteobacteria   \n",
       "94   ASV1122  ASV1122    Actinobacteria               Actinobacteria   \n",
       "95   ASV1125  ASV1125    Proteobacteria          Deltaproteobacteria   \n",
       "98   ASV2108  ASV2108     Bacteroidetes                  Bacteroidia   \n",
       "99   ASV2109  ASV2109    Proteobacteria          Gammaproteobacteria   \n",
       "100  ASV2101  ASV2101    Proteobacteria          Gammaproteobacteria   \n",
       "101  ASV1547  ASV1547    Proteobacteria          Deltaproteobacteria   \n",
       "115   ASV101   ASV101    Proteobacteria          Deltaproteobacteria   \n",
       "118    ASV46    ASV46    Proteobacteria          Gammaproteobacteria   \n",
       "121    ASV43    ASV43    Proteobacteria          Alphaproteobacteria   \n",
       "..       ...      ...               ...                          ...   \n",
       "360  ASV1414  ASV1414     Bacteroidetes                  Bacteroidia   \n",
       "361   ASV222   ASV222    Proteobacteria          Deltaproteobacteria   \n",
       "362   ASV224   ASV224    Proteobacteria          Gammaproteobacteria   \n",
       "363  ASV3140  ASV3140    Proteobacteria          Gammaproteobacteria   \n",
       "364   ASV228   ASV228    Proteobacteria          Deltaproteobacteria   \n",
       "365  ASV1384  ASV1384    Proteobacteria          Gammaproteobacteria   \n",
       "368   ASV737   ASV737    Proteobacteria          Gammaproteobacteria   \n",
       "370   ASV888   ASV888     Bacteroidetes                  Bacteroidia   \n",
       "379  ASV2644  ASV2644    Proteobacteria          Gammaproteobacteria   \n",
       "396   ASV644   ASV644    Proteobacteria          Gammaproteobacteria   \n",
       "405   ASV547   ASV547     Bacteroidetes                  Bacteroidia   \n",
       "406   ASV541   ASV541    Proteobacteria          Deltaproteobacteria   \n",
       "409  ASV4447  ASV4447    Proteobacteria          Deltaproteobacteria   \n",
       "411    ASV99    ASV99    Proteobacteria          Gammaproteobacteria   \n",
       "422     ASV8     ASV8    Proteobacteria          Alphaproteobacteria   \n",
       "425   ASV263   ASV263    Proteobacteria          Gammaproteobacteria   \n",
       "444  ASV1443  ASV1443    Actinobacteria               Actinobacteria   \n",
       "445  ASV1113  ASV1113    Proteobacteria          Deltaproteobacteria   \n",
       "446   ASV602   ASV602      Spirochaetes                 Spirochaetia   \n",
       "450   ASV161   ASV161    Proteobacteria          Gammaproteobacteria   \n",
       "453  ASV2111  ASV2111     Bacteroidetes                  Bacteroidia   \n",
       "457  ASV1231  ASV1231    Proteobacteria          Deltaproteobacteria   \n",
       "463  ASV1652  ASV1652     Acidobacteria  Blastocatellia_(Subgroup_4)   \n",
       "470  ASV1931  ASV1931     Bacteroidetes                  Bacteroidia   \n",
       "471   ASV693   ASV693    Proteobacteria          Deltaproteobacteria   \n",
       "476   ASV597   ASV597    Proteobacteria          Gammaproteobacteria   \n",
       "485   ASV341   ASV341    Proteobacteria          Deltaproteobacteria   \n",
       "488  ASV3875  ASV3875  Gemmatimonadetes             Gemmatimonadetes   \n",
       "489   ASV184   ASV184    Proteobacteria          Gammaproteobacteria   \n",
       "490   ASV187   ASV187    Proteobacteria          Gammaproteobacteria   \n",
       "\n",
       "                    Family                                              Genus  \\\n",
       "4            Haliangiaceae                                         Haliangium   \n",
       "7               Dongiaceae                                             Dongia   \n",
       "11           Haliangiaceae                                         Haliangium   \n",
       "13        Burkholderiaceae                                         Leptothrix   \n",
       "17        Burkholderiaceae                                                NaN   \n",
       "18           Haliangiaceae                                         Haliangium   \n",
       "28           Haliangiaceae                                         Haliangium   \n",
       "32          Unknown_Family                                        Acidibacter   \n",
       "33       Sphingomonadaceae                                       Sphingomonas   \n",
       "37         Microscillaceae                                       Ohtaekwangia   \n",
       "47        Xanthomonadaceae                                         Arenimonas   \n",
       "48        Burkholderiaceae                                          Ideonella   \n",
       "63         Microscillaceae                                                NaN   \n",
       "65           Cytophagaceae                                                NaN   \n",
       "72      Herpetosiphonaceae                                      Herpetosiphon   \n",
       "78         Microscillaceae                                                NaN   \n",
       "79        Burkholderiaceae                                                NaN   \n",
       "81     Steroidobacteraceae                                     Steroidobacter   \n",
       "83      Micromonosporaceae                                   Plantactinospora   \n",
       "87         Sandaracinaceae                                       Sandaracinus   \n",
       "89      Rhodanobacteraceae                                                NaN   \n",
       "94      Intrasporangiaceae                                                NaN   \n",
       "95                 BIrii41                                                NaN   \n",
       "98        Chitinophagaceae                                         Terrimonas   \n",
       "99       Nitrosomonadaceae                                          Ellin6067   \n",
       "100       Burkholderiaceae                                                NaN   \n",
       "101          Haliangiaceae                                         Haliangium   \n",
       "115              0319-6G20                                                NaN   \n",
       "118         Unknown_Family                                        Acidibacter   \n",
       "121           Rhizobiaceae  Allorhizobium-Neorhizobium-Pararhizobium-Rhizo...   \n",
       "..                     ...                                                ...   \n",
       "360      Flavobacteriaceae                                     Flavobacterium   \n",
       "361        Sandaracinaceae                                                NaN   \n",
       "362      Nitrosomonadaceae                                          Ellin6067   \n",
       "363  Acidiferrobacteraceae                                      Sulfurifustis   \n",
       "364          Haliangiaceae                                         Haliangium   \n",
       "365       Burkholderiaceae                                 Noviherbaspirillum   \n",
       "368     Rhodanobacteraceae                                                NaN   \n",
       "370          Cytophagaceae                                                NaN   \n",
       "379  Acidiferrobacteraceae                                      Sulfurifustis   \n",
       "396       Burkholderiaceae                                                NaN   \n",
       "405       Chitinophagaceae                                          Niastella   \n",
       "406          Haliangiaceae                                         Haliangium   \n",
       "409          Haliangiaceae                                         Haliangium   \n",
       "411       Burkholderiaceae                                        Rhizobacter   \n",
       "422       Caulobacteraceae                                      Asticcacaulis   \n",
       "425       Burkholderiaceae                                                NaN   \n",
       "444      Streptomycetaceae                                       Streptomyces   \n",
       "445        Sandaracinaceae                                                NaN   \n",
       "446        Spirochaetaceae                                        Salinispira   \n",
       "450       Burkholderiaceae                                                NaN   \n",
       "453        Microscillaceae                                       Chryseolinea   \n",
       "457          Haliangiaceae                                         Haliangium   \n",
       "463       Pyrinomonadaceae                                               RB41   \n",
       "470       Chitinophagaceae                                    Flavihumibacter   \n",
       "471          Haliangiaceae                                         Haliangium   \n",
       "476       Burkholderiaceae                                         Rhodoferax   \n",
       "485          Haliangiaceae                                         Haliangium   \n",
       "488      Gemmatimonadaceae                                                NaN   \n",
       "489         Unknown_Family                                        Acidibacter   \n",
       "490       Burkholderiaceae                                        Rhizobacter   \n",
       "\n",
       "           BP        BS        CH        CS        RH        RT  meanAbund  \\\n",
       "4    0.000176  0.000000  0.000325  0.000014  0.000000  0.000000   0.000176   \n",
       "7    0.000412  0.000037  0.000301  0.000051  0.000078  0.000000   0.000449   \n",
       "11   0.000073  0.000007  0.005937  0.000103  0.000301  0.000095   0.000081   \n",
       "13   0.000544  0.000037  0.003459  0.000012  0.000149  0.000061   0.000581   \n",
       "17   0.000000  0.000000  0.001839  0.000145  0.000000  0.000000   0.000000   \n",
       "18   0.000000  0.000000  0.000638  0.000000  0.000000  0.000000   0.000000   \n",
       "28   0.000000  0.000000  0.000290  0.000000  0.000000  0.000000   0.000000   \n",
       "32   0.004421  0.000623  0.002689  0.000266  0.000401  0.000000   0.005045   \n",
       "33   0.000089  0.000000  0.000445  0.000009  0.000542  0.000000   0.000089   \n",
       "37   0.000079  0.000045  0.000625  0.000039  0.000038  0.000000   0.000125   \n",
       "47   0.000242  0.000090  0.000290  0.000000  0.000000  0.000000   0.000333   \n",
       "48   0.000186  0.000046  0.004734  0.000445  0.001105  0.003055   0.000231   \n",
       "63   0.000000  0.000074  0.000294  0.000041  0.000143  0.000049   0.000074   \n",
       "65   0.000004  0.000000  0.001652  0.000026  0.002235  0.000144   0.000004   \n",
       "72   0.000000  0.000000  0.001564  0.000188  0.000000  0.000000   0.000000   \n",
       "78   0.002595  0.000632  0.001314  0.000365  0.000178  0.000236   0.003228   \n",
       "79   0.000000  0.000000  0.000663  0.000000  0.000000  0.000000   0.000000   \n",
       "81   0.000267  0.000006  0.000634  0.000025  0.000048  0.000100   0.000273   \n",
       "83   0.000000  0.000022  0.002858  0.000027  0.000000  0.000000   0.000022   \n",
       "87   0.000012  0.000000  0.000618  0.000000  0.000747  0.000114   0.000012   \n",
       "89   0.000195  0.000027  0.000656  0.000038  0.000000  0.000000   0.000221   \n",
       "94   0.000443  0.000077  0.000566  0.000038  0.000092  0.000000   0.000520   \n",
       "95   0.000970  0.000075  0.000373  0.000000  0.000067  0.000000   0.001045   \n",
       "98   0.000267  0.000000  0.000187  0.000000  0.000051  0.000000   0.000267   \n",
       "99   0.000128  0.000027  0.000709  0.000000  0.000000  0.000000   0.000154   \n",
       "100  0.000000  0.000000  0.000854  0.000000  0.000016  0.000000   0.000000   \n",
       "101  0.000000  0.000000  0.000891  0.000000  0.000122  0.000000   0.000000   \n",
       "115  0.000019  0.000000  0.000356  0.000050  0.008504  0.002939   0.000019   \n",
       "118  0.009776  0.002728  0.006229  0.002639  0.000806  0.000026   0.012503   \n",
       "121  0.000276  0.000418  0.001289  0.000256  0.015225  0.004310   0.000694   \n",
       "..        ...       ...       ...       ...       ...       ...        ...   \n",
       "360  0.000000  0.000000  0.000590  0.000000  0.000000  0.000000   0.000000   \n",
       "361  0.000042  0.000000  0.001571  0.000000  0.003563  0.000381   0.000042   \n",
       "362  0.000093  0.000000  0.007127  0.000280  0.000000  0.000000   0.000093   \n",
       "363  0.000000  0.000000  0.000665  0.000000  0.000000  0.000000   0.000000   \n",
       "364  0.000048  0.000018  0.003646  0.000113  0.002181  0.001061   0.000066   \n",
       "365  0.000000  0.000000  0.001063  0.000208  0.000000  0.000000   0.000000   \n",
       "368  0.000118  0.000064  0.002014  0.000107  0.000040  0.000000   0.000181   \n",
       "370  0.000004  0.000000  0.001714  0.000053  0.000609  0.000075   0.000004   \n",
       "379  0.000000  0.000000  0.000550  0.000000  0.000000  0.000000   0.000000   \n",
       "396  0.000022  0.000000  0.004489  0.000085  0.000366  0.000000   0.000022   \n",
       "405  0.000640  0.000306  0.000851  0.000299  0.000099  0.000297   0.000946   \n",
       "406  0.000049  0.000000  0.002865  0.000044  0.000181  0.000000   0.000049   \n",
       "409  0.000016  0.000000  0.000373  0.000000  0.000011  0.000000   0.000016   \n",
       "411  0.000665  0.000751  0.014062  0.001257  0.000572  0.000000   0.001416   \n",
       "422  0.000339  0.000023  0.001922  0.000378  0.039888  0.074301   0.000362   \n",
       "425  0.000267  0.000187  0.004147  0.000415  0.001757  0.000120   0.000453   \n",
       "444  0.000000  0.000000  0.000734  0.000000  0.000048  0.000000   0.000000   \n",
       "445  0.000000  0.000000  0.000444  0.000000  0.000515  0.000258   0.000000   \n",
       "446  0.000009  0.000012  0.003294  0.000140  0.000393  0.000013   0.000021   \n",
       "450  0.000000  0.000000  0.006203  0.000252  0.003774  0.000048   0.000000   \n",
       "453  0.000329  0.000027  0.000170  0.000000  0.000000  0.000000   0.000355   \n",
       "457  0.000037  0.000000  0.001224  0.000000  0.000000  0.000000   0.000037   \n",
       "463  0.000341  0.000113  0.000232  0.000202  0.000057  0.000000   0.000454   \n",
       "470  0.000274  0.000034  0.000393  0.000021  0.000022  0.000000   0.000308   \n",
       "471  0.000022  0.000094  0.003530  0.000168  0.000035  0.000000   0.000116   \n",
       "476  0.000016  0.000000  0.002571  0.000075  0.000612  0.000032   0.000016   \n",
       "485  0.000000  0.000033  0.004539  0.000125  0.000256  0.000048   0.000033   \n",
       "488  0.000015  0.000095  0.000220  0.000000  0.000000  0.000000   0.000110   \n",
       "489  0.004056  0.000142  0.002350  0.000067  0.000381  0.000059   0.004198   \n",
       "490  0.000448  0.000000  0.001353  0.000000  0.000948  0.005231   0.000448   \n",
       "\n",
       "     sumAbund  CH-BS  CH-CS  RH-CS  \n",
       "4    0.000339    NaN    1.0    NaN  \n",
       "7    0.000467    1.0    1.0    NaN  \n",
       "11   0.006348    1.0    1.0    NaN  \n",
       "13   0.003657    1.0    1.0    NaN  \n",
       "17   0.001985    1.0    1.0    NaN  \n",
       "18   0.000638    1.0    1.0    NaN  \n",
       "28   0.000290    NaN    1.0    NaN  \n",
       "32   0.003980    NaN    1.0    NaN  \n",
       "33   0.000996    1.0    1.0    NaN  \n",
       "37   0.000747    1.0    1.0    NaN  \n",
       "47   0.000380    1.0    1.0    NaN  \n",
       "48   0.006330    1.0    1.0    NaN  \n",
       "63   0.000551    NaN    1.0    NaN  \n",
       "65   0.003913    1.0    1.0    1.0  \n",
       "72   0.001752    1.0    1.0    NaN  \n",
       "78   0.002490    NaN    1.0    NaN  \n",
       "79   0.000663    1.0    1.0    NaN  \n",
       "81   0.000713    NaN    1.0    NaN  \n",
       "83   0.002907    1.0    1.0    NaN  \n",
       "87   0.001366    1.0    1.0    NaN  \n",
       "89   0.000720    1.0    1.0    NaN  \n",
       "94   0.000774    1.0    1.0    NaN  \n",
       "95   0.000515    1.0    1.0    NaN  \n",
       "98   0.000238    1.0    1.0    NaN  \n",
       "99   0.000736    1.0    1.0    NaN  \n",
       "100  0.000870    1.0    1.0    NaN  \n",
       "101  0.001012    1.0    1.0    NaN  \n",
       "115  0.008910    1.0    1.0    1.0  \n",
       "118  0.012403    1.0    1.0    NaN  \n",
       "121  0.017188    1.0    1.0    1.0  \n",
       "..        ...    ...    ...    ...  \n",
       "360  0.000590    NaN    1.0    NaN  \n",
       "361  0.005133    1.0    1.0    1.0  \n",
       "362  0.007407    1.0    1.0    NaN  \n",
       "363  0.000665    1.0    1.0    NaN  \n",
       "364  0.005959    1.0    1.0    NaN  \n",
       "365  0.001271    1.0    1.0    NaN  \n",
       "368  0.002224    1.0    1.0    NaN  \n",
       "370  0.002375    1.0    1.0    NaN  \n",
       "379  0.000550    1.0    1.0    NaN  \n",
       "396  0.004941    1.0    1.0    NaN  \n",
       "405  0.001554    1.0    1.0    NaN  \n",
       "406  0.003091    1.0    1.0    NaN  \n",
       "409  0.000384    1.0    1.0    NaN  \n",
       "411  0.016643    1.0    1.0    NaN  \n",
       "422  0.042211    1.0    1.0    1.0  \n",
       "425  0.006506    1.0    1.0    NaN  \n",
       "444  0.000782    1.0    1.0    NaN  \n",
       "445  0.000959    1.0    1.0    NaN  \n",
       "446  0.003840    1.0    1.0    NaN  \n",
       "450  0.010230    1.0    1.0    1.0  \n",
       "453  0.000196    1.0    1.0    NaN  \n",
       "457  0.001224    1.0    1.0    NaN  \n",
       "463  0.000604    1.0    1.0    NaN  \n",
       "470  0.000470    NaN    1.0    NaN  \n",
       "471  0.003827    1.0    1.0    NaN  \n",
       "476  0.003258    1.0    1.0    NaN  \n",
       "485  0.004953    1.0    1.0    NaN  \n",
       "488  0.000315    NaN    1.0    NaN  \n",
       "489  0.002939    1.0    1.0    NaN  \n",
       "490  0.002301    1.0    1.0    1.0  \n",
       "\n",
       "[122 rows x 17 columns]"
      ]
     },
     "execution_count": 418,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmp = dfM[dfM.ASV.isin(list(sigCHCS.ASV))]\n",
    "tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.loc[df.pident > 97]\n",
    "#iTOL code for bolding ASV names requires\n",
    "#node 9606 will have its label displayed in blue with bold italic font, and with yellow background\n",
    "#9606,label,node,#0000ff,1,bold-italic,#ffff00\n",
    "#ASV,label,node,#000000,1,bold,#ffffff\n",
    "df = df[['qseqid']]\n",
    "df['label'] = 'label'\n",
    "df['node'] = 'node'\n",
    "df['font_color'] = '#000000' #black\n",
    "df['number'] = '1'\n",
    "df['font_type'] = 'bold'\n",
    "df['background_colr'] = '#ffff00'#yellow background\n",
    "\n",
    "#mask = dfR['RH-CS'] == 1.0\n",
    "#df[['ASV','label',]]\n",
    "df.head()\n",
    "df.to_csv(os.path.join(TreeDir, 'labelStyle.txt'), sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Explore"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 354,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ASV</th>\n",
       "      <th>OTU</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>BP</th>\n",
       "      <th>BS</th>\n",
       "      <th>CH</th>\n",
       "      <th>CS</th>\n",
       "      <th>RH</th>\n",
       "      <th>RT</th>\n",
       "      <th>meanAbund</th>\n",
       "      <th>sumAbund</th>\n",
       "      <th>CH-BS</th>\n",
       "      <th>CH-CS</th>\n",
       "      <th>RH-CS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>ASV5271</td>\n",
       "      <td>ASV5271</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Brevibacillus</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000068</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>ASV7471</td>\n",
       "      <td>ASV7471</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.000026</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0.000103</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>ASV302</td>\n",
       "      <td>ASV302</td>\n",
       "      <td>Firmicutes</td>\n",
       "      <td>Bacilli</td>\n",
       "      <td>Paenibacillaceae</td>\n",
       "      <td>Paenibacillus</td>\n",
       "      <td>0.00037</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>0.000165</td>\n",
       "      <td>0.000106</td>\n",
       "      <td>0.000147</td>\n",
       "      <td>0.000134</td>\n",
       "      <td>0.000397</td>\n",
       "      <td>0.000444</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         ASV      OTU      Phylum    Class            Family          Genus  \\\n",
       "49   ASV5271  ASV5271  Firmicutes  Bacilli  Paenibacillaceae  Brevibacillus   \n",
       "67   ASV7471  ASV7471  Firmicutes  Bacilli  Paenibacillaceae  Paenibacillus   \n",
       "435   ASV302   ASV302  Firmicutes  Bacilli  Paenibacillaceae  Paenibacillus   \n",
       "\n",
       "          BP        BS        CH        CS        RH        RT  meanAbund  \\\n",
       "49   0.00000  0.000000  0.000000  0.000000  0.000068  0.000056   0.000000   \n",
       "67   0.00000  0.000046  0.000026  0.000000  0.000031  0.000000   0.000046   \n",
       "435  0.00037  0.000027  0.000165  0.000106  0.000147  0.000134   0.000397   \n",
       "\n",
       "     sumAbund  CH-BS  CH-CS  RH-CS  \n",
       "49   0.000068    NaN    NaN    NaN  \n",
       "67   0.000103    NaN    NaN    NaN  \n",
       "435  0.000444    NaN    NaN    NaN  "
      ]
     },
     "execution_count": 354,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfM.loc[dfM.Family == \"Paenibacillaceae\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ASV</th>\n",
       "      <th>OTU</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>BP</th>\n",
       "      <th>BS</th>\n",
       "      <th>CH</th>\n",
       "      <th>CS</th>\n",
       "      <th>RH</th>\n",
       "      <th>RT</th>\n",
       "      <th>meanAbund</th>\n",
       "      <th>sumAbund</th>\n",
       "      <th>CH-BS</th>\n",
       "      <th>CH-CS</th>\n",
       "      <th>RH-CS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ASV2719</td>\n",
       "      <td>ASV2719</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Alphaproteobacteria</td>\n",
       "      <td>Rhizobiaceae</td>\n",
       "      <td>Aminobacter</td>\n",
       "      <td>0.000038</td>\n",
       "      <td>0.000155</td>\n",
       "      <td>0.000096</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>0.000177</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.000193</td>\n",
       "      <td>0.000453</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BE42</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ASV2717</td>\n",
       "      <td>ASV2717</td>\n",
       "      <td>Armatimonadetes</td>\n",
       "      <td>Fimbriimonadia</td>\n",
       "      <td>Fimbriimonadaceae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000154</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000040</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.000184</td>\n",
       "      <td>0.000286</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BE41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASV2481</td>\n",
       "      <td>ASV2481</td>\n",
       "      <td>Proteobacteria</td>\n",
       "      <td>Deltaproteobacteria</td>\n",
       "      <td>Haliangiaceae</td>\n",
       "      <td>Haliangium</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000325</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000176</td>\n",
       "      <td>0.000339</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       ASV      OTU           Phylum                Class             Family  \\\n",
       "0  ASV2719  ASV2719   Proteobacteria  Alphaproteobacteria       Rhizobiaceae   \n",
       "1     BE42      NaN              NaN                  NaN                NaN   \n",
       "2  ASV2717  ASV2717  Armatimonadetes       Fimbriimonadia  Fimbriimonadaceae   \n",
       "3     BE41      NaN              NaN                  NaN                NaN   \n",
       "4  ASV2481  ASV2481   Proteobacteria  Deltaproteobacteria      Haliangiaceae   \n",
       "\n",
       "         Genus        BP        BS        CH        CS        RH        RT  \\\n",
       "0  Aminobacter  0.000038  0.000155  0.000096  0.000025  0.000177  0.000013   \n",
       "1          NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "2          NaN  0.000154  0.000030  0.000187  0.000040  0.000029  0.000011   \n",
       "3          NaN       NaN       NaN       NaN       NaN       NaN       NaN   \n",
       "4   Haliangium  0.000176  0.000000  0.000325  0.000014  0.000000  0.000000   \n",
       "\n",
       "   meanAbund  sumAbund  CH-BS  CH-CS  RH-CS  \n",
       "0   0.000193  0.000453    NaN    NaN    NaN  \n",
       "1        NaN       NaN    NaN    NaN    NaN  \n",
       "2   0.000184  0.000286    1.0    NaN    NaN  \n",
       "3        NaN       NaN    NaN    NaN    NaN  \n",
       "4   0.000176  0.000339    NaN    1.0    NaN  "
      ]
     },
     "execution_count": 359,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfM.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# closest hits to responders"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 362,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>qseqid</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BE26</td>\n",
       "      <td>ASV193</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>444</td>\n",
       "      <td>696</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BE27</td>\n",
       "      <td>ASV193</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>450</td>\n",
       "      <td>702</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BE28</td>\n",
       "      <td>ASV193</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>452</td>\n",
       "      <td>704</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BE29</td>\n",
       "      <td>ASV39</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>433</td>\n",
       "      <td>685</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BE30</td>\n",
       "      <td>ASV39</td>\n",
       "      <td>100.0</td>\n",
       "      <td>253</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>433</td>\n",
       "      <td>685</td>\n",
       "      <td>1</td>\n",
       "      <td>253</td>\n",
       "      <td>3.000000e-132</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  qseqid  sseqid  pident  length  mismatch  gapopen  qstart  qend  sstart  \\\n",
       "0   BE26  ASV193   100.0     253         0        0     444   696       1   \n",
       "1   BE27  ASV193   100.0     253         0        0     450   702       1   \n",
       "2   BE28  ASV193   100.0     253         0        0     452   704       1   \n",
       "3   BE29   ASV39   100.0     253         0        0     433   685       1   \n",
       "4   BE30   ASV39   100.0     253         0        0     433   685       1   \n",
       "\n",
       "   send         evalue  bitscore  \n",
       "0   253  3.000000e-132       468  \n",
       "1   253  3.000000e-132       468  \n",
       "2   253  3.000000e-132       468  \n",
       "3   253  3.000000e-132       468  \n",
       "4   253  3.000000e-132       468  "
      ]
     },
     "execution_count": 362,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pandas.read_table('Isolate-ASV_blast', sep = \"\\t\", header=None)\n",
    "df.columns = ['qseqid','sseqid',\n",
    "'pident',\n",
    "'length',\n",
    "'mismatch',\n",
    "'gapopen',\n",
    "'qstart',\n",
    "'qend',\n",
    "'sstart',\n",
    "'send',\n",
    "'evalue',\n",
    "'bitscore']\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 370,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "IsolateMatches = list(df.sseqid)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check closest isolate match against CH-CS responders"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 381,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ASV12', 'ASV150', 'ASV1811', 'ASV452', 'ASV58', 'ASV8'}"
      ]
     },
     "execution_count": 381,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sigtab = dfl2fc.loc[(dfl2fc.padj < 0.05) & (dfl2fc.log2FoldChange > 0)]\n",
    "sigCHCS = sigtab.loc[sigtab.Test == 'CH-CS']\n",
    "matchCHCS = sigCHCS.loc[sigCHCS.ASV.isin(IsolateMatches)]\n",
    "matchCHCS = set(list(matchCHCS.ASV))\n",
    "matchCHCS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check closest isolate match against CH-BS responders"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 382,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ASV1030',\n",
       " 'ASV12',\n",
       " 'ASV13',\n",
       " 'ASV150',\n",
       " 'ASV1811',\n",
       " 'ASV2669',\n",
       " 'ASV360',\n",
       " 'ASV452',\n",
       " 'ASV8'}"
      ]
     },
     "execution_count": 382,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sigtab = dfl2fc.loc[(dfl2fc.padj < 0.05) & (dfl2fc.log2FoldChange > 0)]\n",
    "sigCHBS = sigtab.loc[sigtab.Test == 'CH-BS']\n",
    "#sigCHBS.head()\n",
    "matchCHBS = sigCHBS.loc[sigCHBS.ASV.isin(IsolateMatches)]\n",
    "matchCHBS = set(list(matchCHBS.ASV))\n",
    "matchCHBS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 383,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ASV12',\n",
       " 'ASV13',\n",
       " 'ASV171',\n",
       " 'ASV188',\n",
       " 'ASV23',\n",
       " 'ASV233',\n",
       " 'ASV236',\n",
       " 'ASV296',\n",
       " 'ASV350',\n",
       " 'ASV362',\n",
       " 'ASV39',\n",
       " 'ASV430',\n",
       " 'ASV452',\n",
       " 'ASV477',\n",
       " 'ASV481',\n",
       " 'ASV522',\n",
       " 'ASV58',\n",
       " 'ASV687',\n",
       " 'ASV765',\n",
       " 'ASV77',\n",
       " 'ASV792',\n",
       " 'ASV8',\n",
       " 'ASV81',\n",
       " 'ASV891',\n",
       " 'ASV90'}"
      ]
     },
     "execution_count": 383,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sigtab = dfl2fc.loc[(dfl2fc.padj < 0.05) & (dfl2fc.log2FoldChange > 0)]\n",
    "sigRHCS = sigtab.loc[sigtab.Test == 'RH-CS']\n",
    "#sigCHBS.head()\n",
    "matchRHCS = sigRHCS.loc[sigRHCS.ASV.isin(IsolateMatches)]\n",
    "matchRHCS = set(list(matchRHCS.ASV))\n",
    "matchRHCS\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 385,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "25"
      ]
     },
     "execution_count": 385,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(matchRHCS)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
